{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {
    "slideshow": {
     "slide_type": "slide"
    }
   },
   "source": [
    "# HTTP/HTML and Web Services\n",
    "## 10/26/2021\n",
    "<a href=\"?print-pdf\">print view</a>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<script src=\"http://bits.csb.pitt.edu/asker.js/lib/asker.js\"></script>\n",
       "<style>\n",
       ".reveal pre { font-size: 100%}\n",
       "</style>\n",
       "\n",
       "<script>\n",
       "\n",
       "require(['https://cdnjs.cloudflare.com/ajax/libs/Chart.js/2.2.2/Chart.js'], function(Ch){\n",
       " Chart = Ch;\n",
       "});\n",
       "\n",
       "$('head').append('<link rel=\"stylesheet\" href=\"http://bits.csb.pitt.edu/asker.js/themes/asker.default.css\" />');\n",
       "\n",
       "\n",
       "//the callback is provided a canvas object and data \n",
       "var chartmaker = function(canvas, labels, data) {\n",
       "  var ctx = $(canvas).get(0).getContext(\"2d\");\n",
       "     var dataset = {labels: labels,                     \n",
       "    datasets:[{\n",
       "     data: data,\n",
       "     backgroundColor: \"rgba(150,64,150,0.5)\",\n",
       "         fillColor: \"rgba(150,64,150,0.8)\",    \n",
       "  }]};\n",
       "  var myBarChart = new Chart(ctx,{type:'bar',data:dataset,options:{legend: {display:false},\n",
       "        scales: {\n",
       "            yAxes: [{\n",
       "                ticks: {\n",
       "                    min: 0,\n",
       "                }\n",
       "            }]}}});\n",
       "};\n",
       "\n",
       "$(\".jp-InputArea .o:contains(html)\").closest('.jp-InputArea').hide();\n",
       "\n",
       "\n",
       "</script>\n"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%%html\n",
    "<script src=\"http://bits.csb.pitt.edu/asker.js/lib/asker.js\"></script>\n",
    "<style>\n",
    ".reveal pre { font-size: 100%}\n",
    "</style>\n",
    "\n",
    "<script>\n",
    "\n",
    "require(['https://cdnjs.cloudflare.com/ajax/libs/Chart.js/2.2.2/Chart.js'], function(Ch){\n",
    " Chart = Ch;\n",
    "});\n",
    "\n",
    "$('head').append('<link rel=\"stylesheet\" href=\"http://bits.csb.pitt.edu/asker.js/themes/asker.default.css\" />');\n",
    "\n",
    "\n",
    "//the callback is provided a canvas object and data \n",
    "var chartmaker = function(canvas, labels, data) {\n",
    "  var ctx = $(canvas).get(0).getContext(\"2d\");\n",
    "     var dataset = {labels: labels,                     \n",
    "    datasets:[{\n",
    "     data: data,\n",
    "     backgroundColor: \"rgba(150,64,150,0.5)\",\n",
    "         fillColor: \"rgba(150,64,150,0.8)\",    \n",
    "  }]};\n",
    "  var myBarChart = new Chart(ctx,{type:'bar',data:dataset,options:{legend: {display:false},\n",
    "        scales: {\n",
    "            yAxes: [{\n",
    "                ticks: {\n",
    "                    min: 0,\n",
    "                }\n",
    "            }]}}});\n",
    "};\n",
    "\n",
    "$(\".jp-InputArea .o:contains(html)\").closest('.jp-InputArea').hide();\n",
    "\n",
    "\n",
    "</script>\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "slideshow": {
     "slide_type": "slide"
    }
   },
   "source": [
    "<center><img src='imgs/OSI_model.jpg'></center>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "slideshow": {
     "slide_type": "slide"
    }
   },
   "source": [
    "# HTTP"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Hypertext Transfer Protocol\n",
    "\n",
    "A request-response protocol in a client-server framework."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "slideshow": {
     "slide_type": "slide"
    }
   },
   "source": [
    "# HTTP Requests"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The request consists of the following:\n",
    "\n",
    "* A request line with desired method (action)\n",
    "* Request Headers\n",
    "* An empty line.\n",
    "* An optional message body.\n",
    "\n",
    "Example\n",
    "<pre>GET / HTTP/1.1\n",
    "Host: cnn.com\n",
    "Connection: keep-alive\n",
    "Accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,*/*;q=0.8\n",
    "User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_8_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/30.0.1599.69 Safari/537.36\n",
    "Accept-Encoding: gzip,deflate,sdch\n",
    "Accept-Language: en-US,en;q=0.8\n",
    "Cookie: SelectedEdition=www; optimizelyEndUserId=oeu1364949768474r0.014468349516391754; optimizelySegments=%7B%22170962340%22%3A%22false%22%2C%22171657961%22%3A%22gc%22%2C%22172148679%22%3A%22none%22%2C%22172265329%22%3A%22search%22%7D; optimizelyBuckets=%7B%7D; s_vi=[CS]v1|25F54343051D3284-6000013900246AA6[CE]</pre>\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "slideshow": {
     "slide_type": "slide"
    }
   },
   "source": [
    "# HTTP Requests"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    " **GET**\n",
    " \n",
    " Requests a representation of the specified resource. Requests using GET should only retrieve data and should have no other effect.\n",
    " \n",
    "**POST**\n",
    "\n",
    "Submits data to the server in the request body.  Can have side-effects.\n",
    "\n",
    "**HEAD**\n",
    "\n",
    "Asks for the response identical to the one that would correspond to a GET request, but without the response body (just the headers). \n",
    "\n",
    "**OPTIONS, PUT, DELETE, TRACE and CONNECT**\n",
    "\n",
    "HTTP 1.1 methods that are less commonly used."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "slideshow": {
     "slide_type": "slide"
    }
   },
   "source": [
    "# Sending Data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**GET**\n",
    "\n",
    "Data must be in the query string of the URL.  This is the part of the URL after a question mark:\n",
    "\n",
    "    http://server/program/path/?query_string\n",
    "\n",
    "The query_string is made up of name=value pairs separated by &:\n",
    "\n",
    "    http://server/program/path/?field1=value1&field2=value2&field3=value3\n",
    "    \n",
    "URLs have length limits (differs by browser, but generally needs to be <2000 characters)\n",
    "    \n",
    "**POST**\n",
    "\n",
    "The data is sent in the request body. There is no length limit (just patience limit)."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "slideshow": {
     "slide_type": "slide"
    }
   },
   "source": [
    "http://www.rcsb.org/pdb/ngl/ngl.do?pdbid=3ERK&bionumber=1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div id=\"httpgetdata\" style=\"width: 500px\"></div>\n",
       "<script>\n",
       "$('head').append('<link rel=\"stylesheet\" href=\"http://bits.csb.pitt.edu/asker.js/themes/asker.default.css\" />');\n",
       "\n",
       "    var divid = '#httpgetdata';\n",
       "\tjQuery(divid).asker({\n",
       "\t    id: divid,\n",
       "\t    question: \"How many data items are being sent in this URL?\",\n",
       "\t\tanswers: ['0','1','2','3','4'],\n",
       "        server: \"http://bits.csb.pitt.edu/asker.js/example/asker.cgi\",\n",
       "\t\tcharter: chartmaker})\n",
       "    \n",
       "$(\".jp-InputArea .o:contains(html)\").closest('.jp-InputArea').hide();\n",
       "\n",
       "\n",
       "</script>\n"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%%html\n",
    "<div id=\"httpgetdata\" style=\"width: 500px\"></div>\n",
    "<script>\n",
    "$('head').append('<link rel=\"stylesheet\" href=\"http://bits.csb.pitt.edu/asker.js/themes/asker.default.css\" />');\n",
    "\n",
    "    var divid = '#httpgetdata';\n",
    "\tjQuery(divid).asker({\n",
    "\t    id: divid,\n",
    "\t    question: \"How many data items are being sent in this URL?\",\n",
    "\t\tanswers: ['0','1','2','3','4'],\n",
    "        server: \"http://bits.csb.pitt.edu/asker.js/example/asker.cgi\",\n",
    "\t\tcharter: chartmaker})\n",
    "    \n",
    "$(\".jp-InputArea .o:contains(html)\").closest('.jp-InputArea').hide();\n",
    "\n",
    "\n",
    "</script>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "slideshow": {
     "slide_type": "slide"
    }
   },
   "source": [
    "# HTTP Response"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "A response consists of the following:\n",
    "\n",
    "* A Status-Line (includes [response code](https://en.wikipedia.org/wiki/List_of_HTTP_status_codes))\n",
    "* Response Headers\n",
    "* An empty line\n",
    "* An optional message body\n",
    "\n",
    "<br>\n",
    "\n",
    "Example:\n",
    "\n",
    "```\n",
    "HTTP/1.1 200 OK\n",
    "Server: nginx\n",
    "Date: Thu, 10 Oct 2013 14:23:59 GMT\n",
    "Content-Type: text/html\n",
    "Transfer-Encoding: chunked\n",
    "Connection: keep-alive\n",
    "Set-Cookie: CG=US:PA:Pittsburgh; path=/\n",
    "Last-Modified: Thu, 10 Oct 2013 14:23:14 GMT\n",
    "Vary: Accept-Encoding\n",
    "Cache-Control: max-age=60, private\n",
    "Expires: Thu, 10 Oct 2013 14:24:58 GMT\n",
    "Content-Encoding: gzip\n",
    "\n",
    "<!DOCTYPE HTML> <html lang=\"en-US\"> <head> <title>CNN.com - Breaking News, U.S., World, Weather, Entertainment &amp; Video News</title> \n",
    "``` \n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "slideshow": {
     "slide_type": "slide"
    }
   },
   "source": [
    "# requests "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "`requests` is a simple but high-level interface for requesting http data\n",
    "\n",
    "Also:\n",
    " * `urllib2` - another high-level interface\n",
    " * `urllib` - similar to `urllib2` but different, has `urlencode` function\n",
    " * `urllib3` - successor to `urllib2` but different\n",
    " * `httplib` - low-level interface to http request\n",
    " * `mechanize` - much higher level interface for scripting web interactions - use this if you need to submit form data, passwords, etc.\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "slideshow": {
     "slide_type": "slide"
    }
   },
   "source": [
    "# `get`\n",
    "\n",
    "`requests.get` takes a URL and returns a response object that contains the message body.\n",
    "\n",
    "**Note:** `urllib2` and `mechanize` have a `urlopen` method that returns a file-like object."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<!DOCTYPE html PUBLIC \"-//W3C//DTD XHTML 1.0 Transitional//EN\" \"http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd\">\n",
      "<html xmlns=\"http://www.w3.org/1999/xhtml\">\n",
      "<head>\n",
      "<meta http-equiv=\"content-type\" content=\"text/html; charset=utf-8\" />\n",
      "<title>MSCBIO 2025: Introduction to Bioinformatics Programming in Python</title>\n",
      "<link href=\"default.css\" rel=\"stylesheet\" type=\"text/css\" />\n",
      "\n",
      "</head>\n",
      "\n",
      "<body>\n",
      "<div class=\"main\">\n",
      "<img src=\"logo.png\" align=right></img>\n",
      "<h1>Introduction to Bioinformatics Programming in Python</h1>\n",
      "<h3><a href=\"syllabus.pdf\">Syllabus</a> | <a href=\"schedule.html\">Schedule</a> | <a href=\"assignments.html\">Assignments</a> | <a href=\"https://pitt.zoom.us/j/95735576127\">Zoom</a> </h3> \n",
      "\n",
      "<h2>Announcements</h2>\n",
      "\n",
      "Course communication will use Slack. <br> \n",
      "<a href=\"http://mscbio2025.slack.com\"><tt>http://mscbio2025.slack.com</tt></a><br>\n",
      "\n",
      "<p>\n",
      "<h2>Instructor</h2>\n",
      "<a href=\"http://bits.csb.pitt.edu\">David Koes</a><br>\n",
      "748 Murdoch Building<br>\n",
      "Office Hours: 11am Tuesday, after class and by appointment<br>\n",
      "<a href=\"mailto:dkoes@pitt.edu\"><tt>dkoes@pitt.edu</tt></a><p>\n",
      "\n",
      "\n",
      "\n",
      "<h2>General Info</h2>\n",
      " The goal\n",
      "of this course is to provide you with some of the essential computing skills and knowledge you\n",
      "will need to be a successful computational biologist. \n",
      "<p>There are two lectures a week on <b>Tuesdays and Thursdays from 12:00-1:20pm</b> in <a href=\"https://www.csb.pitt.edu/contact/\">Murdoch</a> 814.  Additionally, all lectures will be \n",
      "streamed over <a href=\"https://pitt.zoom.us/j/96033338651\">Zoom</a>.\n",
      "Each lecture will cover the basics of a different area in computational biology in the context of python\n",
      "programming project. \n",
      "<p>\n",
      "We will be using <a href=\"http://www.python.org\">Python</a> for the course assignments.\n",
      "Python includes surprisingly <a href=\"https://docs.python.org/3.8/\">good documentation</a>\n",
      "for learning the language, but if you would like a hard-copy textbook I recommend\n",
      "<a href=\"http://www.amazon.com/Bioinformatics-Programming-Using-Python-Biological/dp/059615450X/ref=sr_1_1?ie=UTF8&qid=1377019134&sr=8-1&keywords=bioinformatics+python\">\n",
      "Bioinformatics Programming Using Python</a> or \n",
      "<a href=\"https://www.amazon.com/Data-Science-Scratch-Principles-Python/dp/149190142X\">Data Science from Scratch: First Principles with Python</a>.\n",
      "<p>\n",
      "Course materials, such as slides and example code, will be linked from the <a href=\"schedule.html\">schedule</a>.\n",
      "Course policies can be found in the <a href=\"syllabus.pdf\">syllabus</a>.\n",
      "</p>\n",
      "</div> \n",
      "</body>\n",
      "</html>\n",
      "\n"
     ]
    }
   ],
   "source": [
    "import requests\n",
    "response = requests.get('http://mscbio2025.csb.pitt.edu')\n",
    "print(response.text)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "slideshow": {
     "slide_type": "slide"
    }
   },
   "source": [
    "**Must include protocol in URL**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "ename": "MissingSchema",
     "evalue": "Invalid URL 'www.cnn.com': No schema supplied. Perhaps you meant http://www.cnn.com?",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mMissingSchema\u001b[0m                             Traceback (most recent call last)",
      "\u001b[0;32m/var/folders/c_/pwm7n7_174724g8zkkqlpr3m0000gn/T/ipykernel_86730/2714011330.py\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mfail\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mrequests\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'www.cnn.com'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;32m/Library/Python/3.8/site-packages/requests/api.py\u001b[0m in \u001b[0;36mget\u001b[0;34m(url, params, **kwargs)\u001b[0m\n\u001b[1;32m     73\u001b[0m     \"\"\"\n\u001b[1;32m     74\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 75\u001b[0;31m     \u001b[0;32mreturn\u001b[0m \u001b[0mrequest\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'get'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0murl\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mparams\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mparams\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     76\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     77\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Library/Python/3.8/site-packages/requests/api.py\u001b[0m in \u001b[0;36mrequest\u001b[0;34m(method, url, **kwargs)\u001b[0m\n\u001b[1;32m     59\u001b[0m     \u001b[0;31m# cases, and look like a memory leak in others.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     60\u001b[0m     \u001b[0;32mwith\u001b[0m \u001b[0msessions\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mSession\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0msession\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 61\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0msession\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mrequest\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mmethod\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mmethod\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0murl\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0murl\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     62\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     63\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Library/Python/3.8/site-packages/requests/sessions.py\u001b[0m in \u001b[0;36mrequest\u001b[0;34m(self, method, url, params, data, headers, cookies, files, auth, timeout, allow_redirects, proxies, hooks, stream, verify, cert, json)\u001b[0m\n\u001b[1;32m    526\u001b[0m             \u001b[0mhooks\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mhooks\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    527\u001b[0m         )\n\u001b[0;32m--> 528\u001b[0;31m         \u001b[0mprep\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mprepare_request\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mreq\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    529\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    530\u001b[0m         \u001b[0mproxies\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mproxies\u001b[0m \u001b[0;32mor\u001b[0m \u001b[0;34m{\u001b[0m\u001b[0;34m}\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Library/Python/3.8/site-packages/requests/sessions.py\u001b[0m in \u001b[0;36mprepare_request\u001b[0;34m(self, request)\u001b[0m\n\u001b[1;32m    454\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    455\u001b[0m         \u001b[0mp\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mPreparedRequest\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 456\u001b[0;31m         p.prepare(\n\u001b[0m\u001b[1;32m    457\u001b[0m             \u001b[0mmethod\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mrequest\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mmethod\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mupper\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    458\u001b[0m             \u001b[0murl\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mrequest\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0murl\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Library/Python/3.8/site-packages/requests/models.py\u001b[0m in \u001b[0;36mprepare\u001b[0;34m(self, method, url, headers, files, data, params, auth, cookies, hooks, json)\u001b[0m\n\u001b[1;32m    314\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    315\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mprepare_method\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mmethod\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 316\u001b[0;31m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mprepare_url\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0murl\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mparams\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    317\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mprepare_headers\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mheaders\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    318\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mprepare_cookies\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcookies\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Library/Python/3.8/site-packages/requests/models.py\u001b[0m in \u001b[0;36mprepare_url\u001b[0;34m(self, url, params)\u001b[0m\n\u001b[1;32m    388\u001b[0m             \u001b[0merror\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0merror\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mformat\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mto_native_string\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0murl\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'utf8'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    389\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 390\u001b[0;31m             \u001b[0;32mraise\u001b[0m \u001b[0mMissingSchema\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0merror\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    391\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    392\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0mhost\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mMissingSchema\u001b[0m: Invalid URL 'www.cnn.com': No schema supplied. Perhaps you meant http://www.cnn.com?"
     ]
    }
   ],
   "source": [
    "fail = requests.get('www.cnn.com')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "slideshow": {
     "slide_type": "slide"
    }
   },
   "source": [
    "# Requesting with Data (POST)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<Response [200]>\n"
     ]
    }
   ],
   "source": [
    "url = 'http://pocketquery.csb.pitt.edu/pocket.cgi'\n",
    "values = {'json' : '{\"pdbid_text\":[\"1ycr\"],\"start\":0,\"num\":24,\"sort\":\"score\",\"dir\":\"desc\",\"cmd\":\"cluster\"}'}\n",
    "\n",
    "data = requests.post(url,data=values)\n",
    "print(data)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "the_page = data.text"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['1YCR,B,178,2,5.856,19;23,PHE;TRP,-6.31,-12.62,-6.61,-6.01,5.20125,10.4025,4.7898,5.6127,133.9,267.8,131.1,136.7,72.3,144.6,64.7,79.9,0,0,0,0,0,0,0,0,0.972816',\n",
       " '1YCR,B,129,1,0,19,PHE,-6.61,-6.61,-6.61,-6.61,5.6127,5.6127,5.6127,5.6127,131.1,131.1,131.1,131.1,79.9,79.9,79.9,79.9,0,0,0,0,0,0,0,0,0.96009']"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    " the_page.split('\\n')[16:18]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "slideshow": {
     "slide_type": "slide"
    }
   },
   "source": [
    "# Requesting with Data (GET)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<!DOCTYPE html><html lang=\"en\"><head><script src=\"https://www.googletagmanager.com/gtag/js?id=UA-3923365-3\" async></script><script>//- global rcsb-config object\n",
      "var RC = {\n",
      "      googleAnalyticsTrackingId: 'UA-3923365-3'\n",
      "    , instance: 'production'\n",
      "    , isProductionServer: true\n",
      "    , dataUrl: 'https://data.rcsb.org/'\n",
      "    , MOLSTAR_IMG_URL: 'https://cdn.rcsb.org/images/structures/'\n",
      "};\n",
      "</script><script src=\"//cdn.rcsb.org/javascript/react/react.min.js\"></script><script src=\"//cdn.rcsb.org/javascript/react/reactDOM.min.js\"></script><script src=\"/search/search-data?ts=5450659\"> </script><script src=\"/js/search/react-search.js?ts=5450659\"></script><script>!function(){if(\"performance\"in window==0&&(window.performance={}),Date.now=Date.now||function(){return(new Date).getTime()},\"now\"in window.performance==0){var n=Date.now();performance.timing&&performance.timing.navigationStart&&(n=performance.timing.navigationStart),window.performance.now=function(){return Date.now()-n}}}();(function(){var h=\"undefined\"!=typeof window&&window===this?this:\"undefined\"!=typeof global&&null!=global?global:this,k=\"function\"==typeof Object.defineProperties?Object.defineProperty:function(a,b,c){a!=Array.prototype&&a!=Object.prototype&&(a[b]=c.value)};function l(){l=function(){};h.Symbol||(h.Symbol=m)}var n=0;function m(a){return\"jscomp_symbol_\"+(a||\"\")+n++}\n",
      "    function p(){l();var a=h.Symbol.iterator;a||(a=h.Symbol.iterator=h.Symbol(\"iterator\"));\"function\"!=typeof Array.prototype[a]&&k(Array.prototype,a,{configurable:!0,writable:!0,value:function(){return q(this)}});p=function(){}}function q(a){var b=0;return r(function(){return b<a.length?{done:!1,value:a[b++]}:{done:!0}})}function r(a){p();a={next:a};a[h.Symbol.iterator]=function(){return this};return a}function t(a){p();var b=a[Symbol.iterator];return b?b.call(a):q(a)}\n",
      "    function u(a){if(!(a instanceof Array)){a=t(a);for(var b,c=[];!(b=a.next()).done;)c.push(b.value);a=c}return a}var v=0;function w(a,b){var c=XMLHttpRequest.prototype.send,d=v++;XMLHttpRequest.prototype.send=function(f){for(var e=[],g=0;g<arguments.length;++g)e[g-0]=arguments[g];var E=this;a(d);this.addEventListener(\"readystatechange\",function(){4===E.readyState&&b(d)});return c.apply(this,e)}}\n",
      "    function x(a,b){var c=fetch;fetch=function(d){for(var f=[],e=0;e<arguments.length;++e)f[e-0]=arguments[e];return new Promise(function(d,e){var g=v++;a(g);c.apply(null,[].concat(u(f))).then(function(a){b(g);d(a)},function(a){b(a);e(a)})})}}var y=\"img script iframe link audio video source\".split(\" \");function z(a,b){a=t(a);for(var c=a.next();!c.done;c=a.next())if(c=c.value,b.includes(c.nodeName.toLowerCase())||z(c.children,b))return!0;return!1}\n",
      "    function A(a){var b=new MutationObserver(function(c){c=t(c);for(var b=c.next();!b.done;b=c.next())b=b.value,\"childList\"==b.type&&z(b.addedNodes,y)?a(b):\"attributes\"==b.type&&y.includes(b.target.tagName.toLowerCase())&&a(b)});b.observe(document,{attributes:!0,childList:!0,subtree:!0,attributeFilter:[\"href\",\"src\"]});return b}\n",
      "    function B(a,b){if(2<a.length)return performance.now();var c=[];b=t(b);for(var d=b.next();!d.done;d=b.next())d=d.value,c.push({timestamp:d.start,type:\"requestStart\"}),c.push({timestamp:d.end,type:\"requestEnd\"});b=t(a);for(d=b.next();!d.done;d=b.next())c.push({timestamp:d.value,type:\"requestStart\"});c.sort(function(a,b){return a.timestamp-b.timestamp});a=a.length;for(b=c.length-1;0<=b;b--)switch(d=c[b],d.type){case \"requestStart\":a--;break;case \"requestEnd\":a++;if(2<a)return d.timestamp;break;default:throw Error(\"Internal Error: This should never happen\");\n",
      "    }return 0}function C(a){a=a?a:{};this.w=!!a.useMutationObserver;this.u=a.minValue||null;a=window.__tti&&window.__tti.e;var b=window.__tti&&window.__tti.o;this.a=a?a.map(function(a){return{start:a.startTime,end:a.startTime+a.duration}}):[];b&&b.disconnect();this.b=[];this.f=new Map;this.j=null;this.v=-Infinity;this.i=!1;this.h=this.c=this.s=null;w(this.m.bind(this),this.l.bind(this));x(this.m.bind(this),this.l.bind(this));D(this);this.w&&(this.h=A(this.B.bind(this)))}\n",
      "    C.prototype.getFirstConsistentlyInteractive=function(){var a=this;return new Promise(function(b){a.s=b;\"complete\"==document.readyState?F(a):window.addEventListener(\"load\",function(){F(a)})})};function F(a){a.i=!0;var b=0<a.a.length?a.a[a.a.length-1].end:0,c=B(a.g,a.b);G(a,Math.max(c+5E3,b))}\n",
      "    function G(a,b){!a.i||a.v>b||(clearTimeout(a.j),a.j=setTimeout(function(){var b=performance.timing.navigationStart,d=B(a.g,a.b),b=(window.a&&window.a.A?1E3*window.a.A().C-b:0)||performance.timing.domContentLoadedEventEnd-b;if(a.u)var f=a.u;else performance.timing.domContentLoadedEventEnd?(f=performance.timing,f=f.domContentLoadedEventEnd-f.navigationStart):f=null;var e=performance.now();null===f&&G(a,Math.max(d+5E3,e+1E3));var g=a.a;5E3>e-d?d=null:(d=g.length?g[g.length-1].end:b,d=5E3>e-d?null:Math.max(d,\n",
      "        f));d&&(a.s(d),clearTimeout(a.j),a.i=!1,a.c&&a.c.disconnect(),a.h&&a.h.disconnect());G(a,performance.now()+1E3)},b-performance.now()),a.v=b)}\n",
      "    function D(a){a.c=new PerformanceObserver(function(b){b=t(b.getEntries());for(var c=b.next();!c.done;c=b.next())if(c=c.value,\"resource\"===c.entryType&&(a.b.push({start:c.fetchStart,end:c.responseEnd}),G(a,B(a.g,a.b)+5E3)),\"longtask\"===c.entryType){var d=c.startTime+c.duration;a.a.push({start:c.startTime,end:d});G(a,d+5E3)}});a.c.observe({entryTypes:[\"longtask\",\"resource\"]})}C.prototype.m=function(a){this.f.set(a,performance.now())};C.prototype.l=function(a){this.f.delete(a)};\n",
      "    C.prototype.B=function(){G(this,performance.now()+5E3)};h.Object.defineProperties(C.prototype,{g:{configurable:!0,enumerable:!0,get:function(){return[].concat(u(this.f.values()))}}});var H={getFirstConsistentlyInteractive:function(a){a=a?a:{};return\"PerformanceLongTaskTiming\"in window?(new C(a)).getFirstConsistentlyInteractive():Promise.resolve(null)}};\n",
      "    \"undefined\"!=typeof module&&module.exports?module.exports=H:\"function\"===typeof define&&define.amd?define(\"ttiPolyfill\",[],function(){return H}):window.ttiPolyfill=H;})();!function(t,e){\"use strict\";function n(t){this.time=t.time,this.target=t.target,this.rootBounds=t.rootBounds,this.boundingClientRect=t.boundingClientRect,this.intersectionRect=t.intersectionRect||a(),this.isIntersecting=!!t.intersectionRect;var e=this.boundingClientRect,n=e.width*e.height,o=this.intersectionRect,i=o.width*o.height;this.intersectionRatio=n?i/n:this.isIntersecting?1:0}function o(t,e){var n=e||{};if(\"function\"!=typeof t)throw new Error(\"callback must be a function\");if(n.root&&1!=n.root.nodeType)throw new Error(\"root must be an Element\");this._checkForIntersections=r(this._checkForIntersections.bind(this),this.THROTTLE_TIMEOUT),this._callback=t,this._observationTargets=[],this._queuedEntries=[],this._rootMarginValues=this._parseRootMargin(n.rootMargin),this.thresholds=this._initThresholds(n.threshold),this.root=n.root||null,this.rootMargin=this._rootMarginValues.map(function(t){return t.value+t.unit}).join(\" \")}function i(){return t.performance&&performance.now&&performance.now()}function r(t,e){var n=null;return function(){n||(n=setTimeout(function(){t(),n=null},e))}}function s(t,e,n,o){\"function\"==typeof t.addEventListener?t.addEventListener(e,n,o||!1):\"function\"==typeof t.attachEvent&&t.attachEvent(\"on\"+e,n)}function h(t,e,n,o){\"function\"==typeof t.removeEventListener?t.removeEventListener(e,n,o||!1):\"function\"==typeof t.detatchEvent&&t.detatchEvent(\"on\"+e,n)}function c(t,e){var n=Math.max(t.top,e.top),o=Math.min(t.bottom,e.bottom),i=Math.max(t.left,e.left),r=Math.min(t.right,e.right),s=r-i,h=o-n;return s>=0&&h>=0&&{top:n,bottom:o,left:i,right:r,width:s,height:h}}function u(t){var e;try{e=t.getBoundingClientRect()}catch(t){}return e?(e.width&&e.height||(e={top:e.top,right:e.right,bottom:e.bottom,left:e.left,width:e.right-e.left,height:e.bottom-e.top}),e):a()}function a(){return{top:0,bottom:0,left:0,right:0,width:0,height:0}}function l(t,e){for(var n=e;n;){if(n==t)return!0;n=p(n)}return!1}function p(t){var e=t.parentNode;return e&&11==e.nodeType&&e.host?e.host:e}if(\"IntersectionObserver\"in t&&\"IntersectionObserverEntry\"in t&&\"intersectionRatio\"in t.IntersectionObserverEntry.prototype)return void(\"isIntersecting\"in t.IntersectionObserverEntry.prototype||Object.defineProperty(t.IntersectionObserverEntry.prototype,\"isIntersecting\",{get:function(){return this.intersectionRatio>0}}));var f=[];o.prototype.THROTTLE_TIMEOUT=100,o.prototype.POLL_INTERVAL=null,o.prototype.USE_MUTATION_OBSERVER=!0,o.prototype.observe=function(t){if(!this._observationTargets.some(function(e){return e.element==t})){if(!t||1!=t.nodeType)throw new Error(\"target must be an Element\");this._registerInstance(),this._observationTargets.push({element:t,entry:null}),this._monitorIntersections(),this._checkForIntersections()}},o.prototype.unobserve=function(t){this._observationTargets=this._observationTargets.filter(function(e){return e.element!=t}),this._observationTargets.length||(this._unmonitorIntersections(),this._unregisterInstance())},o.prototype.disconnect=function(){this._observationTargets=[],this._unmonitorIntersections(),this._unregisterInstance()},o.prototype.takeRecords=function(){var t=this._queuedEntries.slice();return this._queuedEntries=[],t},o.prototype._initThresholds=function(t){var e=t||[0];return Array.isArray(e)||(e=[e]),e.sort().filter(function(t,e,n){if(\"number\"!=typeof t||isNaN(t)||t<0||t>1)throw new Error(\"threshold must be a number between 0 and 1 inclusively\");return t!==n[e-1]})},o.prototype._parseRootMargin=function(t){var e=t||\"0px\",n=e.split(/\\s+/).map(function(t){var e=/^(-?\\d*\\.?\\d+)(px|%)$/.exec(t);if(!e)throw new Error(\"rootMargin must be specified in pixels or percent\");return{value:parseFloat(e[1]),unit:e[2]}});return n[1]=n[1]||n[0],n[2]=n[2]||n[0],n[3]=n[3]||n[1],n},o.prototype._monitorIntersections=function(){this._monitoringIntersections||(this._monitoringIntersections=!0,this.POLL_INTERVAL?this._monitoringInterval=setInterval(this._checkForIntersections,this.POLL_INTERVAL):(s(t,\"resize\",this._checkForIntersections,!0),s(e,\"scroll\",this._checkForIntersections,!0),this.USE_MUTATION_OBSERVER&&\"MutationObserver\"in t&&(this._domObserver=new MutationObserver(this._checkForIntersections),this._domObserver.observe(e,{attributes:!0,childList:!0,characterData:!0,subtree:!0}))))},o.prototype._unmonitorIntersections=function(){this._monitoringIntersections&&(this._monitoringIntersections=!1,clearInterval(this._monitoringInterval),this._monitoringInterval=null,h(t,\"resize\",this._checkForIntersections,!0),h(e,\"scroll\",this._checkForIntersections,!0),this._domObserver&&(this._domObserver.disconnect(),this._domObserver=null))},o.prototype._checkForIntersections=function(){var t=this._rootIsInDom(),e=t?this._getRootRect():a();this._observationTargets.forEach(function(o){var r=o.element,s=u(r),h=this._rootContainsTarget(r),c=o.entry,a=t&&h&&this._computeTargetAndRootIntersection(r,e),l=o.entry=new n({time:i(),target:r,boundingClientRect:s,rootBounds:e,intersectionRect:a});c?t&&h?this._hasCrossedThreshold(c,l)&&this._queuedEntries.push(l):c&&c.isIntersecting&&this._queuedEntries.push(l):this._queuedEntries.push(l)},this),this._queuedEntries.length&&this._callback(this.takeRecords(),this)},o.prototype._computeTargetAndRootIntersection=function(n,o){if(\"none\"!=t.getComputedStyle(n).display){for(var i=u(n),r=i,s=p(n),h=!1;!h;){var a=null,l=1==s.nodeType?t.getComputedStyle(s):{};if(\"none\"==l.display)return;if(s==this.root||s==e?(h=!0,a=o):s!=e.body&&s!=e.documentElement&&\"visible\"!=l.overflow&&(a=u(s)),a&&!(r=c(a,r)))break;s=p(s)}return r}},o.prototype._getRootRect=function(){var t;if(this.root)t=u(this.root);else{var n=e.documentElement,o=e.body;t={top:0,left:0,right:n.clientWidth||o.clientWidth,width:n.clientWidth||o.clientWidth,bottom:n.clientHeight||o.clientHeight,height:n.clientHeight||o.clientHeight}}return this._expandRectByRootMargin(t)},o.prototype._expandRectByRootMargin=function(t){var e=this._rootMarginValues.map(function(e,n){return\"px\"==e.unit?e.value:e.value*(n%2?t.width:t.height)/100}),n={top:t.top-e[0],right:t.right+e[1],bottom:t.bottom+e[2],left:t.left-e[3]};return n.width=n.right-n.left,n.height=n.bottom-n.top,n},o.prototype._hasCrossedThreshold=function(t,e){var n=t&&t.isIntersecting?t.intersectionRatio||0:-1,o=e.isIntersecting?e.intersectionRatio||0:-1;if(n!==o)for(var i=0;i<this.thresholds.length;i++){var r=this.thresholds[i];if(r==n||r==o||r<n!=r<o)return!0}},o.prototype._rootIsInDom=function(){return!this.root||l(e,this.root)},o.prototype._rootContainsTarget=function(t){return l(this.root||e,t)},o.prototype._registerInstance=function(){f.indexOf(this)<0&&f.push(this)},o.prototype._unregisterInstance=function(){var t=f.indexOf(this);-1!=t&&f.splice(t,1)},t.IntersectionObserver=o,t.IntersectionObserverEntry=n}(window,document);function gtag(){dataLayer.push(arguments)}var logger=function(){\"use strict\";function n(n){if(n=JSON.stringify(n),\"sendBeacon\"in navigator)navigator.sendBeacon(t,n);else{var e=new XMLHttpRequest;e.open(\"POST\",t,!1),e.setRequestHeader(\"Content-Type\",\"text/plain;charset=UTF-8\"),e.send(n)}}function e(e,t,i){if(!RC.isProductionServer)switch(e){case\"error\":case\"warn\":case\"info\":i?console[e](t,i):console[e](t);break;default:console.log(t,i)}\"error\"!==e&&\"warn\"!==e||(r.push({level:e,msg:t||\"\",info:Object.assign({},i,{path:window.location.pathname,userAgent:navigator.userAgent})}),r.length>=o&&n(r.splice(0,o)))}var t=\"/analytics\",o=10,r=[];return window.addEventListener(\"unload\",function(){n(r)},!1),{stack:Function.prototype.bind.call(function(n){n&&e(\"error\",n.name+': \"'+n.message+'\"',{type:n.name,description:n.message,trace:n.stack})}),error:Function.prototype.bind.call(function(n,t){e(\"error\",n,t)}),warn:Function.prototype.bind.call(function(n,t){e(\"warn\",n,t)}),info:Function.prototype.bind.call(function(n,t){e(\"info\",n,t)}),verbose:Function.prototype.bind.call(function(n,t){e(\"verbose\",n,t)}),debug:Function.prototype.bind.call(function(n,t){e(\"debug\",n,t)}),silly:Function.prototype.bind.call(function(n,t){e(\"silly\",n,t)})}}();window.dataLayer=window.dataLayer||[],gtag(\"js\",new Date),gtag(\"config\",RC.googleAnalyticsTrackingId),gtag(\"config\",\"G-5JMGYPWJRR\"),window.addEventListener(\"error\",function(n){logger.stack(n.error)}),window.addEventListener(\"unhandledrejection\",function(n){n&&n.reason&&logger.error('Unhandled promise rejection: \"'+n.reason.message+'\"',{type:\"PromiseRejection\",description:n.reason.message,trace:n.reason.stack})});var gtagFirstMeaningfulPaint=function(){var n=!1;return function(){n?logger.warn('Timing for \"firstMeaningfulPaint\" already sent'):(console.log(\"firstMeaningfulPaint\",Math.round(performance.now())),n=!0)}}();!function(){\"use strict\";var n=0;if(\"PerformanceObserver\"in window&&\"PerformancePaintTiming\"in window){new PerformanceObserver(function(e){e.getEntries().forEach(function(e){if(0===n){e.startTime,e.duration}})}).observe({entryTypes:[\"paint\"]});var e=window.__tti={e:[]};e.o=new PerformanceObserver(function(n){e.e=e.e.concat(n.getEntries())}),e.o.observe({entryTypes:[\"longtask\"]}),ttiPolyfill.getFirstConsistentlyInteractive({}).then(function(n){});new PerformanceObserver(function(n){n.getEntries().forEach(function(n){var e=n.attribution[0]||{},t={containerType:e.containerType,containerName:e.containerName,containerId:e.containerId,containerSrc:e.containerSrc,frameOrigin:n.name,durationMs:n.duration,startTimeMs:n.startTime};n.duration>500&&logger.warn(\"Long running task, duration \"+n.duration.toFixed(2)+\"ms.\",t)})}).observe({entryTypes:[\"longtask\"]})}var t=performance.now();window.addEventListener(\"visibilitychange\",function(){document.hidden?t=performance.now():n+=performance.now()-t}),window.addEventListener(\"load\",function(){performance.now()}),window.addEventListener(\"DOMContentLoaded\",function(){performance.now()})}();var lazyImage=function(){var r=new IntersectionObserver(function(r){r.forEach(function(r){r.isIntersecting&&(r.target.src=r.target.getAttribute(\"data-src\"))})},{root:null,rootMargin:\"0px\",threshold:[0]});return{observe:function(e){r.observe(e)},observeArray:function(e){for(var t=0,n=e.length;t<n;++t)r.observe(e[t])}}}();</script><title>RCSB PDB - 1YCR: MDM2 BOUND TO THE TRANSACTIVATION DOMAIN OF P53</title><meta http-equiv=\"Content-Type\" content=\"text/html; charset=UTF-8\"><meta property=\"og:title\" content=\"RCSB PDB - 1YCR: MDM2 BOUND TO THE TRANSACTIVATION DOMAIN OF P53\"><meta property=\"og:description\" content=\"As a member of the wwPDB, the RCSB PDB curates and annotates PDB data according to agreed upon standards. The RCSB PDB also provides a variety of tools and resources. Users can perform simple and advanced searches based on annotations relating to sequence, structure and function. These molecules are visualized, downloaded, and analyzed by users who range from students to specialized scientists.\"><meta property=\"og:image\" content=\"https://cdn.rcsb.org/rcsb-pdb/v2/common/images/rcsb_logo.png\"><meta property=\"og:url\" content=\"\"><meta property=\"og:type\" content=\"\"><meta name=\"twitter:card\" content=\"summary_large_image\"><meta name=\"twitter:title\" content=\"RCSB PDB - 1YCR: MDM2 BOUND TO THE TRANSACTIVATION DOMAIN OF P53\"><meta name=\"twitter:description\" content=\"As a member of the wwPDB, the RCSB PDB curates and annotates PDB data according to agreed upon standards. The RCSB PDB also provides a variety of tools and resources. Users can perform simple and advanced searches based on annotations relating to sequence, structure and function. These molecules are visualized, downloaded, and analyzed by users who range from students to specialized scientists.\"><meta name=\"twitter:image\" content=\"https://cdn.rcsb.org/rcsb-pdb/v2/common/images/rcsb_logo.png\"><meta name=\"description\" content=\"As a member of the wwPDB, the RCSB PDB curates and annotates PDB data according to agreed upon standards. The RCSB PDB also provides a variety of tools and resources. Users can perform simple and advanced searches based on annotations relating to sequence, structure and function. These molecules are visualized, downloaded, and analyzed by users who range from students to specialized scientists.\"><meta name=\"author\" content=\"RCSB Protein Data Bank\"><meta name=\"email\" content=\"info@rcsb.org\"><meta name=\"Charset\" content=\"UTF-8\"><meta name=\"Distribution\" content=\"Global\"><meta name=\"Rating\" content=\"General\"><meta http-equiv=\"content-language\" content=\"en-US\"><meta name=\"viewport\" content=\"width=device-width, initial-scale=1\"><meta name=\"theme-color\" content=\"#5e94c3\"><link rel=\"manifest\" href=\"/manifest.json\"><link rel=\"apple-touch-icon\" href=\"/img/rcsb-apple-touch-icon.jpg\"><link rel=\"stylesheet\" href=\"//cdn.rcsb.org/javascript/bootstrap/latest/css/bootstrap.min.css\"><link rel=\"stylesheet\" href=\"//cdn.rcsb.org/javascript/fontawesome/latest/css/font-awesome.min.css\"><link rel=\"stylesheet\" href=\"/build/css/main.css\"><link rel=\"stylesheet\" href=\"/css/search.css?ts=5450659\"><script>window.addEventListener('DOMContentLoaded', function () {\n",
      "    lazyImage.observeArray(document.querySelectorAll('img[data-src]'));\n",
      "}, false);\n",
      "\n",
      "window.addEventListener('load', function () {\n",
      "    // load the JIRA feedback button only after everything else has been loaded\n",
      "    var script = document.createElement(\"script\");\n",
      "    script.src = \"/js/jira-fdbck.js\";\n",
      "    script.type = \"text/javascript\";\n",
      "    document.getElementsByTagName(\"head\")[0].appendChild(script);\n",
      "    var link = document.createElement(\"link\");\n",
      "    link.rel = \"stylesheet\";\n",
      "    link.href = \"//cdn.rcsb.org/jira-feedback/css/jira-fdbck.css\";\n",
      "    document.getElementsByTagName(\"head\")[0].appendChild(link);\n",
      "    // load the ekko-lightbox css only after everything else has been loaded\n",
      "    var link = document.createElement(\"link\");\n",
      "    link.rel = \"stylesheet\";\n",
      "    link.href = \"/css/ekko-lightbox.css\";\n",
      "    document.getElementsByTagName(\"head\")[0].appendChild(link);\n",
      "});\n",
      "</script><script src=\"//cdn.rcsb.org/javascript/jquery/jquery-3.3.1.min.js\"></script><script src=\"//cdn.rcsb.org/javascript/bootstrap/latest/js/bootstrap.min.js\"></script><script src=\"/js/ekko-lightbox.min.js\" defer></script><script>$(document).delegate('*[data-toggle=\"lightbox\"]',\"click\",function(t){t.preventDefault(),$(this).ekkoLightbox()});$(function(){$(\".nav>.dropdown\").on(\"mouseenter\",function(o){$(this).addClass(\"open\")}),$(\".nav>.dropdown\").on(\"mouseleave\",function(o){$(\".dropdown\").removeClass(\"open\")}),$('[data-toggle=\"tooltip\"]').tooltip()});</script><link rel=\"stylesheet\" href=\"/css/pages/staticpages.css\"><link rel=\"stylesheet\" href=\"/css/structure/structuresummarypage.css\"><link rel=\"stylesheet\" href=\"/css/bootstrap-slider.css\"><link rel=\"stylesheet\" href=\"/css/material-switch.css\"><script type=\"text/javascript\" src=\"/saguaro/app.js\"></script><style>.nav-tabs>li.active>a,.nav-tabs>li.active>a:focus,.nav-tabs>li.active>a:hover{color:#fff;cursor:default;background-color:#5e6973;border:1px solid #ddd;border-bottom-color:transparent}#structuretabs>li.active>a,#structuretabs>li.active>a:focus,#structuretabs>li.active>a:hover{background-color:#325880;border:1px solid transparent;border-bottom:none}#structuretabs{border-bottom:5px solid #325880}#structuretabs>li{margin-bottom:0}#structuretabs>li>a{background-color:#efeded;border-bottom:none}#navbar-collapse-SSP{padding:0;margin:0}.secondaryheader h1{float:left;margin-right:10px}.secondaryheader h4{margin-top:30px}#entitynext{margin-top:10px;background-color:#416d93}.panel-default>.panel-heading{color:#fff;background-color:#5e6973;border-color:#ddd}table .highlight td{background-color:#fff2e0}table td ul{-webkit-padding-start:15px}.noteSearchQuery{margin-bottom:0;padding:5px}.querySearchLink{font-weight:700;text-decoration:underline}table tr td button{margin:4px 1px}table thead .externalannotation>th{background-color:#f0ad4e;color:#fff;border-bottom-width:1px;font-weight:400}#structuretabs .dropdown .dropdown-menu{min-width:100%}@media screen and (max-width:767px){#collapsedHeader>div{padding-left:0;padding-right:0}#SSP-collapsed-tabs{background-color:#efeded;margin-top:0;border:1px solid #ccc}#navbar-collapse-SSP{width:100%;margin:0}#structuretabs{border:0}#structuretabs li{width:100%}#collapsedHeader h1{float:left;width:100%}#ProductPrimaryActions{float:left;width:100%}}</style><script>var structureId = \"1YCR\";\n",
      "console.log(\"CHECK: Summary Structure ID: \" + structureId);\n",
      "var proteinNumber = 0, nonProteinNumber = 0;\n",
      "</script><script>proteinNumber = 2 + proteinNumber,\n",
      "nonProteinNumber = 0 + nonProteinNumber;\n",
      "</script></head><body><nav class=\"navbar navbar-inverse navbar-fixed-top hidden-print\" id=\"nav\" role=\"navigation\"><div class=\"hide hidden-print\" id=\"UnsupportedBrowser\" style=\"width: 100%; border-bottom: 2px solid #FFA500; background-color: #FFFFFF; color: #000000; display: inline-block; font-size: .9em; line-height: 2em; text-align: center; font-weight: bold; font-style: italic;\"><span class=\"label label-warning\" style=\"font-size: 100%;\">Warning</span>        You are using a web browser that we do not support. Our website will not work properly. Please update to a newer version or download a new web browser, such as Chrome or Firefox.</div><div class=\"container\"><div class=\"pull-right\" id=\"mypdb-menu-container\"></div><!-- Brand and toggle get grouped for better mobile display--><div class=\"navbar-header\"><button class=\"navbar-toggle pull-left\" type=\"button\" data-toggle=\"collapse\" data-target=\"#navbar-collapse-RCSB\"><span class=\"sr-only\">Toggle navigation</span><span class=\"icon-bar\"></span><span class=\"icon-bar\"></span><span class=\"icon-bar\"></span></button><a class=\"navbar-brand\" href=\"/\">RCSB PDB</a></div><div class=\"collapse navbar-collapse\" id=\"navbar-collapse-RCSB\" style=\"max-height: 420px;\"><ul class=\"nav navbar-nav pull-left\"><li class=\"dropdown\" id=\"headnav_deposit\"><a class=\"dropdown-toggle\" href=\"#\" data-toggle=\"dropdown\" aria-expanded=\"false\">Deposit&nbsp;<b class=\"caret\"></b></a><div class=\"dropdown-menu multi-column\" id=\"DepositDropdownNav\"><div class=\"row\"><div class=\"col-xs-12 col-sm-2 col-md-2\"><ul class=\"dropdown-menu\" id=\"head-deposit_prepare\"><li><h4>Prepare Data</h4></li><li><a href=\"https://www.wwpdb.org/deposition/preparing-pdbx-mmcif-files\" target=\"_blank\" rel=\"noopener\">PDBx/mmCIF file    </a></li><li><a href=\"http://pdb-extract.wwpdb.org/\" target=\"_blank\" rel=\"noopener\">pdb_extract</a></li><li><a href=\"http://sf-tool.wwpdb.org/\" target=\"_blank\" rel=\"noopener\">SF-Tool</a></li><li><a href=\"http://ligand-expo.rcsb.org/\" target=\"_blank\" rel=\"noopener\">Ligand Expo</a></li><li><a href=\"http://sw-tools.rcsb.org/apps/MAXIT/index.html\" target=\"_blank\" rel=\"noopener\">MAXIT</a></li></ul></div><div class=\"col-xs-12 col-sm-3 col-md-3\"><ul class=\"dropdown-menu\" id=\"head-deposit_validate\"><li><h4>Validate Data</h4></li><li><a href=\"https://validate-rcsb.wwpdb.org\" target=\"_blank\" rel=\"noopener\">Validation Server</a></li><li><a href=\"http://www.wwpdb.org/documentation/journals.php\" target=\"_blank\" rel=\"noopener\">Information for Journals</a></li><li><a href=\"http://www.wwpdb.org/task/validation-task-forces.php\" target=\"_blank\" rel=\"noopener\">Validation Task Forces</a></li></ul></div><div class=\"col-xs-12 col-sm-3 col-md-2\"><ul class=\"dropdown-menu\" id=\"head-deposit_deposit-data\"><li><h4>Deposit Data</h4></li><li><a href=\"http://deposit.wwpdb.org/\" target=\"_blank\" rel=\"noopener\">wwPDB OneDep System</a></li><li><a href=\"https://pdb-dev.wwpdb.org/\" target=\"_blank\" rel=\"noopener\">PDB-Dev</a></li></ul></div><div class=\"col-xs-12 col-sm-2 col-md-3\"><ul class=\"dropdown-menu\" id=\"head-deposit_deposit-help\"><li><h4>Deposition Help & Resources</h4></li><li><a href=\"http://www.wwpdb.org/deposition/faq.php\" target=\"_blank\" rel=\"noopener\">Deposit FAQ</a></li><li><a href=\"http://www.wwpdb.org/deposition/tutorial.php\" target=\"_blank\" rel=\"noopener\">Tutorials</a></li><li><a href=\"http://www.wwpdb.org/documentation/policy\" target=\"_blank\" rel=\"noopener\">Annotation Policies</a></li><li><a href=\"http://www.wwpdb.org/documentation/procedure\" target=\"_blank\" rel=\"noopener\">Processing Procedures</a></li><li class=\"divider hidden-xs\"></li><li><a href=\"http://mmcif.wwpdb.org/\" target=\"_blank\" rel=\"noopener\">PDBx/mmCIF Dictionary</a></li><li><a href=\"http://wwpdb.org/data/ccd\" target=\"_blank\" rel=\"noopener\">Chemical Component Dictionary</a></li><li><a href=\"http://www.wwpdb.org/documentation/file-format.php\" target=\"_blank\" rel=\"noopener\">PDB Format Guide</a></li><li><a href=\"http://biosync.sbkb.org/\" target=\"_blank\" rel=\"noopener\">BioSync Beamlines/Facilities</a></li><li><a href=\"http://sw-tools.rcsb.org/\" target=\"_blank\" rel=\"noopener\">Related Tools</a></li></ul></div></div></div></li><li class=\"dropdown\" id=\"headnav_search\"><a class=\"dropdown-toggle\" href=\"#\" data-toggle=\"dropdown\">Search&nbsp;<b class=\"caret\"></b></a><ul class=\"dropdown-menu\"><li><a href=\"/pages/search_features\">Basic Search</a></li><li><a href=\"/pages/search_features#search_advanced-search\">Advanced Search</a></li><li><a href=\"/pages/search_features#search_sequences\">Sequence Search</a></li><li><a href=\"/chemical-sketch\">Chemical Sketch Tool</a></li><li><a href=\"/pages/search_features#search_unreleased-and-new\">Unreleased & New Entries</a></li><li><a href=\"/pages/search_features#search_annotation\">Browse Annotations</a></li><li><a href=\"/pages/search_features#search_drilldown\">Explore the Archive</a></li><li><a href=\"/pages/search_features#search_statistics\">PDB Statistics</a></li><li><a href=\"/pages/search_features#search_ligands\">Ligands</a></li><li><a href=\"/pages/search_features#search_drugs\">Drugs & Drug Targets</a></li></ul></li><li class=\"dropdown\" id=\"headnav_visualize\"><a class=\"dropdown-toggle\" href=\"#\" data-toggle=\"dropdown\">Visualize&nbsp;<b class=\"caret\"></b></a><ul class=\"dropdown-menu\"><li><a href=\"/pages/visualize_features#visualize_3dviewers\"> Mol* 3D Viewer</a></li><li><a href=\"/pages/visualize_features#visualize_seq-proteinfeatures\">Protein Feature View</a></li><li><a href=\"/pages/visualize_features#visualize_seq-genome\">Genome View</a></li><!-- pathway feature will be disabled until reimplemented with enhanced capabilities - JD 2020-10-30--><!--lia(href='/pages/visualize_features#visualize_pathway') Pathway View--></ul></li><li class=\"dropdown\" id=\"headnav_analyze\"><a class=\"dropdown-toggle\" href=\"#\" data-toggle=\"dropdown\">Analyze&nbsp;<b class=\"caret\"></b></a><ul class=\"dropdown-menu\"><li><a href=\"/pages/analyze_features#Sequence\">Pairwise Structure Alignment</a></li><li><a href=\"/pages/analyze_features#Symmetry\">Protein Symmetry</a></li><li><a href=\"/pages/analyze_features#Quality\">Structure Quality</a></li><li><a href=\"/pages/analyze_features#Genomic\">Map Genomic Position to Protein</a></li><li><a href=\"/stats\">PDB Statistics</a></li><li><a href=\"/pages/analyze_features#Assemblies\">EPPIC Biological Assemblies</a></li><li><a href=\"/pages/analyze_features#CitationMesh\">PDB Citation MeSH Network Explorer </a></li><li class=\"divider hidden-xs\"></li><li><a href=\"/docs/general-help/data-from-external-resources-integrated-into-rcsb-pdb\">Integrated Resources</a></li><li><a href=\"/docs/additional-resources/databases\">Additional Resources</a></li></ul></li><li class=\"dropdown hidden-xs hidden-sm\" id=\"headnav_download\"><a class=\"dropdown-toggle\" href=\"#\" data-toggle=\"dropdown\">Download&nbsp;<b class=\"caret\"></b></a><ul class=\"dropdown-menu\"><li><a href=\"/downloads\">Coordinates & Experimental Data</a></li><li><a href=\"/downloads/fasta\">Sequences</a></li><li><a href=\"/downloads/ligands\">Ligands</a></li><li><a href=\"/docs/programmatic-access/file-download-services\">File Download Services</a></li><li><a href=\"/docs/programmatic-access/web-services-overview\">Web Services</a></li></ul></li><li class=\"dropdown\" id=\"headnav_learn\"><a class=\"dropdown-toggle\" href=\"#\" data-toggle=\"dropdown\">Learn&nbsp;<b class=\"caret\"></b></a><ul class=\"dropdown-menu\"><li class=\"hidden-xs\"><h4><a href=\"http://pdb101.rcsb.org\" target=\"_blank\" rel=\"noopener\"><img class=\"image_submenu\" src=\"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAQAAAC1HAwCAAAAC0lEQVR42mNkYAAAAAYAAjCB0C8AAAAASUVORK5CYII=\" data-src=\"https://cdn.rcsb.org/rcsb-pdb/v2/common/images/pdb101Logo.png\"></a></h4></li><li><a href=\"http://pdb101.rcsb.org/more/about-pdb-101\" target=\"_blank\" rel=\"noopener\">About</a></li><li><a href=\"http://pdb101.rcsb.org/motm/motm-by-title\" target=\"_blank\" rel=\"noopener\">Molecule of the Month</a></li><li><a href=\"http://pdb101.rcsb.org/browse\" target=\"_blank\" rel=\"noopener\">Browse all Resources</a></li><li><a href=\"http://pdb101.rcsb.org/learn/paper-models\" target=\"_blank\" rel=\"noopener\">Educational Resources</a></li><li><a href=\"http://pdb101.rcsb.org/teach/overview\" target=\"_blank\" rel=\"noopener\">Curricula</a></li><li><a href=\"http://pdb101.rcsb.org/learn/guide-to-understanding-pdb-data\" target=\"_blank\" rel=\"noopener\">Guide to PDB Data</a></li><li><a href=\"http://pdb101.rcsb.org/news\" target=\"_blank\" rel=\"noopener\">News and Events</a></li></ul></li><li class=\"dropdown\" id=\"headnav_more\"><a class=\"dropdown-toggle\" href=\"#\" data-toggle=\"dropdown\">More&nbsp;<b class=\"caret\"></b></a><ul class=\"dropdown-menu\"><li><a href=\"/pages/contactus\">Contact Us</a></li><li><a href=\"/pages/policies#References\">Citing Us</a></li><li><a href=\"/pages/policies\">Policies</a></li><li><a href=\"/pages/about-us/index\">About RCSB PDB</a></li><li><a href=\"/pages/about-us/history\">PDB History</a></li><li><a href=\"/pages/pdb50\">PDB50  </a></li><li><a href=\"/pages/team\">Team Members</a></li><li><a href=\"/pages/advisory-committee\">Advisory Committee</a></li><li><a href=\"/pages/news\">News</a></li><li><a href=\"/pages/publications\">Publications</a></li><li><a href=\"https://status.rcsb.org/\" target=\"_blank\">Service Status</a></li></ul></li><li class=\"dropdown\" id=\"headnav_documentation\"><a class=\"dropdown-toggle\" href=\"#\" data-toggle=\"dropdown\">Documentation&nbsp;<b class=\"caret\"></b></a><ul class=\"dropdown-menu\"><li><a href=\"/docs\">General Help</a></li><li class=\"divider\"></li><li><a href=\"/docs/search-and-browse/overview-search-and-browse\">Search and Browse</a></li><li><a href=\"/docs/search-and-browse/basic-search\">Basic Search</a></li><li><a href=\"/docs/search-and-browse/advanced-search/overview-search\">Advanced Search</a></li><li class=\"divider\"></li><li><a href=\"/docs/3d-viewers/mol*/getting-started\">Mol*</a></li><li><a href=\"/docs/sequence-viewers/genome-view\">Sequence Viewers</a></li><li><a href=\"/docs/tools/pairwise-structure-alignment\">Tools</a></li><li><a href=\"/docs/programmatic-access/file-download-services\">Programmatic Access</a></li><li><a href=\"/docs/additional-resources/databases\">Additional Resources</a></li><li><a href=\"/docs/general-help/software-supporters\">Software Supporters</a></li><li class=\"divider\"></li><li><a href=\"/docs/general-help/deposition-resources\">Deposition Resources</a></li><li><a href=\"/docs/general-help/website-faq\">FAQs</a></li><li><a href=\"/docs/general-help/glossary\">Glossary </a></li></ul></li><li><a class=\"dropdown-toggle\" href=\"/pages/jobs\">Careers</a></li><li class=\"dropdown visible-xs\"><div id=\"socialmedia_section_mobile\"><ul class=\"fa-ul hidden-print\"><li><a href=\"https://www.facebook.com/RCSBPDB\" target=\"_blank\" rel=\"noopener\" title=\"Facebook\" data-placement=\"top\" data-toggle=\"tooltip\"><i class=\"fa fa-facebook-square fa-lg\"></i></a></li><li><a href=\"https://twitter.com/buildmodels\" target=\"_blank\" rel=\"noopener\" title=\"Twitter\" data-placement=\"top\" data-toggle=\"tooltip\"><i class=\"fa fa-twitter fa-lg\"></i></a></li><li><a href=\"https://www.youtube.com/user/RCSBProteinDataBank\" target=\"_blank\" rel=\"noopener\" title=\"YouTube\" data-placement=\"top\" data-toggle=\"tooltip\"><i class=\"fa fa-youtube-play fa-lg\"></i></a></li><li><a href=\"https://github.com/rcsb\" target=\"_blank\" rel=\"noopener\" title=\"Github\" data-placement=\"top\" data-toggle=\"tooltip\"><i class=\"fa fa-github fa-lg\"></i></a></li></ul></div></li></ul></div></div></nav><style>.pdb50 {\n",
      "    padding: 0 1px 0 0;\n",
      "    width: 300px; \n",
      "    display: inline-block;\n",
      "    bottom: -30px;\n",
      "    position: absolute;\n",
      "    right: 102px;\n",
      "    margin-left: 5px;\n",
      "    border-top-left-radius: 5px;\n",
      "    border-top-right-radius: 5px;\n",
      "}\n",
      "\n",
      ".pdb50:hover {\n",
      "   opacity: 0.8 !important;\n",
      "   filter: alpha(opacity=70) !important;\n",
      "   box-shadow: 0 0 0px #000000 !important;\n",
      "}\n",
      ".beta-test {\n",
      "    background: #325880;\n",
      "    color: #FE9A2E;\n",
      "    padding: 3px 15px;\n",
      "    height: 26px;\n",
      "    border-top-left-radius: 5px;\n",
      "    border-top-right-radius: 5px;\n",
      "    position: absolute;\n",
      "    bottom: -32px;\n",
      "    right: 110px;\n",
      "    list-style-type: none;\n",
      "}\n",
      ".instance {\n",
      "    background: #325880;\n",
      "    color: #cef;\n",
      "    padding: 3px 15px;\n",
      "    height: 26px;\n",
      "    border-top-left-radius: 5px;\n",
      "    border-top-right-radius: 5px;\n",
      "    position: absolute;\n",
      "    bottom: -32px;\n",
      "    right: 110px;\n",
      "    list-style-type: none;\n",
      "    font-size: 12px;\n",
      "    line-height: 22px;\n",
      "}\n",
      "</style><div id=\"header\"><div class=\"container\"><div class=\"row\" style=\"margin-bottom: 20px;\"><div class=\"col-sm-12 col-md-6 col-lg-5\" id=\"logo_container\"><a href=\"/\"><img id=\"rcsblogo\" src=\"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAQAAAC1HAwCAAAAC0lEQVR42mNkYAAAAAYAAjCB0C8AAAAASUVORK5CYII=\" data-src=\"https://cdn.rcsb.org/rcsb-pdb/v2/common/images/rcsb_logo.png\" alt=\"RCSB PDB\"></a><h2 class=\"hidden-xs\" id=\"tagline\"><a id=\"CurrentStructureCount\" href=\"/search?query=%7B%22type%22%3A%22terminal%22%2C%22service%22%3A%22text%22%2C%22parameters%22%3A%7B%22attribute%22%3A%22rcsb_entry_container_identifiers.entry_id%22%2C%22operator%22%3A%22exists%22%7D%7D\"> 183386 </a>Biological Macromolecular Structures Enabling Breakthroughs in Research and Education</h2><div class=\"clearfix\"></div></div><!-- search bar--><div class=\"col-xs-12 col-sm-12 col-md-6 col-lg-7 hidden-print\" id=\"search-bar-container\"></div></div><ul class=\"hidden-xs hidden-print\" id=\"PDBmembers\"><li><a href=\"https://pdb101.rcsb.org/\"><img id=\"pdb101logo\" src=\"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAQAAAC1HAwCAAAAC0lEQVR42mNkYAAAAAYAAjCB0C8AAAAASUVORK5CYII=\" data-src=\"https://cdn.rcsb.org/rcsb-pdb/v2/common/images/pdb101Logo.png\" alt=\"PDB-101\"></a></li><li><a href=\"https://www.wwpdb.org/\" target=\"_blank\" rel=\"noopener\"><img id=\"wwpdblogo\" src=\"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAQAAAC1HAwCAAAAC0lEQVR42mNkYAAAAAYAAjCB0C8AAAAASUVORK5CYII=\" data-src=\"https://cdn.rcsb.org/rcsb-pdb/v2/common/images/Logo_wwpdb.png\" alt=\"wwPDB\"></a></li><li><a href=\"https://www.emdataresource.org\" target=\"_blank\" rel=\"noopener\"><img id=\"emdatabanklogo\" src=\"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAQAAAC1HAwCAAAAC0lEQVR42mNkYAAAAAYAAjCB0C8AAAAASUVORK5CYII=\" data-src=\"https://cdn.rcsb.org/rcsb-pdb/v2/common/images/EMDataResourcelogo.png\" alt=\"EMDataResource\"></a></li><li><a href=\"http://ndbserver.rutgers.edu/\" target=\"_blank\" rel=\"noopener\"><img id=\"ndblogo\" src=\"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAQAAAC1HAwCAAAAC0lEQVR42mNkYAAAAAYAAjCB0C8AAAAASUVORK5CYII=\" data-src=\"https://cdn.rcsb.org/rcsb-pdb/v2/common/images/Logo_NDB.png\" alt=\"NDB Nucleic Acid Database\"></a></li><li><a href=\"https://foundation.wwpdb.org/\" target=\"_blank\" rel=\"noopener\"><img id=\"wwpdbfoundationlogo\" src=\"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAQAAAC1HAwCAAAAC0lEQVR42mNkYAAAAAYAAjCB0C8AAAAASUVORK5CYII=\" data-src=\"https://cdn.rcsb.org/rcsb-pdb/v2/common/images/Logo_wwPDBFoundation.png\" alt=\"wwPDB Foundation\"></a></li></ul><ul class=\"fa-ul pdb50\"><li><a href=\"/pdb50\" rel=\"noopener\" alt=\"Celebrating 50 Years of PDB Logo\"><img src=\"https://cdn.rcsb.org/rcsb-pdb/v2/home/images/pdb50-branding/logo50-gold-long.svg\"></a></li></ul><div id=\"socialmedia_section\"><ul class=\"fa-ul hidden-print\"><li><a href=\"https://www.facebook.com/RCSBPDB\" target=\"_blank\" rel=\"noopener\" title=\"Facebook\" data-placement=\"top\" data-toggle=\"tooltip\"><i class=\"fa fa-facebook-square fa-lg\"></i></a></li><li><a href=\"https://twitter.com/buildmodels\" target=\"_blank\" rel=\"noopener\" title=\"Twitter\" data-placement=\"top\" data-toggle=\"tooltip\"><i class=\"fa fa-twitter fa-lg\"></i></a></li><li><a href=\"https://www.youtube.com/user/RCSBProteinDataBank\" target=\"_blank\" rel=\"noopener\" title=\"YouTube\" data-placement=\"top\" data-toggle=\"tooltip\"><i class=\"fa fa-youtube-play fa-lg\"></i></a></li><li><a href=\"https://github.com/rcsb\" target=\"_blank\" rel=\"noopener\" title=\"Github\" data-placement=\"top\" data-toggle=\"tooltip\"><i class=\"fa fa-github fa-lg\"></i></a></li></ul></div></div></div><div class=\"container\" id=\"maincontentcontainer\"><button class=\"navbar-toggle pull-left\" id=\"SSP-collapsed-tabs\" type=\"button\" data-toggle=\"collapse\" data-target=\"#navbar-collapse-SSP\"><span class=\"glyphicon glyphicon-menu-hamburger\" aria-hidden=\"true\"></span> Navigation Tabs</button><div class=\"collapse navbar-collapse\" id=\"navbar-collapse-SSP\"><ul class=\"nav nav-tabs hidden-print\" id=\"structuretabs\" role=\"tablist\"><li id=\"structuresummarytab\" role=\"presentation\"><a href=\"/structure/1YCR\">Structure Summary</a></li><li id=\"3dviewtab\" role=\"presentation\"><a href=\"/3d-view/1YCR\">3D View</a></li><li id=\"annotationstab\" role=\"presentation\"><a href=\"/annotations/1YCR\">Annotations</a></li><li id=\"MoreOptions-Method\" role=\"presentation\"><a href=\"/experimental/1YCR\">Experiment</a></li><li id=\"sequencetab\" role=\"presentation\"><a href=\"/sequence/1YCR\">Sequence</a></li><li id=\"genometab\" role=\"presentation\"><a href=\"/genome/1YCR\">Genome</a></li><li id=\"versionstab\" role=\"presentation\"><a href=\"/versions/1YCR\">Versions</a></li><!--+tab( 'Sequence', '/pdb/explore/remediatedSequence.do?structureId=' + entryId, 'Sequence' )--><!--+tab( 'MoreOptions-Method', '/experimental/' + entryId, 'Experiment' )--></ul></div><br><div class=\"tab-content\"><div class=\"tab-pane active\" id=\"summary\" role=\"tabpanel\"><div class=\"row\"><div class=\"col-md-4 col-sm-12 col-xs-12\"><script>// Used for Transmembrane Image Carousel\n",
      "var structureHasEdMaps =  false;\n",
      "var structureFirstLigand = false;\n",
      "var finalImageCount = 2;\n",
      "var isNMR = false;\n",
      "var modelCount = 1\n",
      "</script><div class=\"panel panel-default\" id=\"galleryimagepanel\"><div class=\"panel-body\" id=\"structureimagesection\"><div class=\"carousel slide\" id=\"carousel-structuregallery\" data-ride=\"carousel\" data-interval=\"false\"><div class=\"carousel-inner\" role=\"listbox\"><div class=\"item imageCarouselItem\" id=\"Carousel-BiologicalUnit0\"><div class=\"carousel-header\">Asymmetric Unit</div><div class=\"clearfix\"></div><img class=\"img-responsive center-block mainImage\" src=\"data:image/gif;base64,R0lGODlhAQABAAD/ACwAAAAAAQABAAACADs=\"><a class=\"btn btn-default btn-xs btn-enlargeImage\" type=\"button\" data-toggle=\"lightbox\" data-gallery=\"multiimages0\" href=\"#\" data-title=\"\"><span class=\"glyphicon glyphicon-resize-full\" aria-hidden=\"true\"></span></a><div class=\"galleryNewImages\" id=\"AssemblyNewImage0\"></div><div class=\"clearfix\"></div><div class=\"carousel-footer\"><p><i class=\"fa fa-cube\">&nbsp</i><strong>3D View</strong>:&nbsp<a href=\"/3d-view/1YCR/0\">Structure</a></p></div></div><div class=\"item imageCarouselItem\" id=\"Carousel-BiologicalUnit1\"><div class=\"carousel-header\">Biological Assembly 1 &nbsp<a class=\"hidden-xs hidden-sm hidden-md icon-link\" href=\"http://pdb101.rcsb.org/learn/guide-to-understanding-pdb-data/biological-assemblies\" target=\"_blank\" rel=\"noopener\"><span class=\"glyphicon glyphicon-question-sign\" aria-hidden=\"true\"></span></a></div><div class=\"clearfix\"></div><img class=\"img-responsive center-block mainImage\" src=\"data:image/gif;base64,R0lGODlhAQABAAD/ACwAAAAAAQABAAACADs=\"><a class=\"btn btn-default btn-xs btn-enlargeImage\" type=\"button\" data-toggle=\"lightbox\" data-gallery=\"multiimages1\" href=\"#\" data-title=\"\"><span class=\"glyphicon glyphicon-resize-full\" aria-hidden=\"true\"></span></a><div class=\"galleryNewImages\" id=\"AssemblyNewImage1\"></div><div class=\"clearfix\"></div><div class=\"carousel-footer\"><p><i class=\"fa fa-cube\">&nbsp</i><strong>3D View</strong>:&nbsp<a href=\"/3d-view/1YCR/1\">Structure</a></p><hr><strong>Global Symmetry</strong>: Asymmetric - C1&nbsp<span class=\"glyphicon glyphicon-info-sign hidden-xs hidden-sm hidden-md\" aria-hidden=\"true\" data-toggle=\"tooltip\" data-placement=\"top\" data-original-title=\"Point group symmetry for the biological assembly. Chains are considered equivalent when their sequence identities are above 95%.\"></span><br><strong>Global Stoichiometry</strong>: Hetero 2-mer -&nbsp<span style=\"word-wrap:break-word;\">A1B1&nbsp</span><span class=\"glyphicon glyphicon-info-sign hidden-xs hidden-sm hidden-md\" aria-hidden=\"true\" data-toggle=\"tooltip\" data-placement=\"top\" data-original-title=\"Chains with length below 5 residues or relatively short with respect to other chains are excluded from the stoichiometry and symmetry calculation.\"></span><hr><a href=\"/search?query=%7B%22type%22%3A%22terminal%22%2C%22service%22%3A%22structure%22%2C%22parameters%22%3A%7B%22value%22%3A%7B%22entry_id%22%3A%221YCR%22%2C%22assembly_id%22%3A%221%22%7D%7D%7D&amp;return_type=assembly\">Find Similar Assemblies</a><hr><p>Biological assembly 1&nbspassigned by authors and generated by PISA (software)</p></div></div><a class=\"left carousel-control\" href=\"#carousel-structuregallery\" role=\"button\" data-slide=\"prev\"><span class=\"glyphicon glyphicon-chevron-left\" aria-hidden=\"true\"></span><span class=\"sr-only\">Previous</span></a><a class=\"right carousel-control\" href=\"#carousel-structuregallery\" role=\"button\" data-slide=\"next\"><span class=\"glyphicon glyphicon-chevron-right\" aria-hidden=\"true\"></span><span class=\"sr-only\">Next</span></a></div></div></div></div><!-- Show up only when there is a mix of structure type--><div class=\"well well-sm hidden-sm hidden-xs well-nomar\" id=\"macromoleculeContent\"><p><strong>Macromolecule Content</strong></p><ul><li id=\"contentStructureWeight\">Total Structure Weight: 14.34 kDa&nbsp<span class=\"glyphicon glyphicon-info-sign hidden-xs hidden-sm hidden-md\" aria-hidden=\"true\" data-toggle=\"tooltip\" data-placement=\"top\" data-original-title=\"Molecular weight (in kDa) of all non-water atoms in the deposited model, based on the full deposited sample sequence. Hydrogen atoms are included for the charged state in ARG, HIS &amp; LYS residues.\"></span></li><li id=\"contentAtomSiteCount\">Atom Count: 818&nbsp<span class=\"glyphicon glyphicon-info-sign hidden-xs hidden-sm hidden-md\" aria-hidden=\"true\" data-toggle=\"tooltip\" data-placement=\"top\" data-original-title=\"Number of modeled non-hydrogen atoms in the deposited model.\"></span></li><li id=\"contentResidueCount\">Modelled Residue Count: 98&nbsp<span class=\"glyphicon glyphicon-info-sign hidden-xs hidden-sm hidden-md\" aria-hidden=\"true\" data-toggle=\"tooltip\" data-placement=\"top\" data-original-title=\"Number of modeled polymer monomers in the deposited model.\"></span></li><li id=\"contentResidueCount\">Deposited Residue Count: 124&nbsp<span class=\"glyphicon glyphicon-info-sign hidden-xs hidden-sm hidden-md\" aria-hidden=\"true\" data-toggle=\"tooltip\" data-placement=\"top\" data-original-title=\"Number of all (modeled or unmodeled) polymer monomers in the deposited model.\"></span></li><li id=\"contentProteinChainCount\">Unique protein chains: 2</li></ul></div></div><div class=\"col-md-8 col-sm-12 col-xs-12\"><div class=\"pull-right btn-group\" id=\"ProductPrimaryActions\" role=\"group\"><div class=\"btn-group\" id=\"DisplayFilesButton\" role=\"group\"><button class=\"btn btn-sm btn-default dropdown-toggle\" id=\"dropdownMenuDisplayFiles\" type=\"button\" data-toggle=\"dropdown\" aria-expanded=\"false\"><span class=\"fa fa-file\" aria-hidden=\"true\"></span>&nbsp;Display Files&nbsp;<span class=\"caret\"></span></button><ul class=\"dropdown-menu pull-right\" role=\"menu\" aria-labelledby=\"dropdownMenuDisplayFiles\"><li><a target=\"_blank\" rel=\"noopener\" href=\"/fasta/entry/1YCR/display\">FASTA Sequence</a></li><li class=\"divider\"></li><li><a target=\"_blank\" rel=\"noopener\" href=\"//files.rcsb.org/view/1YCR.pdb\">PDB Format</a></li><li><a target=\"_blank\" rel=\"noopener\" href=\"//files.rcsb.org/header/1YCR.pdb\">PDB Format (Header)</a></li><li class=\"divider\"></li><li><a target=\"_blank\" rel=\"noopener\" href=\"//files.rcsb.org/view/1YCR.cif\">mmCIF Format</a></li><li><a target=\"_blank\" rel=\"noopener\" href=\"//files.rcsb.org/header/1YCR.cif\">mmCIF Format (Header)</a></li></ul></div><div class=\"btn-group\" id=\"DownloadFilesButton\" role=\"group\"><button class=\"btn btn-sm btn-primary dropdown-toggle\" id=\"dropdownMenuDownloadFiles\" type=\"button\" data-toggle=\"dropdown\" aria-expanded=\"false\"><span class=\"fa fa-arrow-circle-o-down\" aria-hidden=\"true\" style=\"font-size: 14px;\"></span>&nbsp;Download Files&nbsp;<span class=\"caret\"></span></button><ul class=\"dropdown-menu pull-right\" role=\"menu\" aria-labelledby=\"dropdownMenuDownloadFiles\"><li><a href=\"/fasta/entry/1YCR\">FASTA Sequence</a></li><li class=\"divider\"></li><li><a href=\"//files.rcsb.org/download/1YCR.pdb\">PDB Format</a></li><li><a href=\"//files.rcsb.org/download/1YCR.pdb.gz\">PDB Format (gz)</a></li><li class=\"divider\"></li><li><a href=\"//files.rcsb.org/download/1YCR.cif\">PDBx/mmCIF Format</a></li><li><a href=\"//files.rcsb.org/download/1YCR.cif.gz\">PDBx/mmCIF Format (gz)</a></li><li class=\"divider\"></li><li><a href=\"//files.rcsb.org/download/1YCR.xml.gz\">PDBML/XML Format (gz)</a></li><li class=\"divider\"></li><li><a href=\"//files.rcsb.org/download/1YCR.pdb1.gz\">Biological Assembly 1</a></li><li class=\"divider\"></li><li><a target=\"_blank\" rel=\"noopener\" href=\"//files.rcsb.org/pub/pdb/validation_reports/yc/1ycr/1ycr_full_validation.pdf\">Validation Full PDF</a></li><li><a target=\"_blank\" rel=\"noopener\" href=\"//files.rcsb.org/pub/pdb/validation_reports/yc/1ycr/1ycr_validation.xml.gz\">Validation XML   </a></li></ul></div></div><h1><span id=\"structureID\">1YCR</span></h1><h4><span id=\"structureTitle\">MDM2 BOUND TO THE TRANSACTIVATION DOMAIN OF P53</span></h4><ul class=\"list-inline\"><li id=\"header_doi\"><strong>DOI:&nbsp</strong><a href=\"http://doi.org/10.2210/pdb1YCR/pdb\" target=\"_blank\" rel=\"noopener\">10.2210/pdb1YCR/pdb</a></li></ul><ul class=\"list-unstyled\"><li id=\"header_classification\"><strong>Classification:&nbsp<a href=\"/search?q=struct_keywords.pdbx_keywords:COMPLEX (ONCOGENE PROTEIN/PEPTIDE)\">COMPLEX (ONCOGENE PROTEIN/PEPTIDE)</a></strong></li><li id=\"header_organism\"><strong>Organism(s):&nbsp</strong><a href=\"/search?q=rcsb_entity_source_organism.taxonomy_lineage.name:Homo sapiens\">Homo sapiens</a></li><li id=\"header_expression-system\"><strong>Expression System:&nbsp</strong><a href=\"/search?q=rcsb_entity_host_organism.ncbi_scientific_name:Escherichia coli\">Escherichia coli</a></li><!-- sum of mutations of all entities            --><li id=\"header_mutation\"><strong>Mutation(s):&nbsp</strong>No&nbsp <span class=\"glyphicon glyphicon-info-sign hidden-xs hidden-sm hidden-md\" aria-hidden=\"true\" data-toggle=\"tooltip\" data-placement=\"right\" data-original-title=\"Indicates sequence mutations.\">          </span></li><br><li id=\"header_deposited-released-dates\"><strong>Deposited:&nbsp</strong>1996-09-30&nbsp<strong>Released:&nbsp</strong>1997-11-19&nbsp</li><li id=\"header_deposition-authors\"><strong>Deposition Author(s):&nbsp</strong><a href=\"/search?q=audit_author.name:Kussie, P.H.\">Kussie, P.H.</a>, <a href=\"/search?q=audit_author.name:Pavletich, N.P.\">Pavletich, N.P.</a></li></ul><hr><div class=\"row\"><div class=\"col-sm-5 col-xs-12\" id=\"header_experimentalDataSnapshot\"><p><strong>Experimental Data Snapshot</strong></p><ul class=\"list-unstyled\" id=\"exp_header_0_snapshot\"><li id=\"exp_header_0_method\"><strong>Method:&nbsp</strong>X-RAY DIFFRACTION</li><li id=\"exp_header_0_diffraction_resolution\"><strong>Resolution:&nbsp</strong>2.60 </li><li id=\"exp_header_0_diffraction_rvalueFree\"><strong>R-Value Free:&nbsp</strong>0.276&nbsp</li><li id=\"exp_header_0_diffraction_rvalueWork\"><strong>R-Value Work:&nbsp</strong>0.200&nbsp</li><li id=\"exp_header_0_diffraction_rvalueObserved\"><strong>R-Value Observed:&nbsp</strong>0.200&nbsp     </li></ul></div><div class=\"col-sm-7 col-xs-12\"><!-- Experimental Validation HTML / PDF--><!-- Experimental Validation HTML / PDF--><style>div.validation-slider { border: 1px solid #ddd; padding: 5px; }\n",
      "div.validation-slider img { max-width: 100%; }\n",
      "</style><p><strong>wwPDB Validation</strong>&nbsp <span class=\"glyphicon glyphicon-info-sign hidden-xs hidden-sm hidden-md\" aria-hidden=\"true\" data-toggle=\"tooltip\" data-placement=\"right\" data-original-title=\"For each metric, two values are noted, one for the percentile rank of the entry compared to the entire PDB archive and one compared with entries determined with the same experimental method. The blue side of the scale is considered &quot;better&quot; than those on the &quot;worse&quot; red side.\">          </span><span class=\"pull-right\"><a class=\"btn btn-default btn-xs\" href=\"/3d-view/1YCR?preset=validationReport\"><i class=\"fa fa-cube\">&nbsp</i>3D Report</a>&nbsp<a class=\"btn btn-default btn-xs\" target=\"_blank\" rel=\"noopener\" href=\"//files.rcsb.org/pub/pdb/validation_reports/yc/1ycr/1ycr_full_validation.pdf\">Full Report</a></span></p><div class=\"validation-slider\"><a target=\"_blank\" rel=\"noopener\" href=\"//files.rcsb.org/pub/pdb/validation_reports/yc/1ycr/1ycr_full_validation.pdf\"><img src=\"//files.rcsb.org/pub/pdb/validation_reports/yc/1ycr/1ycr_multipercentile_validation.png\" width=\"500\"></a></div></div></div><br><div class=\"row\"><div class=\"col-lg-12 col-md-12 col-sm-12 col-xs-12\" id=\"currentVersionSection\"><div class=\"well well-sm well-nomar\">This is version 1.4 of the entry. See complete&nbsp<a href=\"/versions/1YCR\">history</a>.&nbsp</div><br></div></div><br><div class=\"row\"><div class=\"col-lg-12 col-md-12 col-sm-12 col-xs-12\"><div class=\"panel panel-default\" id=\"literaturepanel\"><div class=\"panel-heading\"><div class=\"panel-title\">Literature<div class=\"btn-group pull-right\"><button class=\"btn btn-default btn-xs hidden-print dropdown-toggle\" type=\"button\" data-toggle=\"dropdown\" aria-expanded=\"false\">Download Primary Citation&nbsp<span class=\"caret\"></span></button><ul class=\"dropdown-menu DropdownChangeGallery\" role=\"menu\"><li><a href=\"javascript:document.getElementsByTagName('body')[0].appendChild(document.createElement('script')).setAttribute('src','https://www.mendeley.com/minified/bookmarklet.js');\"><img src=\"//cdn.rcsb.org/rcsb-pdb/explorer/SSPv2/images/MendeleyIcon.png\" width=\"16\">&nbspDownload Mendeley</a></li></ul></div></div></div><div class=\"panel-body\"><div id=\"primarycitation\"><h4>Structure of the MDM2 oncoprotein bound to the p53 tumor suppressor transactivation domain.</h4><a class=\"querySearchLink\" href=\"/search?q=citation.rcsb_authors:Kussie, P.H.\">Kussie, P.H.</a>,&nbsp<a class=\"querySearchLink\" href=\"/search?q=citation.rcsb_authors:Gorina, S.\">Gorina, S.</a>,&nbsp<a class=\"querySearchLink\" href=\"/search?q=citation.rcsb_authors:Marechal, V.\">Marechal, V.</a>,&nbsp<a class=\"querySearchLink\" href=\"/search?q=citation.rcsb_authors:Elenbaas, B.\">Elenbaas, B.</a>,&nbsp<a class=\"querySearchLink\" href=\"/search?q=citation.rcsb_authors:Moreau, J.\">Moreau, J.</a>,&nbsp<a class=\"querySearchLink\" href=\"/search?q=citation.rcsb_authors:Levine, A.J.\">Levine, A.J.</a>,&nbsp<a class=\"querySearchLink\" href=\"/search?q=citation.rcsb_authors:Pavletich, N.P.\">Pavletich, N.P.</a><br><!-- if journal_abbrev already published--><p>(1996) Science&nbsp<strong>274</strong>: 948-953</p><ul class=\"list-unstyled\"><li id=\"pubmedLinks\"><strong>PubMed</strong>:&nbsp<a class=\"querySearchLink\" href=\"/search?q=rcsb_pubmed_container_identifiers.pubmed_id:8875929\">8875929</a>&nbsp<span class=\"label label-external hidden-print\"><a href=\"http://www.ncbi.nlm.nih.gov/pubmed/?term=8875929\" target=\"_blank\" rel=\"noopener\">Search on PubMed</a></span></li><li id=\"pubmedDOI\"><strong>DOI:&nbsp</strong><a href=\"http://dx.doi.org/10.1126/science.274.5289.948\" target=\"_blank\" rel=\"noopener\">10.1126/science.274.5289.948</a></li><li id=\"citationPrimaryRelatedStructures\" style=\"word-wrap:break-word;\">Primary Citation of Related Structures: &nbsp<br><a href=\"/structure/1YCQ\">1YCQ</a>, <a href=\"/structure/1YCR\">1YCR</a></li><!--TODO: add \"Secondary Citation of Related Structures\" when data is available --><br><li id=\"pubmedAbstractText\"><strong>PubMed Abstract:&nbsp</strong><br></li><div class=\"hidden-print\" id=\"abstractPart\"><p>The MDM2 oncoprotein is a cellular inhibitor of the p53 tumor suppressor in that it can bind the transactivation domain of p53 and downregulate its ability to activate transcription. In certain cancers, MDM2 amplification is a common event and contributes to the inactivation of p53 ...<span class=\"glyphicon glyphicon-plus-sign\" id=\"FullAbstract_Show\"></span></p></div><div class=\"hidden-print hide\" id=\"abstractFull\"><p>The MDM2 oncoprotein is a cellular inhibitor of the p53 tumor suppressor in that it can bind the transactivation domain of p53 and downregulate its ability to activate transcription. In certain cancers, MDM2 amplification is a common event and contributes to the inactivation of p53. The crystal structure of the 109-residue amino-terminal domain of MDM2 bound to a 15-residue transactivation domain peptide of p53 revealed that MDM2 has a deep hydrophobic cleft on which the p53 peptide binds as an amphipathic alpha helix. The interface relies on the steric complementarity between the MDM2 cleft and the hydrophobic face of the p53 alpha helix and, in particular, on a triad of p53 amino acids-Phe19, Trp23, and Leu26-which insert deep into the MDM2 cleft. These same p53 residues are also involved in transactivation, supporting the hypothesis that MDM2 inactivates p53 by concealing its transactivation domain. The structure also suggests that the amphipathic alpha helix may be a common structural motif in the binding of a diverse family of transactivation factors to the TATA-binding protein-associated factors.</p><hr><strong>Organizational Affiliation</strong>:&nbsp<p>Cellular Biochemistry and Biophysics Program, Memorial Sloan-Kettering Cancer Center, New York, NY 10021, USA. nikola@xray2.mskcc.org</p><br><button class=\"btn btn-default btn-xs hidden-print\" id=\"FullAbstract_Hide\"><span class=\"glyphicon glyphicon-minus-sign\"></span> Hide Full Abstract</button></div></ul><!--journal_abbrev to be published--></div></div></div></div></div></div></div><br></div><div class=\"panel panel-default\" id=\"macromoleculespanel\"><div class=\"panel-heading\"><div class=\"panel-title\">Macromolecules</div></div><div class=\"panel-body\"><div class=\"row\"><div class=\"col-md-12 col-sm-12 col-xs-12\"><div id=\"MacromoleculeTable\"><div style=\"display: flow-root; margin-bottom: 5px;\">Find similar proteins by:&nbsp<div class=\"btn-group\" role=\"group\"><button class=\"btn-group btn btn-sm btn-default dropdown-toggle\" id=\"dropdownCutOff\" type=\"button\" data-toggle=\"dropdown\" aria-expanded=\"false\" role=\"group\" style=\"color: #337ab7; text-decoration: none;\">Sequence&nbsp;<span class=\"caret\"></span></button><ul class=\"dropdown-menu\" role=\"menu\" aria-labelledby=\"dropdownCutOff\"><li><a href=\"/search?query=%7B%22type%22%3A%22terminal%22%2C%22service%22%3A%22sequence%22%2C%22parameters%22%3A%7B%22target%22%3A%22pdb_protein_sequence%22%2C%22value%22%3A%22SQIPASEQETLVRPKPLLLKLLKSVGAQKDTYTMKEVLFYLGQYIMTKRLYDEKQQHIVYCSNDLLGDLFGVPSFSVKEHRKIYTMIYRNLVVVNQQESSDSGTSVSEN%22%2C%22identity_cutoff%22%3A1%7D%7D&amp;return_type=polymer_entity\">100%</a></li><li><a href=\"/search?query=%7B%22type%22%3A%22terminal%22%2C%22service%22%3A%22sequence%22%2C%22parameters%22%3A%7B%22target%22%3A%22pdb_protein_sequence%22%2C%22value%22%3A%22SQIPASEQETLVRPKPLLLKLLKSVGAQKDTYTMKEVLFYLGQYIMTKRLYDEKQQHIVYCSNDLLGDLFGVPSFSVKEHRKIYTMIYRNLVVVNQQESSDSGTSVSEN%22%2C%22identity_cutoff%22%3A0.95%7D%7D&amp;return_type=polymer_entity\">95%</a></li><li><a href=\"/search?query=%7B%22type%22%3A%22terminal%22%2C%22service%22%3A%22sequence%22%2C%22parameters%22%3A%7B%22target%22%3A%22pdb_protein_sequence%22%2C%22value%22%3A%22SQIPASEQETLVRPKPLLLKLLKSVGAQKDTYTMKEVLFYLGQYIMTKRLYDEKQQHIVYCSNDLLGDLFGVPSFSVKEHRKIYTMIYRNLVVVNQQESSDSGTSVSEN%22%2C%22identity_cutoff%22%3A0.9%7D%7D&amp;return_type=polymer_entity\">90%</a></li><li><a href=\"/search?query=%7B%22type%22%3A%22terminal%22%2C%22service%22%3A%22sequence%22%2C%22parameters%22%3A%7B%22target%22%3A%22pdb_protein_sequence%22%2C%22value%22%3A%22SQIPASEQETLVRPKPLLLKLLKSVGAQKDTYTMKEVLFYLGQYIMTKRLYDEKQQHIVYCSNDLLGDLFGVPSFSVKEHRKIYTMIYRNLVVVNQQESSDSGTSVSEN%22%2C%22identity_cutoff%22%3A0.8%7D%7D&amp;return_type=polymer_entity\">80%</a></li><li><a href=\"/search?query=%7B%22type%22%3A%22terminal%22%2C%22service%22%3A%22sequence%22%2C%22parameters%22%3A%7B%22target%22%3A%22pdb_protein_sequence%22%2C%22value%22%3A%22SQIPASEQETLVRPKPLLLKLLKSVGAQKDTYTMKEVLFYLGQYIMTKRLYDEKQQHIVYCSNDLLGDLFGVPSFSVKEHRKIYTMIYRNLVVVNQQESSDSGTSVSEN%22%2C%22identity_cutoff%22%3A0.7%7D%7D&amp;return_type=polymer_entity\">70%</a></li><li><a href=\"/search?query=%7B%22type%22%3A%22terminal%22%2C%22service%22%3A%22sequence%22%2C%22parameters%22%3A%7B%22target%22%3A%22pdb_protein_sequence%22%2C%22value%22%3A%22SQIPASEQETLVRPKPLLLKLLKSVGAQKDTYTMKEVLFYLGQYIMTKRLYDEKQQHIVYCSNDLLGDLFGVPSFSVKEHRKIYTMIYRNLVVVNQQESSDSGTSVSEN%22%2C%22identity_cutoff%22%3A0.6%7D%7D&amp;return_type=polymer_entity\">60%</a></li><li><a href=\"/search?query=%7B%22type%22%3A%22terminal%22%2C%22service%22%3A%22sequence%22%2C%22parameters%22%3A%7B%22target%22%3A%22pdb_protein_sequence%22%2C%22value%22%3A%22SQIPASEQETLVRPKPLLLKLLKSVGAQKDTYTMKEVLFYLGQYIMTKRLYDEKQQHIVYCSNDLLGDLFGVPSFSVKEHRKIYTMIYRNLVVVNQQESSDSGTSVSEN%22%2C%22identity_cutoff%22%3A0.5%7D%7D&amp;return_type=polymer_entity\">50%</a></li><li><a href=\"/search?query=%7B%22type%22%3A%22terminal%22%2C%22service%22%3A%22sequence%22%2C%22parameters%22%3A%7B%22target%22%3A%22pdb_protein_sequence%22%2C%22value%22%3A%22SQIPASEQETLVRPKPLLLKLLKSVGAQKDTYTMKEVLFYLGQYIMTKRLYDEKQQHIVYCSNDLLGDLFGVPSFSVKEHRKIYTMIYRNLVVVNQQESSDSGTSVSEN%22%2C%22identity_cutoff%22%3A0.4%7D%7D&amp;return_type=polymer_entity\">40%</a></li><li><a href=\"/search?query=%7B%22type%22%3A%22terminal%22%2C%22service%22%3A%22sequence%22%2C%22parameters%22%3A%7B%22target%22%3A%22pdb_protein_sequence%22%2C%22value%22%3A%22SQIPASEQETLVRPKPLLLKLLKSVGAQKDTYTMKEVLFYLGQYIMTKRLYDEKQQHIVYCSNDLLGDLFGVPSFSVKEHRKIYTMIYRNLVVVNQQESSDSGTSVSEN%22%2C%22identity_cutoff%22%3A0.3%7D%7D&amp;return_type=polymer_entity\">30%</a></li></ul></div> (by identity cutoff)&nbsp | &nbsp<a href=\"/search?query=%7B%22type%22%3A%22terminal%22%2C%22service%22%3A%22structure%22%2C%22parameters%22%3A%7B%22value%22%3A%7B%22entry_id%22%3A%221YCR%22%2C%22asym_id%22%3A%22A%22%7D%7D%7D&amp;return_type=polymer_entity\">Structure</a></div><div class=\"table-responsive\"><a class=\"entity-anchor\" name=\"entity-1\"></a><table class=\"table table-bordered table-condensed tableEntity\" id=\"table_macromolecule-protein-entityId-1\"><tbody><tr class=\"info\" style=\"color: #315880\"><th colspan=\"6\" style=\"padding-bottom: 0px; padding-top: 0px\"><h5 style=\"font-weight: bold; font-size: 15px\">Entity ID: 1</h5></th></tr><tr><th class=\"col-sm-2 col-lg-2 pad-by-five\">Molecule</th><th class=\"col-sm-2 col-lg-2 pad-by-five\">Chains<span class=\"glyphicon glyphicon-info-sign hidden-xs hidden-sm hidden-md\" aria-hidden=\"true\" data-toggle=\"tooltip\" data-placement=\"top\" data-original-title=\"If the two PDB chain IDs (label_asym_id; assigned by the PDB) and auth_asym_id (selected by the author) do not coincide, the chain ID is displayed as label_asym_id [auth auth_asym_id]\"></span></th><th class=\"col-sm-2 col-lg-2 pad-by-five\">Sequence Length</th><th class=\"col-sm-2 col-lg-2 pad-by-five\">Organism</th><th class=\"col-sm-2 col-lg-2 pad-by-five\">Details</th><th class=\"col-sm-2 col-lg-2 pad-by-five\">Image</th></tr><tr id=\"macromolecule-entityId-1-rowDescription\"><td>MDM2</td><td style=\"width:200px;\"><a href=\"/sequence/1YCR#A\">A</a></td><td>109</td><td><!-- Check if there is any data inside organism array--><a class=\"querySearchLink\" href=\"/search?q=rcsb_entity_source_organism.taxonomy_lineage.name:Homo sapiens\">Homo sapiens</a></td><td style=\"word-break: break-all;\"><strong>Mutation(s)</strong>: 0&nbsp<span class=\"glyphicon glyphicon-info-sign hidden-xs hidden-sm hidden-md\" aria-hidden=\"true\" data-toggle=\"tooltip\" data-placement=\"top\" data-original-title=\"The number of mutations in this polymer sequence that are engineered from the reference sequence.\"></span><br><strong>Gene Names:&nbsp</strong><a class=\"querySearchLink\" href=\"/search?q=rcsb_entity_source_organism.rcsb_gene_name.value:MDM2\">MDM2</a><br><strong>EC:&nbsp</strong><a class=\"querySearchLink\" href=\"/search?q=rcsb_polymer_entity.rcsb_ec_lineage.id:2.3.2.27\">2.3.2.27</a><br></td><td style=\"text-align: center; vertical-align: middle\"><a data-toggle=\"lightbox\" data-title=\"1YCR_1: Represented by Chain A\" href=\"https://cdn.rcsb.org/images/structures/yc/1ycr/1ycr_chain-A.jpeg\"><img src=\"https://cdn.rcsb.org/images/structures/yc/1ycr/1ycr_chain-A.jpeg\" style=\"width:100px;\"></a></td></tr><tr><td colspan=\"6\" style=\"padding-bottom: 0px; padding-top: 0px; background-color: #f5f5f5;\"><h5 style=\"font-weight: bold; font-size: 15px\">UniProt & NIH Common Fund Data Resources</h5></td></tr><tr><td colspan=\"6\"><div class=\"col-lg-6 col-md-6 text-left\" style=\"padding-left: 0\">Find proteins for&nbsp<a class=\"querySearchLink\" href=\"/search?q=rcsb_polymer_entity_container_identifiers.reference_sequence_identifiers.database_accession:Q00987 AND rcsb_polymer_entity_container_identifiers.reference_sequence_identifiers.database_name:UniProt\">Q00987</a>&nbsp<i>(Homo sapiens)</i></div><div class=\"col-lg-4 col-md-4\">Explore&nbsp<span class=\"label label-rcsb\"><a href=\"/uniprot/Q00987\">Q00987</a></span>&nbsp<span class=\"glyphicon glyphicon-info-sign hidden-xs hidden-sm hidden-md\" aria-hidden=\"true\" data-toggle=\"tooltip\" data-placement=\"top\" data-original-title=\"Explore entries with the same UniProt reference sequence.\"></span></div><div class=\"col-lg-2 col-md-2 text-right\" style=\"padding-right: 0\">Go to UniProtKB: &nbsp<span class=\"label label-external\"><a href=\"https://www.uniprot.org/uniprot/Q00987\" target=\"_blank\" rel=\"noopener\">Q00987</a></span></div></td></tr><tr><td colspan=\"6\"><div class=\"col-lg-4 col-md-4 text-left\" style=\"padding-left: 0\" title=\"Diseases, drugs and related data\"><span>PHAROS: &nbsp<span class=\"label label-external\"><a href=\"https://pharos.nih.gov/targets/Q00987\" target=\"_blank\" rel=\"noopener\" onClick=\"sendGtag(&quot;SSP&quot;, &quot;NIH&quot;, &quot;PHAROS&quot;)\">Q00987</a></span><break></break></span></div><div class=\"col-lg-4 col-md-4\"><span>GTEx: &nbsp<span class=\"label label-external\"><a href=\"https://gtexportal.org/home/gene/ENSG00000135679\" target=\"_blank\" rel=\"noopener\" onClick=\"sendGtag(&quot;SSP&quot;, &quot;NIH&quot;, &quot;GTEx&quot;)\">ENSG00000135679</a></span><break></break></span></div><div class=\"col-lg-4 col-md-4 text-right\" style=\"padding-right: 0\"></div></td></tr><tr><td colspan=\"6\" style=\"padding-bottom: 0px; padding-top: 0px; background-color: #f5f5f5;\"><h5 style=\"font-weight: bold; font-size: 15px; display:inline-block;float:left\">Protein Feature View</h5><h5 style=\"font-weight: bold; font-size: 15px; display:inline-block;float:right;\"><a href=\"/sequence/1YCR#A\">Expand</a></h5></td></tr><tr style=\"border-bottom: 5px solid #ddd\" id=\"RcsbFv\"><td class=\"ProteinFeatureView\" colspan=\"6\"><div id=\"entity\" style=\"margin-top:20px\"><div class=\"col-xs-12 col-sm-12 col-md-4 col-lg-4 form-group\" style=\"margin-bottom: 0; padding-left: 0\"><ul class=\"list-group\" style=\"margin-bottom:0;display:inline-block;width:50%;\"><li class=\"list-group-item\" style=\"border-color:#fff;\"><strong>Reference Sequence</strong></li></ul><div id=\"rcsbWebAppSelect_1\" style=\"display:inline-block;\"></div><div class=\"material-switch pull-right\"></div></div></div><div class=\"clearfix\"></div><div id=\"rcsbWebApp_1\" style=\"margin-bottom: 50px;\"></div><script>setTimeout(function(){\n",
      "    RcsbFvWebApp.buildEntitySummaryFv(\n",
      "        \"rcsbWebApp_1\",\n",
      "        \"rcsbWebAppSelect_1\",\n",
      "        \"1YCR_1\"\n",
      "    );\n",
      "},100)\n",
      "</script></td></tr></tbody></table></div><ul class=\"list-unstyled\"><li class=\"text-muted\">Find similar proteins by:&nbsp Sequence &nbsp | &nbsp Structure<span class=\"fa fa-info-circle\" aria-hidden=\"true\" data-toggle=\"tooltip\" data-placement=\"top\" title=\"Sequence and structure search is not available for sequences with length less than or equal to 20\"></span></li></ul><div class=\"table-responsive\"><a class=\"entity-anchor\" name=\"entity-2\"></a><table class=\"table table-bordered table-condensed tableEntity\" id=\"table_macromolecule-protein-entityId-2\"><tbody><tr class=\"info\" style=\"color: #315880\"><th colspan=\"6\" style=\"padding-bottom: 0px; padding-top: 0px\"><h5 style=\"font-weight: bold; font-size: 15px\">Entity ID: 2</h5></th></tr><tr><th class=\"col-sm-2 col-lg-2 pad-by-five\">Molecule</th><th class=\"col-sm-2 col-lg-2 pad-by-five\">Chains<span class=\"glyphicon glyphicon-info-sign hidden-xs hidden-sm hidden-md\" aria-hidden=\"true\" data-toggle=\"tooltip\" data-placement=\"top\" data-original-title=\"If the two PDB chain IDs (label_asym_id; assigned by the PDB) and auth_asym_id (selected by the author) do not coincide, the chain ID is displayed as label_asym_id [auth auth_asym_id]\"></span></th><th class=\"col-sm-2 col-lg-2 pad-by-five\">Sequence Length</th><th class=\"col-sm-2 col-lg-2 pad-by-five\">Organism</th><th class=\"col-sm-2 col-lg-2 pad-by-five\">Details</th><th class=\"col-sm-2 col-lg-2 pad-by-five\">Image</th></tr><tr id=\"macromolecule-entityId-2-rowDescription\"><td>P53</td><td style=\"width:200px;\"><a href=\"/sequence/1YCR#B\">B</a></td><td>15</td><td><!-- Check if there is any data inside organism array--><small class=\"text-muted\"><i>N/A</i></small></td><td style=\"word-break: break-all;\"><strong>Mutation(s)</strong>: 0&nbsp<span class=\"glyphicon glyphicon-info-sign hidden-xs hidden-sm hidden-md\" aria-hidden=\"true\" data-toggle=\"tooltip\" data-placement=\"top\" data-original-title=\"The number of mutations in this polymer sequence that are engineered from the reference sequence.\"></span><br></td><td style=\"text-align: center; vertical-align: middle\"><a data-toggle=\"lightbox\" data-title=\"1YCR_2: Represented by Chain B\" href=\"https://cdn.rcsb.org/images/structures/yc/1ycr/1ycr_chain-B.jpeg\"><img src=\"https://cdn.rcsb.org/images/structures/yc/1ycr/1ycr_chain-B.jpeg\" style=\"width:100px;\"></a></td></tr><tr><td colspan=\"6\" style=\"padding-bottom: 0px; padding-top: 0px; background-color: #f5f5f5;\"><h5 style=\"font-weight: bold; font-size: 15px\">UniProt & NIH Common Fund Data Resources</h5></td></tr><tr><td colspan=\"6\"><div class=\"col-lg-6 col-md-6 text-left\" style=\"padding-left: 0\">Find proteins for&nbsp<a class=\"querySearchLink\" href=\"/search?q=rcsb_polymer_entity_container_identifiers.reference_sequence_identifiers.database_accession:P04637 AND rcsb_polymer_entity_container_identifiers.reference_sequence_identifiers.database_name:UniProt\">P04637</a>&nbsp<i>(Homo sapiens)</i></div><div class=\"col-lg-4 col-md-4\">Explore&nbsp<span class=\"label label-rcsb\"><a href=\"/uniprot/P04637\">P04637</a></span>&nbsp<span class=\"glyphicon glyphicon-info-sign hidden-xs hidden-sm hidden-md\" aria-hidden=\"true\" data-toggle=\"tooltip\" data-placement=\"top\" data-original-title=\"Explore entries with the same UniProt reference sequence.\"></span></div><div class=\"col-lg-2 col-md-2 text-right\" style=\"padding-right: 0\">Go to UniProtKB: &nbsp<span class=\"label label-external\"><a href=\"https://www.uniprot.org/uniprot/P04637\" target=\"_blank\" rel=\"noopener\">P04637</a></span></div></td></tr><tr><td colspan=\"6\"><div class=\"col-lg-4 col-md-4 text-left\" style=\"padding-left: 0\" title=\"Diseases, drugs and related data\"><span>PHAROS: &nbsp<span class=\"label label-external\"><a href=\"https://pharos.nih.gov/targets/P04637\" target=\"_blank\" rel=\"noopener\" onClick=\"sendGtag(&quot;SSP&quot;, &quot;NIH&quot;, &quot;PHAROS&quot;)\">P04637</a></span><break></break></span></div><div class=\"col-lg-4 col-md-4\"><span>GTEx: &nbsp<span class=\"label label-external\"><a href=\"https://gtexportal.org/home/gene/ENSG00000141510\" target=\"_blank\" rel=\"noopener\" onClick=\"sendGtag(&quot;SSP&quot;, &quot;NIH&quot;, &quot;GTEx&quot;)\">ENSG00000141510</a></span><break></break></span></div><div class=\"col-lg-4 col-md-4 text-right\" style=\"padding-right: 0\"></div></td></tr><tr><td colspan=\"6\" style=\"padding-bottom: 0px; padding-top: 0px; background-color: #f5f5f5;\"><h5 style=\"font-weight: bold; font-size: 15px; display:inline-block;float:left\">Protein Feature View</h5><h5 style=\"font-weight: bold; font-size: 15px; display:inline-block;float:right;\"><a href=\"/sequence/1YCR#B\">Expand</a></h5></td></tr><tr style=\"border-bottom: 5px solid #ddd\" id=\"RcsbFv\"><td class=\"ProteinFeatureView\" colspan=\"6\"><div id=\"entity\" style=\"margin-top:20px\"><div class=\"col-xs-12 col-sm-12 col-md-4 col-lg-4 form-group\" style=\"margin-bottom: 0; padding-left: 0\"><ul class=\"list-group\" style=\"margin-bottom:0;display:inline-block;width:50%;\"><li class=\"list-group-item\" style=\"border-color:#fff;\"><strong>Reference Sequence</strong></li></ul><div id=\"rcsbWebAppSelect_2\" style=\"display:inline-block;\"></div><div class=\"material-switch pull-right\"></div></div></div><div class=\"clearfix\"></div><div id=\"rcsbWebApp_2\" style=\"margin-bottom: 50px;\"></div><script>setTimeout(function(){\n",
      "    RcsbFvWebApp.buildEntitySummaryFv(\n",
      "        \"rcsbWebApp_2\",\n",
      "        \"rcsbWebAppSelect_2\",\n",
      "        \"1YCR_2\"\n",
      "    );\n",
      "},100)\n",
      "</script></td></tr></tbody></table></div></div></div></div></div></div><div class=\"panel panel-default\" id=\"experimentaldata-validation\"><div class=\"panel-heading\"><div class=\"panel-title\">Experimental Data & Validation</div></div><div class=\"panel-body\"><div class=\"row\"><div class=\"col-md-6 col-sm-12 col-xs-12\"><h4>Experimental Data</h4><div class=\"row\"><ul class=\"list-unstyled col-sm-12 col-lg-12 col-md-12\" id=\"experimentaldatabottom\"><li id=\"exp_header_0_method\"><strong>Method:&nbsp</strong>X-RAY DIFFRACTION</li><li id=\"exp_header_0_diffraction_resolution\"><strong>Resolution:&nbsp</strong>2.60 </li><li id=\"exp_header_0_diffraction_rvalueFree\"><strong>R-Value Free:&nbsp</strong>0.276&nbsp</li><li id=\"exp_header_0_diffraction_rvalueWork\"><strong>R-Value Work:&nbsp</strong>0.200&nbsp</li><li id=\"exp_header_0_diffraction_rvalueObserved\"><strong>R-Value Observed:&nbsp</strong>0.200&nbsp</li><li id=\"exp_undefined_xray_spaceGroup\"><strong>Space Group:&nbsp<a class=\"querySearchLink\" href=\"/search?q=symmetry.space_group_name_H_M:C 2 2 21\">C 2 2 2<sub>1</sub></a></strong></li></ul><div class=\"col-lg-12 col-md-12 col-sm-12\"><strong>Unit Cell</strong>:<table class=\"table table-bordered table-condensed\" id=\"unitCellTable\"><thead><tr class=\"active\"><th>Length (  )</th><th>Angle (  )</th></tr></thead><tbody><tr><td>a = 43.414</td><td> = 90</td></tr><tr><td>b = 100.546</td><td> = 90</td></tr><tr><td>c = 54.853</td><td> = 90</td></tr></tbody></table></div></div><strong>Software Package:</strong><table class=\"table table-bordered table-condensed\" id=\"softwarePackages\"><thead><tr class=\"active\"><th>Software Name</th><th>Purpose</th></tr></thead><tbody><tr><td>X-PLOR</td><td>refinement</td></tr><tr><td>TNT</td><td>refinement</td></tr><tr><td>X-PLOR</td><td>model building</td></tr><tr><td>HKL</td><td>data reduction</td></tr><tr><td>X-PLOR</td><td>phasing</td></tr></tbody></table></div><div class=\"col-md-6 col-sm-12 col-xs-12\"><!-- Experimental Validation HTML / PDF--><h4>Structure Validation</h4><!-- NOTE: check pdx/idf/ftp.json file--><p id=\"validationSection\">View&nbsp<a target=\"_blank\" rel=\"noopener\" href=\"//files.rcsb.org/pub/pdb/validation_reports/yc/1ycr/1ycr_full_validation.pdf\">Full Validation Report</a></p><a target=\"_blank\" rel=\"noopener\" href=\"//files.rcsb.org/pub/pdb/validation_reports/yc/1ycr/1ycr_full_validation.pdf\"><img class=\"img-thumbnail\" src=\"//files.rcsb.org/pub/pdb/validation_reports/yc/1ycr/1ycr_multipercentile_validation.png\" width=\"500\"></a><br></div></div><br><div class=\"row\"><div class=\"col-md-6 col-sm-12 col-xs-12\"><a class=\"btn btn-sm btn-default hidden-print\" href=\"/experimental/1YCR\">View more in-depth experimental data</a></div></div></div></div><br><div class=\"panel panel-default\" id=\"entry-history\"><div class=\"panel-heading\"><div class=\"panel-title\">Entry History&nbsp</div></div><div class=\"panel-body\"><div class=\"row\"><div class=\"col-md-6 col-sm-6 col-xs-12 col-xs-12\"><h4>Deposition Data</h4><ul class=\"list-unstyled\" id=\"DepositionDataList\"><li><strong>Deposited Date:&nbsp</strong>1996-09-30&nbsp</li><li><strong>Released Date:&nbsp</strong>1997-11-19&nbsp</li><strong>Deposition Author(s):&nbsp</strong><a href=\"/search?q=audit_author.name:Kussie, P.H.\">Kussie, P.H.</a>, <a href=\"/search?q=audit_author.name:Pavletich, N.P.\">Pavletich, N.P.</a></ul></div><div class=\"col-md-6 col-sm-6 col-xs-12 col-xs-12\"><h4>Revision History &nbsp;<small><a href=\"/versions/1YCR\">(Full details and data files)</a></small></h4><ul id=\"revisionHistorySection\"><li><strong>Version 1.0: 1997-11-19</strong><br>Type: Initial release<br></li><li><strong>Version 1.1: 2008-03-24</strong><br>Changes: Version format compliance </li><li><strong>Version 1.2: 2011-07-13</strong><br>Changes: Version format compliance </li><li><strong>Version 1.3: 2019-07-17</strong><br>Changes: Data collection, Other, Refinement description </li><li><strong>Version 1.4: 2019-08-14</strong><br>Changes: Data collection </li></ul></div></div></div></div></div><div class=\"clearfix\"></div><script>$(\"#structuresummarytab\").addClass(\"active\");</script><script src=\"/js/bootstrap-slider.min.js\"></script><script src=\"/js/jquery.svg.js\"></script><script src=\"/js/structure/helper.js\"></script><script src=\"/js/structure/transmembrane_image_carousel.js\"></script><script src=\"/js/structure/poseview.js\"></script></div><div class=\"hidden-print\" id=\"footer_main\"><div class=\"container\"><div class=\"row\"><div class=\"col-sm-12 col-md-12 col-lg-7 text-center\"><ul class=\"col-xs-4 col-sm-12\"><li><span class=\"row_header\">About</span></li><li><a href=\"/pages/about-us/index\">About Us</a></li><li><a href=\"/pages/policies#References\">Citing Us</a></li><li><a href=\"/pages/publications\">Publications</a></li><li><a href=\"/pages/team\">Team</a></li><li><a href=\"/pages/jobs\">Careers</a></li><li><a href=\"/pages/policies\">Usage & Privacy</a></li></ul><ul class=\"col-xs-4 col-sm-12\"><li><span class=\"row_header\">Help</span></li><li><a href=\"/pages/contactus\">Contact Us</a></li><li><a href=\"/docs/general-help/organization-of-3d-structures-in-the-protein-data-bank\">Documentation</a></li><li><a href=\"/docs/general-help/website-faq\">Website FAQ</a></li><li><a href=\"/docs/general-help/glossary\">Glossary</a></li><li><a href=\"https://status.rcsb.org/\" target=\"_blank\">Service Status    </a></li></ul><ul class=\"col-xs-4 col-sm-12\"><li><span class=\"row_header\">RCSB Partners</span></li><li><a href=\"http://ndbserver.rutgers.edu/\" target=\"_blank\" rel=\"noopener\">Nucleic Acid Database</a></li></ul><ul class=\"col-xs-4 col-sm-12\"><li><span class=\"row_header\"><a href=\"https://www.wwpdb.org/\" target=\"_blank\" rel=\"noopener\" style=\"text-decoration: none; color: black\">wwPDB Partners</a></span></li><li><a href=\"/\">RCSB PDB</a></li><li><a href=\"https://www.ebi.ac.uk/pdbe/\" target=\"_blank\" rel=\"noopener\">PDBe</a></li><li><a href=\"https://www.pdbj.org/\" target=\"_blank\" rel=\"noopener\">PDBj</a></li><li><a href=\"https://bmrb.io/\" target=\"_blank\" rel=\"noopener\">BMRB</a></li><li> <a href=\"https://emdb-empiar.org/\" target=\"_blank\" rel=\"noopener\">EMDB</a></li></ul></div><div class=\"col-xs-12 col-sm-12 col-md-12 col-lg-5 text-center\"><p>RCSB PDB (<a href=\"http://nar.oxfordjournals.org/content/28/1/235.abstract\" target=\"blank\">citation</a>) is hosted by</p><img src=\"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAQAAAC1HAwCAAAAC0lEQVR42mNkYAAAAAYAAjCB0C8AAAAASUVORK5CYII=\" data-src=\"//cdn.rcsb.org/rcsb-pdb//v2/common/images/Logo_Rutgers.png\" alt=\"Logo_Rutgers.png\" usemap=\"#rutgers-map\" height=\"26\" style=\"padding:0px 10px; border-right: 1px solid #000000;\"><img src=\"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAQAAAC1HAwCAAAAC0lEQVR42mNkYAAAAAYAAjCB0C8AAAAASUVORK5CYII=\" data-src=\"//cdn.rcsb.org/rcsb-pdb//v2/common/images/Logo_UCSD-SDSC.png\" alt=\"Logo_UCSD.png\" usemap=\"#UCSD-SDSC-map\" height=\"26\" style=\"padding: 0px 10px; border-right: 1px solid #000000;\"><img src=\"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAQAAAC1HAwCAAAAC0lEQVR42mNkYAAAAAYAAjCB0C8AAAAASUVORK5CYII=\" data-src=\"//cdn.rcsb.org/rcsb-pdb//v2/common/images/Logo_UCSF.png\" alt=\"Logo_UCSF.png\" usemap=\"#UCSF-map\" height=\"26\" style=\"padding-left: 10px;\"><map name=\"rutgers-map\"><area shape=\"rect\" coords=\"0,0,115,26\" href=\"http://www.rutgers.edu/\" alt=\"Rutgers\" target=\"_blank\" rel=\"noopener\"></map><map name=\"UCSD-SDSC-map\"><area shape=\"rect\" coords=\"0,0,150,26\" href=\"http://ucsd.edu/\" alt=\"UCSD\" target=\"_blank\" rel=\"noopener\"><area shape=\"rect\" coords=\"151,0,255,26\" href=\"http://www.sdsc.edu/\" alt=\"SDSC\" target=\"_blank\" rel=\"noopener\"></map><map name=\"UCSF-map\"><area shape=\"rect\" coords=\"0,0,64,26\" href=\"http://www.ucsf.edu/\" alt=\"UCSF\" target=\"_blank\" rel=\"noopener\"></map><p>RCSB PDB is a member of the<span id=\"pdbmembers_footer\"><a href=\"https://www.wwpdb.org\" target=\"_blank\" rel=\"noopener\"><img class=\"wwpdblogo\" src=\"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAQAAAC1HAwCAAAAC0lEQVR42mNkYAAAAAYAAjCB0C8AAAAASUVORK5CYII=\" data-src=\"//cdn.rcsb.org/rcsb-pdb//v2/common/images/Logo_wwpdb.png\" alt=\"wwPDB\"></a><a href=\"https://www.emdataresource.org\" target=\"_blank\" rel=\"noopener\"><img id=\"emdatabanklogo_footer\" src=\"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAQAAAC1HAwCAAAAC0lEQVR42mNkYAAAAAYAAjCB0C8AAAAASUVORK5CYII=\" data-src=\"//cdn.rcsb.org/rcsb-pdb//v2/common/images/EMDataResourcelogo.png\" alt=\"EMDataResource\"></a></span></p></div></div></div></div><div id=\"footer_grant\"><div class=\"container\"><div class=\"row\"><p>RCSB PDB is funded by\n",
      " the <a href=\"http://www.nsf.gov/\" target=\"_blank\" rel=\"noopener\">National Science Foundation</a> (DBI-1832184),\n",
      " the <a href=\"http://science.energy.gov/\" target=\"_blank\" rel=\"noopener\">US Department of Energy</a> (DE-SC0019749),\n",
      " and the <a href=\"http://www.cancer.gov/\" target=\"_blank\" rel=\"noopener\">National Cancer Institute</a>,\n",
      " <a href=\"http://www.niaid.nih.gov/\" target=\"_blank\" rel=\"noopener\">National Institute of Allergy and Infectious Diseases</a>,\n",
      " and <a href=\"http://www.nigms.nih.gov/\" target=\"_blank\" rel=\"noopener\">National Institute of General Medical Sciences</a> of the <a href=\"http://www.nih.gov/\" target=\"_blank\" rel=\"noopener\">National Institutes of Health</a> under grant R01GM133198.</p></div></div></div><div id=\"jira-fdbck\"></div></body></html>\n"
     ]
    }
   ],
   "source": [
    "values = {'structureId' : '1ycr'}\n",
    "data = requests.get('http://www.pdb.org/pdb/explore/explore.do',values)\n",
    "\n",
    "print(data.text)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "slideshow": {
     "slide_type": "slide"
    }
   },
   "source": [
    "# Encoding Data\n",
    "\n",
    "URLs are only allowed to have alphanumeric characters (and a handful of punctuation marks).  This means data needs to be encoded when passing it as a value. requests will do this for you if you pass values as a dictionary."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "values = {'q':\"What's the meaning of life?\"}\n",
    "data = requests.get('http://www.google.com/search',values)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'http://www.google.com/search?q=What%27s+the+meaning+of+life%3F'"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data.url"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "slideshow": {
     "slide_type": "slide"
    }
   },
   "source": [
    "# Faking It\n",
    "\n",
    "<img src=\"imgs/zeldacomp.jpeg\" width=\"500\">\n",
    "\n",
    "...or a python script"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "slideshow": {
     "slide_type": "slide"
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['python-requests 2', 'python-requests 2']"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import requests, re\n",
    "page = requests.get('https://www.whatsmybrowser.org/').text\n",
    "re.findall(r'You&rsquo;re using (.*?)\\.',page)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['Chrome 61', 'Chrome 61']"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "page = requests.get('https://www.whatsmybrowser.org/',headers={'User-Agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/61.0.3163.100 Safari/537.36'}).text\n",
    "re.findall(r'You&rsquo;re using (.*?)\\.',page)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Use `mechanize` if a site is giving you trouble**"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "slideshow": {
     "slide_type": "slide"
    }
   },
   "source": [
    "# Getting Data From the Web"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "If it's on the web, you can get it into python.\n",
    "\n",
    "* *screen scraping* - getting data from your computer screen; often to the extreme of using screen shots and OCR\n",
    "* *web scraping* - downloading HTML content and parsing out what you need\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "slideshow": {
     "slide_type": "slide"
    }
   },
   "source": [
    "# Web Scraping"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The advantage of web scraping is that it always works - if you can see the data in your browser, you can see it in python.\n",
    "\n",
    "Disadvantages:\n",
    "\n",
    "* HTMLParser isn't very sophisticated and requires you to manage context\n",
    "    * see [pyquery](https://pypi.python.org/pypi/pyquery) for a more powerful HTML parsing package inspired by JQuery\n",
    "    * [BeautifulSoup](https://www.crummy.com/software/BeautifulSoup/bs4/doc/) is another popular library for extracting information from HTML\n",
    "    \n",
    "* Scraping is slow and inefficient\n",
    "    * may need to make several requests to get your data (think pagination)\n",
    "    * downloads a lot more than just your data (html)\n",
    "    * your code may be easily broken by changes in the website design\n",
    "    \n",
    "**Only parse raw HTML if there is no better option**"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "slideshow": {
     "slide_type": "slide"
    }
   },
   "source": [
    "# ReSTful Web Services"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "REpresentational State Transfer\n",
    "\n",
    "* Clientserver\n",
    "* Stateless\n",
    "* Cacheable\n",
    "* Layered system\n",
    "* Uniform interface\n",
    "\n",
    "    * basically, resources are identified by their url\n",
    "    * ReST does *not* specify the format of the resources, but they are often provided in **XML**"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "slideshow": {
     "slide_type": "slide"
    }
   },
   "source": [
    "# XML"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "*Extensible Markup Language*. A very general way to express structured data; a generalization of HTML.\n",
    "\n",
    "* **Tag** Key building block of XML - starts and ends with < >\n",
    "    * `<div>` start tag\n",
    "    * `</div>` end tag\n",
    "    * `<br />` empty element tag (not matching end)\n",
    " \n",
    "* **Element** A component of the document; everything between a start and end tag.  May contain child elements.\n",
    "\n",
    "* **Attribute** A name-value pair within a start or empty element tag\n",
    "        <div width=90>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "slideshow": {
     "slide_type": "slide"
    }
   },
   "source": [
    "# XML Example"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Let's consider accessing NCBI's Entrez service to get data from the Gene Expression Omnibus (note we could use BioPython and avoid doing this directly).\n",
    "\n",
    "Let's look for data from humans with between 100 and 500 samples:\n",
    "\n",
    "* (human[Organism]) AND 100:500[Number of Samples]\n",
    "\n",
    "We construct a URL to represent this query.\n",
    "\n",
    "http://eutils.ncbi.nlm.nih.gov/entrez/eutils/esearch.fcgi?db=gds&term=human[Organism]+AND+100:500[Number+of+Samples]\n",
    "\n",
    "db=gds - GEO datasets \n",
    "\n",
    "* GEO DataSets is a study-level database which users can search for studies relevant to their interests. The database stores descriptions of all original submitter-supplied records, as well as curated DataSets.\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "slideshow": {
     "slide_type": "slide"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<?xml version=\"1.0\" encoding=\"UTF-8\" ?>\n",
      "<!DOCTYPE eSearchResult PUBLIC \"-//NLM//DTD esearch 20060628//EN\" \"https://eutils.ncbi.nlm.nih.gov/eutils/dtd/20060628/esearch.dtd\">\n",
      "<eSearchResult><Count>5432</Count><RetMax>20</RetMax><RetStart>0</RetStart><IdList>\n",
      "<Id>200176447</Id>\n",
      "<Id>200186206</Id>\n",
      "<I\n"
     ]
    }
   ],
   "source": [
    "result = requests.get(\\\n",
    "            \"http://eutils.ncbi.nlm.nih.gov/entrez/eutils/esearch.fcgi\",\\\n",
    "             params={'db':'gds',\\\n",
    "             'term':'human[Organism] AND 100:500[Number of Samples]'}).text\n",
    "print(result[:297])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "slideshow": {
     "slide_type": "slide"
    }
   },
   "source": [
    "# `xml.etree.ElementTree`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "eSearchResult {}\n"
     ]
    }
   ],
   "source": [
    "import xml.etree.ElementTree as ET\n",
    "root = ET.fromstring(result)\n",
    "print(root.tag,root.attrib)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "slideshow": {
     "slide_type": "-"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Count\n",
      "RetMax\n",
      "RetStart\n",
      "IdList\n",
      "TranslationSet\n",
      "TranslationStack\n",
      "QueryTranslation\n"
     ]
    }
   ],
   "source": [
    "for child in root:\n",
    "    print(child.tag)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "slideshow": {
     "slide_type": "slide"
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<Element 'IdList' at 0x110965d60>"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "root.find('IdList')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['200176447', '200186206', '200186110', '200182798', '200182503', '200174777', '200148368', '200186001', '200163211', '200145154', '200181982', '200181520', '200185512', '200146323', '200145845', '200145844', '200145825', '200156680', '200157240', '200133655']\n"
     ]
    }
   ],
   "source": [
    "ids = [child.text for child in root.find('IdList')]\n",
    "print(ids)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[]"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "root.findall('Id') #find only considers children"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['200176447', '200186206', '200186110', '200182798', '200182503', '200174777', '200148368', '200186001', '200163211', '200145154', '200181982', '200181520', '200185512', '200146323', '200145845', '200145844', '200145825', '200156680', '200157240', '200133655']\n"
     ]
    }
   ],
   "source": [
    "ids = [elem.text for elem in root.iter('Id')] #iter considers the full tree\n",
    "print(ids)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "slideshow": {
     "slide_type": "slide"
    }
   },
   "source": [
    "# XML Parsing Alternative: Regular Expressions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<?xml version=\"1.0\" encoding=\"UTF-8\" ?>\n",
      "<!DOCTYPE eSearchResult PUBLIC \"-//NLM//DTD esearch 20060628//EN\" \"https://eutils.ncbi.nlm.nih.gov/eutils/dtd/20060628/esearch.dtd\">\n",
      "<eSearchResult><Count>5432</Count><RetMax>20</RetMax><RetStart>0</RetStart><IdList>\n",
      "<Id>200176447</Id>\n",
      "<Id>200186206</Id>\n",
      "<I\n"
     ]
    }
   ],
   "source": [
    "print(result[:297])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "slideshow": {
     "slide_type": "-"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<re.Match object; span=(257, 275), match='<Id>200176447</Id>'>\n"
     ]
    }
   ],
   "source": [
    "import re\n",
    "regex = re.compile(r'<Id>(\\d+)</Id>')\n",
    "print(regex.search(result))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['200176447', '200186206', '200186110', '200182798', '200182503', '200174777', '200148368', '200186001', '200163211', '200145154', '200181982', '200181520', '200185512', '200146323', '200145845', '200145844', '200145825', '200156680', '200157240', '200133655']\n"
     ]
    }
   ],
   "source": [
    "ids = [m.group(1) for m in regex.finditer(result)]\n",
    "print(ids)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "slideshow": {
     "slide_type": "slide"
    }
   },
   "source": [
    "# JSON: JavaScript Object Notation\n",
    "\n",
    "A lightweight data-interchange format.  \n",
    "\n",
    "Essentially, represent data as JavaScript, which is very similar to python dictionaries/lists."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "import json"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'{\"a\": 1, \"b\": [1.2, 1.3, 1.4]}'"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "json.dumps({'a':1, 'b':[1.2,1.3,1.4]})"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "slideshow": {
     "slide_type": "slide"
    }
   },
   "source": [
    "# PDB Restful\n",
    "\n",
    "https://data.rcsb.org/redoc/index.html\n",
    "\n",
    "Access PDB data through endpoints (URLs).  The path of the endpoints starts with https://data.rcsb.org/rest/v1/core, followed by the type of the resource, e.g. entry, polymer_entity, and the identifier. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{\"audit_author\":[{\"name\":\"Fermi, G.\",\"pdbx_ordinal\":1},{\"name\":\"Perutz, M.F.\",\"pdbx_ordinal\":2}],\"cell\":{\"angle_alpha\":90.0,\"angle_beta\":99.34,\"angle_gamma\":90.0,\"length_a\":63.15,\"length_b\":83.59,\"length_c\":53.8,\"zpdb\":4},\"citation\":[{\"country\":\"UK\",\"id\":\"primary\",\"journal_abbrev\":\"J.Mol.Biol.\",\"journal_id_astm\":\"JMOBAK\",\"journal_id_csd\":\"0070\",\"journal_id_issn\":\"0022-2836\",\"journal_volume\":\"175\",\"page_first\":\"159\",\"page_last\":\"174\",\"pdbx_database_id_doi\":\"10.1016/0022-2836(84)90472-8\",\"pdbx_database_id_pub_med\":6726807,\"rcsb_authors\":[\"Fermi, G.\",\"Perutz, M.F.\",\"Shaanan, B.\",\"Fourme, R.\"],\"rcsb_is_primary\":\"Y\",\"rcsb_journal_abbrev\":\"J Mol Biol\",\"title\":\"The crystal structure of human deoxyhaemoglobin at 1.74 A resolution\",\"year\":1984},{\"country\":\"UK\",\"id\":\"1\",\"journal_abbrev\":\"Nature\",\"journal_id_astm\":\"NATUAS\",\"journal_id_csd\":\"0006\",\"journal_id_issn\":\"0028-0836\",\"journal_volume\":\"295\",\"page_first\":\"535\",\"rcsb_authors\":[\"Perutz, M.F.\",\"Hasnain, S.S.\",\"Duke, P.J.\",\"Sessler, J.L.\",\"Hahn, J.E.\"],\"rcsb_is_primary\":\"N\",\"rcsb_journal_abbrev\":\"Nature\",\"title\":\"Stereochemistry of Iron in Deoxyhaemoglobin\",\"year\":1982},{\"country\":\"US\",\"id\":\"3\",\"journal_abbrev\":\"Annu.Rev.Biochem.\",\"journal_id_astm\":\"ARBOAW\",\"journal_id_csd\":\"0413\",\"journal_id_issn\":\"0066-4154\",\"journal_volume\":\"48\",\"page_first\":\"327\",\"rcsb_authors\":[\"Perutz, M.F.\"],\"rcsb_is_primary\":\"N\",\"rcsb_journal_abbrev\":\"Annu Rev Biochem\",\"title\":\"Regulation of Oxygen Affinity of Hemoglobin. Influence of Structure of the Globin on the Heme Iron\",\"year\":1979},{\"country\":\"UK\",\"id\":\"4\",\"journal_abbrev\":\"J.Mol.Biol.\",\"journal_id_astm\":\"JMOBAK\",\"journal_id_csd\":\"0070\",\"journal_id_issn\":\"0022-2836\",\"journal_volume\":\"100\",\"page_first\":\"3\",\"rcsb_authors\":[\"Teneyck, L.F.\",\"Arnone, A.\"],\"rcsb_is_primary\":\"N\",\"rcsb_journal_abbrev\":\"J Mol Biol\",\"title\":\"Three-Dimensional Fourier Synthesis of Human Deoxyhemoglobin at 2.5 Angstroms Resolution, I.X-Ray Analysis\",\"year\":1976},{\"country\":\"UK\",\"id\":\"5\",\"journal_abbrev\":\"J.Mol.Biol.\",\"journal_id_astm\":\"JMOBAK\",\"journal_id_csd\":\"0070\",\"journal_id_issn\":\"0022-2836\",\"journal_volume\":\"97\",\"page_first\":\"237\",\"rcsb_authors\":[\"Fermi, G.\"],\"rcsb_is_primary\":\"N\",\"rcsb_journal_abbrev\":\"J Mol Biol\",\"title\":\"Three-Dimensional Fourier Synthesis of Human Deoxyhaemoglobin at 2.5 Angstroms Resolution, Refinement of the Atomic Model\",\"year\":1975},{\"country\":\"UK\",\"id\":\"6\",\"journal_abbrev\":\"Nature\",\"journal_id_astm\":\"NATUAS\",\"journal_id_csd\":\"0006\",\"journal_id_issn\":\"0028-0836\",\"journal_volume\":\"228\",\"page_first\":\"516\",\"rcsb_authors\":[\"Muirhead, H.\",\"Greer, J.\"],\"rcsb_is_primary\":\"N\",\"rcsb_journal_abbrev\":\"Nature\",\"title\":\"Three-Dimensional Fourier Synthesis of Human Deoxyhaemoglobin at 3.5 Angstroms Resolution\",\"year\":1970},{\"book_publisher\":\"Oxford University Press\",\"id\":\"2\",\"journal_abbrev\":\"Haemoglobin and Myoglobin. Atlas of Molecular Structures in Biology\",\"journal_id_csd\":\"0986\",\"journal_id_issn\":\"0-19-854706-4\",\"journal_volume\":\"2\",\"rcsb_authors\":[\"Fermi, G.\",\"Perutz, M.F.\"],\"rcsb_is_primary\":\"N\",\"rcsb_journal_abbrev\":\"Haemoglobin And Myoglobin Atlas Of Molecular Structures In Biology\",\"year\":1981},{\"book_publisher\":\"National Biomedical Research Foundation, Silver Spring,Md.\",\"id\":\"7\",\"journal_abbrev\":\"Atlas of Protein Sequence and Structure (Data Section)\",\"journal_id_csd\":\"0435\",\"journal_id_issn\":\"0-912466-02-2\",\"journal_volume\":\"5\",\"page_first\":\"56\",\"rcsb_is_primary\":\"N\",\"rcsb_journal_abbrev\":\"Atlas Of Protein Sequence And Structure (data Section)\",\"year\":1972},{\"book_publisher\":\"National Biomedical Research Foundation, Silver Spring,Md.\",\"id\":\"8\",\"journal_abbrev\":\"Atlas of Protein Sequence and Structure (Data Section)\",\"journal_id_csd\":\"0435\",\"journal_id_issn\":\"0-912466-02-2\",\"journal_volume\":\"5\",\"page_first\":\"64\",\"rcsb_is_primary\":\"N\",\"rcsb_journal_abbrev\":\"Atlas Of Protein Sequence And Structure (data Section)\",\"year\":1972}],\"diffrn\":[{\"crystal_id\":\"1\",\"id\":\"1\"}],\"entry\":{\"id\":\"4HHB\"},\"exptl\":[{\"method\":\"X-RAY DIFFRACTION\"}],\"exptl_crystal\":[{\"density_matthews\":2.26,\"density_percent_sol\":45.48,\"id\":\"1\"}],\"pdbx_audit_revision_category\":[{\"category\":\"atom_site\",\"data_content_type\":\"Structure model\",\"ordinal\":1,\"revision_ordinal\":4},{\"category\":\"database_PDB_caveat\",\"data_content_type\":\"Structure model\",\"ordinal\":2,\"revision_ordinal\":4},{\"category\":\"entity\",\"data_content_type\":\"Structure model\",\"ordinal\":3,\"revision_ordinal\":4},{\"category\":\"entity_name_com\",\"data_content_type\":\"Structure model\",\"ordinal\":4,\"revision_ordinal\":4},{\"category\":\"entity_src_gen\",\"data_content_type\":\"Structure model\",\"ordinal\":5,\"revision_ordinal\":4},{\"category\":\"pdbx_database_status\",\"data_content_type\":\"Structure model\",\"ordinal\":6,\"revision_ordinal\":4},{\"category\":\"pdbx_validate_rmsd_angle\",\"data_content_type\":\"Structure model\",\"ordinal\":7,\"revision_ordinal\":4},{\"category\":\"pdbx_validate_rmsd_bond\",\"data_content_type\":\"Structure model\",\"ordinal\":8,\"revision_ordinal\":4},{\"category\":\"struct_ref\",\"data_content_type\":\"Structure model\",\"ordinal\":9,\"revision_ordinal\":4},{\"category\":\"struct_ref_seq\",\"data_content_type\":\"Structure model\",\"ordinal\":10,\"revision_ordinal\":4},{\"category\":\"atom_site\",\"data_content_type\":\"Structure model\",\"ordinal\":11,\"revision_ordinal\":5},{\"category\":\"pdbx_validate_rmsd_angle\",\"data_content_type\":\"Structure model\",\"ordinal\":12,\"revision_ordinal\":5},{\"category\":\"pdbx_validate_rmsd_bond\",\"data_content_type\":\"Structure model\",\"ordinal\":13,\"revision_ordinal\":5},{\"category\":\"struct_site\",\"data_content_type\":\"Structure model\",\"ordinal\":14,\"revision_ordinal\":5}],\"pdbx_audit_revision_details\":[{\"data_content_type\":\"Structure model\",\"ordinal\":1,\"provider\":\"repository\",\"revision_ordinal\":1,\"type\":\"Initial release\"}],\"pdbx_audit_revision_group\":[{\"data_content_type\":\"Structure model\",\"group\":\"Version format compliance\",\"ordinal\":1,\"revision_ordinal\":2},{\"data_content_type\":\"Structure model\",\"group\":\"Advisory\",\"ordinal\":2,\"revision_ordinal\":3},{\"data_content_type\":\"Structure model\",\"group\":\"Version format compliance\",\"ordinal\":3,\"revision_ordinal\":3},{\"data_content_type\":\"Structure model\",\"group\":\"Advisory\",\"ordinal\":4,\"revision_ordinal\":4},{\"data_content_type\":\"Structure model\",\"group\":\"Atomic model\",\"ordinal\":5,\"revision_ordinal\":4},{\"data_content_type\":\"Structure model\",\"group\":\"Data collection\",\"ordinal\":6,\"revision_ordinal\":4},{\"data_content_type\":\"Structure model\",\"group\":\"Database references\",\"ordinal\":7,\"revision_ordinal\":4},{\"data_content_type\":\"Structure model\",\"group\":\"Other\",\"ordinal\":8,\"revision_ordinal\":4},{\"data_content_type\":\"Structure model\",\"group\":\"Source and taxonomy\",\"ordinal\":9,\"revision_ordinal\":4},{\"data_content_type\":\"Structure model\",\"group\":\"Structure summary\",\"ordinal\":10,\"revision_ordinal\":4},{\"data_content_type\":\"Structure model\",\"group\":\"Atomic model\",\"ordinal\":11,\"revision_ordinal\":5},{\"data_content_type\":\"Structure model\",\"group\":\"Data collection\",\"ordinal\":12,\"revision_ordinal\":5},{\"data_content_type\":\"Structure model\",\"group\":\"Derived calculations\",\"ordinal\":13,\"revision_ordinal\":5}],\"pdbx_audit_revision_history\":[{\"data_content_type\":\"Structure model\",\"major_revision\":1,\"minor_revision\":0,\"ordinal\":1,\"revision_date\":\"1984-07-17T00:00:00+0000\"},{\"data_content_type\":\"Structure model\",\"major_revision\":1,\"minor_revision\":1,\"ordinal\":2,\"revision_date\":\"2008-03-03T00:00:00+0000\"},{\"data_content_type\":\"Structure model\",\"major_revision\":1,\"minor_revision\":2,\"ordinal\":3,\"revision_date\":\"2011-07-13T00:00:00+0000\"},{\"data_content_type\":\"Structure model\",\"major_revision\":2,\"minor_revision\":0,\"ordinal\":4,\"revision_date\":\"2020-06-17T00:00:00+0000\"},{\"data_content_type\":\"Structure model\",\"major_revision\":3,\"minor_revision\":0,\"ordinal\":5,\"revision_date\":\"2021-03-31T00:00:00+0000\"}],\"pdbx_audit_revision_item\":[{\"data_content_type\":\"Structure model\",\"item\":\"_atom_site.B_iso_or_equiv\",\"ordinal\":1,\"revision_ordinal\":4},{\"data_content_type\":\"Structure model\",\"item\":\"_atom_site.Cartn_x\",\"ordinal\":2,\"revision_ordinal\":4},{\"data_content_type\":\"Structure model\",\"item\":\"_atom_site.Cartn_y\",\"ordinal\":3,\"revision_ordinal\":4},{\"data_content_type\":\"Structure model\",\"item\":\"_atom_site.Cartn_z\",\"ordinal\":4,\"revision_ordinal\":4},{\"data_content_type\":\"Structure model\",\"item\":\"_entity.pdbx_description\",\"ordinal\":5,\"revision_ordinal\":4},{\"data_content_type\":\"Structure model\",\"item\":\"_entity_src_gen.gene_src_common_name\",\"ordinal\":6,\"revision_ordinal\":4},{\"data_content_type\":\"Structure model\",\"item\":\"_entity_src_gen.pdbx_beg_seq_num\",\"ordinal\":7,\"revision_ordinal\":4},{\"data_content_type\":\"Structure model\",\"item\":\"_entity_src_gen.pdbx_end_seq_num\",\"ordinal\":8,\"revision_ordinal\":4},{\"data_content_type\":\"Structure model\",\"item\":\"_entity_src_gen.pdbx_gene_src_gene\",\"ordinal\":9,\"revision_ordinal\":4},{\"data_content_type\":\"Structure model\",\"item\":\"_entity_src_gen.pdbx_seq_type\",\"ordinal\":10,\"revision_ordinal\":4},{\"data_content_type\":\"Structure model\",\"item\":\"_pdbx_database_status.process_site\",\"ordinal\":11,\"revision_ordinal\":4},{\"data_content_type\":\"Structure model\",\"item\":\"_pdbx_validate_rmsd_angle.angle_deviation\",\"ordinal\":12,\"revision_ordinal\":4},{\"data_content_type\":\"Structure model\",\"item\":\"_pdbx_validate_rmsd_angle.angle_value\",\"ordinal\":13,\"revision_ordinal\":4},{\"data_content_type\":\"Structure model\",\"item\":\"_pdbx_validate_rmsd_bond.bond_deviation\",\"ordinal\":14,\"revision_ordinal\":4},{\"data_content_type\":\"Structure model\",\"item\":\"_pdbx_validate_rmsd_bond.bond_value\",\"ordinal\":15,\"revision_ordinal\":4},{\"data_content_type\":\"Structure model\",\"item\":\"_struct_ref.pdbx_align_begin\",\"ordinal\":16,\"revision_ordinal\":4},{\"data_content_type\":\"Structure model\",\"item\":\"_struct_ref_seq.db_align_beg\",\"ordinal\":17,\"revision_ordinal\":4},{\"data_content_type\":\"Structure model\",\"item\":\"_struct_ref_seq.db_align_end\",\"ordinal\":18,\"revision_ordinal\":4},{\"data_content_type\":\"Structure model\",\"item\":\"_atom_site.B_iso_or_equiv\",\"ordinal\":19,\"revision_ordinal\":5},{\"data_content_type\":\"Structure model\",\"item\":\"_atom_site.Cartn_x\",\"ordinal\":20,\"revision_ordinal\":5},{\"data_content_type\":\"Structure model\",\"item\":\"_atom_site.Cartn_y\",\"ordinal\":21,\"revision_ordinal\":5},{\"data_content_type\":\"Structure model\",\"item\":\"_atom_site.Cartn_z\",\"ordinal\":22,\"revision_ordinal\":5},{\"data_content_type\":\"Structure model\",\"item\":\"_pdbx_validate_rmsd_bond.bond_deviation\",\"ordinal\":23,\"revision_ordinal\":5},{\"data_content_type\":\"Structure model\",\"item\":\"_pdbx_validate_rmsd_bond.bond_value\",\"ordinal\":24,\"revision_ordinal\":5},{\"data_content_type\":\"Structure model\",\"item\":\"_struct_site.pdbx_auth_asym_id\",\"ordinal\":25,\"revision_ordinal\":5},{\"data_content_type\":\"Structure model\",\"item\":\"_struct_site.pdbx_auth_comp_id\",\"ordinal\":26,\"revision_ordinal\":5},{\"data_content_type\":\"Structure model\",\"item\":\"_struct_site.pdbx_auth_seq_id\",\"ordinal\":27,\"revision_ordinal\":5}],\"pdbx_database_pdbobs_spr\":[{\"date\":\"1984-07-17T00:00:00+0000\",\"id\":\"SPRSDE\",\"pdb_id\":\"4HHB\",\"replace_pdb_id\":\"1HHB\"}],\"pdbx_database_related\":[{\"content_type\":\"unspecified\",\"db_id\":\"2HHB\",\"db_name\":\"PDB\",\"details\":\"REFINED BY THE METHOD OF JACK AND LEVITT.  THIS\\n        ENTRY PRESENTS THE BEST ESTIMATE OF THE\\n        COORDINATES.\"},{\"content_type\":\"unspecified\",\"db_id\":\"3HHB\",\"db_name\":\"PDB\",\"details\":\"SYMMETRY AVERAGED ABOUT THE (NON-CRYSTALLOGRAPHIC)\\n        MOLECULAR AXIS AND THEN RE-REGULARIZED BY THE\\n        ENERGY REFINEMENT METHOD OF LEVITT.  THIS ENTRY\\n        PRESENTS COORDINATES THAT ARE ADEQUATE FOR MOST\\n        PURPOSES, SUCH AS COMPARISON WITH OTHER STRUCTURES.\"},{\"content_type\":\"unspecified\",\"db_id\":\"1GLI\",\"db_name\":\"PDB\"}],\"pdbx_database_status\":{\"pdb_format_compatible\":\"Y\",\"process_site\":\"BNL\",\"recvd_initial_deposition_date\":\"1984-03-07T00:00:00+0000\",\"status_code\":\"REL\"},\"pdbx_vrpt_summary\":{\"absolute_percentile_clashscore\":0.0,\"absolute_percentile_percent_ramachandran_outliers\":12.8,\"absolute_percentile_percent_rotamer_outliers\":9.9,\"angles_rmsz\":7.11,\"attempted_validation_steps\":\"molprobity,validation-pack,mogul,buster-report,percentiles,writexml\",\"bonds_rmsz\":9.69,\"clashscore\":141.0,\"high_resol_relative_percentile_clashscore\":1.72,\"high_resol_relative_percentile_percent_ramachandran_outliers\":1.72,\"high_resol_relative_percentile_percent_rotamer_outliers\":1.72,\"ligands_for_buster_report\":\"yes\",\"low_resol_relative_percentile_clashscore\":1.76,\"low_resol_relative_percentile_percent_ramachandran_outliers\":1.76,\"low_resol_relative_percentile_percent_rotamer_outliers\":1.76,\"num_hreduce\":4456.0,\"num_pdbids_absolute_percentile_clashscore\":141614,\"num_pdbids_absolute_percentile_percent_ramachandran_outliers\":138981,\"num_pdbids_absolute_percentile_percent_rotamer_outliers\":138945,\"num_pdbids_relative_percentile_clashscore\":3923,\"num_pdbids_relative_percentile_percent_ramachandran_outliers\":3878,\"num_pdbids_relative_percentile_percent_rotamer_outliers\":3878,\"num_angles_rmsz\":6114,\"num_bonds_rmsz\":4500,\"percent_ramachandran_outliers\":1.24,\"percent_rotamer_outliers\":8.66,\"percentilebins\":[\"all\",\"1.74\",\"xray\"],\"protein_dnarnaentities\":\"1,2\",\"relative_percentile_clashscore\":0.0,\"relative_percentile_percent_ramachandran_outliers\":2.6,\"relative_percentile_percent_rotamer_outliers\":1.1,\"report_creation_date\":\"Mar 23, 2021 -- 08:01 PM EDT\",\"restypes_notchecked_for_bond_angle_geometry\":[\"HEM\",\"PO4\"],\"pdbr\":0.14,\"pdbdeposition_date\":\"1984-03-07T00:00:00+0000\",\"pdbresolution\":1.74,\"pdbrevision_date\":\"2021-03-31T00:00:00+0000\",\"pdbrevision_number\":5.0},\"rcsb_accession_info\":{\"deposit_date\":\"1984-03-07T00:00:00+0000\",\"has_released_experimental_data\":\"N\",\"initial_release_date\":\"1984-07-17T00:00:00+0000\",\"major_revision\":3,\"minor_revision\":0,\"revision_date\":\"2021-03-31T00:00:00+0000\",\"status_code\":\"REL\"},\"rcsb_entry_container_identifiers\":{\"assembly_ids\":[\"1\"],\"entity_ids\":[\"1\",\"2\",\"3\",\"4\",\"5\"],\"entry_id\":\"4HHB\",\"model_ids\":[1],\"non_polymer_entity_ids\":[\"3\",\"4\"],\"polymer_entity_ids\":[\"1\",\"2\"],\"rcsb_id\":\"4HHB\",\"pubmed_id\":6726807},\"rcsb_entry_info\":{\"assembly_count\":1,\"branched_entity_count\":0,\"cis_peptide_count\":0,\"deposited_atom_count\":4779,\"deposited_hydrogen_atom_count\":0,\"deposited_model_count\":1,\"deposited_modeled_polymer_monomer_count\":574,\"deposited_nonpolymer_entity_instance_count\":6,\"deposited_polymer_entity_instance_count\":4,\"deposited_polymer_monomer_count\":574,\"deposited_solvent_atom_count\":221,\"deposited_unmodeled_polymer_monomer_count\":0,\"disulfide_bond_count\":0,\"entity_count\":5,\"experimental_method\":\"X-ray\",\"experimental_method_count\":1,\"inter_mol_covalent_bond_count\":0,\"inter_mol_metalic_bond_count\":4,\"molecular_weight\":64.74,\"na_polymer_entity_types\":\"Other\",\"nonpolymer_bound_components\":[\"HEM\"],\"nonpolymer_entity_count\":2,\"nonpolymer_molecular_weight_maximum\":0.62,\"nonpolymer_molecular_weight_minimum\":0.09,\"polymer_composition\":\"heteromeric protein\",\"polymer_entity_count\":2,\"polymer_entity_count_dna\":0,\"polymer_entity_count_rna\":0,\"polymer_entity_count_nucleic_acid\":0,\"polymer_entity_count_nucleic_acid_hybrid\":0,\"polymer_entity_count_protein\":2,\"polymer_entity_taxonomy_count\":2,\"polymer_molecular_weight_maximum\":15.89,\"polymer_molecular_weight_minimum\":15.15,\"polymer_monomer_count_maximum\":146,\"polymer_monomer_count_minimum\":141,\"resolution_combined\":[1.74],\"selected_polymer_entity_types\":\"Protein (only)\",\"solvent_entity_count\":1,\"diffrn_resolution_high\":{\"provenance_source\":\"From refinement resolution cutoff\",\"value\":1.74}},\"rcsb_primary_citation\":{\"country\":\"UK\",\"id\":\"primary\",\"journal_abbrev\":\"J.Mol.Biol.\",\"journal_id_astm\":\"JMOBAK\",\"journal_id_csd\":\"0070\",\"journal_id_issn\":\"0022-2836\",\"journal_volume\":\"175\",\"page_first\":\"159\",\"page_last\":\"174\",\"pdbx_database_id_doi\":\"10.1016/0022-2836(84)90472-8\",\"pdbx_database_id_pub_med\":6726807,\"rcsb_orcididentifiers\":[\"?\",\"?\",\"?\",\"?\"],\"rcsb_authors\":[\"Fermi, G.\",\"Perutz, M.F.\",\"Shaanan, B.\",\"Fourme, R.\"],\"rcsb_journal_abbrev\":\"J Mol Biol\",\"title\":\"The crystal structure of human deoxyhaemoglobin at 1.74 A resolution\",\"year\":1984},\"refine\":[{\"details\":\"THE COORDINATES GIVEN HERE ARE IN THE ORTHOGONAL ANGSTROM\\nSYSTEM STANDARD FOR HEMOGLOBINS. THE Y AXIS IS THE\\n(NON CRYSTALLOGRAPHIC) MOLECULAR DIAD AND THE X AXIS IS THE\\nPSEUDO DIAD WHICH RELATES THE ALPHA-1 AND BETA-1 CHAINS.\\nTHE TRANSFORMATION GIVEN IN THE *MTRIX* RECORDS BELOW\\nWILL GENERATE COORDINATES FOR THE *C* AND *D* CHAINS FROM\\nTHE *A* AND *B* CHAINS RESPECTIVELY.\",\"ls_rfactor_rwork\":0.135,\"ls_dres_high\":1.74,\"pdbx_diffrn_id\":[\"1\"],\"pdbx_refine_id\":\"X-RAY DIFFRACTION\"}],\"refine_hist\":[{\"cycle_id\":\"LAST\",\"number_atoms_solvent\":221,\"number_atoms_total\":4779,\"pdbx_number_atoms_ligand\":174,\"pdbx_number_atoms_nucleic_acid\":0,\"pdbx_number_atoms_protein\":4384,\"pdbx_refine_id\":\"X-RAY DIFFRACTION\",\"dres_high\":1.74}],\"struct\":{\"pdbx_descriptor\":\"HEMOGLOBIN (DEOXY)\",\"title\":\"THE CRYSTAL STRUCTURE OF HUMAN DEOXYHAEMOGLOBIN AT 1.74 ANGSTROMS RESOLUTION\"},\"struct_keywords\":{\"pdbx_keywords\":\"OXYGEN TRANSPORT\",\"text\":\"OXYGEN TRANSPORT\"},\"symmetry\":{\"int_tables_number\":4,\"space_group_name_hm\":\"P 1 21 1\"},\"rcsb_id\":\"4HHB\"}\n"
     ]
    }
   ],
   "source": [
    "response = requests.get('https://data.rcsb.org/rest/v1/core/entry/4hhb')\n",
    "print(response.text)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {
    "slideshow": {
     "slide_type": "slide"
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'audit_author': [{'name': 'Fermi, G.', 'pdbx_ordinal': 1},\n",
       "  {'name': 'Perutz, M.F.', 'pdbx_ordinal': 2}],\n",
       " 'cell': {'angle_alpha': 90.0,\n",
       "  'angle_beta': 99.34,\n",
       "  'angle_gamma': 90.0,\n",
       "  'length_a': 63.15,\n",
       "  'length_b': 83.59,\n",
       "  'length_c': 53.8,\n",
       "  'zpdb': 4},\n",
       " 'citation': [{'country': 'UK',\n",
       "   'id': 'primary',\n",
       "   'journal_abbrev': 'J.Mol.Biol.',\n",
       "   'journal_id_astm': 'JMOBAK',\n",
       "   'journal_id_csd': '0070',\n",
       "   'journal_id_issn': '0022-2836',\n",
       "   'journal_volume': '175',\n",
       "   'page_first': '159',\n",
       "   'page_last': '174',\n",
       "   'pdbx_database_id_doi': '10.1016/0022-2836(84)90472-8',\n",
       "   'pdbx_database_id_pub_med': 6726807,\n",
       "   'rcsb_authors': ['Fermi, G.', 'Perutz, M.F.', 'Shaanan, B.', 'Fourme, R.'],\n",
       "   'rcsb_is_primary': 'Y',\n",
       "   'rcsb_journal_abbrev': 'J Mol Biol',\n",
       "   'title': 'The crystal structure of human deoxyhaemoglobin at 1.74 A resolution',\n",
       "   'year': 1984},\n",
       "  {'country': 'UK',\n",
       "   'id': '1',\n",
       "   'journal_abbrev': 'Nature',\n",
       "   'journal_id_astm': 'NATUAS',\n",
       "   'journal_id_csd': '0006',\n",
       "   'journal_id_issn': '0028-0836',\n",
       "   'journal_volume': '295',\n",
       "   'page_first': '535',\n",
       "   'rcsb_authors': ['Perutz, M.F.',\n",
       "    'Hasnain, S.S.',\n",
       "    'Duke, P.J.',\n",
       "    'Sessler, J.L.',\n",
       "    'Hahn, J.E.'],\n",
       "   'rcsb_is_primary': 'N',\n",
       "   'rcsb_journal_abbrev': 'Nature',\n",
       "   'title': 'Stereochemistry of Iron in Deoxyhaemoglobin',\n",
       "   'year': 1982},\n",
       "  {'country': 'US',\n",
       "   'id': '3',\n",
       "   'journal_abbrev': 'Annu.Rev.Biochem.',\n",
       "   'journal_id_astm': 'ARBOAW',\n",
       "   'journal_id_csd': '0413',\n",
       "   'journal_id_issn': '0066-4154',\n",
       "   'journal_volume': '48',\n",
       "   'page_first': '327',\n",
       "   'rcsb_authors': ['Perutz, M.F.'],\n",
       "   'rcsb_is_primary': 'N',\n",
       "   'rcsb_journal_abbrev': 'Annu Rev Biochem',\n",
       "   'title': 'Regulation of Oxygen Affinity of Hemoglobin. Influence of Structure of the Globin on the Heme Iron',\n",
       "   'year': 1979},\n",
       "  {'country': 'UK',\n",
       "   'id': '4',\n",
       "   'journal_abbrev': 'J.Mol.Biol.',\n",
       "   'journal_id_astm': 'JMOBAK',\n",
       "   'journal_id_csd': '0070',\n",
       "   'journal_id_issn': '0022-2836',\n",
       "   'journal_volume': '100',\n",
       "   'page_first': '3',\n",
       "   'rcsb_authors': ['Teneyck, L.F.', 'Arnone, A.'],\n",
       "   'rcsb_is_primary': 'N',\n",
       "   'rcsb_journal_abbrev': 'J Mol Biol',\n",
       "   'title': 'Three-Dimensional Fourier Synthesis of Human Deoxyhemoglobin at 2.5 Angstroms Resolution, I.X-Ray Analysis',\n",
       "   'year': 1976},\n",
       "  {'country': 'UK',\n",
       "   'id': '5',\n",
       "   'journal_abbrev': 'J.Mol.Biol.',\n",
       "   'journal_id_astm': 'JMOBAK',\n",
       "   'journal_id_csd': '0070',\n",
       "   'journal_id_issn': '0022-2836',\n",
       "   'journal_volume': '97',\n",
       "   'page_first': '237',\n",
       "   'rcsb_authors': ['Fermi, G.'],\n",
       "   'rcsb_is_primary': 'N',\n",
       "   'rcsb_journal_abbrev': 'J Mol Biol',\n",
       "   'title': 'Three-Dimensional Fourier Synthesis of Human Deoxyhaemoglobin at 2.5 Angstroms Resolution, Refinement of the Atomic Model',\n",
       "   'year': 1975},\n",
       "  {'country': 'UK',\n",
       "   'id': '6',\n",
       "   'journal_abbrev': 'Nature',\n",
       "   'journal_id_astm': 'NATUAS',\n",
       "   'journal_id_csd': '0006',\n",
       "   'journal_id_issn': '0028-0836',\n",
       "   'journal_volume': '228',\n",
       "   'page_first': '516',\n",
       "   'rcsb_authors': ['Muirhead, H.', 'Greer, J.'],\n",
       "   'rcsb_is_primary': 'N',\n",
       "   'rcsb_journal_abbrev': 'Nature',\n",
       "   'title': 'Three-Dimensional Fourier Synthesis of Human Deoxyhaemoglobin at 3.5 Angstroms Resolution',\n",
       "   'year': 1970},\n",
       "  {'book_publisher': 'Oxford University Press',\n",
       "   'id': '2',\n",
       "   'journal_abbrev': 'Haemoglobin and Myoglobin. Atlas of Molecular Structures in Biology',\n",
       "   'journal_id_csd': '0986',\n",
       "   'journal_id_issn': '0-19-854706-4',\n",
       "   'journal_volume': '2',\n",
       "   'rcsb_authors': ['Fermi, G.', 'Perutz, M.F.'],\n",
       "   'rcsb_is_primary': 'N',\n",
       "   'rcsb_journal_abbrev': 'Haemoglobin And Myoglobin Atlas Of Molecular Structures In Biology',\n",
       "   'year': 1981},\n",
       "  {'book_publisher': 'National Biomedical Research Foundation, Silver Spring,Md.',\n",
       "   'id': '7',\n",
       "   'journal_abbrev': 'Atlas of Protein Sequence and Structure (Data Section)',\n",
       "   'journal_id_csd': '0435',\n",
       "   'journal_id_issn': '0-912466-02-2',\n",
       "   'journal_volume': '5',\n",
       "   'page_first': '56',\n",
       "   'rcsb_is_primary': 'N',\n",
       "   'rcsb_journal_abbrev': 'Atlas Of Protein Sequence And Structure (data Section)',\n",
       "   'year': 1972},\n",
       "  {'book_publisher': 'National Biomedical Research Foundation, Silver Spring,Md.',\n",
       "   'id': '8',\n",
       "   'journal_abbrev': 'Atlas of Protein Sequence and Structure (Data Section)',\n",
       "   'journal_id_csd': '0435',\n",
       "   'journal_id_issn': '0-912466-02-2',\n",
       "   'journal_volume': '5',\n",
       "   'page_first': '64',\n",
       "   'rcsb_is_primary': 'N',\n",
       "   'rcsb_journal_abbrev': 'Atlas Of Protein Sequence And Structure (data Section)',\n",
       "   'year': 1972}],\n",
       " 'diffrn': [{'crystal_id': '1', 'id': '1'}],\n",
       " 'entry': {'id': '4HHB'},\n",
       " 'exptl': [{'method': 'X-RAY DIFFRACTION'}],\n",
       " 'exptl_crystal': [{'density_matthews': 2.26,\n",
       "   'density_percent_sol': 45.48,\n",
       "   'id': '1'}],\n",
       " 'pdbx_audit_revision_category': [{'category': 'atom_site',\n",
       "   'data_content_type': 'Structure model',\n",
       "   'ordinal': 1,\n",
       "   'revision_ordinal': 4},\n",
       "  {'category': 'database_PDB_caveat',\n",
       "   'data_content_type': 'Structure model',\n",
       "   'ordinal': 2,\n",
       "   'revision_ordinal': 4},\n",
       "  {'category': 'entity',\n",
       "   'data_content_type': 'Structure model',\n",
       "   'ordinal': 3,\n",
       "   'revision_ordinal': 4},\n",
       "  {'category': 'entity_name_com',\n",
       "   'data_content_type': 'Structure model',\n",
       "   'ordinal': 4,\n",
       "   'revision_ordinal': 4},\n",
       "  {'category': 'entity_src_gen',\n",
       "   'data_content_type': 'Structure model',\n",
       "   'ordinal': 5,\n",
       "   'revision_ordinal': 4},\n",
       "  {'category': 'pdbx_database_status',\n",
       "   'data_content_type': 'Structure model',\n",
       "   'ordinal': 6,\n",
       "   'revision_ordinal': 4},\n",
       "  {'category': 'pdbx_validate_rmsd_angle',\n",
       "   'data_content_type': 'Structure model',\n",
       "   'ordinal': 7,\n",
       "   'revision_ordinal': 4},\n",
       "  {'category': 'pdbx_validate_rmsd_bond',\n",
       "   'data_content_type': 'Structure model',\n",
       "   'ordinal': 8,\n",
       "   'revision_ordinal': 4},\n",
       "  {'category': 'struct_ref',\n",
       "   'data_content_type': 'Structure model',\n",
       "   'ordinal': 9,\n",
       "   'revision_ordinal': 4},\n",
       "  {'category': 'struct_ref_seq',\n",
       "   'data_content_type': 'Structure model',\n",
       "   'ordinal': 10,\n",
       "   'revision_ordinal': 4},\n",
       "  {'category': 'atom_site',\n",
       "   'data_content_type': 'Structure model',\n",
       "   'ordinal': 11,\n",
       "   'revision_ordinal': 5},\n",
       "  {'category': 'pdbx_validate_rmsd_angle',\n",
       "   'data_content_type': 'Structure model',\n",
       "   'ordinal': 12,\n",
       "   'revision_ordinal': 5},\n",
       "  {'category': 'pdbx_validate_rmsd_bond',\n",
       "   'data_content_type': 'Structure model',\n",
       "   'ordinal': 13,\n",
       "   'revision_ordinal': 5},\n",
       "  {'category': 'struct_site',\n",
       "   'data_content_type': 'Structure model',\n",
       "   'ordinal': 14,\n",
       "   'revision_ordinal': 5}],\n",
       " 'pdbx_audit_revision_details': [{'data_content_type': 'Structure model',\n",
       "   'ordinal': 1,\n",
       "   'provider': 'repository',\n",
       "   'revision_ordinal': 1,\n",
       "   'type': 'Initial release'}],\n",
       " 'pdbx_audit_revision_group': [{'data_content_type': 'Structure model',\n",
       "   'group': 'Version format compliance',\n",
       "   'ordinal': 1,\n",
       "   'revision_ordinal': 2},\n",
       "  {'data_content_type': 'Structure model',\n",
       "   'group': 'Advisory',\n",
       "   'ordinal': 2,\n",
       "   'revision_ordinal': 3},\n",
       "  {'data_content_type': 'Structure model',\n",
       "   'group': 'Version format compliance',\n",
       "   'ordinal': 3,\n",
       "   'revision_ordinal': 3},\n",
       "  {'data_content_type': 'Structure model',\n",
       "   'group': 'Advisory',\n",
       "   'ordinal': 4,\n",
       "   'revision_ordinal': 4},\n",
       "  {'data_content_type': 'Structure model',\n",
       "   'group': 'Atomic model',\n",
       "   'ordinal': 5,\n",
       "   'revision_ordinal': 4},\n",
       "  {'data_content_type': 'Structure model',\n",
       "   'group': 'Data collection',\n",
       "   'ordinal': 6,\n",
       "   'revision_ordinal': 4},\n",
       "  {'data_content_type': 'Structure model',\n",
       "   'group': 'Database references',\n",
       "   'ordinal': 7,\n",
       "   'revision_ordinal': 4},\n",
       "  {'data_content_type': 'Structure model',\n",
       "   'group': 'Other',\n",
       "   'ordinal': 8,\n",
       "   'revision_ordinal': 4},\n",
       "  {'data_content_type': 'Structure model',\n",
       "   'group': 'Source and taxonomy',\n",
       "   'ordinal': 9,\n",
       "   'revision_ordinal': 4},\n",
       "  {'data_content_type': 'Structure model',\n",
       "   'group': 'Structure summary',\n",
       "   'ordinal': 10,\n",
       "   'revision_ordinal': 4},\n",
       "  {'data_content_type': 'Structure model',\n",
       "   'group': 'Atomic model',\n",
       "   'ordinal': 11,\n",
       "   'revision_ordinal': 5},\n",
       "  {'data_content_type': 'Structure model',\n",
       "   'group': 'Data collection',\n",
       "   'ordinal': 12,\n",
       "   'revision_ordinal': 5},\n",
       "  {'data_content_type': 'Structure model',\n",
       "   'group': 'Derived calculations',\n",
       "   'ordinal': 13,\n",
       "   'revision_ordinal': 5}],\n",
       " 'pdbx_audit_revision_history': [{'data_content_type': 'Structure model',\n",
       "   'major_revision': 1,\n",
       "   'minor_revision': 0,\n",
       "   'ordinal': 1,\n",
       "   'revision_date': '1984-07-17T00:00:00+0000'},\n",
       "  {'data_content_type': 'Structure model',\n",
       "   'major_revision': 1,\n",
       "   'minor_revision': 1,\n",
       "   'ordinal': 2,\n",
       "   'revision_date': '2008-03-03T00:00:00+0000'},\n",
       "  {'data_content_type': 'Structure model',\n",
       "   'major_revision': 1,\n",
       "   'minor_revision': 2,\n",
       "   'ordinal': 3,\n",
       "   'revision_date': '2011-07-13T00:00:00+0000'},\n",
       "  {'data_content_type': 'Structure model',\n",
       "   'major_revision': 2,\n",
       "   'minor_revision': 0,\n",
       "   'ordinal': 4,\n",
       "   'revision_date': '2020-06-17T00:00:00+0000'},\n",
       "  {'data_content_type': 'Structure model',\n",
       "   'major_revision': 3,\n",
       "   'minor_revision': 0,\n",
       "   'ordinal': 5,\n",
       "   'revision_date': '2021-03-31T00:00:00+0000'}],\n",
       " 'pdbx_audit_revision_item': [{'data_content_type': 'Structure model',\n",
       "   'item': '_atom_site.B_iso_or_equiv',\n",
       "   'ordinal': 1,\n",
       "   'revision_ordinal': 4},\n",
       "  {'data_content_type': 'Structure model',\n",
       "   'item': '_atom_site.Cartn_x',\n",
       "   'ordinal': 2,\n",
       "   'revision_ordinal': 4},\n",
       "  {'data_content_type': 'Structure model',\n",
       "   'item': '_atom_site.Cartn_y',\n",
       "   'ordinal': 3,\n",
       "   'revision_ordinal': 4},\n",
       "  {'data_content_type': 'Structure model',\n",
       "   'item': '_atom_site.Cartn_z',\n",
       "   'ordinal': 4,\n",
       "   'revision_ordinal': 4},\n",
       "  {'data_content_type': 'Structure model',\n",
       "   'item': '_entity.pdbx_description',\n",
       "   'ordinal': 5,\n",
       "   'revision_ordinal': 4},\n",
       "  {'data_content_type': 'Structure model',\n",
       "   'item': '_entity_src_gen.gene_src_common_name',\n",
       "   'ordinal': 6,\n",
       "   'revision_ordinal': 4},\n",
       "  {'data_content_type': 'Structure model',\n",
       "   'item': '_entity_src_gen.pdbx_beg_seq_num',\n",
       "   'ordinal': 7,\n",
       "   'revision_ordinal': 4},\n",
       "  {'data_content_type': 'Structure model',\n",
       "   'item': '_entity_src_gen.pdbx_end_seq_num',\n",
       "   'ordinal': 8,\n",
       "   'revision_ordinal': 4},\n",
       "  {'data_content_type': 'Structure model',\n",
       "   'item': '_entity_src_gen.pdbx_gene_src_gene',\n",
       "   'ordinal': 9,\n",
       "   'revision_ordinal': 4},\n",
       "  {'data_content_type': 'Structure model',\n",
       "   'item': '_entity_src_gen.pdbx_seq_type',\n",
       "   'ordinal': 10,\n",
       "   'revision_ordinal': 4},\n",
       "  {'data_content_type': 'Structure model',\n",
       "   'item': '_pdbx_database_status.process_site',\n",
       "   'ordinal': 11,\n",
       "   'revision_ordinal': 4},\n",
       "  {'data_content_type': 'Structure model',\n",
       "   'item': '_pdbx_validate_rmsd_angle.angle_deviation',\n",
       "   'ordinal': 12,\n",
       "   'revision_ordinal': 4},\n",
       "  {'data_content_type': 'Structure model',\n",
       "   'item': '_pdbx_validate_rmsd_angle.angle_value',\n",
       "   'ordinal': 13,\n",
       "   'revision_ordinal': 4},\n",
       "  {'data_content_type': 'Structure model',\n",
       "   'item': '_pdbx_validate_rmsd_bond.bond_deviation',\n",
       "   'ordinal': 14,\n",
       "   'revision_ordinal': 4},\n",
       "  {'data_content_type': 'Structure model',\n",
       "   'item': '_pdbx_validate_rmsd_bond.bond_value',\n",
       "   'ordinal': 15,\n",
       "   'revision_ordinal': 4},\n",
       "  {'data_content_type': 'Structure model',\n",
       "   'item': '_struct_ref.pdbx_align_begin',\n",
       "   'ordinal': 16,\n",
       "   'revision_ordinal': 4},\n",
       "  {'data_content_type': 'Structure model',\n",
       "   'item': '_struct_ref_seq.db_align_beg',\n",
       "   'ordinal': 17,\n",
       "   'revision_ordinal': 4},\n",
       "  {'data_content_type': 'Structure model',\n",
       "   'item': '_struct_ref_seq.db_align_end',\n",
       "   'ordinal': 18,\n",
       "   'revision_ordinal': 4},\n",
       "  {'data_content_type': 'Structure model',\n",
       "   'item': '_atom_site.B_iso_or_equiv',\n",
       "   'ordinal': 19,\n",
       "   'revision_ordinal': 5},\n",
       "  {'data_content_type': 'Structure model',\n",
       "   'item': '_atom_site.Cartn_x',\n",
       "   'ordinal': 20,\n",
       "   'revision_ordinal': 5},\n",
       "  {'data_content_type': 'Structure model',\n",
       "   'item': '_atom_site.Cartn_y',\n",
       "   'ordinal': 21,\n",
       "   'revision_ordinal': 5},\n",
       "  {'data_content_type': 'Structure model',\n",
       "   'item': '_atom_site.Cartn_z',\n",
       "   'ordinal': 22,\n",
       "   'revision_ordinal': 5},\n",
       "  {'data_content_type': 'Structure model',\n",
       "   'item': '_pdbx_validate_rmsd_bond.bond_deviation',\n",
       "   'ordinal': 23,\n",
       "   'revision_ordinal': 5},\n",
       "  {'data_content_type': 'Structure model',\n",
       "   'item': '_pdbx_validate_rmsd_bond.bond_value',\n",
       "   'ordinal': 24,\n",
       "   'revision_ordinal': 5},\n",
       "  {'data_content_type': 'Structure model',\n",
       "   'item': '_struct_site.pdbx_auth_asym_id',\n",
       "   'ordinal': 25,\n",
       "   'revision_ordinal': 5},\n",
       "  {'data_content_type': 'Structure model',\n",
       "   'item': '_struct_site.pdbx_auth_comp_id',\n",
       "   'ordinal': 26,\n",
       "   'revision_ordinal': 5},\n",
       "  {'data_content_type': 'Structure model',\n",
       "   'item': '_struct_site.pdbx_auth_seq_id',\n",
       "   'ordinal': 27,\n",
       "   'revision_ordinal': 5}],\n",
       " 'pdbx_database_pdbobs_spr': [{'date': '1984-07-17T00:00:00+0000',\n",
       "   'id': 'SPRSDE',\n",
       "   'pdb_id': '4HHB',\n",
       "   'replace_pdb_id': '1HHB'}],\n",
       " 'pdbx_database_related': [{'content_type': 'unspecified',\n",
       "   'db_id': '2HHB',\n",
       "   'db_name': 'PDB',\n",
       "   'details': 'REFINED BY THE METHOD OF JACK AND LEVITT.  THIS\\n        ENTRY PRESENTS THE BEST ESTIMATE OF THE\\n        COORDINATES.'},\n",
       "  {'content_type': 'unspecified',\n",
       "   'db_id': '3HHB',\n",
       "   'db_name': 'PDB',\n",
       "   'details': 'SYMMETRY AVERAGED ABOUT THE (NON-CRYSTALLOGRAPHIC)\\n        MOLECULAR AXIS AND THEN RE-REGULARIZED BY THE\\n        ENERGY REFINEMENT METHOD OF LEVITT.  THIS ENTRY\\n        PRESENTS COORDINATES THAT ARE ADEQUATE FOR MOST\\n        PURPOSES, SUCH AS COMPARISON WITH OTHER STRUCTURES.'},\n",
       "  {'content_type': 'unspecified', 'db_id': '1GLI', 'db_name': 'PDB'}],\n",
       " 'pdbx_database_status': {'pdb_format_compatible': 'Y',\n",
       "  'process_site': 'BNL',\n",
       "  'recvd_initial_deposition_date': '1984-03-07T00:00:00+0000',\n",
       "  'status_code': 'REL'},\n",
       " 'pdbx_vrpt_summary': {'absolute_percentile_clashscore': 0.0,\n",
       "  'absolute_percentile_percent_ramachandran_outliers': 12.8,\n",
       "  'absolute_percentile_percent_rotamer_outliers': 9.9,\n",
       "  'angles_rmsz': 7.11,\n",
       "  'attempted_validation_steps': 'molprobity,validation-pack,mogul,buster-report,percentiles,writexml',\n",
       "  'bonds_rmsz': 9.69,\n",
       "  'clashscore': 141.0,\n",
       "  'high_resol_relative_percentile_clashscore': 1.72,\n",
       "  'high_resol_relative_percentile_percent_ramachandran_outliers': 1.72,\n",
       "  'high_resol_relative_percentile_percent_rotamer_outliers': 1.72,\n",
       "  'ligands_for_buster_report': 'yes',\n",
       "  'low_resol_relative_percentile_clashscore': 1.76,\n",
       "  'low_resol_relative_percentile_percent_ramachandran_outliers': 1.76,\n",
       "  'low_resol_relative_percentile_percent_rotamer_outliers': 1.76,\n",
       "  'num_hreduce': 4456.0,\n",
       "  'num_pdbids_absolute_percentile_clashscore': 141614,\n",
       "  'num_pdbids_absolute_percentile_percent_ramachandran_outliers': 138981,\n",
       "  'num_pdbids_absolute_percentile_percent_rotamer_outliers': 138945,\n",
       "  'num_pdbids_relative_percentile_clashscore': 3923,\n",
       "  'num_pdbids_relative_percentile_percent_ramachandran_outliers': 3878,\n",
       "  'num_pdbids_relative_percentile_percent_rotamer_outliers': 3878,\n",
       "  'num_angles_rmsz': 6114,\n",
       "  'num_bonds_rmsz': 4500,\n",
       "  'percent_ramachandran_outliers': 1.24,\n",
       "  'percent_rotamer_outliers': 8.66,\n",
       "  'percentilebins': ['all', '1.74', 'xray'],\n",
       "  'protein_dnarnaentities': '1,2',\n",
       "  'relative_percentile_clashscore': 0.0,\n",
       "  'relative_percentile_percent_ramachandran_outliers': 2.6,\n",
       "  'relative_percentile_percent_rotamer_outliers': 1.1,\n",
       "  'report_creation_date': 'Mar 23, 2021 -- 08:01 PM EDT',\n",
       "  'restypes_notchecked_for_bond_angle_geometry': ['HEM', 'PO4'],\n",
       "  'pdbr': 0.14,\n",
       "  'pdbdeposition_date': '1984-03-07T00:00:00+0000',\n",
       "  'pdbresolution': 1.74,\n",
       "  'pdbrevision_date': '2021-03-31T00:00:00+0000',\n",
       "  'pdbrevision_number': 5.0},\n",
       " 'rcsb_accession_info': {'deposit_date': '1984-03-07T00:00:00+0000',\n",
       "  'has_released_experimental_data': 'N',\n",
       "  'initial_release_date': '1984-07-17T00:00:00+0000',\n",
       "  'major_revision': 3,\n",
       "  'minor_revision': 0,\n",
       "  'revision_date': '2021-03-31T00:00:00+0000',\n",
       "  'status_code': 'REL'},\n",
       " 'rcsb_entry_container_identifiers': {'assembly_ids': ['1'],\n",
       "  'entity_ids': ['1', '2', '3', '4', '5'],\n",
       "  'entry_id': '4HHB',\n",
       "  'model_ids': [1],\n",
       "  'non_polymer_entity_ids': ['3', '4'],\n",
       "  'polymer_entity_ids': ['1', '2'],\n",
       "  'rcsb_id': '4HHB',\n",
       "  'pubmed_id': 6726807},\n",
       " 'rcsb_entry_info': {'assembly_count': 1,\n",
       "  'branched_entity_count': 0,\n",
       "  'cis_peptide_count': 0,\n",
       "  'deposited_atom_count': 4779,\n",
       "  'deposited_hydrogen_atom_count': 0,\n",
       "  'deposited_model_count': 1,\n",
       "  'deposited_modeled_polymer_monomer_count': 574,\n",
       "  'deposited_nonpolymer_entity_instance_count': 6,\n",
       "  'deposited_polymer_entity_instance_count': 4,\n",
       "  'deposited_polymer_monomer_count': 574,\n",
       "  'deposited_solvent_atom_count': 221,\n",
       "  'deposited_unmodeled_polymer_monomer_count': 0,\n",
       "  'disulfide_bond_count': 0,\n",
       "  'entity_count': 5,\n",
       "  'experimental_method': 'X-ray',\n",
       "  'experimental_method_count': 1,\n",
       "  'inter_mol_covalent_bond_count': 0,\n",
       "  'inter_mol_metalic_bond_count': 4,\n",
       "  'molecular_weight': 64.74,\n",
       "  'na_polymer_entity_types': 'Other',\n",
       "  'nonpolymer_bound_components': ['HEM'],\n",
       "  'nonpolymer_entity_count': 2,\n",
       "  'nonpolymer_molecular_weight_maximum': 0.62,\n",
       "  'nonpolymer_molecular_weight_minimum': 0.09,\n",
       "  'polymer_composition': 'heteromeric protein',\n",
       "  'polymer_entity_count': 2,\n",
       "  'polymer_entity_count_dna': 0,\n",
       "  'polymer_entity_count_rna': 0,\n",
       "  'polymer_entity_count_nucleic_acid': 0,\n",
       "  'polymer_entity_count_nucleic_acid_hybrid': 0,\n",
       "  'polymer_entity_count_protein': 2,\n",
       "  'polymer_entity_taxonomy_count': 2,\n",
       "  'polymer_molecular_weight_maximum': 15.89,\n",
       "  'polymer_molecular_weight_minimum': 15.15,\n",
       "  'polymer_monomer_count_maximum': 146,\n",
       "  'polymer_monomer_count_minimum': 141,\n",
       "  'resolution_combined': [1.74],\n",
       "  'selected_polymer_entity_types': 'Protein (only)',\n",
       "  'solvent_entity_count': 1,\n",
       "  'diffrn_resolution_high': {'provenance_source': 'From refinement resolution cutoff',\n",
       "   'value': 1.74}},\n",
       " 'rcsb_primary_citation': {'country': 'UK',\n",
       "  'id': 'primary',\n",
       "  'journal_abbrev': 'J.Mol.Biol.',\n",
       "  'journal_id_astm': 'JMOBAK',\n",
       "  'journal_id_csd': '0070',\n",
       "  'journal_id_issn': '0022-2836',\n",
       "  'journal_volume': '175',\n",
       "  'page_first': '159',\n",
       "  'page_last': '174',\n",
       "  'pdbx_database_id_doi': '10.1016/0022-2836(84)90472-8',\n",
       "  'pdbx_database_id_pub_med': 6726807,\n",
       "  'rcsb_orcididentifiers': ['?', '?', '?', '?'],\n",
       "  'rcsb_authors': ['Fermi, G.', 'Perutz, M.F.', 'Shaanan, B.', 'Fourme, R.'],\n",
       "  'rcsb_journal_abbrev': 'J Mol Biol',\n",
       "  'title': 'The crystal structure of human deoxyhaemoglobin at 1.74 A resolution',\n",
       "  'year': 1984},\n",
       " 'refine': [{'details': 'THE COORDINATES GIVEN HERE ARE IN THE ORTHOGONAL ANGSTROM\\nSYSTEM STANDARD FOR HEMOGLOBINS. THE Y AXIS IS THE\\n(NON CRYSTALLOGRAPHIC) MOLECULAR DIAD AND THE X AXIS IS THE\\nPSEUDO DIAD WHICH RELATES THE ALPHA-1 AND BETA-1 CHAINS.\\nTHE TRANSFORMATION GIVEN IN THE *MTRIX* RECORDS BELOW\\nWILL GENERATE COORDINATES FOR THE *C* AND *D* CHAINS FROM\\nTHE *A* AND *B* CHAINS RESPECTIVELY.',\n",
       "   'ls_rfactor_rwork': 0.135,\n",
       "   'ls_dres_high': 1.74,\n",
       "   'pdbx_diffrn_id': ['1'],\n",
       "   'pdbx_refine_id': 'X-RAY DIFFRACTION'}],\n",
       " 'refine_hist': [{'cycle_id': 'LAST',\n",
       "   'number_atoms_solvent': 221,\n",
       "   'number_atoms_total': 4779,\n",
       "   'pdbx_number_atoms_ligand': 174,\n",
       "   'pdbx_number_atoms_nucleic_acid': 0,\n",
       "   'pdbx_number_atoms_protein': 4384,\n",
       "   'pdbx_refine_id': 'X-RAY DIFFRACTION',\n",
       "   'dres_high': 1.74}],\n",
       " 'struct': {'pdbx_descriptor': 'HEMOGLOBIN (DEOXY)',\n",
       "  'title': 'THE CRYSTAL STRUCTURE OF HUMAN DEOXYHAEMOGLOBIN AT 1.74 ANGSTROMS RESOLUTION'},\n",
       " 'struct_keywords': {'pdbx_keywords': 'OXYGEN TRANSPORT',\n",
       "  'text': 'OXYGEN TRANSPORT'},\n",
       " 'symmetry': {'int_tables_number': 4, 'space_group_name_hm': 'P 1 21 1'},\n",
       " 'rcsb_id': '4HHB'}"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pdbinfo = json.loads(response.text)\n",
    "pdbinfo"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'assembly_count': 1,\n",
       " 'branched_entity_count': 0,\n",
       " 'cis_peptide_count': 0,\n",
       " 'deposited_atom_count': 4779,\n",
       " 'deposited_hydrogen_atom_count': 0,\n",
       " 'deposited_model_count': 1,\n",
       " 'deposited_modeled_polymer_monomer_count': 574,\n",
       " 'deposited_nonpolymer_entity_instance_count': 6,\n",
       " 'deposited_polymer_entity_instance_count': 4,\n",
       " 'deposited_polymer_monomer_count': 574,\n",
       " 'deposited_solvent_atom_count': 221,\n",
       " 'deposited_unmodeled_polymer_monomer_count': 0,\n",
       " 'disulfide_bond_count': 0,\n",
       " 'entity_count': 5,\n",
       " 'experimental_method': 'X-ray',\n",
       " 'experimental_method_count': 1,\n",
       " 'inter_mol_covalent_bond_count': 0,\n",
       " 'inter_mol_metalic_bond_count': 4,\n",
       " 'molecular_weight': 64.74,\n",
       " 'na_polymer_entity_types': 'Other',\n",
       " 'nonpolymer_bound_components': ['HEM'],\n",
       " 'nonpolymer_entity_count': 2,\n",
       " 'nonpolymer_molecular_weight_maximum': 0.62,\n",
       " 'nonpolymer_molecular_weight_minimum': 0.09,\n",
       " 'polymer_composition': 'heteromeric protein',\n",
       " 'polymer_entity_count': 2,\n",
       " 'polymer_entity_count_dna': 0,\n",
       " 'polymer_entity_count_rna': 0,\n",
       " 'polymer_entity_count_nucleic_acid': 0,\n",
       " 'polymer_entity_count_nucleic_acid_hybrid': 0,\n",
       " 'polymer_entity_count_protein': 2,\n",
       " 'polymer_entity_taxonomy_count': 2,\n",
       " 'polymer_molecular_weight_maximum': 15.89,\n",
       " 'polymer_molecular_weight_minimum': 15.15,\n",
       " 'polymer_monomer_count_maximum': 146,\n",
       " 'polymer_monomer_count_minimum': 141,\n",
       " 'resolution_combined': [1.74],\n",
       " 'selected_polymer_entity_types': 'Protein (only)',\n",
       " 'solvent_entity_count': 1,\n",
       " 'diffrn_resolution_high': {'provenance_source': 'From refinement resolution cutoff',\n",
       "  'value': 1.74}}"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pdbinfo['rcsb_entry_info']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {
    "slideshow": {
     "slide_type": "slide"
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'rcsb_cluster_membership': [{'cluster_id': 93, 'identity': 100},\n",
       "  {'cluster_id': 115, 'identity': 95},\n",
       "  {'cluster_id': 119, 'identity': 90},\n",
       "  {'cluster_id': 31, 'identity': 70},\n",
       "  {'cluster_id': 8, 'identity': 50},\n",
       "  {'cluster_id': 13, 'identity': 30}],\n",
       " 'entity_poly': {'nstd_linkage': 'no',\n",
       "  'nstd_monomer': 'no',\n",
       "  'pdbx_seq_one_letter_code': 'VLSPADKTNVKAAWGKVGAHAGEYGAEALERMFLSFPTTKTYFPHFDLSHGSAQVKGHGKKVADALTNAVAHVDDMPNALSALSDLHAHKLRVDPVNFKLLSHCLLVTLAAHLPAEFTPAVHASLDKFLASVSTVLTSKYR',\n",
       "  'pdbx_seq_one_letter_code_can': 'VLSPADKTNVKAAWGKVGAHAGEYGAEALERMFLSFPTTKTYFPHFDLSHGSAQVKGHGKKVADALTNAVAHVDDMPNALSALSDLHAHKLRVDPVNFKLLSHCLLVTLAAHLPAEFTPAVHASLDKFLASVSTVLTSKYR',\n",
       "  'pdbx_strand_id': 'A,C',\n",
       "  'rcsb_artifact_monomer_count': 0,\n",
       "  'rcsb_conflict_count': 0,\n",
       "  'rcsb_deletion_count': 0,\n",
       "  'rcsb_entity_polymer_type': 'Protein',\n",
       "  'rcsb_insertion_count': 0,\n",
       "  'rcsb_mutation_count': 0,\n",
       "  'rcsb_non_std_monomer_count': 0,\n",
       "  'rcsb_sample_sequence_length': 141,\n",
       "  'type': 'polypeptide(L)'},\n",
       " 'entity_src_gen': [{'gene_src_common_name': 'Human',\n",
       "   'gene_src_genus': 'Homo',\n",
       "   'pdbx_alt_source_flag': 'sample',\n",
       "   'pdbx_beg_seq_num': 1,\n",
       "   'pdbx_end_seq_num': 141,\n",
       "   'pdbx_gene_src_gene': 'HBA1, HBA2',\n",
       "   'pdbx_gene_src_ncbi_taxonomy_id': '9606',\n",
       "   'pdbx_gene_src_scientific_name': 'Homo sapiens',\n",
       "   'pdbx_seq_type': 'Biological sequence',\n",
       "   'pdbx_src_id': 1}],\n",
       " 'rcsb_entity_host_organism': [{'beg_seq_num': 1,\n",
       "   'end_seq_num': 141,\n",
       "   'pdbx_src_id': '1',\n",
       "   'provenance_source': 'PDB Primary Data'}],\n",
       " 'rcsb_entity_source_organism': [{'beg_seq_num': 1,\n",
       "   'common_name': 'Human',\n",
       "   'end_seq_num': 141,\n",
       "   'ncbi_common_names': ['human'],\n",
       "   'ncbi_parent_scientific_name': 'Eukaryota',\n",
       "   'ncbi_scientific_name': 'Homo sapiens',\n",
       "   'ncbi_taxonomy_id': 9606,\n",
       "   'pdbx_src_id': '1',\n",
       "   'provenance_source': 'PDB Primary Data',\n",
       "   'scientific_name': 'Homo sapiens',\n",
       "   'source_type': 'genetically engineered',\n",
       "   'taxonomy_lineage': [{'depth': 1,\n",
       "     'id': '131567',\n",
       "     'name': 'cellular organisms'},\n",
       "    {'depth': 1, 'id': '131567', 'name': 'biota'},\n",
       "    {'depth': 2, 'id': '2759', 'name': 'Eukaryota'},\n",
       "    {'depth': 2, 'id': '2759', 'name': 'Eucarya'},\n",
       "    {'depth': 2, 'id': '2759', 'name': 'Eucaryotae'},\n",
       "    {'depth': 2, 'id': '2759', 'name': 'Eukarya'},\n",
       "    {'depth': 2, 'id': '2759', 'name': 'Eukaryotae'},\n",
       "    {'depth': 2, 'id': '2759', 'name': 'eucaryotes'},\n",
       "    {'depth': 2, 'id': '2759', 'name': 'eukaryotes'},\n",
       "    {'depth': 3, 'id': '33154', 'name': 'Opisthokonta'},\n",
       "    {'depth': 3, 'id': '33154', 'name': 'Fungi/Metazoa group'},\n",
       "    {'depth': 3, 'id': '33154', 'name': 'opisthokonts'},\n",
       "    {'depth': 4, 'id': '33208', 'name': 'Metazoa'},\n",
       "    {'depth': 4, 'id': '33208', 'name': 'Animalia'},\n",
       "    {'depth': 4, 'id': '33208', 'name': 'metazoans'},\n",
       "    {'depth': 4, 'id': '33208', 'name': 'multicellular animals'},\n",
       "    {'depth': 5, 'id': '6072', 'name': 'Eumetazoa'},\n",
       "    {'depth': 6, 'id': '33213', 'name': 'Bilateria'},\n",
       "    {'depth': 7, 'id': '33511', 'name': 'Deuterostomia'},\n",
       "    {'depth': 7, 'id': '33511', 'name': 'deuterostomes'},\n",
       "    {'depth': 8, 'id': '7711', 'name': 'Chordata'},\n",
       "    {'depth': 8, 'id': '7711', 'name': 'chordates'},\n",
       "    {'depth': 9, 'id': '89593', 'name': 'Craniata'},\n",
       "    {'depth': 10, 'id': '7742', 'name': 'Vertebrata'},\n",
       "    {'depth': 10, 'id': '7742', 'name': 'vertebrates'},\n",
       "    {'depth': 11, 'id': '7776', 'name': 'Gnathostomata'},\n",
       "    {'depth': 11, 'id': '7776', 'name': 'jawed vertebrates'},\n",
       "    {'depth': 12, 'id': '117570', 'name': 'Teleostomi'},\n",
       "    {'depth': 13, 'id': '117571', 'name': 'Euteleostomi'},\n",
       "    {'depth': 13, 'id': '117571', 'name': 'bony vertebrates'},\n",
       "    {'depth': 14, 'id': '8287', 'name': 'Sarcopterygii'},\n",
       "    {'depth': 15, 'id': '1338369', 'name': 'Dipnotetrapodomorpha'},\n",
       "    {'depth': 16, 'id': '32523', 'name': 'Tetrapoda'},\n",
       "    {'depth': 16, 'id': '32523', 'name': 'tetrapods'},\n",
       "    {'depth': 17, 'id': '32524', 'name': 'Amniota'},\n",
       "    {'depth': 17, 'id': '32524', 'name': 'amniotes'},\n",
       "    {'depth': 18, 'id': '40674', 'name': 'Mammalia'},\n",
       "    {'depth': 18, 'id': '40674', 'name': 'mammals'},\n",
       "    {'depth': 19, 'id': '32525', 'name': 'Theria'},\n",
       "    {'depth': 20, 'id': '9347', 'name': 'Eutheria'},\n",
       "    {'depth': 20, 'id': '9347', 'name': 'Placentalia'},\n",
       "    {'depth': 20, 'id': '9347', 'name': 'eutherian mammals'},\n",
       "    {'depth': 20, 'id': '9347', 'name': 'placental mammals'},\n",
       "    {'depth': 20, 'id': '9347', 'name': 'placentals'},\n",
       "    {'depth': 21, 'id': '1437010', 'name': 'Boreoeutheria'},\n",
       "    {'depth': 21, 'id': '1437010', 'name': 'Boreotheria'},\n",
       "    {'depth': 22, 'id': '314146', 'name': 'Euarchontoglires'},\n",
       "    {'depth': 23, 'id': '9443', 'name': 'Primates'},\n",
       "    {'depth': 23, 'id': '9443', 'name': 'Primata'},\n",
       "    {'depth': 23, 'id': '9443', 'name': 'primate'},\n",
       "    {'depth': 24, 'id': '376913', 'name': 'Haplorrhini'},\n",
       "    {'depth': 25, 'id': '314293', 'name': 'Simiiformes'},\n",
       "    {'depth': 25, 'id': '314293', 'name': 'Anthropoidea'},\n",
       "    {'depth': 26, 'id': '9526', 'name': 'Catarrhini'},\n",
       "    {'depth': 27, 'id': '314295', 'name': 'Hominoidea'},\n",
       "    {'depth': 27, 'id': '314295', 'name': 'ape'},\n",
       "    {'depth': 27, 'id': '314295', 'name': 'apes'},\n",
       "    {'depth': 28, 'id': '9604', 'name': 'Hominidae'},\n",
       "    {'depth': 28, 'id': '9604', 'name': 'Pongidae'},\n",
       "    {'depth': 28, 'id': '9604', 'name': 'great apes'},\n",
       "    {'depth': 29, 'id': '207598', 'name': 'Homininae'},\n",
       "    {'depth': 29, 'id': '207598', 'name': 'Homo/Pan/Gorilla group'},\n",
       "    {'depth': 30, 'id': '9605', 'name': 'Homo'},\n",
       "    {'depth': 30, 'id': '9605', 'name': 'humans'},\n",
       "    {'depth': 31, 'id': '9606', 'name': 'Homo sapiens'},\n",
       "    {'depth': 31, 'id': '9606', 'name': 'human'}],\n",
       "   'rcsb_gene_name': [{'provenance_source': 'PDB Primary Data',\n",
       "     'value': 'HBA1'},\n",
       "    {'provenance_source': 'PDB Primary Data', 'value': 'HBA2'},\n",
       "    {'provenance_source': 'UniProt', 'value': 'HBA1'},\n",
       "    {'provenance_source': 'UniProt', 'value': 'HBA2'}]}],\n",
       " 'rcsb_polymer_entity': {'formula_weight': 15.15,\n",
       "  'pdbx_description': 'Hemoglobin subunit alpha',\n",
       "  'pdbx_number_of_molecules': 2,\n",
       "  'rcsb_multiple_source_flag': 'N',\n",
       "  'rcsb_source_part_count': 1,\n",
       "  'rcsb_source_taxonomy_count': 1,\n",
       "  'src_method': 'man',\n",
       "  'rcsb_macromolecular_names_combined': [{'name': 'Hemoglobin subunit alpha',\n",
       "    'provenance_code': 'ECO:0000304',\n",
       "    'provenance_source': 'PDB Preferred Name'},\n",
       "   {'name': 'Alpha-globin',\n",
       "    'provenance_code': 'ECO:0000303',\n",
       "    'provenance_source': 'PDB Synonym'},\n",
       "   {'name': 'Hemoglobin alpha chain',\n",
       "    'provenance_code': 'ECO:0000303',\n",
       "    'provenance_source': 'PDB Synonym'}]},\n",
       " 'rcsb_polymer_entity_align': [{'provenance_source': 'SIFTS',\n",
       "   'reference_database_accession': 'P69905',\n",
       "   'reference_database_name': 'UniProt',\n",
       "   'aligned_regions': [{'entity_beg_seq_id': 1,\n",
       "     'length': 141,\n",
       "     'ref_beg_seq_id': 2}]}],\n",
       " 'rcsb_polymer_entity_annotation': [{'annotation_id': 'PF00042',\n",
       "   'assignment_version': '34.0',\n",
       "   'name': 'Globin (Globin)',\n",
       "   'provenance_source': 'Pfam',\n",
       "   'type': 'Pfam'},\n",
       "  {'annotation_id': 'GO:0072562',\n",
       "   'assignment_version': '193',\n",
       "   'name': 'blood microparticle',\n",
       "   'provenance_source': 'UniProt',\n",
       "   'type': 'GO',\n",
       "   'annotation_lineage': [{'id': 'GO:0005575', 'name': 'cellular_component'},\n",
       "    {'id': 'GO:0005576', 'name': 'extracellular region'},\n",
       "    {'id': 'GO:0005615', 'name': 'extracellular space'},\n",
       "    {'id': 'GO:0072562', 'name': 'blood microparticle'},\n",
       "    {'id': 'GO:0110165', 'name': 'cellular anatomical entity'}]},\n",
       "  {'annotation_id': 'GO:0005829',\n",
       "   'assignment_version': '193',\n",
       "   'name': 'cytosol',\n",
       "   'provenance_source': 'UniProt',\n",
       "   'type': 'GO',\n",
       "   'annotation_lineage': [{'id': 'GO:0005575', 'name': 'cellular_component'},\n",
       "    {'id': 'GO:0005622', 'name': 'intracellular anatomical structure'},\n",
       "    {'id': 'GO:0005737', 'name': 'cytoplasm'},\n",
       "    {'id': 'GO:0005829', 'name': 'cytosol'},\n",
       "    {'id': 'GO:0110165', 'name': 'cellular anatomical entity'}]},\n",
       "  {'annotation_id': 'GO:0022627',\n",
       "   'assignment_version': '193',\n",
       "   'name': 'cytosolic small ribosomal subunit',\n",
       "   'provenance_source': 'UniProt',\n",
       "   'type': 'GO',\n",
       "   'annotation_lineage': [{'id': 'GO:0005575', 'name': 'cellular_component'},\n",
       "    {'id': 'GO:0005622', 'name': 'intracellular anatomical structure'},\n",
       "    {'id': 'GO:0005737', 'name': 'cytoplasm'},\n",
       "    {'id': 'GO:0005829', 'name': 'cytosol'},\n",
       "    {'id': 'GO:0005840', 'name': 'ribosome'},\n",
       "    {'id': 'GO:0015935', 'name': 'small ribosomal subunit'},\n",
       "    {'id': 'GO:0022626', 'name': 'cytosolic ribosome'},\n",
       "    {'id': 'GO:0022627', 'name': 'cytosolic small ribosomal subunit'},\n",
       "    {'id': 'GO:0032991', 'name': 'protein-containing complex'},\n",
       "    {'id': 'GO:0043226', 'name': 'organelle'},\n",
       "    {'id': 'GO:0043228', 'name': 'non-membrane-bounded organelle'},\n",
       "    {'id': 'GO:0043229', 'name': 'intracellular organelle'},\n",
       "    {'id': 'GO:0043232',\n",
       "     'name': 'intracellular non-membrane-bounded organelle'},\n",
       "    {'id': 'GO:0044391', 'name': 'ribosomal subunit'},\n",
       "    {'id': 'GO:0110165', 'name': 'cellular anatomical entity'},\n",
       "    {'id': 'GO:1990904', 'name': 'ribonucleoprotein complex'}]},\n",
       "  {'annotation_id': 'GO:0071682',\n",
       "   'assignment_version': '193',\n",
       "   'name': 'endocytic vesicle lumen',\n",
       "   'provenance_source': 'UniProt',\n",
       "   'type': 'GO',\n",
       "   'annotation_lineage': [{'id': 'GO:0005575', 'name': 'cellular_component'},\n",
       "    {'id': 'GO:0005622', 'name': 'intracellular anatomical structure'},\n",
       "    {'id': 'GO:0005737', 'name': 'cytoplasm'},\n",
       "    {'id': 'GO:0030139', 'name': 'endocytic vesicle'},\n",
       "    {'id': 'GO:0031410', 'name': 'cytoplasmic vesicle'},\n",
       "    {'id': 'GO:0031974', 'name': 'membrane-enclosed lumen'},\n",
       "    {'id': 'GO:0031982', 'name': 'vesicle'},\n",
       "    {'id': 'GO:0043226', 'name': 'organelle'},\n",
       "    {'id': 'GO:0043227', 'name': 'membrane-bounded organelle'},\n",
       "    {'id': 'GO:0043229', 'name': 'intracellular organelle'},\n",
       "    {'id': 'GO:0043231', 'name': 'intracellular membrane-bounded organelle'},\n",
       "    {'id': 'GO:0043233', 'name': 'organelle lumen'},\n",
       "    {'id': 'GO:0070013', 'name': 'intracellular organelle lumen'},\n",
       "    {'id': 'GO:0071682', 'name': 'endocytic vesicle lumen'},\n",
       "    {'id': 'GO:0097708', 'name': 'intracellular vesicle'},\n",
       "    {'id': 'GO:0110165', 'name': 'cellular anatomical entity'}]},\n",
       "  {'annotation_id': 'GO:0070062',\n",
       "   'assignment_version': '193',\n",
       "   'name': 'extracellular exosome',\n",
       "   'provenance_source': 'UniProt',\n",
       "   'type': 'GO',\n",
       "   'annotation_lineage': [{'id': 'GO:0005575', 'name': 'cellular_component'},\n",
       "    {'id': 'GO:0005576', 'name': 'extracellular region'},\n",
       "    {'id': 'GO:0005615', 'name': 'extracellular space'},\n",
       "    {'id': 'GO:0031982', 'name': 'vesicle'},\n",
       "    {'id': 'GO:0043226', 'name': 'organelle'},\n",
       "    {'id': 'GO:0043227', 'name': 'membrane-bounded organelle'},\n",
       "    {'id': 'GO:0043230', 'name': 'extracellular organelle'},\n",
       "    {'id': 'GO:0065010', 'name': 'extracellular membrane-bounded organelle'},\n",
       "    {'id': 'GO:0070062', 'name': 'extracellular exosome'},\n",
       "    {'id': 'GO:0110165', 'name': 'cellular anatomical entity'},\n",
       "    {'id': 'GO:1903561', 'name': 'extracellular vesicle'}]},\n",
       "  {'annotation_id': 'GO:0005576',\n",
       "   'assignment_version': '193',\n",
       "   'name': 'extracellular region',\n",
       "   'provenance_source': 'UniProt',\n",
       "   'type': 'GO',\n",
       "   'annotation_lineage': [{'id': 'GO:0005575', 'name': 'cellular_component'},\n",
       "    {'id': 'GO:0005576', 'name': 'extracellular region'},\n",
       "    {'id': 'GO:0110165', 'name': 'cellular anatomical entity'}]},\n",
       "  {'annotation_id': 'GO:0005615',\n",
       "   'assignment_version': '193',\n",
       "   'name': 'extracellular space',\n",
       "   'provenance_source': 'UniProt',\n",
       "   'type': 'GO',\n",
       "   'annotation_lineage': [{'id': 'GO:0005575', 'name': 'cellular_component'},\n",
       "    {'id': 'GO:0005576', 'name': 'extracellular region'},\n",
       "    {'id': 'GO:0005615', 'name': 'extracellular space'},\n",
       "    {'id': 'GO:0110165', 'name': 'cellular anatomical entity'}]},\n",
       "  {'annotation_id': 'GO:0031838',\n",
       "   'assignment_version': '193',\n",
       "   'name': 'haptoglobin-hemoglobin complex',\n",
       "   'provenance_source': 'UniProt',\n",
       "   'type': 'GO',\n",
       "   'annotation_lineage': [{'id': 'GO:0005575', 'name': 'cellular_component'},\n",
       "    {'id': 'GO:0031838', 'name': 'haptoglobin-hemoglobin complex'},\n",
       "    {'id': 'GO:0032991', 'name': 'protein-containing complex'}]},\n",
       "  {'annotation_id': 'GO:0005833',\n",
       "   'assignment_version': '193',\n",
       "   'name': 'hemoglobin complex',\n",
       "   'provenance_source': 'UniProt',\n",
       "   'type': 'GO',\n",
       "   'annotation_lineage': [{'id': 'GO:0005575', 'name': 'cellular_component'},\n",
       "    {'id': 'GO:0005622', 'name': 'intracellular anatomical structure'},\n",
       "    {'id': 'GO:0005737', 'name': 'cytoplasm'},\n",
       "    {'id': 'GO:0005829', 'name': 'cytosol'},\n",
       "    {'id': 'GO:0005833', 'name': 'hemoglobin complex'},\n",
       "    {'id': 'GO:0032991', 'name': 'protein-containing complex'},\n",
       "    {'id': 'GO:0110165', 'name': 'cellular anatomical entity'}]},\n",
       "  {'annotation_id': 'GO:0016020',\n",
       "   'assignment_version': '193',\n",
       "   'name': 'membrane',\n",
       "   'provenance_source': 'UniProt',\n",
       "   'type': 'GO',\n",
       "   'annotation_lineage': [{'id': 'GO:0005575', 'name': 'cellular_component'},\n",
       "    {'id': 'GO:0016020', 'name': 'membrane'},\n",
       "    {'id': 'GO:0110165', 'name': 'cellular anatomical entity'}]},\n",
       "  {'annotation_id': 'GO:0020037',\n",
       "   'assignment_version': '193',\n",
       "   'name': 'heme binding',\n",
       "   'provenance_source': 'UniProt',\n",
       "   'type': 'GO',\n",
       "   'annotation_lineage': [{'id': 'GO:0003674', 'name': 'molecular_function'},\n",
       "    {'id': 'GO:0005488', 'name': 'binding'},\n",
       "    {'id': 'GO:0020037', 'name': 'heme binding'},\n",
       "    {'id': 'GO:0046906', 'name': 'tetrapyrrole binding'},\n",
       "    {'id': 'GO:0097159', 'name': 'organic cyclic compound binding'},\n",
       "    {'id': 'GO:1901363', 'name': 'heterocyclic compound binding'}]},\n",
       "  {'annotation_id': 'GO:0005506',\n",
       "   'assignment_version': '193',\n",
       "   'name': 'iron ion binding',\n",
       "   'provenance_source': 'UniProt',\n",
       "   'type': 'GO',\n",
       "   'annotation_lineage': [{'id': 'GO:0003674', 'name': 'molecular_function'},\n",
       "    {'id': 'GO:0005488', 'name': 'binding'},\n",
       "    {'id': 'GO:0005506', 'name': 'iron ion binding'},\n",
       "    {'id': 'GO:0043167', 'name': 'ion binding'},\n",
       "    {'id': 'GO:0043169', 'name': 'cation binding'},\n",
       "    {'id': 'GO:0046872', 'name': 'metal ion binding'},\n",
       "    {'id': 'GO:0046914', 'name': 'transition metal ion binding'}]},\n",
       "  {'annotation_id': 'GO:0043177',\n",
       "   'assignment_version': '193',\n",
       "   'name': 'organic acid binding',\n",
       "   'provenance_source': 'UniProt',\n",
       "   'type': 'GO',\n",
       "   'annotation_lineage': [{'id': 'GO:0003674', 'name': 'molecular_function'},\n",
       "    {'id': 'GO:0005488', 'name': 'binding'},\n",
       "    {'id': 'GO:0036094', 'name': 'small molecule binding'},\n",
       "    {'id': 'GO:0043177', 'name': 'organic acid binding'}]},\n",
       "  {'annotation_id': 'GO:0019825',\n",
       "   'assignment_version': '193',\n",
       "   'name': 'oxygen binding',\n",
       "   'provenance_source': 'UniProt',\n",
       "   'type': 'GO',\n",
       "   'annotation_lineage': [{'id': 'GO:0003674', 'name': 'molecular_function'},\n",
       "    {'id': 'GO:0005488', 'name': 'binding'},\n",
       "    {'id': 'GO:0019825', 'name': 'oxygen binding'},\n",
       "    {'id': 'GO:0036094', 'name': 'small molecule binding'}]},\n",
       "  {'annotation_id': 'GO:0005344',\n",
       "   'assignment_version': '193',\n",
       "   'name': 'oxygen carrier activity',\n",
       "   'provenance_source': 'UniProt',\n",
       "   'type': 'GO',\n",
       "   'annotation_lineage': [{'id': 'GO:0003674', 'name': 'molecular_function'},\n",
       "    {'id': 'GO:0005344', 'name': 'oxygen carrier activity'},\n",
       "    {'id': 'GO:0140104', 'name': 'molecular carrier activity'}]},\n",
       "  {'annotation_id': 'GO:0015701',\n",
       "   'assignment_version': '193',\n",
       "   'name': 'bicarbonate transport',\n",
       "   'provenance_source': 'UniProt',\n",
       "   'type': 'GO',\n",
       "   'annotation_lineage': [{'id': 'GO:0006810', 'name': 'transport'},\n",
       "    {'id': 'GO:0006811', 'name': 'ion transport'},\n",
       "    {'id': 'GO:0006820', 'name': 'anion transport'},\n",
       "    {'id': 'GO:0008150', 'name': 'biological_process'},\n",
       "    {'id': 'GO:0015701', 'name': 'bicarbonate transport'},\n",
       "    {'id': 'GO:0015711', 'name': 'organic anion transport'},\n",
       "    {'id': 'GO:0051179', 'name': 'localization'},\n",
       "    {'id': 'GO:0051234', 'name': 'establishment of localization'},\n",
       "    {'id': 'GO:0071702', 'name': 'organic substance transport'}]},\n",
       "  {'annotation_id': 'GO:0098869',\n",
       "   'assignment_version': '193',\n",
       "   'name': 'cellular oxidant detoxification',\n",
       "   'provenance_source': 'UniProt',\n",
       "   'type': 'GO',\n",
       "   'annotation_lineage': [{'id': 'GO:0008150', 'name': 'biological_process'},\n",
       "    {'id': 'GO:0009636', 'name': 'response to toxic substance'},\n",
       "    {'id': 'GO:0009987', 'name': 'cellular process'},\n",
       "    {'id': 'GO:0042221', 'name': 'response to chemical'},\n",
       "    {'id': 'GO:0050896', 'name': 'response to stimulus'},\n",
       "    {'id': 'GO:0051716', 'name': 'cellular response to stimulus'},\n",
       "    {'id': 'GO:0070887', 'name': 'cellular response to chemical stimulus'},\n",
       "    {'id': 'GO:0097237', 'name': 'cellular response to toxic substance'},\n",
       "    {'id': 'GO:0098754', 'name': 'detoxification'},\n",
       "    {'id': 'GO:0098869', 'name': 'cellular oxidant detoxification'},\n",
       "    {'id': 'GO:1990748', 'name': 'cellular detoxification'}]},\n",
       "  {'annotation_id': 'GO:0042744',\n",
       "   'assignment_version': '193',\n",
       "   'name': 'hydrogen peroxide catabolic process',\n",
       "   'provenance_source': 'UniProt',\n",
       "   'type': 'GO',\n",
       "   'annotation_lineage': [{'id': 'GO:0008150', 'name': 'biological_process'},\n",
       "    {'id': 'GO:0008152', 'name': 'metabolic process'},\n",
       "    {'id': 'GO:0009056', 'name': 'catabolic process'},\n",
       "    {'id': 'GO:0009987', 'name': 'cellular process'},\n",
       "    {'id': 'GO:0042743', 'name': 'hydrogen peroxide metabolic process'},\n",
       "    {'id': 'GO:0042744', 'name': 'hydrogen peroxide catabolic process'},\n",
       "    {'id': 'GO:0044237', 'name': 'cellular metabolic process'},\n",
       "    {'id': 'GO:0044248', 'name': 'cellular catabolic process'},\n",
       "    {'id': 'GO:0072593',\n",
       "     'name': 'reactive oxygen species metabolic process'}]},\n",
       "  {'annotation_id': 'GO:0015671',\n",
       "   'assignment_version': '193',\n",
       "   'name': 'oxygen transport',\n",
       "   'provenance_source': 'UniProt',\n",
       "   'type': 'GO',\n",
       "   'annotation_lineage': [{'id': 'GO:0006810', 'name': 'transport'},\n",
       "    {'id': 'GO:0008150', 'name': 'biological_process'},\n",
       "    {'id': 'GO:0015669', 'name': 'gas transport'},\n",
       "    {'id': 'GO:0015671', 'name': 'oxygen transport'},\n",
       "    {'id': 'GO:0051179', 'name': 'localization'},\n",
       "    {'id': 'GO:0051234', 'name': 'establishment of localization'}]},\n",
       "  {'annotation_id': 'GO:0010942',\n",
       "   'assignment_version': '193',\n",
       "   'name': 'positive regulation of cell death',\n",
       "   'provenance_source': 'UniProt',\n",
       "   'type': 'GO',\n",
       "   'annotation_lineage': [{'id': 'GO:0008150', 'name': 'biological_process'},\n",
       "    {'id': 'GO:0008219', 'name': 'cell death'},\n",
       "    {'id': 'GO:0009987', 'name': 'cellular process'},\n",
       "    {'id': 'GO:0010941', 'name': 'regulation of cell death'},\n",
       "    {'id': 'GO:0010942', 'name': 'positive regulation of cell death'},\n",
       "    {'id': 'GO:0048518', 'name': 'positive regulation of biological process'},\n",
       "    {'id': 'GO:0048522', 'name': 'positive regulation of cellular process'},\n",
       "    {'id': 'GO:0050789', 'name': 'regulation of biological process'},\n",
       "    {'id': 'GO:0050794', 'name': 'regulation of cellular process'},\n",
       "    {'id': 'GO:0065007', 'name': 'biological regulation'}]},\n",
       "  {'annotation_id': 'GO:0006898',\n",
       "   'assignment_version': '193',\n",
       "   'name': 'receptor-mediated endocytosis',\n",
       "   'provenance_source': 'UniProt',\n",
       "   'type': 'GO',\n",
       "   'annotation_lineage': [{'id': 'GO:0006810', 'name': 'transport'},\n",
       "    {'id': 'GO:0006897', 'name': 'endocytosis'},\n",
       "    {'id': 'GO:0006898', 'name': 'receptor-mediated endocytosis'},\n",
       "    {'id': 'GO:0008150', 'name': 'biological_process'},\n",
       "    {'id': 'GO:0009987', 'name': 'cellular process'},\n",
       "    {'id': 'GO:0016192', 'name': 'vesicle-mediated transport'},\n",
       "    {'id': 'GO:0051179', 'name': 'localization'},\n",
       "    {'id': 'GO:0051234', 'name': 'establishment of localization'}]},\n",
       "  {'annotation_id': 'GO:0042542',\n",
       "   'assignment_version': '193',\n",
       "   'name': 'response to hydrogen peroxide',\n",
       "   'provenance_source': 'UniProt',\n",
       "   'type': 'GO',\n",
       "   'annotation_lineage': [{'id': 'GO:0000302',\n",
       "     'name': 'response to reactive oxygen species'},\n",
       "    {'id': 'GO:0006950', 'name': 'response to stress'},\n",
       "    {'id': 'GO:0006979', 'name': 'response to oxidative stress'},\n",
       "    {'id': 'GO:0008150', 'name': 'biological_process'},\n",
       "    {'id': 'GO:0010035', 'name': 'response to inorganic substance'},\n",
       "    {'id': 'GO:0042221', 'name': 'response to chemical'},\n",
       "    {'id': 'GO:0042542', 'name': 'response to hydrogen peroxide'},\n",
       "    {'id': 'GO:0050896', 'name': 'response to stimulus'},\n",
       "    {'id': 'GO:1901700', 'name': 'response to oxygen-containing compound'}]},\n",
       "  {'annotation_id': 'IPR000971',\n",
       "   'assignment_version': '193',\n",
       "   'name': 'Globin',\n",
       "   'provenance_source': 'UniProt',\n",
       "   'type': 'InterPro',\n",
       "   'annotation_lineage': [{'depth': 1, 'id': 'IPR000971', 'name': 'Globin'}]},\n",
       "  {'annotation_id': 'IPR009050',\n",
       "   'assignment_version': '193',\n",
       "   'name': 'Globin-like superfamily',\n",
       "   'provenance_source': 'UniProt',\n",
       "   'type': 'InterPro',\n",
       "   'annotation_lineage': [{'depth': 1,\n",
       "     'id': 'IPR009050',\n",
       "     'name': 'Globin-like superfamily'}]},\n",
       "  {'annotation_id': 'IPR012292',\n",
       "   'assignment_version': '193',\n",
       "   'name': 'Globin/Protoglobin',\n",
       "   'provenance_source': 'UniProt',\n",
       "   'type': 'InterPro',\n",
       "   'annotation_lineage': [{'depth': 1,\n",
       "     'id': 'IPR012292',\n",
       "     'name': 'Globin/Protoglobin'}]},\n",
       "  {'annotation_id': 'IPR002338',\n",
       "   'assignment_version': '193',\n",
       "   'name': 'Haemoglobin, alpha-type',\n",
       "   'provenance_source': 'UniProt',\n",
       "   'type': 'InterPro',\n",
       "   'annotation_lineage': [{'depth': 1,\n",
       "     'id': 'IPR002338',\n",
       "     'name': 'Haemoglobin, alpha-type'}]},\n",
       "  {'annotation_id': 'IPR002339',\n",
       "   'assignment_version': '193',\n",
       "   'name': 'Haemoglobin, pi',\n",
       "   'provenance_source': 'UniProt',\n",
       "   'type': 'InterPro',\n",
       "   'annotation_lineage': [{'depth': 1,\n",
       "     'id': 'IPR002338',\n",
       "     'name': 'Haemoglobin, alpha-type'},\n",
       "    {'depth': 2, 'id': 'IPR002339', 'name': 'Haemoglobin, pi'}]},\n",
       "  {'annotation_id': 'P69905',\n",
       "   'assignment_version': '1.0',\n",
       "   'name': 'Glycoprotein',\n",
       "   'provenance_source': 'PDB',\n",
       "   'type': 'GlyGen'}],\n",
       " 'rcsb_polymer_entity_container_identifiers': {'asym_ids': ['A', 'C'],\n",
       "  'auth_asym_ids': ['A', 'C'],\n",
       "  'chem_comp_monomers': ['ALA',\n",
       "   'ARG',\n",
       "   'ASN',\n",
       "   'ASP',\n",
       "   'CYS',\n",
       "   'GLN',\n",
       "   'GLU',\n",
       "   'GLY',\n",
       "   'HIS',\n",
       "   'LEU',\n",
       "   'LYS',\n",
       "   'MET',\n",
       "   'PHE',\n",
       "   'PRO',\n",
       "   'SER',\n",
       "   'THR',\n",
       "   'TRP',\n",
       "   'TYR',\n",
       "   'VAL'],\n",
       "  'entity_id': '1',\n",
       "  'entry_id': '4HHB',\n",
       "  'rcsb_id': '4HHB_1',\n",
       "  'reference_sequence_identifiers': [{'database_accession': 'P69905',\n",
       "    'database_name': 'UniProt',\n",
       "    'provenance_source': 'SIFTS'}],\n",
       "  'uniprot_ids': ['P69905']},\n",
       " 'rcsb_polymer_entity_feature': [{'assignment_version': '34.0',\n",
       "   'feature_id': 'PF00042',\n",
       "   'name': 'Globin (Globin)',\n",
       "   'provenance_source': 'Pfam',\n",
       "   'type': 'Pfam',\n",
       "   'feature_positions': [{'beg_seq_id': 6, 'end_seq_id': 106}]},\n",
       "  {'name': 'Hydropathy values',\n",
       "   'provenance_source': 'biojava-aa-prop-6.0.0-alpha7',\n",
       "   'type': 'hydropathy',\n",
       "   'feature_positions': [{'beg_seq_id': 5,\n",
       "     'values': [-0.47,\n",
       "      -0.47,\n",
       "      -1.32,\n",
       "      -1.03,\n",
       "      -0.66,\n",
       "      -0.96,\n",
       "      -0.61,\n",
       "      -0.61,\n",
       "      -0.07,\n",
       "      0.28,\n",
       "      0.01,\n",
       "      0.09,\n",
       "      0.09,\n",
       "      -0.16,\n",
       "      -0.44,\n",
       "      -0.54,\n",
       "      -0.16,\n",
       "      -0.42,\n",
       "      -0.77,\n",
       "      -0.77,\n",
       "      0.01,\n",
       "      -0.58,\n",
       "      -1.03,\n",
       "      -0.43,\n",
       "      0.02,\n",
       "      0.49,\n",
       "      0.2,\n",
       "      0.9,\n",
       "      0.52,\n",
       "      0.02,\n",
       "      0.33,\n",
       "      0.4,\n",
       "      0.11,\n",
       "      -0.34,\n",
       "      -0.46,\n",
       "      -0.54,\n",
       "      -1.21,\n",
       "      -0.72,\n",
       "      -1.03,\n",
       "      -0.53,\n",
       "      -0.19,\n",
       "      -0.47,\n",
       "      -0.37,\n",
       "      -0.77,\n",
       "      -0.39,\n",
       "      -0.42,\n",
       "      -0.27,\n",
       "      -0.31,\n",
       "      -0.78,\n",
       "      -1.04,\n",
       "      -0.73,\n",
       "      -1.12,\n",
       "      -1.47,\n",
       "      -1.2,\n",
       "      -0.61,\n",
       "      -1.47,\n",
       "      -0.83,\n",
       "      -0.37,\n",
       "      -0.09,\n",
       "      -0.43,\n",
       "      0.2,\n",
       "      1.1,\n",
       "      0.83,\n",
       "      0.28,\n",
       "      1.13,\n",
       "      0.54,\n",
       "      -0.27,\n",
       "      0.02,\n",
       "      0.23,\n",
       "      -0.36,\n",
       "      -0.62,\n",
       "      -0.4,\n",
       "      -0.13,\n",
       "      -0.4,\n",
       "      0.41,\n",
       "      0.71,\n",
       "      0.11,\n",
       "      0.71,\n",
       "      0.74,\n",
       "      0.74,\n",
       "      -0.03,\n",
       "      -0.38,\n",
       "      -0.16,\n",
       "      -1.08,\n",
       "      -0.52,\n",
       "      -0.52,\n",
       "      -1.12,\n",
       "      -0.3,\n",
       "      -0.89,\n",
       "      -0.22,\n",
       "      -0.22,\n",
       "      -0.22,\n",
       "      0.7,\n",
       "      0.14,\n",
       "      0.18,\n",
       "      0.63,\n",
       "      0.59,\n",
       "      1.4,\n",
       "      1.56,\n",
       "      1.91,\n",
       "      1.91,\n",
       "      1.69,\n",
       "      1.98,\n",
       "      1.98,\n",
       "      2.12,\n",
       "      1.52,\n",
       "      1.3,\n",
       "      0.44,\n",
       "      0.83,\n",
       "      0.33,\n",
       "      -0.04,\n",
       "      -0.04,\n",
       "      0.78,\n",
       "      0.0,\n",
       "      0.38,\n",
       "      0.09,\n",
       "      0.9,\n",
       "      0.2,\n",
       "      -0.16,\n",
       "      0.33,\n",
       "      0.56,\n",
       "      0.29,\n",
       "      0.56,\n",
       "      0.82,\n",
       "      0.82,\n",
       "      0.32,\n",
       "      1.18,\n",
       "      2.03,\n",
       "      1.64,\n",
       "      1.13,\n",
       "      0.5,\n",
       "      0.44,\n",
       "      -0.52]}]},\n",
       "  {'name': 'Disordered binding sites',\n",
       "   'provenance_source': 'Anchor2',\n",
       "   'type': 'disorder_binding',\n",
       "   'feature_positions': [{'beg_seq_id': 1,\n",
       "     'values': [0.39,\n",
       "      0.38,\n",
       "      0.37,\n",
       "      0.36,\n",
       "      0.36,\n",
       "      0.36,\n",
       "      0.35,\n",
       "      0.35,\n",
       "      0.35,\n",
       "      0.35,\n",
       "      0.36,\n",
       "      0.37,\n",
       "      0.38,\n",
       "      0.39,\n",
       "      0.39,\n",
       "      0.39,\n",
       "      0.39,\n",
       "      0.39,\n",
       "      0.4,\n",
       "      0.4,\n",
       "      0.39,\n",
       "      0.39,\n",
       "      0.4,\n",
       "      0.41,\n",
       "      0.42,\n",
       "      0.42,\n",
       "      0.4,\n",
       "      0.39,\n",
       "      0.37,\n",
       "      0.36,\n",
       "      0.36,\n",
       "      0.35,\n",
       "      0.35,\n",
       "      0.35,\n",
       "      0.36,\n",
       "      0.37,\n",
       "      0.38,\n",
       "      0.39,\n",
       "      0.4,\n",
       "      0.41,\n",
       "      0.41,\n",
       "      0.41,\n",
       "      0.4,\n",
       "      0.39,\n",
       "      0.39,\n",
       "      0.38,\n",
       "      0.38,\n",
       "      0.38,\n",
       "      0.38,\n",
       "      0.38,\n",
       "      0.39,\n",
       "      0.4,\n",
       "      0.4,\n",
       "      0.41,\n",
       "      0.42,\n",
       "      0.43,\n",
       "      0.42,\n",
       "      0.42,\n",
       "      0.4,\n",
       "      0.4,\n",
       "      0.39,\n",
       "      0.38,\n",
       "      0.38,\n",
       "      0.37,\n",
       "      0.36,\n",
       "      0.36,\n",
       "      0.36,\n",
       "      0.36,\n",
       "      0.36,\n",
       "      0.36,\n",
       "      0.36,\n",
       "      0.37,\n",
       "      0.37,\n",
       "      0.37,\n",
       "      0.38,\n",
       "      0.39,\n",
       "      0.39,\n",
       "      0.4,\n",
       "      0.41,\n",
       "      0.42,\n",
       "      0.42,\n",
       "      0.43,\n",
       "      0.43,\n",
       "      0.42,\n",
       "      0.41,\n",
       "      0.4,\n",
       "      0.4,\n",
       "      0.41,\n",
       "      0.41,\n",
       "      0.42,\n",
       "      0.43,\n",
       "      0.42,\n",
       "      0.4,\n",
       "      0.38,\n",
       "      0.37,\n",
       "      0.35,\n",
       "      0.34,\n",
       "      0.32,\n",
       "      0.32,\n",
       "      0.31,\n",
       "      0.31,\n",
       "      0.31,\n",
       "      0.31,\n",
       "      0.3,\n",
       "      0.3,\n",
       "      0.29,\n",
       "      0.28,\n",
       "      0.26,\n",
       "      0.25,\n",
       "      0.23,\n",
       "      0.21,\n",
       "      0.2,\n",
       "      0.19,\n",
       "      0.17,\n",
       "      0.15,\n",
       "      0.14,\n",
       "      0.13,\n",
       "      0.12,\n",
       "      0.12,\n",
       "      0.11,\n",
       "      0.11,\n",
       "      0.11,\n",
       "      0.1,\n",
       "      0.09,\n",
       "      0.09,\n",
       "      0.09,\n",
       "      0.09,\n",
       "      0.08,\n",
       "      0.08,\n",
       "      0.08,\n",
       "      0.07,\n",
       "      0.07,\n",
       "      0.07,\n",
       "      0.07,\n",
       "      0.07,\n",
       "      0.08,\n",
       "      0.08,\n",
       "      0.08,\n",
       "      0.08,\n",
       "      0.08,\n",
       "      0.09]}]},\n",
       "  {'name': 'Disordered regions',\n",
       "   'provenance_source': 'IUPred2(short)',\n",
       "   'type': 'disorder',\n",
       "   'feature_positions': [{'beg_seq_id': 1,\n",
       "     'values': [0.9,\n",
       "      0.85,\n",
       "      0.79,\n",
       "      0.68,\n",
       "      0.64,\n",
       "      0.6,\n",
       "      0.5,\n",
       "      0.46,\n",
       "      0.38,\n",
       "      0.34,\n",
       "      0.27,\n",
       "      0.31,\n",
       "      0.37,\n",
       "      0.28,\n",
       "      0.26,\n",
       "      0.26,\n",
       "      0.27,\n",
       "      0.24,\n",
       "      0.18,\n",
       "      0.19,\n",
       "      0.26,\n",
       "      0.23,\n",
       "      0.18,\n",
       "      0.16,\n",
       "      0.21,\n",
       "      0.15,\n",
       "      0.17,\n",
       "      0.22,\n",
       "      0.22,\n",
       "      0.26,\n",
       "      0.25,\n",
       "      0.21,\n",
       "      0.14,\n",
       "      0.15,\n",
       "      0.24,\n",
       "      0.16,\n",
       "      0.19,\n",
       "      0.12,\n",
       "      0.13,\n",
       "      0.19,\n",
       "      0.17,\n",
       "      0.17,\n",
       "      0.18,\n",
       "      0.27,\n",
       "      0.28,\n",
       "      0.29,\n",
       "      0.37,\n",
       "      0.35,\n",
       "      0.35,\n",
       "      0.35,\n",
       "      0.35,\n",
       "      0.31,\n",
       "      0.36,\n",
       "      0.44,\n",
       "      0.38,\n",
       "      0.31,\n",
       "      0.38,\n",
       "      0.39,\n",
       "      0.44,\n",
       "      0.39,\n",
       "      0.37,\n",
       "      0.38,\n",
       "      0.32,\n",
       "      0.35,\n",
       "      0.33,\n",
       "      0.35,\n",
       "      0.38,\n",
       "      0.39,\n",
       "      0.36,\n",
       "      0.28,\n",
       "      0.29,\n",
       "      0.26,\n",
       "      0.22,\n",
       "      0.24,\n",
       "      0.24,\n",
       "      0.18,\n",
       "      0.25,\n",
       "      0.21,\n",
       "      0.2,\n",
       "      0.23,\n",
       "      0.18,\n",
       "      0.21,\n",
       "      0.14,\n",
       "      0.18,\n",
       "      0.21,\n",
       "      0.15,\n",
       "      0.18,\n",
       "      0.08,\n",
       "      0.08,\n",
       "      0.05,\n",
       "      0.05,\n",
       "      0.05,\n",
       "      0.06,\n",
       "      0.06,\n",
       "      0.03,\n",
       "      0.02,\n",
       "      0.02,\n",
       "      0.02,\n",
       "      0.01,\n",
       "      0.01,\n",
       "      0.01,\n",
       "      0.01,\n",
       "      0.01,\n",
       "      0.01,\n",
       "      0.01,\n",
       "      0.01,\n",
       "      0.01,\n",
       "      0.01,\n",
       "      0.02,\n",
       "      0.01,\n",
       "      0.01,\n",
       "      0.02,\n",
       "      0.02,\n",
       "      0.02,\n",
       "      0.01,\n",
       "      0.03,\n",
       "      0.05,\n",
       "      0.04,\n",
       "      0.03,\n",
       "      0.04,\n",
       "      0.04,\n",
       "      0.03,\n",
       "      0.03,\n",
       "      0.06,\n",
       "      0.03,\n",
       "      0.03,\n",
       "      0.02,\n",
       "      0.05,\n",
       "      0.05,\n",
       "      0.03,\n",
       "      0.03,\n",
       "      0.07,\n",
       "      0.1,\n",
       "      0.16,\n",
       "      0.2,\n",
       "      0.34,\n",
       "      0.38,\n",
       "      0.41,\n",
       "      0.57,\n",
       "      0.69,\n",
       "      0.76]}]}],\n",
       " 'rcsb_polymer_entity_feature_summary': [{'count': 0,\n",
       "   'coverage': 0.0,\n",
       "   'type': 'CARD_MODEL'},\n",
       "  {'count': 0, 'coverage': 0.0, 'type': 'IMGT_ANTIBODY_DESCRIPTION'},\n",
       "  {'count': 0, 'coverage': 0.0, 'type': 'IMGT_ANTIBODY_DOMAIN_NAME'},\n",
       "  {'count': 0, 'coverage': 0.0, 'type': 'IMGT_ANTIBODY_GENE_ALLELE_NAME'},\n",
       "  {'count': 0, 'coverage': 0.0, 'type': 'IMGT_ANTIBODY_ORGANISM_NAME'},\n",
       "  {'count': 0, 'coverage': 0.0, 'type': 'IMGT_ANTIBODY_PROTEIN_NAME'},\n",
       "  {'count': 0, 'coverage': 0.0, 'type': 'IMGT_ANTIBODY_RECEPTOR_DESCRIPTION'},\n",
       "  {'count': 0, 'coverage': 0.0, 'type': 'IMGT_ANTIBODY_RECEPTOR_TYPE'},\n",
       "  {'count': 1, 'coverage': 0.71631, 'type': 'Pfam'},\n",
       "  {'count': 0, 'coverage': 0.0, 'type': 'SABDAB_ANTIBODY_ANTIGEN_NAME'},\n",
       "  {'count': 0, 'coverage': 0.0, 'type': 'SABDAB_ANTIBODY_NAME'},\n",
       "  {'count': 0, 'coverage': 0.0, 'type': 'SABDAB_ANTIBODY_TARGET'},\n",
       "  {'count': 0, 'coverage': 0.0, 'type': 'artifact'},\n",
       "  {'count': 0, 'coverage': 0.0, 'type': 'modified_monomer'},\n",
       "  {'count': 0, 'coverage': 0.0, 'type': 'mutation'}],\n",
       " 'rcsb_polymer_entity_name_com': [{'name': 'Alpha-globin,Hemoglobin alpha chain'}],\n",
       " 'rcsb_related_target_references': [{'related_resource_name': 'ChEMBL',\n",
       "   'related_resource_version': '29',\n",
       "   'related_target_id': 'CHEMBL2095168',\n",
       "   'target_taxonomy_id': 9606,\n",
       "   'aligned_target': [{'entity_beg_seq_id': 1,\n",
       "     'length': 140,\n",
       "     'target_beg_seq_id': 2}]},\n",
       "  {'related_resource_name': 'ChEMBL',\n",
       "   'related_resource_version': '29',\n",
       "   'related_target_id': 'CHEMBL2887',\n",
       "   'target_taxonomy_id': 9606,\n",
       "   'aligned_target': [{'entity_beg_seq_id': 1,\n",
       "     'length': 140,\n",
       "     'target_beg_seq_id': 2}]},\n",
       "  {'related_resource_name': 'DrugBank',\n",
       "   'related_resource_version': '5.1',\n",
       "   'related_target_id': 'P69905',\n",
       "   'target_taxonomy_id': 9606,\n",
       "   'aligned_target': [{'entity_beg_seq_id': 1,\n",
       "     'length': 140,\n",
       "     'target_beg_seq_id': 2}]},\n",
       "  {'related_resource_name': 'Pharos',\n",
       "   'related_resource_version': '6.11.0',\n",
       "   'related_target_id': '6231',\n",
       "   'target_taxonomy_id': 9606,\n",
       "   'aligned_target': [{'entity_beg_seq_id': 1,\n",
       "     'length': 140,\n",
       "     'target_beg_seq_id': 2}]}],\n",
       " 'rcsb_target_cofactors': [{'cofactor_name': 'Iron Dextran',\n",
       "   'cofactor_resource_id': 'DB00893',\n",
       "   'mechanism_of_action': 'After iron dextran is injected, the circulating iron dextran is removed from the plasma by cells of the reticuloendothelial system, which split the complex into its components of iron and dextran. The iron is immediately bound to the available protein moieties to form hemosiderin or ferritin, the physiological forms of iron, or to a lesser extent to transferrin. This iron which is subject to physiological control replenishes hemoglobin and depleted iron stores.',\n",
       "   'neighbor_flag': 'N',\n",
       "   'pubmed_ids': [17139284, 17016423, 11752352],\n",
       "   'resource_name': 'DrugBank',\n",
       "   'resource_version': '5.1',\n",
       "   'target_resource_id': 'P69905'},\n",
       "  {'cofactor_in_ch_ikey': 'XEEYBQQBJWHFJM-UHFFFAOYSA-N',\n",
       "   'cofactor_smiles': '[Fe]',\n",
       "   'cofactor_name': 'Iron',\n",
       "   'cofactor_resource_id': 'DB01592',\n",
       "   'mechanism_of_action': 'Iron is necessary for the production of hemoglobin. Iron-deficiency can lead to decreased production of hemoglobin and a microcytic, hypochromic anemia.',\n",
       "   'neighbor_flag': 'N',\n",
       "   'pubmed_ids': [16901899],\n",
       "   'resource_name': 'DrugBank',\n",
       "   'resource_version': '5.1',\n",
       "   'target_resource_id': 'P69905'},\n",
       "  {'cofactor_in_ch_ikey': 'HCHKCACWOHOZIP-UHFFFAOYSA-N',\n",
       "   'cofactor_smiles': '[Zn]',\n",
       "   'cofactor_name': 'Zinc',\n",
       "   'cofactor_resource_id': 'DB01593',\n",
       "   'mechanism_of_action': '**Zinc has three primary biological roles**: _catalytic_, _structural_, and _regulatory_. The catalytic and structural role of zinc is well established, and there are various noteworthy reviews on these functions. For example, zinc is a structural constituent in numerous proteins, inclusive of growth factors, cytokines, receptors, enzymes, and transcription factors for different cellular signaling pathways. It is implicated in numerous cellular processes as a cofactor for approximately 3000 human proteins including enzymes, nuclear factors, and hormones [L2096].\\r\\n\\r\\nZinc promotes resistance to epithelial apoptosis through cell protection (cytoprotection) against reactive oxygen species and bacterial toxins, likely through the antioxidant activity of the cysteine-rich metallothioneins [A32419].\\r\\n\\r\\nIn HL-60 cells (promyelocytic leukemia cell line), zinc enhances the up-regulation of A20 mRNA, which, via TRAF pathway, decreases NF-kappaB activation, leading to decreased gene expression and generation of tumor necrosis factor-alpha (TNF-alpha), IL-1beta, and IL-8 [A32418].\\r\\n\\r\\nThere are several mechanisms of action of zinc on acute diarrhea. Various mechanisms are specific to the gastrointestinal system: zinc restores mucosal barrier integrity and enterocyte brush-border enzyme activity, it promotes the production of antibodies and circulating lymphocytes against intestinal pathogens, and has a direct effect on ion channels, acting as a potassium channel blocker of adenosine 3-5-cyclic monophosphate-mediated chlorine secretion. Cochrane researchers examined the evidence available up to 30 September 2016 [L2106].\\r\\n\\r\\nZinc deficiency in humans decreases the activity of serum _thymulin_ (a hormone of the thymus), which is necessary for the maturation of T-helper cells. T-helper 1 (Th(1)) cytokines are decreased but T-helper 2 (Th(2)) cytokines are not affected by zinc deficiency in humans [A342417].\\r\\n\\r\\nThe change of _Th(1)_ to _Th(2)_ function leads to cell-mediated immune dysfunction. Because IL-2 production (Th(1) cytokine) is decreased, this causes decreased activity of natural-killer-cell (NK cell) and T cytolytic cells, normally involved in killing viruses, bacteria, and malignant cells [A3424]. \\r\\n\\r\\nIn humans, zinc deficiency may lead to the generation of new CD4+ T cells, produced in the thymus. In cell culture studies (HUT-78, a Th(0) human malignant lymphoblastoid cell line), as a result of zinc deficiency, nuclear factor-kappaB (NF-kappaB) activation, phosphorylation of IkappaB, and binding of NF-kappaB to DNA are decreased and this results in decreased Th(1) cytokine production [A32417].\\r\\n\\r\\nIn another study, zinc supplementation in human subjects suppressed the gene expression and production of pro-inflammatory cytokines and decreased oxidative stress markers [A3424]. In HL-60 cells (a human pro-myelocytic leukemia cell line), zinc deficiency increased the levels of TNF-alpha, IL-1beta, and IL-8 cytokines and mRNA. In such cells, zinc was found to induce A20, a zinc finger protein that inhibited NF-kappaB activation by the tumor necrosis factor receptor-associated factor pathway. This process decreased gene expression of pro-inflammatory cytokines and oxidative stress markers [A32417].\\r\\n\\r\\nThe exact mechanism of zinc in acne treatment is poorly understood. However, zinc is considered to act directly on microbial inflammatory equilibrium and facilitate antibiotic absorption when used in combination with other agents. Topical zinc alone as well as in combination with other agents may be efficacious because of its anti-inflammatory activity and ability to reduce P. acnes bacteria by the inhibition of P. acnes lipases and free fatty acid levels [L2102].',\n",
       "   'neighbor_flag': 'N',\n",
       "   'pubmed_ids': [23896426],\n",
       "   'resource_name': 'DrugBank',\n",
       "   'resource_version': '5.1',\n",
       "   'target_resource_id': 'P69905'},\n",
       "  {'cofactor_in_ch_ikey': 'HAEJSGLKJYIYTB-ZZXKWVIFSA-N',\n",
       "   'cofactor_smiles': 'OC(=O)\\\\C=C\\\\C1=CC=C(C=C1)C(O)=O',\n",
       "   'cofactor_chem_comp_id': 'CIN',\n",
       "   'cofactor_name': '4-Carboxycinnamic Acid',\n",
       "   'cofactor_resource_id': 'DB02126',\n",
       "   'neighbor_flag': 'N',\n",
       "   'pubmed_ids': [10592235],\n",
       "   'resource_name': 'DrugBank',\n",
       "   'resource_version': '5.1',\n",
       "   'target_resource_id': 'P69905'},\n",
       "  {'cofactor_in_ch_ikey': 'TZRXHJWUDPFEEY-UHFFFAOYSA-N',\n",
       "   'cofactor_smiles': '[O-][N+](=O)OCC(CO[N+]([O-])=O)(CO[N+]([O-])=O)CO[N+]([O-])=O',\n",
       "   'cofactor_name': 'Pentaerythritol tetranitrate',\n",
       "   'cofactor_resource_id': 'DB06154',\n",
       "   'mechanism_of_action': 'Pentaerythritol tetranitrate is the lipid soluble polyol ester of nitric acid belonging to the family of _nitro-vasodilators_. Pentaerythritol tetranitrate releases free nitric oxide (NO) after the denitration reaction, which triggers NO-dependent signaling transduction involving soluble _guanylate cyclase (sGC_). Nitric oxide binds reversibly to the ferrous-heme center of sGC, causing conformational change and activating the enzyme. This enzyme activation results in increased cellular concentrations of _cyclic guanosine monophosphate _(cGMP) within the vascular smooth muscle, resulting in vasodilation mediated by cGMP-dependent protein kinases. Additionally, this agent causes dose-dependent arterial and venous bed [L2393].',\n",
       "   'neighbor_flag': 'N',\n",
       "   'resource_name': 'DrugBank',\n",
       "   'resource_version': '5.1',\n",
       "   'target_resource_id': 'P69905'},\n",
       "  {'cofactor_in_ch_ikey': 'AVXQTLSOXWQOHO-UHFFFAOYSA-N',\n",
       "   'cofactor_smiles': 'COC1=C(OCC2=NC=CC=C2)C=C(C)C=C1',\n",
       "   'cofactor_chem_comp_id': 'B77',\n",
       "   'cofactor_name': '2-[(2-methoxy-5-methylphenoxy)methyl]pyridine',\n",
       "   'cofactor_resource_id': 'DB07427',\n",
       "   'neighbor_flag': 'Y',\n",
       "   'pubmed_ids': [10592235],\n",
       "   'resource_name': 'DrugBank',\n",
       "   'resource_version': '5.1',\n",
       "   'target_resource_id': 'P69905'},\n",
       "  {'cofactor_in_ch_ikey': 'MBHBRRBLXCXQKV-UHFFFAOYSA-N',\n",
       "   'cofactor_smiles': 'COC1=CC(OCC2=CC=NC=C2)=C(C)C=C1',\n",
       "   'cofactor_chem_comp_id': 'B78',\n",
       "   'cofactor_name': '4-[(5-methoxy-2-methylphenoxy)methyl]pyridine',\n",
       "   'cofactor_resource_id': 'DB07428',\n",
       "   'neighbor_flag': 'Y',\n",
       "   'pubmed_ids': [10592235],\n",
       "   'resource_name': 'DrugBank',\n",
       "   'resource_version': '5.1',\n",
       "   'target_resource_id': 'P69905'},\n",
       "  {'cofactor_in_ch_ikey': 'CXMXRPHRNRROMY-UHFFFAOYSA-N',\n",
       "   'cofactor_smiles': 'OC(=O)CCCCCCCCC(O)=O',\n",
       "   'cofactor_chem_comp_id': 'DEC',\n",
       "   'cofactor_name': 'Sebacic acid',\n",
       "   'cofactor_resource_id': 'DB07645',\n",
       "   'neighbor_flag': 'N',\n",
       "   'pubmed_ids': [10592235],\n",
       "   'resource_name': 'DrugBank',\n",
       "   'resource_version': '5.1',\n",
       "   'target_resource_id': 'P69905'},\n",
       "  {'cofactor_in_ch_ikey': 'OYJPTSMWFKGZJM-UHFFFAOYSA-N',\n",
       "   'cofactor_smiles': 'CC(C)(OC1=CC=C(NC(=O)NC2=CC(Cl)=CC(Cl)=C2)C=C1)C(O)=O',\n",
       "   'cofactor_chem_comp_id': 'L35',\n",
       "   'cofactor_name': '2-[4-({[(3,5-DICHLOROPHENYL)AMINO]CARBONYL}AMINO)PHENOXY]-2-METHYLPROPANOIC ACID',\n",
       "   'cofactor_resource_id': 'DB08077',\n",
       "   'neighbor_flag': 'Y',\n",
       "   'pubmed_ids': [10592235],\n",
       "   'resource_name': 'DrugBank',\n",
       "   'resource_version': '5.1',\n",
       "   'target_resource_id': 'P69905'},\n",
       "  {'cofactor_in_ch_ikey': 'RXOHFPCZGPKIRD-UHFFFAOYSA-N',\n",
       "   'cofactor_smiles': 'OC(=O)C1=CC2=CC=C(C=C2C=C1)C(O)=O',\n",
       "   'cofactor_chem_comp_id': 'NDD',\n",
       "   'cofactor_name': '2,6-dicarboxynaphthalene',\n",
       "   'cofactor_resource_id': 'DB08262',\n",
       "   'neighbor_flag': 'N',\n",
       "   'pubmed_ids': [10592235],\n",
       "   'resource_name': 'DrugBank',\n",
       "   'resource_version': '5.1',\n",
       "   'target_resource_id': 'P69905'},\n",
       "  {'cofactor_in_ch_ikey': 'BNFRJXLZYUTIII-UHFFFAOYSA-N',\n",
       "   'cofactor_smiles': 'CC1=CC(NC(=O)CC2=CC=C(OC(C)(C)C(O)=O)C=C2)=CC(C)=C1',\n",
       "   'cofactor_chem_comp_id': 'RQ3',\n",
       "   'cofactor_name': 'Efaproxiral',\n",
       "   'cofactor_resource_id': 'DB08486',\n",
       "   'neighbor_flag': 'Y',\n",
       "   'pubmed_ids': [10592235],\n",
       "   'resource_name': 'DrugBank',\n",
       "   'resource_version': '5.1',\n",
       "   'target_resource_id': 'P69905'},\n",
       "  {'cofactor_in_ch_ikey': 'QMKYBPDZANOJGF-UHFFFAOYSA-N',\n",
       "   'cofactor_smiles': 'OC(=O)C1=CC(=CC(=C1)C(O)=O)C(O)=O',\n",
       "   'cofactor_chem_comp_id': 'TMM',\n",
       "   'cofactor_name': 'Trimesic acid',\n",
       "   'cofactor_resource_id': 'DB08632',\n",
       "   'neighbor_flag': 'N',\n",
       "   'pubmed_ids': [10592235],\n",
       "   'resource_name': 'DrugBank',\n",
       "   'resource_version': '5.1',\n",
       "   'target_resource_id': 'P69905'},\n",
       "  {'cofactor_in_ch_ikey': 'IOVCWXUNBOPUCH-UHFFFAOYSA-N',\n",
       "   'cofactor_smiles': 'ON=O',\n",
       "   'cofactor_name': 'Nitrous acid',\n",
       "   'cofactor_resource_id': 'DB09112',\n",
       "   'mechanism_of_action': 'Cyanide has a high affinity for the oxidized form of iron (Fe3+) such as that found in cytochrome oxidase a3 [A19441]. Cyanide binds to and inhibits cytochrome oxidase a3, preventing oxidative phophorylation from occuring. The resultant lack of ATP cannot support normal cellular processes, particularly in the brain. Compensatory increases in anaerobic respiration result in rising levels of lactic acid and subsequent acidosis. \\r\\n\\r\\nNitrite primarily acts by oxidizing hemoglobin to methemoglobin [A19440]. The now oxidized Fe3+ in methemoglobin also binds cyanide with high affinity and accepts cyanide from cytochrome a3. This leaves cytochrome a3 free to resume its function in oxidative phosphorylation. The slow dissociation of cyanide from methemoglobin allows hepatic enzymes such as rhodanese to detoxify the compound without further systemic toxicity occuring. Methemoglobin is reduced back to hemoglobin by methemoglobin reductase allowing the affected blood cells to resume normal functioning.\\r\\n\\r\\nThe reduction of nitrite by hemoglobin results in the formation of nitric oxide [A19442]. Nitric oxide acts as a powerful vasodilator, producing vascular smooth muscle relaxation through activation of soluble guanylate cyclase and the subsequent cyclic guanylyl triphosphate mediated signalling cascade [A19443].',\n",
       "   'neighbor_flag': 'N',\n",
       "   'pubmed_ids': [1569239],\n",
       "   'resource_name': 'DrugBank',\n",
       "   'resource_version': '5.1',\n",
       "   'target_resource_id': 'P69905'},\n",
       "  {'cofactor_in_ch_ikey': 'RYGMFSIKBFXOCR-UHFFFAOYSA-N',\n",
       "   'cofactor_smiles': '[Cu]',\n",
       "   'cofactor_name': 'Copper',\n",
       "   'cofactor_resource_id': 'DB09130',\n",
       "   'mechanism_of_action': \"Copper is absorbed from the gut via high affinity copper uptake protein and likely through low affinity copper uptake protein and natural resistance-associated macrophage protein-2 [A19528]. It is believed that copper is reduced to the Cu1+ form prior to transport. Once inside the enterocyte, it is bound to copper transport protein ATOX1 which shuttles the ion to copper transporting ATPase-1 on the golgi membrane which take up copper into the golgi apparatus. Once copper has been secreted by enterocytes into the systemic circulation it remain largely bound by ceruloplasmin (65-90%), albumin (18%), and alpha 2-macroglobulin (12%). \\r\\n\\r\\nCopper is an essential element in the body and is incorporated into many oxidase enzymes as a cofactor [A19518]. It is also a component of zinc/copper super oxide dismutase, giving it an anti-oxidant role. Copper defiency occurs in Occipital Horn Syndrome and Menke's disease both of which are associated with impaired development of connective tissue due to the lack of copper to act as a cofactor in protein-lysine-6-oxidase. Menke's disease is also associated with progressive neurological impairment leading to death in infancy. The precise mechanisms of the effects of copper deficiency are vague due to the wide range of enzymes which use the ion as a cofactor.\\r\\n\\r\\nCopper appears to reduce the viabilty and motility of spermatozoa [A19526]. This reduces the likelyhood of fertilization with a copper IUD, producing copper's contraceptive effect [A19526]. The exact mechanism of copper's effect on sperm are unknown.\",\n",
       "   'neighbor_flag': 'N',\n",
       "   'pubmed_ids': [23896426],\n",
       "   'resource_name': 'DrugBank',\n",
       "   'resource_version': '5.1',\n",
       "   'target_resource_id': 'P69905'},\n",
       "  {'cofactor_in_ch_ikey': 'MYMOFIZGZYHOMD-UHFFFAOYSA-N',\n",
       "   'cofactor_smiles': 'O=O',\n",
       "   'cofactor_chem_comp_id': 'OXY',\n",
       "   'cofactor_name': 'Oxygen',\n",
       "   'cofactor_resource_id': 'DB09140',\n",
       "   'mechanism_of_action': 'Oxygen therapy increases the arterial pressure of oxygen and is effective in improving gas exchange and oxygen delivery to tissues, provided that there are functional alveolar units. Oxygen plays a critical role as an electron acceptor during oxidative phosphorylation in the electron transport chain through activation of cytochrome c oxidase (terminal enzyme of the electron transport chain). This process achieves successful aerobic respiration in organisms to generate ATP molecules as an energy source in many tissues. Oxygen supplementation acts to restore normal cellular activity at the mitochondrial level and reduce metabolic acidosis. There is also evidence that oxygen may interact with O2-sensitive voltage-gated potassium channels in glomus cells and cause hyperpolarization of mitochondrial membrane [A19120]. ',\n",
       "   'neighbor_flag': 'Y',\n",
       "   'pubmed_ids': [1634355],\n",
       "   'resource_name': 'DrugBank',\n",
       "   'resource_version': '5.1',\n",
       "   'target_resource_id': 'P69905'},\n",
       "  {'cofactor_in_ch_ikey': 'CADNYOZXMIKYPR-UHFFFAOYSA-B',\n",
       "   'cofactor_smiles': '[Fe+3].[Fe+3].[Fe+3].[Fe+3].[O-]P([O-])(=O)OP([O-])([O-])=O.[O-]P([O-])(=O)OP([O-])([O-])=O.[O-]P([O-])(=O)OP([O-])([O-])=O',\n",
       "   'cofactor_name': 'Ferric pyrophosphate',\n",
       "   'cofactor_resource_id': 'DB09147',\n",
       "   'mechanism_of_action': 'The usage of ferric pyrophosphate is based on the strong complex formation between these two species. Besides, the capacity of pyrophosphate to trigger iron removal from transferrin, enhance iron transfer from transferrin to ferritin and promote iron exchange between transferrin molecules. These properties make it a very suitable compound for parenteral administration, iron delivery into circulation and incorporation into hemoglobin.[A31979]',\n",
       "   'neighbor_flag': 'N',\n",
       "   'pubmed_ids': [10231452],\n",
       "   'resource_name': 'DrugBank',\n",
       "   'resource_version': '5.1',\n",
       "   'target_resource_id': 'P69905'},\n",
       "  {'cofactor_in_ch_ikey': 'MQBDAEHWGRMADS-XNHLMZCASA-M',\n",
       "   'cofactor_smiles': '[O--].[O--].[O--].[Na+].[Fe+3].[Fe+3].OC[C@@H](O)[C@@H](O)[C@H](O)[C@@H](O)C([O-])=O.OC[C@H]1O[C@@](CO)(O[C@H]2O[C@H](CO)[C@@H](O)[C@H](O)[C@H]2O)[C@@H](O)[C@@H]1O.OC[C@H]1O[C@@](CO)(O[C@H]2O[C@H](CO)[C@@H](O)[C@H](O)[C@H]2O)[C@@H](O)[C@@H]1O.OC[C@H]1O[C@@](CO)(O[C@H]2O[C@H](CO)[C@@H](O)[C@H](O)[C@H]2O)[C@@H](O)[C@@H]1O.OC[C@H]1O[C@@](CO)(O[C@H]2O[C@H](CO)[C@@H](O)[C@H](O)[C@H]2O)[C@@H](O)[C@@H]1O.OC[C@H]1O[C@@](CO)(O[C@H]2O[C@H](CO)[C@@H](O)[C@H](O)[C@H]2O)[C@@H](O)[C@@H]1O',\n",
       "   'cofactor_name': 'Sodium ferric gluconate complex',\n",
       "   'cofactor_resource_id': 'DB09517',\n",
       "   'mechanism_of_action': 'The complex is endocytosed by macrophages of the reticuloendothelial system. Within an endosome of the macrophage , lysosome fuses with the endosome creating an acidic environment leading to the cleavage of the complex from iron. Iron is then incorporated in ferritin, transferrin or hemoglobin. Sodium ferric gluconate also normalizes RBC production by binding with hemoglobin ',\n",
       "   'neighbor_flag': 'N',\n",
       "   'resource_name': 'DrugBank',\n",
       "   'resource_version': '5.1',\n",
       "   'target_resource_id': 'P69905'},\n",
       "  {'cofactor_in_ch_ikey': 'BAUYGSIQEAFULO-UHFFFAOYSA-L',\n",
       "   'cofactor_smiles': '[Fe++].[O-]S([O-])(=O)=O',\n",
       "   'cofactor_name': 'Ferrous sulfate anhydrous',\n",
       "   'cofactor_resource_id': 'DB13257',\n",
       "   'mechanism_of_action': 'Iron is required to maintain optimal health, particularly for helping to form red blood cells (RBC) that carry oxygen around the body. A deficiency in iron indicates that the body cannot produce enough normal red blood cells.[A32514,L11800] Iron deficiency anemia occurs when body stores of iron decrease to very low levels, and the stored iron is insufficient to support normal red blood cell (RBC) production. Insufficient dietary iron, impaired iron absorption, bleeding, pregnancy, or loss of iron through the urine can lead to iron deficiency.[A32514,L11794]  Symptoms of iron deficiency anemia include fatigue, breathlessness, palpitations, dizziness, and headache.\\r\\n\\r\\nTaking iron in supplement form, such as ferrous sulfate, allows for more rapid increases in iron levels when dietary supply and stores are not sufficient.[L2175] Iron is transported by the divalent metal transporter 1 (DMT1) across the endolysosomal membrane to enter the macrophage. It can then can be incorporated into ferritin and be stored in the macrophage or carried of the macrophage by ferroportin. This exported iron is  oxidized by the enzyme to ceruloplasmin to Fe3+, followed by sequestration by transferrin for transport in the serum to various sites, including the bone marrow for hemoglobin synthesis or into the liver.[A32524] Iron combines with porphyrin and globin chains to form hemoglobin, which is critical for oxygen delivery from the lungs to other tissues.[L2263]',\n",
       "   'neighbor_flag': 'N',\n",
       "   'pubmed_ids': [24310424, 21694802, 18954837],\n",
       "   'resource_name': 'DrugBank',\n",
       "   'resource_version': '5.1',\n",
       "   'target_resource_id': 'P69905'},\n",
       "  {'cofactor_in_ch_ikey': 'SXAWSYZURCZSDX-UHFFFAOYSA-B',\n",
       "   'cofactor_smiles': '[Fe+3].[Fe+3].[Fe+3].[Fe+3].OP(O)(=O)OP(O)(O)=O.OP(O)(=O)OP(O)(O)=O.OP([O-])(=O)OP([O-])([O-])=O.OC(CC([O-])=O)(CC([O-])=O)C([O-])=O.OC(CC([O-])=O)(CC([O-])=O)C([O-])=O.OC(CC([O-])=O)(CC([O-])=O)C([O-])=O',\n",
       "   'cofactor_name': 'Ferric pyrophosphate citrate',\n",
       "   'cofactor_resource_id': 'DB13995',\n",
       "   'mechanism_of_action': 'The usage of ferric pyrophosphate is based on the strong complex formation between these two species. Besides, the capacity of pyrophosphate to trigger iron removal from transferrin, enhance iron transfer from transferrin to ferritin and promote iron exchange between transferrin molecules. These properties make it a very suitable compound for parenteral administration, iron delivery into circulation and incorporation into hemoglobin.[A31979]',\n",
       "   'neighbor_flag': 'N',\n",
       "   'pubmed_ids': [10231452],\n",
       "   'resource_name': 'DrugBank',\n",
       "   'resource_version': '5.1',\n",
       "   'target_resource_id': 'P69905'},\n",
       "  {'cofactor_in_ch_ikey': 'DJWUNCQRNNEAKC-UHFFFAOYSA-L',\n",
       "   'cofactor_smiles': '[Zn++].CC([O-])=O.CC([O-])=O',\n",
       "   'cofactor_name': 'Zinc acetate',\n",
       "   'cofactor_resource_id': 'DB14487',\n",
       "   'mechanism_of_action': '**Zinc has three primary biological roles**: _catalytic_, _structural_, and _regulatory_. The catalytic and structural role of zinc is well established, and there are various noteworthy reviews on these functions. For example, zinc is a structural constituent in numerous proteins, inclusive of growth factors, cytokines, receptors, enzymes, and transcription factors for different cellular signaling pathways. It is implicated in numerous cellular processes as a cofactor for approximately 3000 human proteins including enzymes, nuclear factors, and hormones [L2096].\\r\\n\\r\\nZinc promotes resistance to epithelial apoptosis through cell protection (cytoprotection) against reactive oxygen species and bacterial toxins, likely through the antioxidant activity of the cysteine-rich metallothioneins [A32419].\\r\\n\\r\\nIn HL-60 cells (promyelocytic leukemia cell line), zinc enhances the up-regulation of A20 mRNA, which, via TRAF pathway, decreases NF-kappaB activation, leading to decreased gene expression and generation of tumor necrosis factor-alpha (TNF-alpha), IL-1beta, and IL-8 [A32418].\\r\\n\\r\\nThere are several mechanisms of action of zinc on acute diarrhea. Various mechanisms are specific to the gastrointestinal system: zinc restores mucosal barrier integrity and enterocyte brush-border enzyme activity, it promotes the production of antibodies and circulating lymphocytes against intestinal pathogens, and has a direct effect on ion channels, acting as a potassium channel blocker of adenosine 3-5-cyclic monophosphate-mediated chlorine secretion. Cochrane researchers examined the evidence available up to 30 September 2016 [L2106].\\r\\n\\r\\nZinc deficiency in humans decreases the activity of serum _thymulin_ (a hormone of the thymus), which is necessary for the maturation of T-helper cells. T-helper 1 (Th(1)) cytokines are decreased but T-helper 2 (Th(2)) cytokines are not affected by zinc deficiency in humans [A342417].\\r\\n\\r\\nThe change of _Th(1)_ to _Th(2)_ function leads to cell-mediated immune dysfunction. Because IL-2 production (Th(1) cytokine) is decreased, this causes decreased activity of natural-killer-cell (NK cell) and T cytolytic cells, normally involved in killing viruses, bacteria, and malignant cells [A3424]. \\r\\n\\r\\nIn humans, zinc deficiency may lead to the generation of new CD4+ T cells, produced in the thymus. In cell culture studies (HUT-78, a Th(0) human malignant lymphoblastoid cell line), as a result of zinc deficiency, nuclear factor-kappaB (NF-kappaB) activation, phosphorylation of IkappaB, and binding of NF-kappaB to DNA are decreased and this results in decreased Th(1) cytokine production [A32417].\\r\\n\\r\\nIn another study, zinc supplementation in human subjects suppressed the gene expression and production of pro-inflammatory cytokines and decreased oxidative stress markers [A3424]. In HL-60 cells (a human pro-myelocytic leukemia cell line), zinc deficiency increased the levels of TNF-alpha, IL-1beta, and IL-8 cytokines and mRNA. In such cells, zinc was found to induce A20, a zinc finger protein that inhibited NF-kappaB activation by the tumor necrosis factor receptor-associated factor pathway. This process decreased gene expression of pro-inflammatory cytokines and oxidative stress markers [A32417].\\r\\n\\r\\nThe exact mechanism of zinc in acne treatment is poorly understood. However, zinc is considered to act directly on microbial inflammatory equilibrium and facilitate antibiotic absorption when used in combination with other agents. Topical zinc alone as well as in combination with other agents may be efficacious because of its anti-inflammatory activity and ability to reduce P. acnes bacteria by the inhibition of P. acnes lipases and free fatty acid levels [L2102].',\n",
       "   'neighbor_flag': 'N',\n",
       "   'pubmed_ids': [23896426],\n",
       "   'resource_name': 'DrugBank',\n",
       "   'resource_version': '5.1',\n",
       "   'target_resource_id': 'P69905'},\n",
       "  {'cofactor_in_ch_ikey': 'VRIVJOXICYMTAG-IYEMJOQQSA-L',\n",
       "   'cofactor_smiles': '[Fe++].[H][C@@](O)(CO)[C@@]([H])(O)[C@]([H])(O)[C@@]([H])(O)C([O-])=O.[H][C@@](O)(CO)[C@@]([H])(O)[C@]([H])(O)[C@@]([H])(O)C([O-])=O',\n",
       "   'cofactor_name': 'Ferrous gluconate',\n",
       "   'cofactor_resource_id': 'DB14488',\n",
       "   'mechanism_of_action': 'Iron is necessary for the production of hemoglobin. Iron-deficiency can lead to decreased production of hemoglobin and a microcytic, hypochromic anemia.',\n",
       "   'neighbor_flag': 'N',\n",
       "   'pubmed_ids': [16901899],\n",
       "   'resource_name': 'DrugBank',\n",
       "   'resource_version': '5.1',\n",
       "   'target_resource_id': 'P69905'},\n",
       "  {'cofactor_in_ch_ikey': 'MDXRFOWKIZPNTA-UHFFFAOYSA-L',\n",
       "   'cofactor_smiles': '[Fe++].[O-]C(=O)CCC([O-])=O',\n",
       "   'cofactor_name': 'Ferrous succinate',\n",
       "   'cofactor_resource_id': 'DB14489',\n",
       "   'mechanism_of_action': 'Iron is necessary for the production of hemoglobin. Iron-deficiency can lead to decreased production of hemoglobin and a microcytic, hypochromic anemia.',\n",
       "   'neighbor_flag': 'N',\n",
       "   'pubmed_ids': [16901899],\n",
       "   'resource_name': 'DrugBank',\n",
       "   'resource_version': '5.1',\n",
       "   'target_resource_id': 'P69905'},\n",
       "  {'cofactor_in_ch_ikey': 'RFBYLSCVRUTUSB-ZZMNMWMASA-L',\n",
       "   'cofactor_smiles': '[Fe++].[H][C@](O)(CO)[C@@]1([H])OC(=O)C(O)=C1O.[H][C@](O)(CO)[C@@]1([H])OC(=O)C([O-])=C1[O-]',\n",
       "   'cofactor_name': 'Ferrous ascorbate',\n",
       "   'cofactor_resource_id': 'DB14490',\n",
       "   'mechanism_of_action': 'Iron is necessary for the production of hemoglobin. Iron-deficiency can lead to decreased production of hemoglobin and a microcytic, hypochromic anemia.',\n",
       "   'neighbor_flag': 'N',\n",
       "   'pubmed_ids': [16901899],\n",
       "   'resource_name': 'DrugBank',\n",
       "   'resource_version': '5.1',\n",
       "   'target_resource_id': 'P69905'},\n",
       "  {'cofactor_in_ch_ikey': 'PMVSDNDAUGGCCE-TYYBGVCCSA-L',\n",
       "   'cofactor_smiles': '[Fe++].[H]\\\\C(=C(\\\\[H])C([O-])=O)C([O-])=O',\n",
       "   'cofactor_name': 'Ferrous fumarate',\n",
       "   'cofactor_resource_id': 'DB14491',\n",
       "   'mechanism_of_action': 'Iron is necessary for the production of hemoglobin. Iron-deficiency can lead to decreased production of hemoglobin and a microcytic, hypochromic anemia.',\n",
       "   'neighbor_flag': 'N',\n",
       "   'pubmed_ids': [16901899],\n",
       "   'resource_name': 'DrugBank',\n",
       "   'resource_version': '5.1',\n",
       "   'target_resource_id': 'P69905'},\n",
       "  {'cofactor_in_ch_ikey': 'YJYOLOWXCPIBSY-UHFFFAOYSA-L',\n",
       "   'cofactor_smiles': '[Fe++].NCC([O-])=O.NCC([O-])=O.OS(O)(=O)=O',\n",
       "   'cofactor_name': 'Ferrous glycine sulfate',\n",
       "   'cofactor_resource_id': 'DB14501',\n",
       "   'mechanism_of_action': 'Iron is necessary for the production of hemoglobin. Iron-deficiency can lead to decreased production of hemoglobin and a microcytic, hypochromic anemia.',\n",
       "   'neighbor_flag': 'N',\n",
       "   'pubmed_ids': [16901899],\n",
       "   'resource_name': 'DrugBank',\n",
       "   'resource_version': '5.1',\n",
       "   'target_resource_id': 'P69905'},\n",
       "  {'cofactor_in_ch_ikey': 'JIAARYAFYJHUJI-UHFFFAOYSA-L',\n",
       "   'cofactor_smiles': '[Cl-].[Cl-].[Zn++]',\n",
       "   'cofactor_name': 'Zinc chloride',\n",
       "   'cofactor_resource_id': 'DB14533',\n",
       "   'mechanism_of_action': 'Zinc performs catalytic, structural, and regulatory roles in the body. Zinc is a component of approximately 3000 human proteins.[A204104]\\r\\n\\r\\nZinc is cytoprotective against reactive oxygen species mediated apoptosis through the action of metallothioneins.[A32419]\\r\\n\\r\\nIn a promyelocytic leukemia cell line, zinc enhances the up-regulation of A20 mRNA, which, via the TRAF pathway, decreases NF-kappaB activation, leading to decreased gene expression and generation of TNF-, IL-1, and IL-8 [A32418].\\r\\n\\r\\nIn patients with diarrhea, zinc restores mucosal barrier integrity, restores enterocyte brush-border enzyme activity, promotes the production of antibodies, and promotes the production of circulating lymphocytes against intestinal pathogens.[A204101] Zinc also directly affects ion channels as a potassium channel blocker of cAMP-mediated chlorine secretion.[A204101]\\r\\n\\r\\nZinc deficiency decreases thymulin, inhibiting T-helper cell maturation and decreased Th-1 cytokines like IL-2.[A32417] Decreased IL-2 decreases the activity of NK cells and CD8+ T cells.[A32417] Zinc deficiency also leads to the generation of CD4+ T cells, decreased NF-B activation, decreased phosphorylation of IB, and decreased binding of NF-B to DNA.[A32417]',\n",
       "   'neighbor_flag': 'N',\n",
       "   'pubmed_ids': [646791],\n",
       "   'resource_name': 'DrugBank',\n",
       "   'resource_version': '5.1',\n",
       "   'target_resource_id': 'P69905'},\n",
       "  {'cofactor_name': 'Zinc sulfate, unspecified form',\n",
       "   'cofactor_resource_id': 'DB14548',\n",
       "   'neighbor_flag': 'N',\n",
       "   'pubmed_ids': [646791],\n",
       "   'resource_name': 'DrugBank',\n",
       "   'resource_version': '5.1',\n",
       "   'target_resource_id': 'P69905'},\n",
       "  {'cofactor_in_ch_ikey': 'FWCVZAQENIZVMY-UHFFFAOYSA-N',\n",
       "   'cofactor_smiles': 'CC(C)N1N=CC=C1C1=C(COC2=CC=CC(O)=C2C=O)C=CC=N1',\n",
       "   'cofactor_name': 'Voxelotor',\n",
       "   'cofactor_resource_id': 'DB14975',\n",
       "   'mechanism_of_action': 'Deoxygenated sickle hemoglobin (HbS) polymerization is the causal factor for sickle cell disease.  The genetic mutation associated with this disease leads to the formation of abnormal, sickle shaped red blood cells that aggregate and block blood vessels throughout the body, causing vaso-occlusive crises.[T734] Voxelotor binds irreversibly with the Nterminal valine of the chain of hemoglobin, leading to an allosteric modification of Hb20, which increases the affinity for oxygen. Oxygenated HbS does not polymerize.[A188126,A188129] By directly blocking HbS polymerization, voxelotor can successfully treat sickle cell disease by preventing the formation of abnormally shaped cells, which eventually cause lack of oxygenation and blood flow to organs.[A188123,A188138,T734]',\n",
       "   'neighbor_flag': 'N',\n",
       "   'pubmed_ids': [30655275],\n",
       "   'resource_name': 'DrugBank',\n",
       "   'resource_version': '5.1',\n",
       "   'target_resource_id': 'P69905'},\n",
       "  {'cofactor_in_ch_ikey': 'JTQTXQSGPZRXJF-DOJSGGEQSA-N',\n",
       "   'cofactor_smiles': '[Fe+3].OC[C@H](O)[C@@H](O)[C@H](O)[C@H](O)CO[C@H]1O[C@H](CO[C@H]2O[C@H](CO)[C@@H](O)[C@H](O)[C@H]2O)[C@@H](O)[C@H](O)[C@H]1O',\n",
       "   'cofactor_name': 'Ferric derisomaltose',\n",
       "   'cofactor_resource_id': 'DB15617',\n",
       "   'mechanism_of_action': 'This drug is a complex made of iron (III) hydroxide and derisomaltose, which is an iron carbohydrate oligosaccharide that works to releases iron. The released iron then binds to the transport protein, transferrin, and is taken to erythroid precursor cells[A190528] for incorporation into the hemoglobin molecule.[L11581,L11617]',\n",
       "   'neighbor_flag': 'N',\n",
       "   'pubmed_ids': [29261547, 24310424],\n",
       "   'resource_name': 'DrugBank',\n",
       "   'resource_version': '5.1',\n",
       "   'target_resource_id': 'P69905'},\n",
       "  {'binding_assay_value': 5.09,\n",
       "   'binding_assay_value_type': 'pEC50',\n",
       "   'cofactor_smiles': 'CC(C)N1N=CC=C1C1=C(COC2=C(C=O)C(O)=CC=C2)C=CC=N1',\n",
       "   'cofactor_resource_id': 'CHEMBL4101807',\n",
       "   'mechanism_of_action': 'Voxelotor is a hemoglobin S (HbS) polymerization inhibitor that binds to HbS with a 1:1 stoichiometry and exhibits preferential partitioning to red blood cells (RBCs). By increasing the affinity of Hb for oxygen, voxelotor demonstrates dose-dependent inhibition of HbS polymerization. Nonclinical studies suggest that voxelotor may inhibit RBC sickling, improve RBC deformability, and reduce whole blood viscosity.',\n",
       "   'neighbor_flag': 'N',\n",
       "   'resource_name': 'Pharos',\n",
       "   'resource_version': '6.11.0',\n",
       "   'target_resource_id': '6231'}],\n",
       " 'rcsb_id': '4HHB_1',\n",
       " 'rcsb_genomic_lineage': [{'id': '9606',\n",
       "   'name': 'Homo sapiens (man)',\n",
       "   'depth': 0},\n",
       "  {'id': '9606:16', 'name': 'Chromosome 16', 'depth': 1},\n",
       "  {'id': '9606:16:hemoglobin_subunit_alpha_2',\n",
       "   'name': 'hemoglobin subunit alpha 2',\n",
       "   'depth': 2},\n",
       "  {'id': '9606:16:hemoglobin_subunit_alpha_1',\n",
       "   'name': 'hemoglobin subunit alpha 1',\n",
       "   'depth': 2}],\n",
       " 'rcsb_cluster_flexibility': {'link': 'http://pdbflex.org/cluster.html#!/1babA/252/4hhbA',\n",
       "  'label': 'Low',\n",
       "  'avg_rmsd': 0.738,\n",
       "  'max_rmsd': 4.392,\n",
       "  'provenance_code': 'PDBFlex'},\n",
       " 'rcsb_latest_revision': {'major_revision': 3, 'minor_revision': 0}}"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "response = requests.get('https://data.rcsb.org/rest/v1/core/polymer_entity/4hhb/1')\n",
    "json.loads(response.text)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {
    "slideshow": {
     "slide_type": "slide"
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'rcsb_ligand_neighbors': [{'atom_id': 'NE2',\n",
       "   'auth_seq_id': 87,\n",
       "   'comp_id': 'HIS',\n",
       "   'distance': 2.143,\n",
       "   'ligand_asym_id': 'E',\n",
       "   'ligand_atom_id': 'FE',\n",
       "   'ligand_comp_id': 'HEM',\n",
       "   'ligand_entity_id': '3',\n",
       "   'ligand_is_bound': 'Y',\n",
       "   'ligand_model_id': 1,\n",
       "   'seq_id': 87}],\n",
       " 'rcsb_polymer_entity_instance_container_identifiers': {'asym_id': 'A',\n",
       "  'auth_asym_id': 'A',\n",
       "  'auth_to_entity_poly_seq_mapping': ['1',\n",
       "   '2',\n",
       "   '3',\n",
       "   '4',\n",
       "   '5',\n",
       "   '6',\n",
       "   '7',\n",
       "   '8',\n",
       "   '9',\n",
       "   '10',\n",
       "   '11',\n",
       "   '12',\n",
       "   '13',\n",
       "   '14',\n",
       "   '15',\n",
       "   '16',\n",
       "   '17',\n",
       "   '18',\n",
       "   '19',\n",
       "   '20',\n",
       "   '21',\n",
       "   '22',\n",
       "   '23',\n",
       "   '24',\n",
       "   '25',\n",
       "   '26',\n",
       "   '27',\n",
       "   '28',\n",
       "   '29',\n",
       "   '30',\n",
       "   '31',\n",
       "   '32',\n",
       "   '33',\n",
       "   '34',\n",
       "   '35',\n",
       "   '36',\n",
       "   '37',\n",
       "   '38',\n",
       "   '39',\n",
       "   '40',\n",
       "   '41',\n",
       "   '42',\n",
       "   '43',\n",
       "   '44',\n",
       "   '45',\n",
       "   '46',\n",
       "   '47',\n",
       "   '48',\n",
       "   '49',\n",
       "   '50',\n",
       "   '51',\n",
       "   '52',\n",
       "   '53',\n",
       "   '54',\n",
       "   '55',\n",
       "   '56',\n",
       "   '57',\n",
       "   '58',\n",
       "   '59',\n",
       "   '60',\n",
       "   '61',\n",
       "   '62',\n",
       "   '63',\n",
       "   '64',\n",
       "   '65',\n",
       "   '66',\n",
       "   '67',\n",
       "   '68',\n",
       "   '69',\n",
       "   '70',\n",
       "   '71',\n",
       "   '72',\n",
       "   '73',\n",
       "   '74',\n",
       "   '75',\n",
       "   '76',\n",
       "   '77',\n",
       "   '78',\n",
       "   '79',\n",
       "   '80',\n",
       "   '81',\n",
       "   '82',\n",
       "   '83',\n",
       "   '84',\n",
       "   '85',\n",
       "   '86',\n",
       "   '87',\n",
       "   '88',\n",
       "   '89',\n",
       "   '90',\n",
       "   '91',\n",
       "   '92',\n",
       "   '93',\n",
       "   '94',\n",
       "   '95',\n",
       "   '96',\n",
       "   '97',\n",
       "   '98',\n",
       "   '99',\n",
       "   '100',\n",
       "   '101',\n",
       "   '102',\n",
       "   '103',\n",
       "   '104',\n",
       "   '105',\n",
       "   '106',\n",
       "   '107',\n",
       "   '108',\n",
       "   '109',\n",
       "   '110',\n",
       "   '111',\n",
       "   '112',\n",
       "   '113',\n",
       "   '114',\n",
       "   '115',\n",
       "   '116',\n",
       "   '117',\n",
       "   '118',\n",
       "   '119',\n",
       "   '120',\n",
       "   '121',\n",
       "   '122',\n",
       "   '123',\n",
       "   '124',\n",
       "   '125',\n",
       "   '126',\n",
       "   '127',\n",
       "   '128',\n",
       "   '129',\n",
       "   '130',\n",
       "   '131',\n",
       "   '132',\n",
       "   '133',\n",
       "   '134',\n",
       "   '135',\n",
       "   '136',\n",
       "   '137',\n",
       "   '138',\n",
       "   '139',\n",
       "   '140',\n",
       "   '141'],\n",
       "  'entity_id': '1',\n",
       "  'entry_id': '4HHB',\n",
       "  'rcsb_id': '4HHB.A'},\n",
       " 'rcsb_polymer_instance_annotation': [{'annotation_id': '1.10.490.10',\n",
       "   'assignment_version': 'v4_2_0',\n",
       "   'name': 'Globins',\n",
       "   'ordinal': 1,\n",
       "   'provenance_source': 'CATH',\n",
       "   'type': 'CATH',\n",
       "   'annotation_lineage': [{'depth': 1, 'id': '1', 'name': 'Mainly Alpha'},\n",
       "    {'depth': 2, 'id': '1.10', 'name': 'Orthogonal Bundle'},\n",
       "    {'depth': 3, 'id': '1.10.490', 'name': 'Globin-like'},\n",
       "    {'depth': 4, 'id': '1.10.490.10', 'name': 'Globins'}]},\n",
       "  {'annotation_id': 'd4hhba_',\n",
       "   'assignment_version': '2.08-stable',\n",
       "   'name': 'Hemoglobin, alpha-chain',\n",
       "   'ordinal': 5,\n",
       "   'provenance_source': 'SCOPe',\n",
       "   'type': 'SCOP',\n",
       "   'annotation_lineage': [{'depth': 1,\n",
       "     'id': '46456',\n",
       "     'name': 'All alpha proteins'},\n",
       "    {'depth': 2, 'id': '46457', 'name': 'Globin-like'},\n",
       "    {'depth': 3, 'id': '46458', 'name': 'Globin-like'},\n",
       "    {'depth': 4, 'id': '46463', 'name': 'Globins'},\n",
       "    {'depth': 5, 'id': '46486', 'name': 'Hemoglobin, alpha-chain'}]},\n",
       "  {'annotation_id': '8039836',\n",
       "   'assignment_version': '2021-09-29',\n",
       "   'name': 'Globin-like',\n",
       "   'ordinal': 9,\n",
       "   'provenance_source': 'SCOP2B',\n",
       "   'type': 'SCOP2',\n",
       "   'annotation_lineage': [{'depth': 1, 'id': '1', 'name': 'Globular proteins'},\n",
       "    {'depth': 2, 'id': '1000000', 'name': 'All alpha proteins'},\n",
       "    {'depth': 3, 'id': '2000002', 'name': 'Globin-like'},\n",
       "    {'depth': 4, 'id': '3000554', 'name': 'Globin-like'}]},\n",
       "  {'annotation_id': 'e4hhbA1',\n",
       "   'assignment_version': '1.6',\n",
       "   'name': 'Globin',\n",
       "   'ordinal': 13,\n",
       "   'provenance_source': 'ECOD',\n",
       "   'type': 'ECOD',\n",
       "   'annotation_lineage': [{'depth': 1,\n",
       "     'id': '100006',\n",
       "     'name': 'A: alpha arrays'},\n",
       "    {'depth': 2, 'id': '200282', 'name': 'X: Globin-like (From Topology)'},\n",
       "    {'depth': 3, 'id': '300614', 'name': 'H: Globin-like (From Topology)'},\n",
       "    {'depth': 4, 'id': '400686', 'name': 'T: Globin-like'},\n",
       "    {'depth': 5, 'id': '504160', 'name': 'F: Globin'}]}],\n",
       " 'rcsb_polymer_instance_feature': [{'assignment_version': 'v4_2_0',\n",
       "   'feature_id': '1.10.490.10',\n",
       "   'name': 'Globins',\n",
       "   'ordinal': 1,\n",
       "   'provenance_source': 'CATH',\n",
       "   'reference_scheme': 'PDB entity',\n",
       "   'type': 'CATH',\n",
       "   'feature_positions': [{'beg_seq_id': 1, 'end_seq_id': 141}],\n",
       "   'additional_properties': [{'name': 'CATH_NAME', 'values': ['Globins']},\n",
       "    {'name': 'CATH_DOMAIN_ID', 'values': ['4hhbA00']}]},\n",
       "  {'assignment_version': '2.08-stable',\n",
       "   'feature_id': 'd4hhba_',\n",
       "   'name': 'Hemoglobin, alpha-chain',\n",
       "   'ordinal': 5,\n",
       "   'provenance_source': 'SCOPe',\n",
       "   'reference_scheme': 'PDB entity',\n",
       "   'type': 'SCOP',\n",
       "   'feature_positions': [{'beg_seq_id': 1, 'end_seq_id': 141}],\n",
       "   'additional_properties': [{'name': 'SCOP_NAME',\n",
       "     'values': ['Hemoglobin', ' alpha-chain']},\n",
       "    {'name': 'SCOP_DOMAIN_ID', 'values': ['d4hhba_']},\n",
       "    {'name': 'SCOP_SUN_ID', 'values': ['46486']}]},\n",
       "  {'assignment_version': '2021-09-29',\n",
       "   'feature_id': '8039836',\n",
       "   'name': 'Globin-like',\n",
       "   'ordinal': 9,\n",
       "   'provenance_source': 'SCOP2B',\n",
       "   'reference_scheme': 'PDB entity',\n",
       "   'type': 'SCOP2B_SUPERFAMILY',\n",
       "   'feature_positions': [{'beg_seq_id': 2, 'end_seq_id': 141}],\n",
       "   'additional_properties': [{'name': 'SCOP2_SUPERFAMILY_NAME',\n",
       "     'values': ['Globin-like']},\n",
       "    {'name': 'SCOP2_DOMAIN_ID', 'values': ['8039836']},\n",
       "    {'name': 'SCOP2_SUPERFAMILY_ID', 'values': ['3000554']}]},\n",
       "  {'assignment_version': '1.6',\n",
       "   'feature_id': 'e4hhbA1',\n",
       "   'name': 'Globin',\n",
       "   'ordinal': 13,\n",
       "   'provenance_source': 'ECOD',\n",
       "   'reference_scheme': 'PDB entity',\n",
       "   'type': 'ECOD',\n",
       "   'feature_positions': [{'beg_seq_id': 1, 'end_seq_id': 141}],\n",
       "   'additional_properties': [{'name': 'ECOD_FAMILY_NAME',\n",
       "     'values': ['Globin']},\n",
       "    {'name': 'ECOD_DOMAIN_ID', 'values': ['e4hhbA1']}]},\n",
       "  {'assignment_version': 'V1.0',\n",
       "   'feature_id': 'HELX_P1',\n",
       "   'name': 'helix',\n",
       "   'ordinal': 17,\n",
       "   'provenance_source': 'PROMOTIF',\n",
       "   'reference_scheme': 'PDB entity',\n",
       "   'type': 'HELIX_P',\n",
       "   'feature_positions': [{'beg_seq_id': 3, 'end_seq_id': 18}]},\n",
       "  {'assignment_version': 'V1.0',\n",
       "   'feature_id': 'HELX_P2',\n",
       "   'name': 'helix',\n",
       "   'ordinal': 18,\n",
       "   'provenance_source': 'PROMOTIF',\n",
       "   'reference_scheme': 'PDB entity',\n",
       "   'type': 'HELIX_P',\n",
       "   'feature_positions': [{'beg_seq_id': 20, 'end_seq_id': 35}]},\n",
       "  {'assignment_version': 'V1.0',\n",
       "   'feature_id': 'HELX_P3',\n",
       "   'name': 'helix',\n",
       "   'ordinal': 19,\n",
       "   'provenance_source': 'PROMOTIF',\n",
       "   'reference_scheme': 'PDB entity',\n",
       "   'type': 'HELIX_P',\n",
       "   'feature_positions': [{'beg_seq_id': 36, 'end_seq_id': 42}]},\n",
       "  {'assignment_version': 'V1.0',\n",
       "   'feature_id': 'HELX_P4',\n",
       "   'name': 'helix',\n",
       "   'ordinal': 20,\n",
       "   'provenance_source': 'PROMOTIF',\n",
       "   'reference_scheme': 'PDB entity',\n",
       "   'type': 'HELIX_P',\n",
       "   'feature_positions': [{'beg_seq_id': 50, 'end_seq_id': 51}]},\n",
       "  {'assignment_version': 'V1.0',\n",
       "   'feature_id': 'HELX_P5',\n",
       "   'name': 'helix',\n",
       "   'ordinal': 21,\n",
       "   'provenance_source': 'PROMOTIF',\n",
       "   'reference_scheme': 'PDB entity',\n",
       "   'type': 'HELIX_P',\n",
       "   'feature_positions': [{'beg_seq_id': 52, 'end_seq_id': 71}]},\n",
       "  {'assignment_version': 'V1.0',\n",
       "   'feature_id': 'HELX_P6',\n",
       "   'name': 'helix',\n",
       "   'ordinal': 22,\n",
       "   'provenance_source': 'PROMOTIF',\n",
       "   'reference_scheme': 'PDB entity',\n",
       "   'type': 'HELIX_P',\n",
       "   'feature_positions': [{'beg_seq_id': 80, 'end_seq_id': 88}]},\n",
       "  {'assignment_version': 'V1.0',\n",
       "   'feature_id': 'HELX_P7',\n",
       "   'name': 'helix',\n",
       "   'ordinal': 23,\n",
       "   'provenance_source': 'PROMOTIF',\n",
       "   'reference_scheme': 'PDB entity',\n",
       "   'type': 'HELIX_P',\n",
       "   'feature_positions': [{'beg_seq_id': 94, 'end_seq_id': 112}]},\n",
       "  {'assignment_version': 'V1.0',\n",
       "   'feature_id': 'HELX_P8',\n",
       "   'name': 'helix',\n",
       "   'ordinal': 24,\n",
       "   'provenance_source': 'PROMOTIF',\n",
       "   'reference_scheme': 'PDB entity',\n",
       "   'type': 'HELIX_P',\n",
       "   'feature_positions': [{'beg_seq_id': 118, 'end_seq_id': 138}]},\n",
       "  {'assignment_version': 'V1.0',\n",
       "   'feature_id': '1',\n",
       "   'name': 'unassigned secondary structure',\n",
       "   'ordinal': 49,\n",
       "   'provenance_source': 'PROMOTIF',\n",
       "   'reference_scheme': 'PDB entity',\n",
       "   'type': 'UNASSIGNED_SEC_STRUCT',\n",
       "   'feature_positions': [{'beg_seq_id': 1, 'end_seq_id': 2},\n",
       "    {'beg_seq_id': 19, 'end_seq_id': 19},\n",
       "    {'beg_seq_id': 43, 'end_seq_id': 49},\n",
       "    {'beg_seq_id': 72, 'end_seq_id': 79},\n",
       "    {'beg_seq_id': 89, 'end_seq_id': 93},\n",
       "    {'beg_seq_id': 113, 'end_seq_id': 117},\n",
       "    {'beg_seq_id': 139, 'end_seq_id': 141}]},\n",
       "  {'assignment_version': 'V1.0',\n",
       "   'description': 'Software generated binding site for ligand entity 3 component HEM instance E chain A',\n",
       "   'feature_id': 'AC3',\n",
       "   'name': 'ligand HEM',\n",
       "   'ordinal': 54,\n",
       "   'provenance_source': 'PDB',\n",
       "   'reference_scheme': 'PDB entity',\n",
       "   'type': 'BINDING_SITE',\n",
       "   'feature_positions': [{'beg_comp_id': 'TYR', 'beg_seq_id': 42},\n",
       "    {'beg_comp_id': 'PHE', 'beg_seq_id': 43},\n",
       "    {'beg_comp_id': 'HIS', 'beg_seq_id': 45},\n",
       "    {'beg_comp_id': 'PHE', 'beg_seq_id': 46},\n",
       "    {'beg_comp_id': 'HIS', 'beg_seq_id': 58},\n",
       "    {'beg_comp_id': 'LYS', 'beg_seq_id': 61},\n",
       "    {'beg_comp_id': 'LEU', 'beg_seq_id': 86},\n",
       "    {'beg_comp_id': 'HIS', 'beg_seq_id': 87},\n",
       "    {'beg_comp_id': 'LEU', 'beg_seq_id': 91},\n",
       "    {'beg_comp_id': 'VAL', 'beg_seq_id': 93},\n",
       "    {'beg_comp_id': 'ASN', 'beg_seq_id': 97},\n",
       "    {'beg_comp_id': 'PHE', 'beg_seq_id': 98},\n",
       "    {'beg_comp_id': 'LEU', 'beg_seq_id': 101},\n",
       "    {'beg_comp_id': 'LEU', 'beg_seq_id': 136}]},\n",
       "  {'assignment_version': 'V1.0',\n",
       "   'description': 'Software generated binding site for ligand entity 3 component HEM instance G chain B',\n",
       "   'feature_id': 'AC4',\n",
       "   'name': 'ligand HEM',\n",
       "   'ordinal': 55,\n",
       "   'provenance_source': 'PDB',\n",
       "   'reference_scheme': 'PDB entity',\n",
       "   'type': 'BINDING_SITE',\n",
       "   'feature_positions': [{'beg_comp_id': 'ALA', 'beg_seq_id': 53}]},\n",
       "  {'assignment_version': 'V1.0',\n",
       "   'description': 'Molprobity bond angle outlier in instance A model 1',\n",
       "   'feature_id': 'ANGLE_OUTLIER_1',\n",
       "   'name': 'Molprobity bond angle outlier',\n",
       "   'ordinal': 59,\n",
       "   'provenance_source': 'PDB',\n",
       "   'reference_scheme': 'PDB entity',\n",
       "   'type': 'ANGLE_OUTLIER',\n",
       "   'feature_positions': [{'beg_comp_id': 'VAL', 'beg_seq_id': 1},\n",
       "    {'beg_comp_id': 'VAL', 'beg_seq_id': 1},\n",
       "    {'beg_comp_id': 'VAL', 'beg_seq_id': 1},\n",
       "    {'beg_comp_id': 'VAL', 'beg_seq_id': 1},\n",
       "    {'beg_comp_id': 'VAL', 'beg_seq_id': 1},\n",
       "    {'beg_comp_id': 'LEU', 'beg_seq_id': 2},\n",
       "    {'beg_comp_id': 'LEU', 'beg_seq_id': 2},\n",
       "    {'beg_comp_id': 'LEU', 'beg_seq_id': 2},\n",
       "    {'beg_comp_id': 'LEU', 'beg_seq_id': 2},\n",
       "    {'beg_comp_id': 'LEU', 'beg_seq_id': 2},\n",
       "    {'beg_comp_id': 'LEU', 'beg_seq_id': 2},\n",
       "    {'beg_comp_id': 'SER', 'beg_seq_id': 3},\n",
       "    {'beg_comp_id': 'SER', 'beg_seq_id': 3},\n",
       "    {'beg_comp_id': 'SER', 'beg_seq_id': 3},\n",
       "    {'beg_comp_id': 'PRO', 'beg_seq_id': 4},\n",
       "    {'beg_comp_id': 'PRO', 'beg_seq_id': 4},\n",
       "    {'beg_comp_id': 'PRO', 'beg_seq_id': 4},\n",
       "    {'beg_comp_id': 'PRO', 'beg_seq_id': 4},\n",
       "    {'beg_comp_id': 'PRO', 'beg_seq_id': 4},\n",
       "    {'beg_comp_id': 'PRO', 'beg_seq_id': 4},\n",
       "    {'beg_comp_id': 'ALA', 'beg_seq_id': 5},\n",
       "    {'beg_comp_id': 'ALA', 'beg_seq_id': 5},\n",
       "    {'beg_comp_id': 'ASP', 'beg_seq_id': 6},\n",
       "    {'beg_comp_id': 'ASP', 'beg_seq_id': 6},\n",
       "    {'beg_comp_id': 'ASP', 'beg_seq_id': 6},\n",
       "    {'beg_comp_id': 'ASP', 'beg_seq_id': 6},\n",
       "    {'beg_comp_id': 'LYS', 'beg_seq_id': 7},\n",
       "    {'beg_comp_id': 'LYS', 'beg_seq_id': 7},\n",
       "    {'beg_comp_id': 'LYS', 'beg_seq_id': 7},\n",
       "    {'beg_comp_id': 'LYS', 'beg_seq_id': 7},\n",
       "    {'beg_comp_id': 'LYS', 'beg_seq_id': 7},\n",
       "    {'beg_comp_id': 'LYS', 'beg_seq_id': 7},\n",
       "    {'beg_comp_id': 'THR', 'beg_seq_id': 8},\n",
       "    {'beg_comp_id': 'THR', 'beg_seq_id': 8},\n",
       "    {'beg_comp_id': 'THR', 'beg_seq_id': 8},\n",
       "    {'beg_comp_id': 'THR', 'beg_seq_id': 8},\n",
       "    {'beg_comp_id': 'THR', 'beg_seq_id': 8},\n",
       "    {'beg_comp_id': 'VAL', 'beg_seq_id': 10},\n",
       "    {'beg_comp_id': 'VAL', 'beg_seq_id': 10},\n",
       "    {'beg_comp_id': 'LYS', 'beg_seq_id': 11},\n",
       "    {'beg_comp_id': 'LYS', 'beg_seq_id': 11},\n",
       "    {'beg_comp_id': 'LYS', 'beg_seq_id': 11},\n",
       "    {'beg_comp_id': 'ALA', 'beg_seq_id': 12},\n",
       "    {'beg_comp_id': 'ALA', 'beg_seq_id': 12},\n",
       "    {'beg_comp_id': 'ALA', 'beg_seq_id': 12},\n",
       "    {'beg_comp_id': 'ALA', 'beg_seq_id': 12},\n",
       "    {'beg_comp_id': 'ALA', 'beg_seq_id': 13},\n",
       "    {'beg_comp_id': 'ALA', 'beg_seq_id': 13},\n",
       "    {'beg_comp_id': 'ALA', 'beg_seq_id': 13},\n",
       "    {'beg_comp_id': 'TRP', 'beg_seq_id': 14},\n",
       "    {'beg_comp_id': 'TRP', 'beg_seq_id': 14},\n",
       "    {'beg_comp_id': 'TRP', 'beg_seq_id': 14},\n",
       "    {'beg_comp_id': 'TRP', 'beg_seq_id': 14},\n",
       "    {'beg_comp_id': 'TRP', 'beg_seq_id': 14},\n",
       "    {'beg_comp_id': 'TRP', 'beg_seq_id': 14},\n",
       "    {'beg_comp_id': 'TRP', 'beg_seq_id': 14},\n",
       "    {'beg_comp_id': 'TRP', 'beg_seq_id': 14},\n",
       "    {'beg_comp_id': 'TRP', 'beg_seq_id': 14},\n",
       "    {'beg_comp_id': 'TRP', 'beg_seq_id': 14},\n",
       "    {'beg_comp_id': 'GLY', 'beg_seq_id': 15},\n",
       "    {'beg_comp_id': 'GLY', 'beg_seq_id': 15},\n",
       "    {'beg_comp_id': 'GLY', 'beg_seq_id': 15},\n",
       "    {'beg_comp_id': 'GLY', 'beg_seq_id': 15},\n",
       "    {'beg_comp_id': 'GLY', 'beg_seq_id': 15},\n",
       "    {'beg_comp_id': 'LYS', 'beg_seq_id': 16},\n",
       "    {'beg_comp_id': 'LYS', 'beg_seq_id': 16},\n",
       "    {'beg_comp_id': 'LYS', 'beg_seq_id': 16},\n",
       "    {'beg_comp_id': 'LYS', 'beg_seq_id': 16},\n",
       "    {'beg_comp_id': 'LYS', 'beg_seq_id': 16},\n",
       "    {'beg_comp_id': 'LYS', 'beg_seq_id': 16},\n",
       "    {'beg_comp_id': 'VAL', 'beg_seq_id': 17},\n",
       "    {'beg_comp_id': 'VAL', 'beg_seq_id': 17},\n",
       "    {'beg_comp_id': 'VAL', 'beg_seq_id': 17},\n",
       "    {'beg_comp_id': 'VAL', 'beg_seq_id': 17},\n",
       "    {'beg_comp_id': 'VAL', 'beg_seq_id': 17},\n",
       "    {'beg_comp_id': 'GLY', 'beg_seq_id': 18},\n",
       "    {'beg_comp_id': 'GLY', 'beg_seq_id': 18},\n",
       "    {'beg_comp_id': 'GLY', 'beg_seq_id': 18},\n",
       "    {'beg_comp_id': 'ALA', 'beg_seq_id': 19},\n",
       "    {'beg_comp_id': 'ALA', 'beg_seq_id': 19},\n",
       "    {'beg_comp_id': 'ALA', 'beg_seq_id': 19},\n",
       "    {'beg_comp_id': 'ALA', 'beg_seq_id': 19},\n",
       "    {'beg_comp_id': 'ALA', 'beg_seq_id': 19},\n",
       "    {'beg_comp_id': 'HIS', 'beg_seq_id': 20},\n",
       "    {'beg_comp_id': 'HIS', 'beg_seq_id': 20},\n",
       "    {'beg_comp_id': 'HIS', 'beg_seq_id': 20},\n",
       "    {'beg_comp_id': 'HIS', 'beg_seq_id': 20},\n",
       "    {'beg_comp_id': 'ALA', 'beg_seq_id': 21},\n",
       "    {'beg_comp_id': 'ALA', 'beg_seq_id': 21},\n",
       "    {'beg_comp_id': 'ALA', 'beg_seq_id': 21},\n",
       "    {'beg_comp_id': 'ALA', 'beg_seq_id': 21},\n",
       "    {'beg_comp_id': 'ALA', 'beg_seq_id': 21},\n",
       "    {'beg_comp_id': 'ALA', 'beg_seq_id': 21},\n",
       "    {'beg_comp_id': 'GLY', 'beg_seq_id': 22},\n",
       "    {'beg_comp_id': 'GLY', 'beg_seq_id': 22},\n",
       "    {'beg_comp_id': 'GLY', 'beg_seq_id': 22},\n",
       "    {'beg_comp_id': 'GLU', 'beg_seq_id': 23},\n",
       "    {'beg_comp_id': 'GLU', 'beg_seq_id': 23},\n",
       "    {'beg_comp_id': 'GLU', 'beg_seq_id': 23},\n",
       "    {'beg_comp_id': 'GLU', 'beg_seq_id': 23},\n",
       "    {'beg_comp_id': 'TYR', 'beg_seq_id': 24},\n",
       "    {'beg_comp_id': 'TYR', 'beg_seq_id': 24},\n",
       "    {'beg_comp_id': 'TYR', 'beg_seq_id': 24},\n",
       "    {'beg_comp_id': 'TYR', 'beg_seq_id': 24},\n",
       "    {'beg_comp_id': 'TYR', 'beg_seq_id': 24},\n",
       "    {'beg_comp_id': 'TYR', 'beg_seq_id': 24},\n",
       "    {'beg_comp_id': 'TYR', 'beg_seq_id': 24},\n",
       "    {'beg_comp_id': 'TYR', 'beg_seq_id': 24},\n",
       "    {'beg_comp_id': 'GLY', 'beg_seq_id': 25},\n",
       "    {'beg_comp_id': 'GLY', 'beg_seq_id': 25},\n",
       "    {'beg_comp_id': 'ALA', 'beg_seq_id': 26},\n",
       "    {'beg_comp_id': 'ALA', 'beg_seq_id': 26},\n",
       "    {'beg_comp_id': 'ALA', 'beg_seq_id': 26},\n",
       "    {'beg_comp_id': 'ALA', 'beg_seq_id': 26},\n",
       "    {'beg_comp_id': 'ALA', 'beg_seq_id': 26},\n",
       "    {'beg_comp_id': 'GLU', 'beg_seq_id': 27},\n",
       "    {'beg_comp_id': 'GLU', 'beg_seq_id': 27},\n",
       "    {'beg_comp_id': 'ALA', 'beg_seq_id': 28},\n",
       "    {'beg_comp_id': 'ALA', 'beg_seq_id': 28},\n",
       "    {'beg_comp_id': 'ALA', 'beg_seq_id': 28},\n",
       "    {'beg_comp_id': 'LEU', 'beg_seq_id': 29},\n",
       "    {'beg_comp_id': 'GLU', 'beg_seq_id': 30},\n",
       "    {'beg_comp_id': 'GLU', 'beg_seq_id': 30},\n",
       "    {'beg_comp_id': 'GLU', 'beg_seq_id': 30},\n",
       "    {'beg_comp_id': 'ARG', 'beg_seq_id': 31},\n",
       "    {'beg_comp_id': 'ARG', 'beg_seq_id': 31},\n",
       "    {'beg_comp_id': 'ARG', 'beg_seq_id': 31},\n",
       "    {'beg_comp_id': 'MET', 'beg_seq_id': 32},\n",
       "    {'beg_comp_id': 'PHE', 'beg_seq_id': 33},\n",
       "    {'beg_comp_id': 'PHE', 'beg_seq_id': 33},\n",
       "    {'beg_comp_id': 'PHE', 'beg_seq_id': 33},\n",
       "    {'beg_comp_id': 'PHE', 'beg_seq_id': 33},\n",
       "    {'beg_comp_id': 'PHE', 'beg_seq_id': 33},\n",
       "    {'beg_comp_id': 'PHE', 'beg_seq_id': 33},\n",
       "    {'beg_comp_id': 'PHE', 'beg_seq_id': 33},\n",
       "    {'beg_comp_id': 'PHE', 'beg_seq_id': 33},\n",
       "    {'beg_comp_id': 'LEU', 'beg_seq_id': 34},\n",
       "    {'beg_comp_id': 'LEU', 'beg_seq_id': 34},\n",
       "    {'beg_comp_id': 'SER', 'beg_seq_id': 35},\n",
       "    {'beg_comp_id': 'PHE', 'beg_seq_id': 36},\n",
       "    {'beg_comp_id': 'PHE', 'beg_seq_id': 36},\n",
       "    {'beg_comp_id': 'THR', 'beg_seq_id': 38},\n",
       "    {'beg_comp_id': 'THR', 'beg_seq_id': 39},\n",
       "    {'beg_comp_id': 'THR', 'beg_seq_id': 39},\n",
       "    {'beg_comp_id': 'LYS', 'beg_seq_id': 40},\n",
       "    {'beg_comp_id': 'THR', 'beg_seq_id': 41},\n",
       "    {'beg_comp_id': 'THR', 'beg_seq_id': 41},\n",
       "    {'beg_comp_id': 'TYR', 'beg_seq_id': 42},\n",
       "    {'beg_comp_id': 'TYR', 'beg_seq_id': 42},\n",
       "    {'beg_comp_id': 'TYR', 'beg_seq_id': 42},\n",
       "    {'beg_comp_id': 'TYR', 'beg_seq_id': 42},\n",
       "    {'beg_comp_id': 'TYR', 'beg_seq_id': 42},\n",
       "    {'beg_comp_id': 'PHE', 'beg_seq_id': 43},\n",
       "    {'beg_comp_id': 'PHE', 'beg_seq_id': 43},\n",
       "    {'beg_comp_id': 'PHE', 'beg_seq_id': 43},\n",
       "    {'beg_comp_id': 'PHE', 'beg_seq_id': 43},\n",
       "    {'beg_comp_id': 'PHE', 'beg_seq_id': 43},\n",
       "    {'beg_comp_id': 'PHE', 'beg_seq_id': 43},\n",
       "    {'beg_comp_id': 'PRO', 'beg_seq_id': 44},\n",
       "    {'beg_comp_id': 'PRO', 'beg_seq_id': 44},\n",
       "    {'beg_comp_id': 'HIS', 'beg_seq_id': 45},\n",
       "    {'beg_comp_id': 'HIS', 'beg_seq_id': 45},\n",
       "    {'beg_comp_id': 'PHE', 'beg_seq_id': 46},\n",
       "    {'beg_comp_id': 'PHE', 'beg_seq_id': 46},\n",
       "    {'beg_comp_id': 'PHE', 'beg_seq_id': 46},\n",
       "    {'beg_comp_id': 'PHE', 'beg_seq_id': 46},\n",
       "    {'beg_comp_id': 'PHE', 'beg_seq_id': 46},\n",
       "    {'beg_comp_id': 'PHE', 'beg_seq_id': 46},\n",
       "    {'beg_comp_id': 'ASP', 'beg_seq_id': 47},\n",
       "    {'beg_comp_id': 'ASP', 'beg_seq_id': 47},\n",
       "    {'beg_comp_id': 'ASP', 'beg_seq_id': 47},\n",
       "    {'beg_comp_id': 'ASP', 'beg_seq_id': 47},\n",
       "    {'beg_comp_id': 'LEU', 'beg_seq_id': 48},\n",
       "    {'beg_comp_id': 'LEU', 'beg_seq_id': 48},\n",
       "    {'beg_comp_id': 'LEU', 'beg_seq_id': 48},\n",
       "    {'beg_comp_id': 'SER', 'beg_seq_id': 49},\n",
       "    {'beg_comp_id': 'SER', 'beg_seq_id': 49},\n",
       "    {'beg_comp_id': 'SER', 'beg_seq_id': 49},\n",
       "    {'beg_comp_id': 'SER', 'beg_seq_id': 49},\n",
       "    {'beg_comp_id': 'HIS', 'beg_seq_id': 50},\n",
       "    {'beg_comp_id': 'HIS', 'beg_seq_id': 50},\n",
       "    {'beg_comp_id': 'HIS', 'beg_seq_id': 50},\n",
       "    {'beg_comp_id': 'HIS', 'beg_seq_id': 50},\n",
       "    {'beg_comp_id': 'HIS', 'beg_seq_id': 50},\n",
       "    {'beg_comp_id': 'HIS', 'beg_seq_id': 50},\n",
       "    {'beg_comp_id': 'HIS', 'beg_seq_id': 50},\n",
       "    {'beg_comp_id': 'HIS', 'beg_seq_id': 50},\n",
       "    {'beg_comp_id': 'GLY', 'beg_seq_id': 51},\n",
       "    {'beg_comp_id': 'GLY', 'beg_seq_id': 51},\n",
       "    {'beg_comp_id': 'SER', 'beg_seq_id': 52},\n",
       "    {'beg_comp_id': 'SER', 'beg_seq_id': 52},\n",
       "    {'beg_comp_id': 'SER', 'beg_seq_id': 52},\n",
       "    {'beg_comp_id': 'SER', 'beg_seq_id': 52},\n",
       "    {'beg_comp_id': 'ALA', 'beg_seq_id': 53},\n",
       "    {'beg_comp_id': 'GLN', 'beg_seq_id': 54},\n",
       "    {'beg_comp_id': 'GLN', 'beg_seq_id': 54},\n",
       "    {'beg_comp_id': 'GLN', 'beg_seq_id': 54},\n",
       "    {'beg_comp_id': 'VAL', 'beg_seq_id': 55},\n",
       "    {'beg_comp_id': 'LYS', 'beg_seq_id': 56},\n",
       "    {'beg_comp_id': 'LYS', 'beg_seq_id': 56},\n",
       "    {'beg_comp_id': 'LYS', 'beg_seq_id': 56},\n",
       "    {'beg_comp_id': 'LYS', 'beg_seq_id': 56},\n",
       "    {'beg_comp_id': 'LYS', 'beg_seq_id': 56},\n",
       "    {'beg_comp_id': 'LYS', 'beg_seq_id': 56},\n",
       "    {'beg_comp_id': 'GLY', 'beg_seq_id': 57},\n",
       "    {'beg_comp_id': 'GLY', 'beg_seq_id': 57},\n",
       "    {'beg_comp_id': 'HIS', 'beg_seq_id': 58},\n",
       "    {'beg_comp_id': 'HIS', 'beg_seq_id': 58},\n",
       "    {'beg_comp_id': 'HIS', 'beg_seq_id': 58},\n",
       "    {'beg_comp_id': 'HIS', 'beg_seq_id': 58},\n",
       "    {'beg_comp_id': 'HIS', 'beg_seq_id': 58},\n",
       "    {'beg_comp_id': 'HIS', 'beg_seq_id': 58},\n",
       "    {'beg_comp_id': 'GLY', 'beg_seq_id': 59},\n",
       "    {'beg_comp_id': 'GLY', 'beg_seq_id': 59},\n",
       "    {'beg_comp_id': 'LYS', 'beg_seq_id': 60},\n",
       "    {'beg_comp_id': 'LYS', 'beg_seq_id': 60},\n",
       "    {'beg_comp_id': 'LYS', 'beg_seq_id': 60},\n",
       "    {'beg_comp_id': 'LYS', 'beg_seq_id': 60},\n",
       "    {'beg_comp_id': 'LYS', 'beg_seq_id': 61},\n",
       "    {'beg_comp_id': 'LYS', 'beg_seq_id': 61},\n",
       "    {'beg_comp_id': 'LYS', 'beg_seq_id': 61},\n",
       "    {'beg_comp_id': 'LYS', 'beg_seq_id': 61},\n",
       "    {'beg_comp_id': 'LYS', 'beg_seq_id': 61},\n",
       "    {'beg_comp_id': 'LYS', 'beg_seq_id': 61},\n",
       "    {'beg_comp_id': 'VAL', 'beg_seq_id': 62},\n",
       "    {'beg_comp_id': 'VAL', 'beg_seq_id': 62},\n",
       "    {'beg_comp_id': 'VAL', 'beg_seq_id': 62},\n",
       "    {'beg_comp_id': 'VAL', 'beg_seq_id': 62},\n",
       "    {'beg_comp_id': 'VAL', 'beg_seq_id': 62},\n",
       "    {'beg_comp_id': 'ALA', 'beg_seq_id': 63},\n",
       "    {'beg_comp_id': 'ALA', 'beg_seq_id': 63},\n",
       "    {'beg_comp_id': 'ASP', 'beg_seq_id': 64},\n",
       "    {'beg_comp_id': 'ASP', 'beg_seq_id': 64},\n",
       "    {'beg_comp_id': 'ASP', 'beg_seq_id': 64},\n",
       "    {'beg_comp_id': 'ASP', 'beg_seq_id': 64},\n",
       "    {'beg_comp_id': 'ASP', 'beg_seq_id': 64},\n",
       "    {'beg_comp_id': 'ASP', 'beg_seq_id': 64},\n",
       "    {'beg_comp_id': 'ASP', 'beg_seq_id': 64},\n",
       "    {'beg_comp_id': 'ASP', 'beg_seq_id': 64},\n",
       "    {'beg_comp_id': 'ALA', 'beg_seq_id': 65},\n",
       "    {'beg_comp_id': 'ALA', 'beg_seq_id': 65},\n",
       "    {'beg_comp_id': 'THR', 'beg_seq_id': 67},\n",
       "    {'beg_comp_id': 'THR', 'beg_seq_id': 67},\n",
       "    {'beg_comp_id': 'ASN', 'beg_seq_id': 68},\n",
       "    {'beg_comp_id': 'ALA', 'beg_seq_id': 69},\n",
       "    {'beg_comp_id': 'VAL', 'beg_seq_id': 70},\n",
       "    {'beg_comp_id': 'VAL', 'beg_seq_id': 70},\n",
       "    {'beg_comp_id': 'ALA', 'beg_seq_id': 71},\n",
       "    {'beg_comp_id': 'ALA', 'beg_seq_id': 71},\n",
       "    {'beg_comp_id': 'ALA', 'beg_seq_id': 71},\n",
       "    {'beg_comp_id': 'ALA', 'beg_seq_id': 71},\n",
       "    {'beg_comp_id': 'ALA', 'beg_seq_id': 71},\n",
       "    {'beg_comp_id': 'ALA', 'beg_seq_id': 71},\n",
       "    {'beg_comp_id': 'HIS', 'beg_seq_id': 72},\n",
       "    {'beg_comp_id': 'HIS', 'beg_seq_id': 72},\n",
       "    {'beg_comp_id': 'HIS', 'beg_seq_id': 72},\n",
       "    {'beg_comp_id': 'HIS', 'beg_seq_id': 72},\n",
       "    {'beg_comp_id': 'HIS', 'beg_seq_id': 72},\n",
       "    {'beg_comp_id': 'HIS', 'beg_seq_id': 72},\n",
       "    {'beg_comp_id': 'VAL', 'beg_seq_id': 73},\n",
       "    {'beg_comp_id': 'VAL', 'beg_seq_id': 73},\n",
       "    {'beg_comp_id': 'VAL', 'beg_seq_id': 73},\n",
       "    {'beg_comp_id': 'VAL', 'beg_seq_id': 73},\n",
       "    {'beg_comp_id': 'VAL', 'beg_seq_id': 73},\n",
       "    {'beg_comp_id': 'ASP', 'beg_seq_id': 74},\n",
       "    {'beg_comp_id': 'ASP', 'beg_seq_id': 74},\n",
       "    {'beg_comp_id': 'ASP', 'beg_seq_id': 74},\n",
       "    {'beg_comp_id': 'ASP', 'beg_seq_id': 74},\n",
       "    {'beg_comp_id': 'ASP', 'beg_seq_id': 74},\n",
       "    {'beg_comp_id': 'ASP', 'beg_seq_id': 75},\n",
       "    {'beg_comp_id': 'ASP', 'beg_seq_id': 75},\n",
       "    {'beg_comp_id': 'ASP', 'beg_seq_id': 75},\n",
       "    {'beg_comp_id': 'ASP', 'beg_seq_id': 75},\n",
       "    {'beg_comp_id': 'ASP', 'beg_seq_id': 75},\n",
       "    {'beg_comp_id': 'ASP', 'beg_seq_id': 75},\n",
       "    {'beg_comp_id': 'ASP', 'beg_seq_id': 75},\n",
       "    {'beg_comp_id': 'ASP', 'beg_seq_id': 75},\n",
       "    {'beg_comp_id': 'MET', 'beg_seq_id': 76},\n",
       "    {'beg_comp_id': 'MET', 'beg_seq_id': 76},\n",
       "    {'beg_comp_id': 'MET', 'beg_seq_id': 76},\n",
       "    {'beg_comp_id': 'PRO', 'beg_seq_id': 77},\n",
       "    {'beg_comp_id': 'PRO', 'beg_seq_id': 77},\n",
       "    {'beg_comp_id': 'PRO', 'beg_seq_id': 77},\n",
       "    {'beg_comp_id': 'PRO', 'beg_seq_id': 77},\n",
       "    {'beg_comp_id': 'ASN', 'beg_seq_id': 78},\n",
       "    {'beg_comp_id': 'ASN', 'beg_seq_id': 78},\n",
       "    {'beg_comp_id': 'ASN', 'beg_seq_id': 78},\n",
       "    {'beg_comp_id': 'ASN', 'beg_seq_id': 78},\n",
       "    {'beg_comp_id': 'ALA', 'beg_seq_id': 79},\n",
       "    {'beg_comp_id': 'SER', 'beg_seq_id': 81},\n",
       "    {'beg_comp_id': 'SER', 'beg_seq_id': 81},\n",
       "    {'beg_comp_id': 'SER', 'beg_seq_id': 81},\n",
       "    {'beg_comp_id': 'SER', 'beg_seq_id': 81},\n",
       "    {'beg_comp_id': 'SER', 'beg_seq_id': 81},\n",
       "    {'beg_comp_id': 'SER', 'beg_seq_id': 81},\n",
       "    {'beg_comp_id': 'ALA', 'beg_seq_id': 82},\n",
       "    {'beg_comp_id': 'ALA', 'beg_seq_id': 82},\n",
       "    {'beg_comp_id': 'ALA', 'beg_seq_id': 82},\n",
       "    {'beg_comp_id': 'LEU', 'beg_seq_id': 83},\n",
       "    {'beg_comp_id': 'LEU', 'beg_seq_id': 83},\n",
       "    {'beg_comp_id': 'LEU', 'beg_seq_id': 83},\n",
       "    {'beg_comp_id': 'LEU', 'beg_seq_id': 83},\n",
       "    {'beg_comp_id': 'LEU', 'beg_seq_id': 83},\n",
       "    {'beg_comp_id': 'LEU', 'beg_seq_id': 83},\n",
       "    {'beg_comp_id': 'SER', 'beg_seq_id': 84},\n",
       "    {'beg_comp_id': 'SER', 'beg_seq_id': 84},\n",
       "    {'beg_comp_id': 'SER', 'beg_seq_id': 84},\n",
       "    {'beg_comp_id': 'SER', 'beg_seq_id': 84},\n",
       "    {'beg_comp_id': 'SER', 'beg_seq_id': 84},\n",
       "    {'beg_comp_id': 'ASP', 'beg_seq_id': 85},\n",
       "    {'beg_comp_id': 'ASP', 'beg_seq_id': 85},\n",
       "    {'beg_comp_id': 'ASP', 'beg_seq_id': 85},\n",
       "    {'beg_comp_id': 'ASP', 'beg_seq_id': 85},\n",
       "    {'beg_comp_id': 'ASP', 'beg_seq_id': 85},\n",
       "    {'beg_comp_id': 'LEU', 'beg_seq_id': 86},\n",
       "    {'beg_comp_id': 'LEU', 'beg_seq_id': 86},\n",
       "    {'beg_comp_id': 'LEU', 'beg_seq_id': 86},\n",
       "    {'beg_comp_id': 'HIS', 'beg_seq_id': 87},\n",
       "    {'beg_comp_id': 'HIS', 'beg_seq_id': 87},\n",
       "    {'beg_comp_id': 'ALA', 'beg_seq_id': 88},\n",
       "    {'beg_comp_id': 'ALA', 'beg_seq_id': 88},\n",
       "    {'beg_comp_id': 'ALA', 'beg_seq_id': 88},\n",
       "    {'beg_comp_id': 'ALA', 'beg_seq_id': 88},\n",
       "    {'beg_comp_id': 'HIS', 'beg_seq_id': 89},\n",
       "    {'beg_comp_id': 'HIS', 'beg_seq_id': 89},\n",
       "    {'beg_comp_id': 'HIS', 'beg_seq_id': 89},\n",
       "    {'beg_comp_id': 'HIS', 'beg_seq_id': 89},\n",
       "    {'beg_comp_id': 'HIS', 'beg_seq_id': 89},\n",
       "    {'beg_comp_id': 'HIS', 'beg_seq_id': 89},\n",
       "    {'beg_comp_id': 'HIS', 'beg_seq_id': 89},\n",
       "    {'beg_comp_id': 'HIS', 'beg_seq_id': 89},\n",
       "    {'beg_comp_id': 'HIS', 'beg_seq_id': 89},\n",
       "    {'beg_comp_id': 'LYS', 'beg_seq_id': 90},\n",
       "    {'beg_comp_id': 'LYS', 'beg_seq_id': 90},\n",
       "    {'beg_comp_id': 'LYS', 'beg_seq_id': 90},\n",
       "    {'beg_comp_id': 'LYS', 'beg_seq_id': 90},\n",
       "    {'beg_comp_id': 'ARG', 'beg_seq_id': 92},\n",
       "    {'beg_comp_id': 'ARG', 'beg_seq_id': 92},\n",
       "    {'beg_comp_id': 'ARG', 'beg_seq_id': 92},\n",
       "    {'beg_comp_id': 'ARG', 'beg_seq_id': 92},\n",
       "    {'beg_comp_id': 'ARG', 'beg_seq_id': 92},\n",
       "    {'beg_comp_id': 'ARG', 'beg_seq_id': 92},\n",
       "    {'beg_comp_id': 'ARG', 'beg_seq_id': 92},\n",
       "    {'beg_comp_id': 'VAL', 'beg_seq_id': 93},\n",
       "    {'beg_comp_id': 'ASP', 'beg_seq_id': 94},\n",
       "    {'beg_comp_id': 'PRO', 'beg_seq_id': 95},\n",
       "    {'beg_comp_id': 'VAL', 'beg_seq_id': 96},\n",
       "    {'beg_comp_id': 'VAL', 'beg_seq_id': 96},\n",
       "    {'beg_comp_id': 'VAL', 'beg_seq_id': 96},\n",
       "    {'beg_comp_id': 'PHE', 'beg_seq_id': 98},\n",
       "    {'beg_comp_id': 'PHE', 'beg_seq_id': 98},\n",
       "    {'beg_comp_id': 'PHE', 'beg_seq_id': 98},\n",
       "    {'beg_comp_id': 'PHE', 'beg_seq_id': 98},\n",
       "    {'beg_comp_id': 'PHE', 'beg_seq_id': 98},\n",
       "    {'beg_comp_id': 'LYS', 'beg_seq_id': 99},\n",
       "    {'beg_comp_id': 'LYS', 'beg_seq_id': 99},\n",
       "    {'beg_comp_id': 'LYS', 'beg_seq_id': 99},\n",
       "    {'beg_comp_id': 'LEU', 'beg_seq_id': 100},\n",
       "    {'beg_comp_id': 'LEU', 'beg_seq_id': 101},\n",
       "    {'beg_comp_id': 'LEU', 'beg_seq_id': 101},\n",
       "    {'beg_comp_id': 'HIS', 'beg_seq_id': 103},\n",
       "    {'beg_comp_id': 'HIS', 'beg_seq_id': 103},\n",
       "    {'beg_comp_id': 'HIS', 'beg_seq_id': 103},\n",
       "    {'beg_comp_id': 'HIS', 'beg_seq_id': 103},\n",
       "    {'beg_comp_id': 'CYS', 'beg_seq_id': 104},\n",
       "    {'beg_comp_id': 'LEU', 'beg_seq_id': 105},\n",
       "    {'beg_comp_id': 'LEU', 'beg_seq_id': 105},\n",
       "    {'beg_comp_id': 'LEU', 'beg_seq_id': 105},\n",
       "    {'beg_comp_id': 'LEU', 'beg_seq_id': 106},\n",
       "    {'beg_comp_id': 'LEU', 'beg_seq_id': 106},\n",
       "    {'beg_comp_id': 'LEU', 'beg_seq_id': 106},\n",
       "    {'beg_comp_id': 'LEU', 'beg_seq_id': 106},\n",
       "    {'beg_comp_id': 'THR', 'beg_seq_id': 108},\n",
       "    {'beg_comp_id': 'LEU', 'beg_seq_id': 109},\n",
       "    {'beg_comp_id': 'LEU', 'beg_seq_id': 109},\n",
       "    {'beg_comp_id': 'LEU', 'beg_seq_id': 109},\n",
       "    {'beg_comp_id': 'ALA', 'beg_seq_id': 110},\n",
       "    {'beg_comp_id': 'ALA', 'beg_seq_id': 110},\n",
       "    {'beg_comp_id': 'ALA', 'beg_seq_id': 110},\n",
       "    {'beg_comp_id': 'ALA', 'beg_seq_id': 111},\n",
       "    {'beg_comp_id': 'HIS', 'beg_seq_id': 112},\n",
       "    {'beg_comp_id': 'HIS', 'beg_seq_id': 112},\n",
       "    {'beg_comp_id': 'HIS', 'beg_seq_id': 112},\n",
       "    {'beg_comp_id': 'HIS', 'beg_seq_id': 112},\n",
       "    {'beg_comp_id': 'HIS', 'beg_seq_id': 112},\n",
       "    {'beg_comp_id': 'LEU', 'beg_seq_id': 113},\n",
       "    {'beg_comp_id': 'LEU', 'beg_seq_id': 113},\n",
       "    {'beg_comp_id': 'PRO', 'beg_seq_id': 114},\n",
       "    {'beg_comp_id': 'PRO', 'beg_seq_id': 114},\n",
       "    {'beg_comp_id': 'GLU', 'beg_seq_id': 116},\n",
       "    {'beg_comp_id': 'GLU', 'beg_seq_id': 116},\n",
       "    {'beg_comp_id': 'GLU', 'beg_seq_id': 116},\n",
       "    {'beg_comp_id': 'PHE', 'beg_seq_id': 117},\n",
       "    {'beg_comp_id': 'PHE', 'beg_seq_id': 117},\n",
       "    {'beg_comp_id': 'PHE', 'beg_seq_id': 117},\n",
       "    {'beg_comp_id': 'PHE', 'beg_seq_id': 117},\n",
       "    {'beg_comp_id': 'PHE', 'beg_seq_id': 117},\n",
       "    {'beg_comp_id': 'PHE', 'beg_seq_id': 117},\n",
       "    {'beg_comp_id': 'THR', 'beg_seq_id': 118},\n",
       "    {'beg_comp_id': 'PRO', 'beg_seq_id': 119},\n",
       "    {'beg_comp_id': 'ALA', 'beg_seq_id': 120},\n",
       "    {'beg_comp_id': 'HIS', 'beg_seq_id': 122},\n",
       "    {'beg_comp_id': 'LEU', 'beg_seq_id': 125},\n",
       "    {'beg_comp_id': 'ASP', 'beg_seq_id': 126},\n",
       "    {'beg_comp_id': 'ASP', 'beg_seq_id': 126},\n",
       "    {'beg_comp_id': 'ASP', 'beg_seq_id': 126},\n",
       "    {'beg_comp_id': 'ASP', 'beg_seq_id': 126},\n",
       "    {'beg_comp_id': 'LYS', 'beg_seq_id': 127},\n",
       "    {'beg_comp_id': 'LYS', 'beg_seq_id': 127},\n",
       "    {'beg_comp_id': 'PHE', 'beg_seq_id': 128},\n",
       "    {'beg_comp_id': 'PHE', 'beg_seq_id': 128},\n",
       "    {'beg_comp_id': 'PHE', 'beg_seq_id': 128},\n",
       "    {'beg_comp_id': 'PHE', 'beg_seq_id': 128},\n",
       "    {'beg_comp_id': 'LEU', 'beg_seq_id': 129},\n",
       "    {'beg_comp_id': 'SER', 'beg_seq_id': 131},\n",
       "    {'beg_comp_id': 'SER', 'beg_seq_id': 131},\n",
       "    {'beg_comp_id': 'SER', 'beg_seq_id': 131},\n",
       "    {'beg_comp_id': 'VAL', 'beg_seq_id': 132},\n",
       "    {'beg_comp_id': 'VAL', 'beg_seq_id': 132},\n",
       "    {'beg_comp_id': 'VAL', 'beg_seq_id': 132},\n",
       "    {'beg_comp_id': 'THR', 'beg_seq_id': 134},\n",
       "    {'beg_comp_id': 'THR', 'beg_seq_id': 134},\n",
       "    {'beg_comp_id': 'THR', 'beg_seq_id': 134},\n",
       "    {'beg_comp_id': 'THR', 'beg_seq_id': 134},\n",
       "    {'beg_comp_id': 'VAL', 'beg_seq_id': 135},\n",
       "    {'beg_comp_id': 'LEU', 'beg_seq_id': 136},\n",
       "    {'beg_comp_id': 'LEU', 'beg_seq_id': 136},\n",
       "    {'beg_comp_id': 'THR', 'beg_seq_id': 137},\n",
       "    {'beg_comp_id': 'THR', 'beg_seq_id': 137},\n",
       "    {'beg_comp_id': 'SER', 'beg_seq_id': 138},\n",
       "    {'beg_comp_id': 'SER', 'beg_seq_id': 138},\n",
       "    {'beg_comp_id': 'SER', 'beg_seq_id': 138},\n",
       "    {'beg_comp_id': 'LYS', 'beg_seq_id': 139},\n",
       "    {'beg_comp_id': 'LYS', 'beg_seq_id': 139},\n",
       "    {'beg_comp_id': 'TYR', 'beg_seq_id': 140},\n",
       "    {'beg_comp_id': 'TYR', 'beg_seq_id': 140},\n",
       "    {'beg_comp_id': 'TYR', 'beg_seq_id': 140},\n",
       "    {'beg_comp_id': 'TYR', 'beg_seq_id': 140},\n",
       "    {'beg_comp_id': 'ARG', 'beg_seq_id': 141},\n",
       "    {'beg_comp_id': 'ARG', 'beg_seq_id': 141},\n",
       "    {'beg_comp_id': 'ARG', 'beg_seq_id': 141},\n",
       "    {'beg_comp_id': 'ARG', 'beg_seq_id': 141},\n",
       "    {'beg_comp_id': 'ARG', 'beg_seq_id': 141},\n",
       "    {'beg_comp_id': 'ARG', 'beg_seq_id': 141}]},\n",
       "  {'assignment_version': 'V1.0',\n",
       "   'description': 'Molprobity bond distance outlier in instance A model 1',\n",
       "   'feature_id': 'BOND_OUTLIER_2',\n",
       "   'name': 'Molprobity bond distance outlier',\n",
       "   'ordinal': 60,\n",
       "   'provenance_source': 'PDB',\n",
       "   'reference_scheme': 'PDB entity',\n",
       "   'type': 'BOND_OUTLIER',\n",
       "   'feature_positions': [{'beg_comp_id': 'VAL', 'beg_seq_id': 1},\n",
       "    {'beg_comp_id': 'VAL', 'beg_seq_id': 1},\n",
       "    {'beg_comp_id': 'VAL', 'beg_seq_id': 1},\n",
       "    {'beg_comp_id': 'VAL', 'beg_seq_id': 1},\n",
       "    {'beg_comp_id': 'LEU', 'beg_seq_id': 2},\n",
       "    {'beg_comp_id': 'LEU', 'beg_seq_id': 2},\n",
       "    {'beg_comp_id': 'LEU', 'beg_seq_id': 2},\n",
       "    {'beg_comp_id': 'SER', 'beg_seq_id': 3},\n",
       "    {'beg_comp_id': 'SER', 'beg_seq_id': 3},\n",
       "    {'beg_comp_id': 'SER', 'beg_seq_id': 3},\n",
       "    {'beg_comp_id': 'SER', 'beg_seq_id': 3},\n",
       "    {'beg_comp_id': 'PRO', 'beg_seq_id': 4},\n",
       "    {'beg_comp_id': 'PRO', 'beg_seq_id': 4},\n",
       "    {'beg_comp_id': 'PRO', 'beg_seq_id': 4},\n",
       "    {'beg_comp_id': 'ALA', 'beg_seq_id': 5},\n",
       "    {'beg_comp_id': 'ALA', 'beg_seq_id': 5},\n",
       "    {'beg_comp_id': 'ALA', 'beg_seq_id': 5},\n",
       "    {'beg_comp_id': 'ASP', 'beg_seq_id': 6},\n",
       "    {'beg_comp_id': 'LYS', 'beg_seq_id': 7},\n",
       "    {'beg_comp_id': 'LYS', 'beg_seq_id': 7},\n",
       "    {'beg_comp_id': 'THR', 'beg_seq_id': 8},\n",
       "    {'beg_comp_id': 'THR', 'beg_seq_id': 8},\n",
       "    {'beg_comp_id': 'THR', 'beg_seq_id': 8},\n",
       "    {'beg_comp_id': 'THR', 'beg_seq_id': 8},\n",
       "    {'beg_comp_id': 'THR', 'beg_seq_id': 8},\n",
       "    {'beg_comp_id': 'ASN', 'beg_seq_id': 9},\n",
       "    {'beg_comp_id': 'ASN', 'beg_seq_id': 9},\n",
       "    {'beg_comp_id': 'VAL', 'beg_seq_id': 10},\n",
       "    {'beg_comp_id': 'VAL', 'beg_seq_id': 10},\n",
       "    {'beg_comp_id': 'LYS', 'beg_seq_id': 11},\n",
       "    {'beg_comp_id': 'LYS', 'beg_seq_id': 11},\n",
       "    {'beg_comp_id': 'LYS', 'beg_seq_id': 11},\n",
       "    {'beg_comp_id': 'LYS', 'beg_seq_id': 11},\n",
       "    {'beg_comp_id': 'LYS', 'beg_seq_id': 11},\n",
       "    {'beg_comp_id': 'ALA', 'beg_seq_id': 12},\n",
       "    {'beg_comp_id': 'ALA', 'beg_seq_id': 12},\n",
       "    {'beg_comp_id': 'ALA', 'beg_seq_id': 12},\n",
       "    {'beg_comp_id': 'ALA', 'beg_seq_id': 13},\n",
       "    {'beg_comp_id': 'TRP', 'beg_seq_id': 14},\n",
       "    {'beg_comp_id': 'TRP', 'beg_seq_id': 14},\n",
       "    {'beg_comp_id': 'TRP', 'beg_seq_id': 14},\n",
       "    {'beg_comp_id': 'TRP', 'beg_seq_id': 14},\n",
       "    {'beg_comp_id': 'TRP', 'beg_seq_id': 14},\n",
       "    {'beg_comp_id': 'TRP', 'beg_seq_id': 14},\n",
       "    {'beg_comp_id': 'TRP', 'beg_seq_id': 14},\n",
       "    {'beg_comp_id': 'TRP', 'beg_seq_id': 14},\n",
       "    {'beg_comp_id': 'TRP', 'beg_seq_id': 14},\n",
       "    {'beg_comp_id': 'GLY', 'beg_seq_id': 15},\n",
       "    {'beg_comp_id': 'GLY', 'beg_seq_id': 15},\n",
       "    {'beg_comp_id': 'GLY', 'beg_seq_id': 15},\n",
       "    {'beg_comp_id': 'LYS', 'beg_seq_id': 16},\n",
       "    {'beg_comp_id': 'LYS', 'beg_seq_id': 16},\n",
       "    {'beg_comp_id': 'LYS', 'beg_seq_id': 16},\n",
       "    {'beg_comp_id': 'LYS', 'beg_seq_id': 16},\n",
       "    {'beg_comp_id': 'LYS', 'beg_seq_id': 16},\n",
       "    {'beg_comp_id': 'VAL', 'beg_seq_id': 17},\n",
       "    {'beg_comp_id': 'VAL', 'beg_seq_id': 17},\n",
       "    {'beg_comp_id': 'VAL', 'beg_seq_id': 17},\n",
       "    {'beg_comp_id': 'VAL', 'beg_seq_id': 17},\n",
       "    {'beg_comp_id': 'VAL', 'beg_seq_id': 17},\n",
       "    {'beg_comp_id': 'GLY', 'beg_seq_id': 18},\n",
       "    {'beg_comp_id': 'GLY', 'beg_seq_id': 18},\n",
       "    {'beg_comp_id': 'ALA', 'beg_seq_id': 19},\n",
       "    {'beg_comp_id': 'ALA', 'beg_seq_id': 19},\n",
       "    {'beg_comp_id': 'HIS', 'beg_seq_id': 20},\n",
       "    {'beg_comp_id': 'HIS', 'beg_seq_id': 20},\n",
       "    {'beg_comp_id': 'HIS', 'beg_seq_id': 20},\n",
       "    {'beg_comp_id': 'HIS', 'beg_seq_id': 20},\n",
       "    {'beg_comp_id': 'HIS', 'beg_seq_id': 20},\n",
       "    {'beg_comp_id': 'HIS', 'beg_seq_id': 20},\n",
       "    {'beg_comp_id': 'HIS', 'beg_seq_id': 20},\n",
       "    {'beg_comp_id': 'ALA', 'beg_seq_id': 21},\n",
       "    {'beg_comp_id': 'ALA', 'beg_seq_id': 21},\n",
       "    {'beg_comp_id': 'ALA', 'beg_seq_id': 21},\n",
       "    {'beg_comp_id': 'ALA', 'beg_seq_id': 21},\n",
       "    {'beg_comp_id': 'GLY', 'beg_seq_id': 22},\n",
       "    {'beg_comp_id': 'GLY', 'beg_seq_id': 22},\n",
       "    {'beg_comp_id': 'GLU', 'beg_seq_id': 23},\n",
       "    {'beg_comp_id': 'GLU', 'beg_seq_id': 23},\n",
       "    {'beg_comp_id': 'GLU', 'beg_seq_id': 23},\n",
       "    {'beg_comp_id': 'TYR', 'beg_seq_id': 24},\n",
       "    {'beg_comp_id': 'TYR', 'beg_seq_id': 24},\n",
       "    {'beg_comp_id': 'TYR', 'beg_seq_id': 24},\n",
       "    {'beg_comp_id': 'TYR', 'beg_seq_id': 24},\n",
       "    {'beg_comp_id': 'TYR', 'beg_seq_id': 24},\n",
       "    {'beg_comp_id': 'GLY', 'beg_seq_id': 25},\n",
       "    {'beg_comp_id': 'ALA', 'beg_seq_id': 26},\n",
       "    {'beg_comp_id': 'ALA', 'beg_seq_id': 26},\n",
       "    {'beg_comp_id': 'ALA', 'beg_seq_id': 26},\n",
       "    {'beg_comp_id': 'GLU', 'beg_seq_id': 27},\n",
       "    {'beg_comp_id': 'GLU', 'beg_seq_id': 27},\n",
       "    {'beg_comp_id': 'GLU', 'beg_seq_id': 27},\n",
       "    {'beg_comp_id': 'ALA', 'beg_seq_id': 28},\n",
       "    {'beg_comp_id': 'LEU', 'beg_seq_id': 29},\n",
       "    {'beg_comp_id': 'LEU', 'beg_seq_id': 29},\n",
       "    {'beg_comp_id': 'GLU', 'beg_seq_id': 30},\n",
       "    {'beg_comp_id': 'GLU', 'beg_seq_id': 30},\n",
       "    {'beg_comp_id': 'GLU', 'beg_seq_id': 30},\n",
       "    {'beg_comp_id': 'ARG', 'beg_seq_id': 31},\n",
       "    {'beg_comp_id': 'PHE', 'beg_seq_id': 33},\n",
       "    {'beg_comp_id': 'PHE', 'beg_seq_id': 33},\n",
       "    {'beg_comp_id': 'PHE', 'beg_seq_id': 33},\n",
       "    {'beg_comp_id': 'PHE', 'beg_seq_id': 33},\n",
       "    {'beg_comp_id': 'PHE', 'beg_seq_id': 33},\n",
       "    {'beg_comp_id': 'PHE', 'beg_seq_id': 33},\n",
       "    {'beg_comp_id': 'PHE', 'beg_seq_id': 33},\n",
       "    {'beg_comp_id': 'LEU', 'beg_seq_id': 34},\n",
       "    {'beg_comp_id': 'LEU', 'beg_seq_id': 34},\n",
       "    {'beg_comp_id': 'SER', 'beg_seq_id': 35},\n",
       "    {'beg_comp_id': 'SER', 'beg_seq_id': 35},\n",
       "    {'beg_comp_id': 'PHE', 'beg_seq_id': 36},\n",
       "    {'beg_comp_id': 'PHE', 'beg_seq_id': 36},\n",
       "    {'beg_comp_id': 'PRO', 'beg_seq_id': 37},\n",
       "    {'beg_comp_id': 'PRO', 'beg_seq_id': 37},\n",
       "    {'beg_comp_id': 'PRO', 'beg_seq_id': 37},\n",
       "    {'beg_comp_id': 'THR', 'beg_seq_id': 39},\n",
       "    {'beg_comp_id': 'LYS', 'beg_seq_id': 40},\n",
       "    {'beg_comp_id': 'LYS', 'beg_seq_id': 40},\n",
       "    {'beg_comp_id': 'LYS', 'beg_seq_id': 40},\n",
       "    {'beg_comp_id': 'THR', 'beg_seq_id': 41},\n",
       "    {'beg_comp_id': 'THR', 'beg_seq_id': 41},\n",
       "    {'beg_comp_id': 'TYR', 'beg_seq_id': 42},\n",
       "    {'beg_comp_id': 'TYR', 'beg_seq_id': 42},\n",
       "    {'beg_comp_id': 'TYR', 'beg_seq_id': 42},\n",
       "    {'beg_comp_id': 'PHE', 'beg_seq_id': 43},\n",
       "    {'beg_comp_id': 'PHE', 'beg_seq_id': 43},\n",
       "    {'beg_comp_id': 'PHE', 'beg_seq_id': 43},\n",
       "    {'beg_comp_id': 'PHE', 'beg_seq_id': 43},\n",
       "    {'beg_comp_id': 'PRO', 'beg_seq_id': 44},\n",
       "    {'beg_comp_id': 'PRO', 'beg_seq_id': 44},\n",
       "    {'beg_comp_id': 'PRO', 'beg_seq_id': 44},\n",
       "    {'beg_comp_id': 'PRO', 'beg_seq_id': 44},\n",
       "    {'beg_comp_id': 'HIS', 'beg_seq_id': 45},\n",
       "    {'beg_comp_id': 'HIS', 'beg_seq_id': 45},\n",
       "    {'beg_comp_id': 'HIS', 'beg_seq_id': 45},\n",
       "    {'beg_comp_id': 'PHE', 'beg_seq_id': 46},\n",
       "    {'beg_comp_id': 'PHE', 'beg_seq_id': 46},\n",
       "    {'beg_comp_id': 'PHE', 'beg_seq_id': 46},\n",
       "    {'beg_comp_id': 'PHE', 'beg_seq_id': 46},\n",
       "    {'beg_comp_id': 'PHE', 'beg_seq_id': 46},\n",
       "    {'beg_comp_id': 'PHE', 'beg_seq_id': 46},\n",
       "    {'beg_comp_id': 'PHE', 'beg_seq_id': 46},\n",
       "    {'beg_comp_id': 'ASP', 'beg_seq_id': 47},\n",
       "    {'beg_comp_id': 'ASP', 'beg_seq_id': 47},\n",
       "    {'beg_comp_id': 'ASP', 'beg_seq_id': 47},\n",
       "    {'beg_comp_id': 'LEU', 'beg_seq_id': 48},\n",
       "    {'beg_comp_id': 'LEU', 'beg_seq_id': 48},\n",
       "    {'beg_comp_id': 'LEU', 'beg_seq_id': 48},\n",
       "    {'beg_comp_id': 'SER', 'beg_seq_id': 49},\n",
       "    {'beg_comp_id': 'SER', 'beg_seq_id': 49},\n",
       "    {'beg_comp_id': 'SER', 'beg_seq_id': 49},\n",
       "    {'beg_comp_id': 'SER', 'beg_seq_id': 49},\n",
       "    {'beg_comp_id': 'HIS', 'beg_seq_id': 50},\n",
       "    {'beg_comp_id': 'HIS', 'beg_seq_id': 50},\n",
       "    {'beg_comp_id': 'HIS', 'beg_seq_id': 50},\n",
       "    {'beg_comp_id': 'HIS', 'beg_seq_id': 50},\n",
       "    {'beg_comp_id': 'HIS', 'beg_seq_id': 50},\n",
       "    {'beg_comp_id': 'GLY', 'beg_seq_id': 51},\n",
       "    {'beg_comp_id': 'SER', 'beg_seq_id': 52},\n",
       "    {'beg_comp_id': 'SER', 'beg_seq_id': 52},\n",
       "    {'beg_comp_id': 'SER', 'beg_seq_id': 52},\n",
       "    {'beg_comp_id': 'SER', 'beg_seq_id': 52},\n",
       "    {'beg_comp_id': 'ALA', 'beg_seq_id': 53},\n",
       "    {'beg_comp_id': 'GLN', 'beg_seq_id': 54},\n",
       "    {'beg_comp_id': 'GLN', 'beg_seq_id': 54},\n",
       "    {'beg_comp_id': 'LYS', 'beg_seq_id': 56},\n",
       "    {'beg_comp_id': 'LYS', 'beg_seq_id': 56},\n",
       "    {'beg_comp_id': 'LYS', 'beg_seq_id': 56},\n",
       "    {'beg_comp_id': 'LYS', 'beg_seq_id': 56},\n",
       "    {'beg_comp_id': 'GLY', 'beg_seq_id': 57},\n",
       "    {'beg_comp_id': 'HIS', 'beg_seq_id': 58},\n",
       "    {'beg_comp_id': 'HIS', 'beg_seq_id': 58},\n",
       "    {'beg_comp_id': 'HIS', 'beg_seq_id': 58},\n",
       "    {'beg_comp_id': 'GLY', 'beg_seq_id': 59},\n",
       "    {'beg_comp_id': 'GLY', 'beg_seq_id': 59},\n",
       "    {'beg_comp_id': 'GLY', 'beg_seq_id': 59},\n",
       "    {'beg_comp_id': 'LYS', 'beg_seq_id': 60},\n",
       "    {'beg_comp_id': 'LYS', 'beg_seq_id': 60},\n",
       "    {'beg_comp_id': 'LYS', 'beg_seq_id': 60},\n",
       "    {'beg_comp_id': 'LYS', 'beg_seq_id': 60},\n",
       "    {'beg_comp_id': 'LYS', 'beg_seq_id': 60},\n",
       "    {'beg_comp_id': 'LYS', 'beg_seq_id': 60},\n",
       "    {'beg_comp_id': 'LYS', 'beg_seq_id': 60},\n",
       "    {'beg_comp_id': 'LYS', 'beg_seq_id': 61},\n",
       "    {'beg_comp_id': 'LYS', 'beg_seq_id': 61},\n",
       "    {'beg_comp_id': 'LYS', 'beg_seq_id': 61},\n",
       "    {'beg_comp_id': 'LYS', 'beg_seq_id': 61},\n",
       "    {'beg_comp_id': 'VAL', 'beg_seq_id': 62},\n",
       "    {'beg_comp_id': 'VAL', 'beg_seq_id': 62},\n",
       "    {'beg_comp_id': 'ALA', 'beg_seq_id': 63},\n",
       "    {'beg_comp_id': 'ALA', 'beg_seq_id': 63},\n",
       "    {'beg_comp_id': 'ALA', 'beg_seq_id': 63},\n",
       "    {'beg_comp_id': 'ASP', 'beg_seq_id': 64},\n",
       "    {'beg_comp_id': 'ASP', 'beg_seq_id': 64},\n",
       "    {'beg_comp_id': 'ASP', 'beg_seq_id': 64},\n",
       "    {'beg_comp_id': 'ASP', 'beg_seq_id': 64},\n",
       "    {'beg_comp_id': 'ASP', 'beg_seq_id': 64},\n",
       "    {'beg_comp_id': 'THR', 'beg_seq_id': 67},\n",
       "    {'beg_comp_id': 'THR', 'beg_seq_id': 67},\n",
       "    {'beg_comp_id': 'ASN', 'beg_seq_id': 68},\n",
       "    {'beg_comp_id': 'ASN', 'beg_seq_id': 68},\n",
       "    {'beg_comp_id': 'ASN', 'beg_seq_id': 68},\n",
       "    {'beg_comp_id': 'ALA', 'beg_seq_id': 69},\n",
       "    {'beg_comp_id': 'ALA', 'beg_seq_id': 69},\n",
       "    {'beg_comp_id': 'VAL', 'beg_seq_id': 70},\n",
       "    {'beg_comp_id': 'VAL', 'beg_seq_id': 70},\n",
       "    {'beg_comp_id': 'ALA', 'beg_seq_id': 71},\n",
       "    {'beg_comp_id': 'ALA', 'beg_seq_id': 71},\n",
       "    {'beg_comp_id': 'ALA', 'beg_seq_id': 71},\n",
       "    {'beg_comp_id': 'ALA', 'beg_seq_id': 71},\n",
       "    {'beg_comp_id': 'HIS', 'beg_seq_id': 72},\n",
       "    {'beg_comp_id': 'HIS', 'beg_seq_id': 72},\n",
       "    {'beg_comp_id': 'HIS', 'beg_seq_id': 72},\n",
       "    {'beg_comp_id': 'HIS', 'beg_seq_id': 72},\n",
       "    {'beg_comp_id': 'HIS', 'beg_seq_id': 72},\n",
       "    {'beg_comp_id': 'HIS', 'beg_seq_id': 72},\n",
       "    {'beg_comp_id': 'HIS', 'beg_seq_id': 72},\n",
       "    {'beg_comp_id': 'VAL', 'beg_seq_id': 73},\n",
       "    {'beg_comp_id': 'VAL', 'beg_seq_id': 73},\n",
       "    {'beg_comp_id': 'VAL', 'beg_seq_id': 73},\n",
       "    {'beg_comp_id': 'ASP', 'beg_seq_id': 74},\n",
       "    {'beg_comp_id': 'ASP', 'beg_seq_id': 74},\n",
       "    {'beg_comp_id': 'ASP', 'beg_seq_id': 74},\n",
       "    {'beg_comp_id': 'ASP', 'beg_seq_id': 74},\n",
       "    {'beg_comp_id': 'ASP', 'beg_seq_id': 74},\n",
       "    {'beg_comp_id': 'ASP', 'beg_seq_id': 74},\n",
       "    {'beg_comp_id': 'ASP', 'beg_seq_id': 75},\n",
       "    {'beg_comp_id': 'ASP', 'beg_seq_id': 75},\n",
       "    {'beg_comp_id': 'ASP', 'beg_seq_id': 75},\n",
       "    {'beg_comp_id': 'ASP', 'beg_seq_id': 75},\n",
       "    {'beg_comp_id': 'ASP', 'beg_seq_id': 75},\n",
       "    {'beg_comp_id': 'ASP', 'beg_seq_id': 75},\n",
       "    {'beg_comp_id': 'MET', 'beg_seq_id': 76},\n",
       "    {'beg_comp_id': 'MET', 'beg_seq_id': 76},\n",
       "    {'beg_comp_id': 'MET', 'beg_seq_id': 76},\n",
       "    {'beg_comp_id': 'MET', 'beg_seq_id': 76},\n",
       "    {'beg_comp_id': 'MET', 'beg_seq_id': 76},\n",
       "    {'beg_comp_id': 'PRO', 'beg_seq_id': 77},\n",
       "    {'beg_comp_id': 'PRO', 'beg_seq_id': 77},\n",
       "    {'beg_comp_id': 'PRO', 'beg_seq_id': 77},\n",
       "    {'beg_comp_id': 'PRO', 'beg_seq_id': 77},\n",
       "    {'beg_comp_id': 'ASN', 'beg_seq_id': 78},\n",
       "    {'beg_comp_id': 'ASN', 'beg_seq_id': 78},\n",
       "    {'beg_comp_id': 'ASN', 'beg_seq_id': 78},\n",
       "    {'beg_comp_id': 'ASN', 'beg_seq_id': 78},\n",
       "    {'beg_comp_id': 'ASN', 'beg_seq_id': 78},\n",
       "    {'beg_comp_id': 'ALA', 'beg_seq_id': 79},\n",
       "    {'beg_comp_id': 'ALA', 'beg_seq_id': 79},\n",
       "    {'beg_comp_id': 'ALA', 'beg_seq_id': 79},\n",
       "    {'beg_comp_id': 'LEU', 'beg_seq_id': 80},\n",
       "    {'beg_comp_id': 'SER', 'beg_seq_id': 81},\n",
       "    {'beg_comp_id': 'SER', 'beg_seq_id': 81},\n",
       "    {'beg_comp_id': 'LEU', 'beg_seq_id': 83},\n",
       "    {'beg_comp_id': 'SER', 'beg_seq_id': 84},\n",
       "    {'beg_comp_id': 'SER', 'beg_seq_id': 84},\n",
       "    {'beg_comp_id': 'SER', 'beg_seq_id': 84},\n",
       "    {'beg_comp_id': 'ASP', 'beg_seq_id': 85},\n",
       "    {'beg_comp_id': 'ASP', 'beg_seq_id': 85},\n",
       "    {'beg_comp_id': 'ASP', 'beg_seq_id': 85},\n",
       "    {'beg_comp_id': 'ASP', 'beg_seq_id': 85},\n",
       "    {'beg_comp_id': 'ASP', 'beg_seq_id': 85},\n",
       "    {'beg_comp_id': 'LEU', 'beg_seq_id': 86},\n",
       "    {'beg_comp_id': 'LEU', 'beg_seq_id': 86},\n",
       "    {'beg_comp_id': 'HIS', 'beg_seq_id': 87},\n",
       "    {'beg_comp_id': 'HIS', 'beg_seq_id': 87},\n",
       "    {'beg_comp_id': 'HIS', 'beg_seq_id': 87},\n",
       "    {'beg_comp_id': 'ALA', 'beg_seq_id': 88},\n",
       "    {'beg_comp_id': 'ALA', 'beg_seq_id': 88},\n",
       "    {'beg_comp_id': 'ALA', 'beg_seq_id': 88},\n",
       "    {'beg_comp_id': 'HIS', 'beg_seq_id': 89},\n",
       "    {'beg_comp_id': 'LYS', 'beg_seq_id': 90},\n",
       "    {'beg_comp_id': 'LYS', 'beg_seq_id': 90},\n",
       "    {'beg_comp_id': 'LYS', 'beg_seq_id': 90},\n",
       "    {'beg_comp_id': 'LYS', 'beg_seq_id': 90},\n",
       "    {'beg_comp_id': 'LYS', 'beg_seq_id': 90},\n",
       "    {'beg_comp_id': 'LYS', 'beg_seq_id': 90},\n",
       "    {'beg_comp_id': 'LEU', 'beg_seq_id': 91},\n",
       "    {'beg_comp_id': 'LEU', 'beg_seq_id': 91},\n",
       "    {'beg_comp_id': 'ARG', 'beg_seq_id': 92},\n",
       "    {'beg_comp_id': 'ARG', 'beg_seq_id': 92},\n",
       "    {'beg_comp_id': 'ARG', 'beg_seq_id': 92},\n",
       "    {'beg_comp_id': 'ARG', 'beg_seq_id': 92},\n",
       "    {'beg_comp_id': 'ARG', 'beg_seq_id': 92},\n",
       "    {'beg_comp_id': 'ARG', 'beg_seq_id': 92},\n",
       "    {'beg_comp_id': 'ARG', 'beg_seq_id': 92},\n",
       "    {'beg_comp_id': 'ARG', 'beg_seq_id': 92},\n",
       "    {'beg_comp_id': 'VAL', 'beg_seq_id': 93},\n",
       "    {'beg_comp_id': 'VAL', 'beg_seq_id': 93},\n",
       "    {'beg_comp_id': 'ASP', 'beg_seq_id': 94},\n",
       "    {'beg_comp_id': 'PRO', 'beg_seq_id': 95},\n",
       "    {'beg_comp_id': 'VAL', 'beg_seq_id': 96},\n",
       "    {'beg_comp_id': 'VAL', 'beg_seq_id': 96},\n",
       "    {'beg_comp_id': 'VAL', 'beg_seq_id': 96},\n",
       "    {'beg_comp_id': 'ASN', 'beg_seq_id': 97},\n",
       "    {'beg_comp_id': 'PHE', 'beg_seq_id': 98},\n",
       "    {'beg_comp_id': 'PHE', 'beg_seq_id': 98},\n",
       "    {'beg_comp_id': 'LYS', 'beg_seq_id': 99},\n",
       "    {'beg_comp_id': 'LYS', 'beg_seq_id': 99},\n",
       "    {'beg_comp_id': 'LYS', 'beg_seq_id': 99},\n",
       "    {'beg_comp_id': 'LYS', 'beg_seq_id': 99},\n",
       "    {'beg_comp_id': 'LYS', 'beg_seq_id': 99},\n",
       "    {'beg_comp_id': 'CYS', 'beg_seq_id': 104},\n",
       "    {'beg_comp_id': 'LEU', 'beg_seq_id': 105},\n",
       "    {'beg_comp_id': 'LEU', 'beg_seq_id': 105},\n",
       "    {'beg_comp_id': 'LEU', 'beg_seq_id': 105},\n",
       "    {'beg_comp_id': 'LEU', 'beg_seq_id': 106},\n",
       "    {'beg_comp_id': 'LEU', 'beg_seq_id': 106},\n",
       "    {'beg_comp_id': 'LEU', 'beg_seq_id': 106},\n",
       "    {'beg_comp_id': 'VAL', 'beg_seq_id': 107},\n",
       "    {'beg_comp_id': 'VAL', 'beg_seq_id': 107},\n",
       "    {'beg_comp_id': 'THR', 'beg_seq_id': 108},\n",
       "    {'beg_comp_id': 'LEU', 'beg_seq_id': 109},\n",
       "    {'beg_comp_id': 'LEU', 'beg_seq_id': 109},\n",
       "    {'beg_comp_id': 'HIS', 'beg_seq_id': 112},\n",
       "    {'beg_comp_id': 'HIS', 'beg_seq_id': 112},\n",
       "    {'beg_comp_id': 'HIS', 'beg_seq_id': 112},\n",
       "    {'beg_comp_id': 'HIS', 'beg_seq_id': 112},\n",
       "    {'beg_comp_id': 'LEU', 'beg_seq_id': 113},\n",
       "    {'beg_comp_id': 'LEU', 'beg_seq_id': 113},\n",
       "    {'beg_comp_id': 'PRO', 'beg_seq_id': 114},\n",
       "    {'beg_comp_id': 'PRO', 'beg_seq_id': 114},\n",
       "    {'beg_comp_id': 'PRO', 'beg_seq_id': 114},\n",
       "    {'beg_comp_id': 'PRO', 'beg_seq_id': 114},\n",
       "    {'beg_comp_id': 'ALA', 'beg_seq_id': 115},\n",
       "    {'beg_comp_id': 'ALA', 'beg_seq_id': 115},\n",
       "    {'beg_comp_id': 'GLU', 'beg_seq_id': 116},\n",
       "    {'beg_comp_id': 'PHE', 'beg_seq_id': 117},\n",
       "    {'beg_comp_id': 'PHE', 'beg_seq_id': 117},\n",
       "    {'beg_comp_id': 'PRO', 'beg_seq_id': 119},\n",
       "    {'beg_comp_id': 'ALA', 'beg_seq_id': 120},\n",
       "    {'beg_comp_id': 'ALA', 'beg_seq_id': 120},\n",
       "    {'beg_comp_id': 'VAL', 'beg_seq_id': 121},\n",
       "    {'beg_comp_id': 'HIS', 'beg_seq_id': 122},\n",
       "    {'beg_comp_id': 'HIS', 'beg_seq_id': 122},\n",
       "    {'beg_comp_id': 'ALA', 'beg_seq_id': 123},\n",
       "    {'beg_comp_id': 'LYS', 'beg_seq_id': 127},\n",
       "    {'beg_comp_id': 'LYS', 'beg_seq_id': 127},\n",
       "    {'beg_comp_id': 'LYS', 'beg_seq_id': 127},\n",
       "    {'beg_comp_id': 'PHE', 'beg_seq_id': 128},\n",
       "    {'beg_comp_id': 'PHE', 'beg_seq_id': 128},\n",
       "    {'beg_comp_id': 'PHE', 'beg_seq_id': 128},\n",
       "    {'beg_comp_id': 'LEU', 'beg_seq_id': 129},\n",
       "    {'beg_comp_id': 'LEU', 'beg_seq_id': 129},\n",
       "    {'beg_comp_id': 'SER', 'beg_seq_id': 131},\n",
       "    {'beg_comp_id': 'SER', 'beg_seq_id': 131},\n",
       "    {'beg_comp_id': 'VAL', 'beg_seq_id': 132},\n",
       "    {'beg_comp_id': 'VAL', 'beg_seq_id': 132},\n",
       "    {'beg_comp_id': 'THR', 'beg_seq_id': 134},\n",
       "    {'beg_comp_id': 'THR', 'beg_seq_id': 134},\n",
       "    {'beg_comp_id': 'VAL', 'beg_seq_id': 135},\n",
       "    {'beg_comp_id': 'THR', 'beg_seq_id': 137},\n",
       "    {'beg_comp_id': 'THR', 'beg_seq_id': 137},\n",
       "    {'beg_comp_id': 'THR', 'beg_seq_id': 137},\n",
       "    {'beg_comp_id': 'THR', 'beg_seq_id': 137},\n",
       "    {'beg_comp_id': 'THR', 'beg_seq_id': 137},\n",
       "    {'beg_comp_id': 'SER', 'beg_seq_id': 138},\n",
       "    {'beg_comp_id': 'SER', 'beg_seq_id': 138},\n",
       "    {'beg_comp_id': 'LYS', 'beg_seq_id': 139},\n",
       "    {'beg_comp_id': 'LYS', 'beg_seq_id': 139},\n",
       "    {'beg_comp_id': 'LYS', 'beg_seq_id': 139},\n",
       "    {'beg_comp_id': 'TYR', 'beg_seq_id': 140},\n",
       "    {'beg_comp_id': 'TYR', 'beg_seq_id': 140},\n",
       "    {'beg_comp_id': 'TYR', 'beg_seq_id': 140},\n",
       "    {'beg_comp_id': 'TYR', 'beg_seq_id': 140},\n",
       "    {'beg_comp_id': 'TYR', 'beg_seq_id': 140},\n",
       "    {'beg_comp_id': 'ARG', 'beg_seq_id': 141},\n",
       "    {'beg_comp_id': 'ARG', 'beg_seq_id': 141},\n",
       "    {'beg_comp_id': 'ARG', 'beg_seq_id': 141},\n",
       "    {'beg_comp_id': 'ARG', 'beg_seq_id': 141},\n",
       "    {'beg_comp_id': 'ARG', 'beg_seq_id': 141}]},\n",
       "  {'assignment_version': 'V1.0',\n",
       "   'description': 'Molprobity Ramachandran outlier in instance A model 1',\n",
       "   'feature_id': 'RAMACHANDRAN_OUTLIER_3',\n",
       "   'name': 'Molprobity Ramachandran outlier',\n",
       "   'ordinal': 61,\n",
       "   'provenance_source': 'PDB',\n",
       "   'reference_scheme': 'PDB entity',\n",
       "   'type': 'RAMACHANDRAN_OUTLIER',\n",
       "   'feature_positions': [{'beg_comp_id': 'SER', 'beg_seq_id': 3},\n",
       "    {'beg_comp_id': 'LYS', 'beg_seq_id': 16},\n",
       "    {'beg_comp_id': 'ALA', 'beg_seq_id': 21}]},\n",
       "  {'assignment_version': 'V1.0',\n",
       "   'description': 'Molprobity rotamer outlier in instance A model 1',\n",
       "   'feature_id': 'ROTAMER_OUTLIER_4',\n",
       "   'name': 'Molprobity rotamer outlier',\n",
       "   'ordinal': 62,\n",
       "   'provenance_source': 'PDB',\n",
       "   'reference_scheme': 'PDB entity',\n",
       "   'type': 'ROTAMER_OUTLIER',\n",
       "   'feature_positions': [{'beg_comp_id': 'LEU', 'beg_seq_id': 2},\n",
       "    {'beg_comp_id': 'PRO', 'beg_seq_id': 4},\n",
       "    {'beg_comp_id': 'HIS', 'beg_seq_id': 45},\n",
       "    {'beg_comp_id': 'SER', 'beg_seq_id': 52},\n",
       "    {'beg_comp_id': 'ASP', 'beg_seq_id': 75},\n",
       "    {'beg_comp_id': 'SER', 'beg_seq_id': 84},\n",
       "    {'beg_comp_id': 'PRO', 'beg_seq_id': 95},\n",
       "    {'beg_comp_id': 'SER', 'beg_seq_id': 138}]},\n",
       "  {'assignment_version': 'V1.0',\n",
       "   'description': 'STEREO_OUTLIER in instance A model 1',\n",
       "   'feature_id': 'STEREO_OUTLIER_5',\n",
       "   'name': 'STEREO_OUTLIER',\n",
       "   'ordinal': 63,\n",
       "   'provenance_source': 'PDB',\n",
       "   'reference_scheme': 'PDB entity',\n",
       "   'type': 'STEREO_OUTLIER',\n",
       "   'feature_positions': [{'beg_comp_id': 'THR', 'beg_seq_id': 137}]}],\n",
       " 'rcsb_polymer_instance_feature_summary': [{'count': 2,\n",
       "   'coverage': 0.10638,\n",
       "   'maximum_length': 0,\n",
       "   'maximum_value': 0.0,\n",
       "   'minimum_length': 0,\n",
       "   'minimum_value': 0.0,\n",
       "   'type': 'BINDING_SITE'},\n",
       "  {'count': 0,\n",
       "   'coverage': 0.0,\n",
       "   'maximum_length': 0,\n",
       "   'maximum_value': 0.0,\n",
       "   'minimum_length': 0,\n",
       "   'minimum_value': 0.0,\n",
       "   'type': 'C-MANNOSYLATION_SITE'},\n",
       "  {'count': 1,\n",
       "   'coverage': 1.0,\n",
       "   'maximum_length': 141,\n",
       "   'maximum_value': 0.0,\n",
       "   'minimum_length': 141,\n",
       "   'minimum_value': 0.0,\n",
       "   'type': 'CATH'},\n",
       "  {'count': 0,\n",
       "   'coverage': 0.0,\n",
       "   'maximum_length': 0,\n",
       "   'maximum_value': 0.0,\n",
       "   'minimum_length': 0,\n",
       "   'minimum_value': 0.0,\n",
       "   'type': 'CIS-PEPTIDE'},\n",
       "  {'count': 1,\n",
       "   'coverage': 1.0,\n",
       "   'maximum_length': 0,\n",
       "   'maximum_value': 0.0,\n",
       "   'minimum_length': 0,\n",
       "   'minimum_value': 0.0,\n",
       "   'type': 'ECOD'},\n",
       "  {'count': 8,\n",
       "   'coverage': 0.78014,\n",
       "   'maximum_length': 21,\n",
       "   'maximum_value': 0.0,\n",
       "   'minimum_length': 2,\n",
       "   'minimum_value': 0.0,\n",
       "   'type': 'HELIX_P'},\n",
       "  {'count': 0,\n",
       "   'coverage': 0.0,\n",
       "   'maximum_length': 0,\n",
       "   'maximum_value': 0.0,\n",
       "   'minimum_length': 0,\n",
       "   'minimum_value': 0.0,\n",
       "   'type': 'MEMBRANE_SEGMENT'},\n",
       "  {'count': 0,\n",
       "   'coverage': 0.0,\n",
       "   'maximum_length': 0,\n",
       "   'maximum_value': 0.0,\n",
       "   'minimum_length': 0,\n",
       "   'minimum_value': 0.0,\n",
       "   'type': 'N-GLYCOSYLATION_SITE'},\n",
       "  {'count': 0,\n",
       "   'coverage': 0.0,\n",
       "   'maximum_length': 0,\n",
       "   'maximum_value': 0.0,\n",
       "   'minimum_length': 0,\n",
       "   'minimum_value': 0.0,\n",
       "   'type': 'O-GLYCOSYLATION_SITE'},\n",
       "  {'count': 0,\n",
       "   'coverage': 0.0,\n",
       "   'maximum_length': 0,\n",
       "   'maximum_value': 0.0,\n",
       "   'minimum_length': 0,\n",
       "   'minimum_value': 0.0,\n",
       "   'type': 'S-GLYCOSYLATION_SITE'},\n",
       "  {'count': 0,\n",
       "   'coverage': 0.0,\n",
       "   'maximum_length': 0,\n",
       "   'maximum_value': 0.0,\n",
       "   'minimum_length': 0,\n",
       "   'minimum_value': 0.0,\n",
       "   'type': 'SABDAB_ANTIBODY_HEAVY_CHAIN_SUBCLASS'},\n",
       "  {'count': 0,\n",
       "   'coverage': 0.0,\n",
       "   'maximum_length': 0,\n",
       "   'maximum_value': 0.0,\n",
       "   'minimum_length': 0,\n",
       "   'minimum_value': 0.0,\n",
       "   'type': 'SABDAB_ANTIBODY_LIGHT_CHAIN_SUBCLASS'},\n",
       "  {'count': 0,\n",
       "   'coverage': 0.0,\n",
       "   'maximum_length': 0,\n",
       "   'maximum_value': 0.0,\n",
       "   'minimum_length': 0,\n",
       "   'minimum_value': 0.0,\n",
       "   'type': 'SABDAB_ANTIBODY_LIGHT_CHAIN_TYPE'},\n",
       "  {'count': 0,\n",
       "   'coverage': 0.0,\n",
       "   'maximum_length': 0,\n",
       "   'maximum_value': 0.0,\n",
       "   'minimum_length': 0,\n",
       "   'minimum_value': 0.0,\n",
       "   'type': 'SAbDab Antibody Heavy Chain Subclass'},\n",
       "  {'count': 0,\n",
       "   'coverage': 0.0,\n",
       "   'maximum_length': 0,\n",
       "   'maximum_value': 0.0,\n",
       "   'minimum_length': 0,\n",
       "   'minimum_value': 0.0,\n",
       "   'type': 'SAbDab Antibody Light Chain Subclass'},\n",
       "  {'count': 0,\n",
       "   'coverage': 0.0,\n",
       "   'maximum_length': 0,\n",
       "   'maximum_value': 0.0,\n",
       "   'minimum_length': 0,\n",
       "   'minimum_value': 0.0,\n",
       "   'type': 'SAbDab Antibody Light Chain Type'},\n",
       "  {'count': 1,\n",
       "   'coverage': 1.0,\n",
       "   'maximum_length': 141,\n",
       "   'maximum_value': 0.0,\n",
       "   'minimum_length': 141,\n",
       "   'minimum_value': 0.0,\n",
       "   'type': 'SCOP'},\n",
       "  {'count': 0,\n",
       "   'coverage': 0.0,\n",
       "   'maximum_length': 0,\n",
       "   'maximum_value': 0.0,\n",
       "   'minimum_length': 0,\n",
       "   'minimum_value': 0.0,\n",
       "   'type': 'SCOP2 Family'},\n",
       "  {'count': 0,\n",
       "   'coverage': 0.0,\n",
       "   'maximum_length': 0,\n",
       "   'maximum_value': 0.0,\n",
       "   'minimum_length': 0,\n",
       "   'minimum_value': 0.0,\n",
       "   'type': 'SCOP2 Superfamily'},\n",
       "  {'count': 0,\n",
       "   'coverage': 0.0,\n",
       "   'maximum_length': 0,\n",
       "   'maximum_value': 0.0,\n",
       "   'minimum_length': 0,\n",
       "   'minimum_value': 0.0,\n",
       "   'type': 'SCOP2B Superfamily'},\n",
       "  {'count': 1,\n",
       "   'coverage': 0.99291,\n",
       "   'maximum_length': 0,\n",
       "   'maximum_value': 0.0,\n",
       "   'minimum_length': 0,\n",
       "   'minimum_value': 0.0,\n",
       "   'type': 'SCOP2B_SUPERFAMILY'},\n",
       "  {'count': 0,\n",
       "   'coverage': 0.0,\n",
       "   'maximum_length': 0,\n",
       "   'maximum_value': 0.0,\n",
       "   'minimum_length': 0,\n",
       "   'minimum_value': 0.0,\n",
       "   'type': 'SCOP2_FAMILY'},\n",
       "  {'count': 0,\n",
       "   'coverage': 0.0,\n",
       "   'maximum_length': 0,\n",
       "   'maximum_value': 0.0,\n",
       "   'minimum_length': 0,\n",
       "   'minimum_value': 0.0,\n",
       "   'type': 'SCOP2_SUPERFAMILY'},\n",
       "  {'count': 0,\n",
       "   'coverage': 0.0,\n",
       "   'maximum_length': 0,\n",
       "   'maximum_value': 0.0,\n",
       "   'minimum_length': 0,\n",
       "   'minimum_value': 0.0,\n",
       "   'type': 'SHEET'},\n",
       "  {'count': 1,\n",
       "   'coverage': 0.21986,\n",
       "   'maximum_length': 31,\n",
       "   'maximum_value': 0.0,\n",
       "   'minimum_length': 31,\n",
       "   'minimum_value': 0.0,\n",
       "   'type': 'UNASSIGNED_SEC_STRUCT'},\n",
       "  {'count': 0,\n",
       "   'coverage': 0.0,\n",
       "   'maximum_length': 0,\n",
       "   'maximum_value': 0.0,\n",
       "   'minimum_length': 0,\n",
       "   'minimum_value': 0.0,\n",
       "   'type': 'UNOBSERVED_ATOM_XYZ'},\n",
       "  {'count': 0,\n",
       "   'coverage': 0.0,\n",
       "   'maximum_length': 0,\n",
       "   'maximum_value': 0.0,\n",
       "   'minimum_length': 0,\n",
       "   'minimum_value': 0.0,\n",
       "   'type': 'UNOBSERVED_RESIDUE_XYZ'},\n",
       "  {'count': 0,\n",
       "   'coverage': 0.0,\n",
       "   'maximum_length': 0,\n",
       "   'maximum_value': 0.0,\n",
       "   'minimum_length': 0,\n",
       "   'minimum_value': 0.0,\n",
       "   'type': 'ZERO_OCCUPANCY_ATOM_XYZ'},\n",
       "  {'count': 0,\n",
       "   'coverage': 0.0,\n",
       "   'maximum_length': 0,\n",
       "   'maximum_value': 0.0,\n",
       "   'minimum_length': 0,\n",
       "   'minimum_value': 0.0,\n",
       "   'type': 'ZERO_OCCUPANCY_RESIDUE_XYZ'},\n",
       "  {'count': 444, 'coverage': 3.14894, 'type': 'ANGLE_OUTLIER'},\n",
       "  {'count': 370, 'coverage': 2.62411, 'type': 'BOND_OUTLIER'},\n",
       "  {'count': 0, 'coverage': 0.0, 'type': 'MOGUL_ANGLE_OUTLIER'},\n",
       "  {'count': 0, 'coverage': 0.0, 'type': 'MOGUL_BOND_OUTLIER'},\n",
       "  {'count': 3, 'coverage': 0.02128, 'type': 'RAMACHANDRAN_OUTLIER'},\n",
       "  {'count': 8, 'coverage': 0.05674, 'type': 'ROTAMER_OUTLIER'},\n",
       "  {'count': 0, 'coverage': 0.0, 'type': 'RSCC_OUTLIER'},\n",
       "  {'count': 0, 'coverage': 0.0, 'type': 'RSRZ_OUTLIER'},\n",
       "  {'count': 1, 'coverage': 0.00709, 'type': 'STEREO_OUTLIER'}],\n",
       " 'rcsb_polymer_struct_conn': [{'connect_type': 'metal coordination',\n",
       "   'dist_value': 2.143,\n",
       "   'id': 'metalc1',\n",
       "   'ordinal_id': 1,\n",
       "   'connect_target': {'auth_seq_id': '87',\n",
       "    'label_asym_id': 'A',\n",
       "    'label_atom_id': 'NE2',\n",
       "    'label_comp_id': 'HIS',\n",
       "    'label_seq_id': 87,\n",
       "    'symmetry': '1_555'},\n",
       "   'connect_partner': {'label_asym_id': 'E',\n",
       "    'label_atom_id': 'FE',\n",
       "    'label_comp_id': 'HEM',\n",
       "    'symmetry': '1_555'}}],\n",
       " 'rcsb_id': '4HHB.A',\n",
       " 'rcsb_latest_revision': {'major_revision': 3, 'minor_revision': 0}}"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "response = requests.get('https://data.rcsb.org/rest/v1/core/polymer_entity_instance/4HHB/A')\n",
    "json.loads(response.text)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "slideshow": {
     "slide_type": "slide"
    }
   },
   "source": [
    "# BLAST URLAPI"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "[http://www.ncbi.nlm.nih.gov/BLAST/Doc/urlapi.html](http://www.ncbi.nlm.nih.gov/BLAST/Doc/urlapi.html)\n",
    "\n",
    "A URLAPI request looks like this:\n",
    "\n",
    "`http://www.ncbi.nlm.nih.gov/blast/Blast.cgi?CMD=<command>&{<name>=<value>}`\n",
    "\n",
    "where CMD can be\n",
    "\n",
    "* **PUT** add a job to the queue\n",
    "* **GET** get formated results from a job\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "slideshow": {
     "slide_type": "slide"
    }
   },
   "source": [
    "# BLAST PUT"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Some common attributes to PUT:\n",
    "\n",
    "* DATABASE - what database to search, *mandatory*\n",
    "* PROGRAM - what program to use (blastn, blastp, blastx, tblastn, tblastx)\n",
    "* QUERY - Accession(s), gi(s), or FASTA sequence(s), *mandatory*\n",
    "* MATRIX_NAME - matrix to use (default BLOSUM62)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {
    "slideshow": {
     "slide_type": "slide"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div id=\"webblast\" style=\"width: 500px\"></div>\n",
       "<script>\n",
       "$('head').append('<link rel=\"stylesheet\" href=\"http://bits.csb.pitt.edu/asker.js/themes/asker.default.css\" />');\n",
       "\n",
       "    var divid = '#webblast';\n",
       "\tjQuery(divid).asker({\n",
       "\t    id: divid,\n",
       "\t    question: \"When making a BLAST request with a long sequence, which HTTP request type should you use?\",\n",
       "\t\tanswers: ['GET','POST','REQUEST','Does not matter'],\n",
       "        server: \"http://bits.csb.pitt.edu/asker.js/example/asker.cgi\",\n",
       "\t\tcharter: chartmaker})\n",
       "    \n",
       "$(\".jp-InputArea .o:contains(html)\").closest('.jp-InputArea').hide();\n",
       "\n",
       "\n",
       "</script>\n"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%%html\n",
    "<div id=\"webblast\" style=\"width: 500px\"></div>\n",
    "<script>\n",
    "$('head').append('<link rel=\"stylesheet\" href=\"http://bits.csb.pitt.edu/asker.js/themes/asker.default.css\" />');\n",
    "\n",
    "    var divid = '#webblast';\n",
    "\tjQuery(divid).asker({\n",
    "\t    id: divid,\n",
    "\t    question: \"When making a BLAST request with a long sequence, which HTTP request type should you use?\",\n",
    "\t\tanswers: ['GET','POST','REQUEST','Does not matter'],\n",
    "        server: \"http://bits.csb.pitt.edu/asker.js/example/asker.cgi\",\n",
    "\t\tcharter: chartmaker})\n",
    "    \n",
    "$(\".jp-InputArea .o:contains(html)\").closest('.jp-InputArea').hide();\n",
    "\n",
    "\n",
    "</script>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {
    "slideshow": {
     "slide_type": "slide"
    }
   },
   "outputs": [],
   "source": [
    "response = requests.post('https://www.ncbi.nlm.nih.gov/blast/Blast.cgi',\\\n",
    "                        data={'CMD': 'PUT', 'DATABASE': 'nr', 'PROGRAM':'blastp', \\\n",
    "                              'QUERY': 'SQETFSDLWKLLPEN'})"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "https://blast.ncbi.nlm.nih.gov/Blast.cgi?CMD=PUT&DATABASE=nr&QUERY=SQETFSDLWKLLPEN&PROGRAM=blastp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {
    "slideshow": {
     "slide_type": "slide"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<!DOCTYPE html PUBLIC \"-//W3C//DTD XHTML 1.0 Transitional//EN\" \"http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd\">\n",
      "<html xmlns=\"http://www.w3.org/1999/xhtml\" xml:lang=\"en\" lang=\"en\">\n",
      "<head>\n",
      "<meta http-equiv=\"Content-Type\" content=\"text/html; charset=UTF-8\" />\n",
      "<meta name=\"jig\" content=\"ncbitoggler ncbiautocomplete\"/>\n",
      "<meta name=\"ncbi_app\" content=\"static\" />\n",
      "<meta name=\"ncbi_pdid\" content=\"blastformatreq\" />\n",
      "<meta name=\"ncbi_stat\" content=\"false\" />\n",
      "<meta name=\"ncbi_sessionid\" content=\"5AAB1B8417760541_0000SID\" />\n",
      "<meta name=\"ncbi_phid\" content=\"5AAB1B84177605410000000000000001\" />\n",
      "<title>NCBI Blast</title>\n",
      "<meta http-equiv=\"Pragma\" content=\"no-cache\">\n",
      "<link rel=\"stylesheet\" type=\"text/css\" href=\"css/uswds.min.css\" media=\"screen\" />\n",
      "<link rel=\"stylesheet\"  type=\"text/css\" href=\"https://www.ncbi.nlm.nih.gov/style-guide/static/nwds/css/nwds.css\"/>\n",
      "<link rel=\"stylesheet\" href=\"css/headerNew.css\"/>\n",
      "<link rel=\"stylesheet\" href=\"https://use.fontawesome.com/releases/v5.5.0/css/all.css\" crossorigin=\"anonymous\"> <!-- Font Awesome icons -->\n",
      "<link rel=\"stylesheet\" type=\"text/css\" href=\"css/footerNew.css\" media=\"screen\" />\n",
      "<link rel=\"stylesheet\" type=\"text/css\" href=\"css/main.css\" media=\"screen\" />\n",
      "<link rel=\"stylesheet\" type=\"text/css\" href=\"css/common.css\" media=\"screen\" />\n",
      "<link rel=\"stylesheet\" type=\"text/css\" href=\"css/blastReq.css\" media=\"screen\" />\n",
      "<!--[if IE]>\n",
      "<link rel=\"stylesheet\" type=\"text/css\" href=\"css/blastReqIE.css\" media=\"screen\" />\n",
      "<![endif]-->\n",
      "<link rel=\"stylesheet\" type=\"text/css\" href=\"css/print.css\" media=\"print\" />\n",
      "\n",
      "\n",
      "<!--[if lte IE 6]>\n",
      "<link rel=\"stylesheet\" type=\"text/css\" href=\"css/ie6_or_less.css\" />\n",
      "<![endif]-->\n",
      "<script type=\"text/javascript\" src=\"/core/jig/1.15.2/js/jig.min.js             \"></script>   \n",
      "<script type=\"text/javascript\" src=\"js/utils.js\"></script>\n",
      "<script type=\"text/javascript\" src=\"js/blast.js\"></script>\n",
      "<script type=\"text/javascript\" src=\"js/format.js\"></script>\n",
      "\n",
      "</head>\n",
      "\n",
      "<body id=\"type-a\">\n",
      "\n",
      "<div id=\"wrap\">\n",
      "\t\t<ul class=\"msg\" id=\"msgServ\"><li class=\"warning\">\n",
      "<p class=\"warning\">\n",
      "This URL will be replaced with <a href=\"https://blast.ncbi.nlm.nih.gov\" title=\"BLAST\">blast.ncbi.nlm.nih.gov</a> on 12/1/2014.\n",
      "Please change your bookmark now.\n",
      "</p>\n",
      "</li></ul>    \t\n",
      "<header class=\"ncbi-header\" role=\"banner\" data-section=\"Header\">\n",
      "<ul class=\"skip_nav\">\n",
      "    <li><a accesskey=\"1\" title=\"NCBI Homepage\" href=\"https://www.ncbi.nlm.nih.gov/\">NCBI Home</a></li>\n",
      "    <li><a accesskey=\"2\" title=\"MyNCBI\" href=\"https://www.ncbi.nlm.nih.gov/myncbi\">Sign in to NCBI</a></li>\n",
      "    <li><a accesskey=\"3\" title=\"Skip to main content\" href=\"#mainCont\">Skip to Main Content</a></li>\n",
      "    <li><a accesskey=\"4\" title=\"Skip to navigation\" href=\"#navcontent\">Skip to Navigation</a></li>\n",
      "    <li><a accesskey=\"0\" title=\"About NCBI Accesskeys\" href=\"https://www.ncbi.nlm.nih.gov/guide/browsers/#accesskeys\">About NCBI Accesskeys</a></li>\n",
      "</ul>\n",
      "<div class=\"usa-grid\">\n",
      "    <div class=\"usa-width-one-whole\">\n",
      "        <div class=\"agencies\">\n",
      "            <div class=\"nih\"><span class=\"usa-sr-only\">National Institutes of Health</span></div>\n",
      "            <div class=\"nlm\">U.S. National Library of Medicine</div>\n",
      "            <div class=\"ncbi\">National Center for Biotechnology Information</div>\n",
      "            <a href=\"https://www.ncbi.nlm.nih.gov/\" class=\"homepage\"><span class=\"usa-sr-only\" id=\"header_ncbi\">NCBI homepage</span></a>\n",
      "        </div>\n",
      "\n",
      "        <div class=\"account\">\n",
      "            <a id=\"account_login\" href=\"https://www.ncbi.nlm.nih.gov/account/?back_url=https%3A%2F%2Fwww%2Encbi%2Enlm%2Enih%2Egov%2Fblast%2FBlast%2Ecgi%3FCMD%3DPUT%26DATABASE%3Dnr%26ENTREZ%5FQUERY%3D%26FULL%5FDBNAME%3Dnr%26JOB%5FTITLE%3DProtein%2BSequence%2B%26MYNCBI%5FUSER%3D9177145548%26MYNCBI%5FUSER%3D9177145548%26ORG%5FDBS%3Dgiless%5Fdbvers5%26PROGRAM%3Dblastp%26QUERY%5FINFO%3DProtein%2BSequence%2B%26QUERY%5FLENGTH%3D15%26RID%3DRES2M90F013%26RTOE%3D57%26USER%5FTYPE%3D2%26USER%5FTYPE%3D2\" class=\"usa-button header-button\">Log in</a>\n",
      "            <button id=\"account_info\" class=\"header-button\" aria-controls=\"account_popup\">\n",
      "                <span class=\"fa fa-user\" aria-hidden=\"true\"></span>\n",
      "                <span class=\"username desktop-only\" aria-hidden=\"true\" id=\"uname_short\"></span>\n",
      "                <span class=\"sr-only\">Show account info</span>\n",
      "            </button>\n",
      "        </div>\n",
      "\n",
      "        <div class=\"ncbi-popup-anchor\">\n",
      "            <div class=\"ncbi-popup account-popup\" id=\"account_popup\" aria-hidden=\"true\" role=\"dialog\" aria-labelledby=\"account-popup-header\">\n",
      "                <div class=\"ncbi-popup-head\">\n",
      "                    <button class=\"ncbi-close-button\"><span class=\"fa fa-window-close\"></span><span class=\"usa-sr-only\">Close</span></button>\n",
      "                    <h4>Account</h4>\n",
      "                </div>\n",
      "                <div class=\"account-user-info\">\n",
      "                    Logged in as:<br>\n",
      "                    <b><span class=\"username\" id=\"uname_long\">username</span></b>\n",
      "                </div>\n",
      "                <div class=\"account-links\">\n",
      "                    <ul class=\"usa-unstyled-list\">\n",
      "                        <li><a id=\"account_myncbi\" href=\"https://www.ncbi.nlm.nih.gov/myncbi/\">Dashboard</a> <span class=\"ncbi-text-small-light\">(My NCBI)</span></li>\n",
      "                        <li><a id=\"account_pubs\" href=\"https://www.ncbi.nlm.nih.gov/myncbi/collections/bibliography/\">Publications</a> <span class=\"ncbi-text-small-light\">(My Bibliography)</span></li>\n",
      "                        <li><a id=\"account_settings\" href=\"https://www.ncbi.nlm.nih.gov/account/settings/\">Account settings</a></li>\n",
      "                        <li><a id=\"account_logout\" href=\"https://www.ncbi.nlm.nih.gov/account/signout/?back_url=https%3A%2F%2Fwww%2Encbi%2Enlm%2Enih%2Egov%2Fblast%2FBlast%2Ecgi%3FCMD%3DPUT%26DATABASE%3Dnr%26ENTREZ%5FQUERY%3D%26FULL%5FDBNAME%3Dnr%26JOB%5FTITLE%3DProtein%2BSequence%2B%26MYNCBI%5FUSER%3D9177145548%26MYNCBI%5FUSER%3D9177145548%26ORG%5FDBS%3Dgiless%5Fdbvers5%26PROGRAM%3Dblastp%26QUERY%5FINFO%3DProtein%2BSequence%2B%26QUERY%5FLENGTH%3D15%26RID%3DRES2M90F013%26RTOE%3D57%26USER%5FTYPE%3D2%26USER%5FTYPE%3D2\">Log out</a></li>\n",
      "                    </ul>\n",
      "                </div>\n",
      "            </div>\n",
      "        </div>\n",
      "\n",
      "    </div>\n",
      "</div>\n",
      "</header>\n",
      "<nav class=\"ncbi-topnav\" id=\"navcontent\">\n",
      "    <div class=\"usa-grid\">\n",
      "        <a class=\"ncbi-topnav-root\" href=\"Blast.cgi\">BLAST <sup>&reg;</sup></a> <span id=\"brc\"><span class=\"brcrmbsign\">&raquo;</span> blastp suite</span>\n",
      "        <ul class=\"rf ncbi-topnav-list\" id=\"topnav-list\">\n",
      "            <li class=\"first \"><a href=\"Blast.cgi?CMD=Web&amp;PAGE_TYPE=BlastHome\" title=\"BLAST Home\">Home</a></li>\n",
      "            <li class=\"recent \"><a href=\"Blast.cgi?CMD=GetSaved&amp;RECENT_RESULTS=on\" title=\"Unexpired BLAST jobs\">Recent Results</a></li>                \n",
      "            <li class=\"saved \"><a href=\"Blast.cgi?CMD=GetSaved\" title=\"Saved sets of BLAST search parameters\">Saved Strategies</a></li>\n",
      "            <li  class= \"last documentation \"> <a href=\"Blast.cgi?CMD=Web&amp;PAGE_TYPE=BlastDocs\" title=\"BLAST documentation\">Help</a></li>                            \n",
      "        </ul>\n",
      "    </div>\n",
      "</nav>\n",
      "\n",
      "\n",
      "        <div id=\"content-wrap\">\n",
      "\n",
      "\t\t<div class=\"pageTitle\">                            \n",
      "                   Format Request\n",
      "                   <span id=\"frmRequestPrTr\"></span>                   \n",
      "                </div>\n",
      "\t\t\t\t<!-- Do errors this way -->\t\t\t\t\n",
      "\t\t\t\t<!--<ul class=\"msg\"><li class=\"\"><p></p></li></ul>-->\n",
      "\t\t\t\t<ul id=\"msgR\" class=\"msg\"><li class=\"\"></li></ul>\n",
      "                <div id=\"content\">\n",
      "\t\t\t\t<form action=\"Blast.cgi\" enctype=\"application/x-www-form-urlencoded\" method=\"post\" name=\"FormatForm\" id=\"FormatForm\">\t\t\t\t\n",
      "\n",
      "<script language=\"JavaScript\">\n",
      "\n",
      " <!--\n",
      "\n",
      "//document.images['BlastHeaderGif'].src = 'html/head_formating.gif';\n",
      "\n",
      "// -->\n",
      "\n",
      "</script>\n",
      "\n",
      "\n",
      "\n",
      "<!--\n",
      "                <p class='info'>\n",
      "<strong>Job submitted.</strong>\n",
      "We estimate that results will be ready in 16 seconds or less.\n",
      "\n",
      "</p>\n",
      "-->\n",
      "\n",
      "<div class=\"fbtn\">\n",
      "<!--\n",
      "<a href=\"javascript:document.forms[0].submit();\">\n",
      "<img align=\"middle\" alt=\"Format button\" border=\"0\" src=\"FormatPage_files/format_but.gif\">\n",
      "</a>\n",
      "-->\n",
      "</div>\n",
      "\n",
      "<dl class=\"summary  query title db\">\n",
      "<dd>\n",
      "</dd>\n",
      "\n",
      "<!-- <span class=\" query title db\">-->\n",
      "<dt class=\"hidden query\">Query</dt><dd class=\"hidden query\">Protein Sequence</dd>\n",
      "<dt class=\"hidden db\">Database</dt><dd class=\"hidden db\">nr</dd>\n",
      "<dt class=\"hidden title\">Job title</dt><dd class=\"hidden title\">Protein Sequence</dd>\n",
      "<dt class=\"hidden entrez\">Entrez Query</dt><dd class=\"hidden entrez\"><span class=\"note entrez\">Note: Your search is limited to records matching this Entrez query</span></dd>\n",
      "<!-- </span> -->\n",
      "<dt><label for=\"rid\">Request ID</label></dt><dd><input name=\"RID\" size=\"50\" type=\"text\" value=\"RES2M90F013\" id=\"rid\" />\n",
      "<input type=\"submit\" value=\"View report\" name=\"ViewReport\" class=\"button\" />\n",
      "<!-- <img border=\"0\" id=\"viewRpButton\" src=\"images/veiwRpButton.jpg\" class=\"viewReport\"  alt=\"View report\"  mouseovImg=\"images/veiwRpButtonOver.jpg\" mouseoutImg=\"images/veiwRpButton.jpg\" mousedownImg=\"images/veiwRpButtonDown.jpg\" mouseupImg=\"images/veiwRpButtonOver.jpg\"  />-->\n",
      "<input type=\"checkbox\" name=\"NEWWINRES\"  form=\"FormatForm\" winType=\"const\" id=\"nw\" class=\"newwin\"  />\n",
      "<label for=\"nw\">Show results in a new window</label>\n",
      "</dd>\n",
      "<dt>Format<br/>\n",
      "<!--<a class='help' href=\"#\">[Help]</a></dt> -->\n",
      "\n",
      "<dd>\n",
      "<table id=\"filterResults\" class=\"options blastp \">\n",
      "\n",
      "<tr class=\"paramSet xgl\">\n",
      "<td class=\"hd\"><label for=\"FORMAT_OBJECT\">Show</label></td>\n",
      "<td>\n",
      "<div class=\"fi\">\n",
      "<select id=\"FORMAT_OBJECT\" class=\"reset\" name=\"FORMAT_OBJECT\" defVal=\"Alignment\">\n",
      "<option value=\"Alignment\" >Alignment</option>\n",
      "<option value=\"PSSM_Scoremat\" >PssmWithParameters</option>\n",
      "<option value=\"Bioseq\"  >Bioseq</option>\n",
      "</select>\n",
      "<label for=\"FORMAT_TYPE\">as</label>\n",
      "<select name=\"FORMAT_TYPE\" id=\"FORMAT_TYPE\" class=\"reset\" defVal=\"HTML\">\n",
      "<option value=\"HTML\"  >HTML</option>\n",
      "<option value=\"Text\"  >Plain text</option>\n",
      "<option value=\"ASN.1\"  >ASN.1</option>\n",
      "<option value=\"XML\"  >XML</option>\n",
      "<option value=\"XML2\"  >XML2</option>\n",
      "<option value=\"JSON2\"  >JSON2</option>\n",
      "<option value=\"XML2_S\"  >XML2_S</option>\n",
      "<option value=\"JSON2_S\"  >JSON2_S</option>\n",
      "<option value=\"SAM_SQ\"  >SAM_SQ</option>\n",
      "<option value=\"SAM\"  >SAM</option>\n",
      "</select>\n",
      "<input name=\"PSSM_FORMAT_TYPE\" value=\"Text\" size=\"3\" id=\"pssmFormat\" type=\"text\" class=\"hidden dispType\" />\n",
      "<input name=\"BIOSEQ_FORMAT_TYPE\" value=\"ASN.1\" size=\"3\" id=\"bioseqFormat\" type=\"text\" class=\"hidden dispType\" />\n",
      "<input name=\"PSSM_SC_FORMAT_TYPE\" value=\"ASN.1\" size=\"3\" id=\"pssmScFormat\" type=\"text\" class=\"hidden dispType\" />\n",
      "<a class=\"resetAll\" id=\"resetAll\" >Reset form to defaults</a>\n",
      "<a class=\"helplink  ui-ncbitoggler\" data-jig=\"ncbitoggler\" title=\"Alignments object formatting help\" id=\"formatHelp\" href=\"#\"><i class=\"fas fa-question-circle\"></i> <span class=\"usa-sr-only\">Help</span></a>\n",
      "<div class=\"ui-helper-reset\" aria-live=\"assertive\" >\n",
      "<p class=\"helpbox ui-ncbitoggler-slave\" id=\"hlp1\">\n",
      "These options control formatting of alignments in results pages. The\n",
      "default is HTML, but other formats (including plain text) are available.\n",
      "PSSM and PssmWithParameters are representations of Position Specific Scoring Matrices and are only available for PSI-BLAST. \n",
      "The Advanced view option allows the database descriptions to be sorted by various indices in a table.\n",
      "</p>\n",
      "</div><!-- ARIA -->\n",
      "</div>\n",
      "</td>\n",
      "</tr>\n",
      "\n",
      "<tr class=\"odd paramSet\">\n",
      "<td class=\"hd\"><label for=\"ALIGNMENT_VIEW\">Alignment View</label></td>\n",
      "<td>\n",
      "<div class=\"fi\">\n",
      "<select name=\"ALIGNMENT_VIEW\" id=\"ALIGNMENT_VIEW\" defVal=\"Pairwise\" class=\"reset\">\n",
      "<option value=\"Pairwise\"  >Pairwise</option>\n",
      "<option value=\"PairwiseWithIdentities\"  >Pairwise with dots for identities</option>\n",
      "<option value=\"QueryAnchored\"  >Query-anchored with dots for identities</option>\n",
      "<option value=\"QueryAnchoredNoIdentities\"  >Query-anchored with letters for identities</option>\n",
      "<option value=\"FlatQueryAnchored\"  >Flat query-anchored with dots for identities</option>\n",
      "<option value=\"FlatQueryAnchoredNoIdentities\"  >Flat query-anchored with letters for identities</option>\n",
      "<option value=\"Tabular\"  >Hit Table</option>\n",
      "</select>\n",
      "\n",
      "<a class=\"helplink  ui-ncbitoggler\" data-jig=\"ncbitoggler\" title=\"Alignments view options help\" id=\"alnViewHelp\" href=\"#\"><i class=\"fas fa-question-circle\"></i> <span class=\"usa-sr-only\">Help</span></a>\n",
      "<div class=\"ui-helper-reset\" aria-live=\"assertive\" >\n",
      "<p class=\"helpbox ui-ncbitoggler-slave\" id=\"hlp2\">\n",
      "Choose how to view alignments.\n",
      "The default \"pairwise\" view shows how each subject sequence aligns\n",
      "individually to the query sequence. The \"query-anchored\" view shows how\n",
      "all subject sequences align to the query sequence. For each view type,\n",
      "you can choose to show \"identities\" (matching residues) as letters or\n",
      "dots.\n",
      "<a href=\"Blast.cgi?CMD=Web&amp;PAGE_TYPE=BlastDocs&amp;DOC_TYPE=BlastHelp#alignment_view\" target=\"helpWin\" title=\"Additional alignments view options help\">more...</a>\n",
      "</p>\n",
      "</div><!-- ARIA -->\n",
      "</div>\n",
      "</td>\n",
      "</tr>\n",
      "\n",
      "<tr class=\"paramSet\">\n",
      "<td class=\"hd\"><label>Display</label></td>\n",
      "<td class=\"cb\">\n",
      "<div class=\"fi\">\n",
      "<input name=\"SHOW_OVERVIEW\" id=\"SHOW_OVERVIEW\" type=\"checkbox\" class=\"cb reset\" defVal=\"checked\" checked=\"checked\" />\n",
      "<label class=\"rb\" for=\"SHOW_OVERVIEW\">Graphical Overview</label>\n",
      "\n",
      "<span id=\"shl\" >\n",
      "<input name=\"SHOW_LINKOUT\" id=\"SHOW_LINKOUT\" type=\"checkbox\" class=\"cb reset\" defVal=\"checked\" checked=\"checked\" />\n",
      "<label class=\"rb\" for=\"SHOW_LINKOUT\">Linkout</label>\n",
      "</span>\n",
      "<span id=\"gts\" >\n",
      "<input name=\"GET_SEQUENCE\" id=\"GET_SEQUENCE\" type=\"checkbox\" class=\"cb reset\" defVal=\"checked\" checked=\"checked\" />\n",
      "<label class=\"rb\" for=\"GET_SEQUENCE\">Sequence Retrieval</label>\n",
      "</span>\n",
      "\n",
      "<input name=\"NCBI_GI\" id=\"NCBI_GI\" type=\"checkbox\" class=\"cb reset hidden\" defVal=\"unchecked\"  />\n",
      "<label class=\"rb hidden\" for=\"NCBI_GI\">NCBI-gi</label>\n",
      "<span id=\"scf\" >\n",
      "<input name=\"SHOW_CDS_FEATURE\" id=\"SHOW_CDS_FEATURE\" type=\"checkbox\" class=\"cb reset blastn\" defVal=\"unchecked\"  />\n",
      "<label for=\"SHOW_CDS_FEATURE\" class=\"blastn\">CDS feature</label>\n",
      "</span>\n",
      "<a class=\"helplink  ui-ncbitoggler\" data-jig=\"ncbitoggler\" title=\"Alignments display options help\" id=\"displayHelp\" href=\"#\"><i class=\"fas fa-question-circle\"></i> <span class=\"usa-sr-only\">Help</span></a>\n",
      "<div class=\"ui-helper-reset\" aria-live=\"assertive\" >\n",
      "<ul class=\"helpbox ui-ncbitoggler-slave\" id=\"hlp3\">\n",
      "<li>Graphical Overview: Graphical Overview: Show graph of similar sequence regions aligned to  query.\n",
      "<a href=\"Blast.cgi?CMD=Web&amp;PAGE_TYPE=BlastDocs&amp;DOC_TYPE=BlastHelp#show_overview\" target=\"helpWin\" title=\"Graphical Overview help\">more...</a>\n",
      "</li>\n",
      "<li>NCBI-gi: Show NCBI gi identifiers.\n",
      "</li>\n",
      "<li>CDS feature: Show annotated coding region and translation.\n",
      "<a href=\"Blast.cgi?CMD=Web&amp;PAGE_TYPE=BlastDocs&amp;DOC_TYPE=BlastHelp#show_cds_feature\" title=\"CDS feature help\" target=\"helpWin\" >more...</a>\n",
      "</li></ul>\n",
      "</div><!-- ARIA -->\n",
      "</div>\n",
      "</td>\n",
      "</tr>\n",
      "\n",
      "\n",
      "<tr class=\"paramSet odd xgl\">\n",
      "<td class=\"hd\"><label>Masking</label></td>\n",
      "<td>\n",
      "<div class=\"fi\">\n",
      "<label for=\"MASK_CHAR\"> Character: </label>\n",
      "<select name=\"MASK_CHAR\" id=\"MASK_CHAR\"  class=\"reset\" defVal=\"2\">\n",
      "<option value=\"0\"  >X for protein, n for nucleotide</option>\n",
      "<option value=\"2\" selected=\"selected\" >Lower Case</option>\n",
      "</select>\n",
      "<label for=\"MASK_COLOR\"> Color:</label>\n",
      "<select name=\"MASK_COLOR\" id=\"MASK_COLOR\" class=\"reset\" defVal=\"1\">\n",
      "<option value=\"0\"  >Black\n",
      "</option>\n",
      "\n",
      "<option value=\"1\" selected=\"selected\" >Grey\n",
      "</option>\n",
      "\n",
      "<option value=\"2\"  >Red\n",
      "</option>\n",
      "\n",
      "</select>\n",
      "<a class=\"helplink  ui-ncbitoggler\" data-jig=\"ncbitoggler\" title=\"Alignments masking help\" id=\"maskingHelp\" href=\"#\"><i class=\"fas fa-question-circle\"></i> <span class=\"usa-sr-only\">Help</span></a>\n",
      "<div class=\"ui-helper-reset\" aria-live=\"assertive\" >\n",
      "<ul class=\"helpbox ui-ncbitoggler-slave\" id=\"hlp4\">\n",
      "<li>Masking Character: Display masked (filtered) sequence regions as lower-case or as specific letters (N for nucleotide, P for protein).\n",
      "</li>\n",
      "<li>Masking Color: Display masked sequence regions in the given color.</li>\n",
      "</ul>\n",
      "</div><!-- ARIA -->\n",
      "</div>\n",
      "</td>\n",
      "</tr>\n",
      "\n",
      "\n",
      "<tr id=\"lr\" class=\"paramSet xgl\">\n",
      "<td class=\"hd\"><label>Limit results</label></td>\n",
      "<td>\n",
      "<div class=\"fi\">\n",
      "<label for=\"FRM_DESCRIPTIONS\">Descriptions:</label>\n",
      "<select name=\"DESCRIPTIONS\" id=\"FRM_DESCRIPTIONS\" class=\"reset\" defVal=\"100\">\n",
      "<option value=\"0\"      >0</option>\n",
      "<option value=\"10\"     >10</option>\n",
      "<option value=\"50\"     >50</option>\n",
      "<option value=\"100\"   selected=\"selected\" >100</option>\n",
      "<option value=\"250\"    >250</option>\n",
      "<option value=\"500\"    >500</option>\n",
      "<option value=\"1000\"   >1000</option>\n",
      "<option value=\"5000\"   >5000</option>\n",
      "<option value=\"10000\"  >10000</option>\n",
      "<option value=\"20000\"  >20000</option>\n",
      "</select>\n",
      "\n",
      "<label for=\"FRM_NUM_OVERVIEW\">Graphical overview:</label>\n",
      "<select name=\"NUM_OVERVIEW\" id=\"FRM_NUM_OVERVIEW\" class=\"reset\" defVal=\"100\">\n",
      "<option value=\"0\"     >0</option>\n",
      "<option value=\"10\"    >10</option>\n",
      "<option value=\"50\"    >50</option>\n",
      "<option value=\"100\"  selected=\"selected\" >100</option>\n",
      "<option value=\"250\"   >250</option>\n",
      "<option value=\"500\"  >500</option>\n",
      "<option value=\"1000\"  >1000</option>\n",
      "</select>\n",
      "<span id=\"frmAln\">\n",
      "<label for=\"FRM_ALIGNMENTS\">Alignments:</label>\n",
      "<select name=\"ALIGNMENTS\" id=\"FRM_ALIGNMENTS\" class=\"reset\" defVal=\"100\">\n",
      "<option value=\"0\"      >0</option>\n",
      "<option value=\"10\"     >10</option>\n",
      "<option value=\"50\"     >50</option>\n",
      "<option value=\"100\"   selected=\"selected\" >100</option>\n",
      "<option value=\"250\"    >250</option>\n",
      "<option value=\"500\"    >500</option>\n",
      "<option value=\"1000\"   >1000</option>\n",
      "<option value=\"5000\"   >5000</option>\n",
      "<option value=\"10000\"  >10000</option>\n",
      "<option value=\"20000\"  >20000</option>\n",
      "</select>\n",
      "</span>\n",
      "<label for=\"FRM_LINE_LENGTH\">Line length:</label>\n",
      "<select name=\"LINE_LENGTH\" id=\"FRM_LINE_LENGTH\" class=\"reset\" defVal=\"60\">\n",
      "<option value=\"60\"     >60</option>\n",
      "<option value=\"90\"     >90</option>\n",
      "<option value=\"120\"     >120</option>\n",
      "<option value=\"150\"     >150</option>\n",
      "</select>\n",
      "<a class=\"helplink  ui-ncbitoggler\" data-jig=\"ncbitoggler\" title=\"Limit number of descriptions/alignments help\" id=\"numHelp\" href=\"#\"><i class=\"fas fa-question-circle\"></i> <span class=\"usa-sr-only\">Help</span></a>\n",
      "<div class=\"ui-helper-reset\" aria-live=\"assertive\" >\n",
      "<ul class=\"helpbox ui-ncbitoggler-slave\" id=\"hlp5\">\n",
      "<li>Descriptions: Show short descriptions for up to the given number of  sequences.</li> \n",
      "<li>Alignments:  Show alignments for up to the given number of sequences, in order of statistical significance.</li>\n",
      "<li>Line lenghth:  Number of letters to show on one line in an alignment.</li>\n",
      "</ul>\n",
      "</div><!-- ARIA -->\n",
      "</div>\n",
      "</td>\n",
      "</tr>\n",
      "\n",
      "<tr class=\"paramSet odd xgl \">\n",
      "<td class=\"hd\"></td>\n",
      "<td>\n",
      "<div class=\"\">\n",
      "<label for=\"qorganism\">Organism</label>\n",
      "<span class=\"instr\">Type common name, binomial, taxid, or group name. Only 20 top taxa will be shown.</span><br/>\n",
      "<input name=\"FORMAT_ORGANISM\" size=\"55\"  type=\"text\" id=\"qorganism\" value=\"\" data-jigconfig=\"dictionary:'taxids_sg',isCrossDomain:false\" autocomplete=\"off\" data-jig=\"ncbiautocomplete\" class=\"reset\">\n",
      "<input type=\"hidden\" value = \"1\" name=\"FORMAT_NUM_ORG\" id=\"numOrg\" />\n",
      "<input type=\"checkbox\" name=\"FORMAT_ORG_EXCLUDE\"  class=\"oExclR cb\" id=\"orgExcl\"/>        \n",
      "<label for=\"orgExcl\" class=\"right\">exclude</label>\n",
      "<a href=\"#\" title=\"Add organism\" id=\"addOrg\"><img border=\"0\" src=\"css/images/addOrg.jpg\" id=\"addOrgIm\"   alt=\"Add organism\"  mouseovImg=\"css/images/addOrgOver.jpg\" mouseoutImg=\"css/images/addOrg.jpg\" mousedownImg=\"css/images/addOrgDown.jpg\" mouseupImg=\"css/images/addOrgOver.jpg\"  /></a>\n",
      "<div id=\"orgs\">\n",
      "\n",
      "</div>\n",
      "<div class=\"fi\">\n",
      "<a class=\"helplink  ui-ncbitoggler\" data-jig=\"ncbitoggler\" title=\"Limit results by organism help\" id=\"organismHelp\" href=\"#\"><i class=\"fas fa-question-circle\"></i> <span class=\"usa-sr-only\">Help</span></a>\n",
      "<div class=\"ui-helper-reset\" aria-live=\"assertive\" >\n",
      "<p class=\"helpbox ui-ncbitoggler-slave\" id=\"hlp6\">\n",
      "Show only sequences from the given organism.\n",
      "</p>\n",
      "</div><!-- ARIA -->\n",
      "</div>\n",
      "</div>\n",
      "</td>\n",
      "</tr>\n",
      "\n",
      "<tr class=\"paramSet xgl hidden\">\n",
      "<td class=\"hd\"></td>\n",
      "<td>\n",
      "<div class=\"fi\">\n",
      "<label for=\"FORMAT_EQ_TEXT\">Entrez query:</label>\n",
      "<input name=\"FORMAT_EQ_TEXT\" id=\"FORMAT_EQ_TEXT\" size=\"60\" type=\"text\" value=\"\" class=\"reset\" />\n",
      "<a class=\"helplink  ui-ncbitoggler\" data-jig=\"ncbitoggler\" title=\"Limit results by Entrez query help\" id=\"entrezHelp\" href=\"#\"><i class=\"fas fa-question-circle\"></i> <span class=\"usa-sr-only\">Help</span></a>\n",
      "<div class=\"ui-helper-reset\" aria-live=\"assertive\" >\n",
      "<p class=\"helpbox ui-ncbitoggler-slave\" id=\"hlp7\">\n",
      "Show only those sequences that match the given Entrez query.\n",
      "<a href=\"Blast.cgi?CMD=Web&amp;PAGE_TYPE=BlastDocs&amp;DOC_TYPE=BlastHelp#limit_result\" target=\"helpWin\" title=\"Additional limit results by Entrez query help\"  target=\"helpWin\">more...</a>\n",
      "</p>\n",
      "</div><!-- ARIA -->\n",
      "</div>\n",
      "</td>\n",
      "</tr>\n",
      "\n",
      "  \n",
      "<tr class=\"paramSet odd xgl\">\n",
      "<td class=\"hd\"></td>\n",
      "<td>\n",
      "<div class=\"fi\">\n",
      "<label for=\"EXPECT_LOW\">Expect Min:</label> <input name=\"EXPECT_LOW\" id=\"EXPECT_LOW\" size=\"10\" type=\"text\" value=\"\" class=\"reset\"/>\n",
      "<label for=\"EXPECT_HIGH\">Expect Max:</label> <input name=\"EXPECT_HIGH\" id=\"EXPECT_HIGH\" size=\"10\" type=\"text\" value=\"\" class=\"reset\" />\n",
      "<a class=\"helplink  ui-ncbitoggler\" data-jig=\"ncbitoggler\" title=\"Limit results by expect value range help\" id=\"expectHelp\" href=\"#\"><i class=\"fas fa-question-circle\"></i> <span class=\"usa-sr-only\">Help</span></a>\n",
      "<div class=\"ui-helper-reset\" aria-live=\"assertive\" >\n",
      "<p class=\"helpbox ui-ncbitoggler-slave\" id=\"hlp8\">\n",
      "Show only sequences with expect values in the given range.\n",
      "<a href=\"Blast.cgi?CMD=Web&amp;PAGE_TYPE=BlastDocs&amp;DOC_TYPE=BlastHelp#expect_range\" target=\"helpWin\" title=\"Additional limit results by expect value range help\">more...</a>\n",
      "</p>\n",
      "</div><!-- ARIA -->\n",
      "</div>\n",
      "</td>\n",
      "</tr>\n",
      "<tr class=\"paramSet xgl\">\n",
      "<td class=\"hd\"></td>\n",
      "<td>\n",
      " <div class=\"fi\">\n",
      "<label for=\"PERC_IDENT_LOW\">Percent Identity Min:</label> <input name=\"PERC_IDENT_LOW\" id=\"PERC_IDENT_LOW\" size=\"10\" type=\"text\" value=\"\" class=\"reset\"/>\n",
      "<label for=\"PERC_IDENT_HIGH\">Percent Identity Max:</label> <input name=\"PERC_IDENT_HIGH\" id=\"PERC_IDENT_HIGH\" size=\"10\" type=\"text\" value=\"\" class=\"reset\" />\n",
      "<a class=\"helplink  ui-ncbitoggler\" data-jig=\"ncbitoggler\" title=\"Limit results by percent identity range help\" id=\"percIdentHelp\" href=\"#\"><i class=\"fas fa-question-circle\"></i> <span class=\"usa-sr-only\">Help</span></a>\n",
      "<div class=\"ui-helper-reset\" aria-live=\"assertive\" >\n",
      "<p class=\"helpbox ui-ncbitoggler-slave\" id=\"hlp10\">\n",
      " Show only sequences with percent identity values in the given range.  \n",
      "</p>\n",
      "</div><!-- ARIA -->\n",
      "</div>\n",
      "</td>\n",
      "</tr>      \n",
      "<tr class=\"psiBlast odd paramSet xgl \">\n",
      "<td class=\"hd\"><label>Format for</label></td>\n",
      "<td>\n",
      "<div class=\"fi\">\n",
      "<input name=\"RUN_PSIBLAST_FORM\" id=\"RUN_PSIBLAST\" type=\"checkbox\" class=\"cb psiBlast hidden\"  />\n",
      "<label for=\"I_THRESH\">PSI-BLAST with inclusion threshold:</label>\n",
      "<input name=\"I_THRESH\" id=\"I_THRESH\" size=\"10\" type=\"text\" value=\"\" defVal=\"0.005\" />\n",
      "<a class=\"helplink  ui-ncbitoggler\" data-jig=\"ncbitoggler\" title=\"PSI BLAST formatting help\" id=\"psiHelp\" href=\"#\"><i class=\"fas fa-question-circle\"></i> <span class=\"usa-sr-only\">Help</span></a>\n",
      "<div class=\"ui-helper-reset\" aria-live=\"assertive\" >\n",
      "<ul class=\"helpbox ui-ncbitoggler-slave\" id=\"hlp9\">\n",
      "<li>Format for PSI-BLAST: The Position-Specific Iterated BLAST (PSI-BLAST) program performs iterative searches with a protein query, \n",
      "in which sequences found in one round of search are used to build a custom score model for the next round.\n",
      "<a href=\"Blast.cgi?CMD=Web&amp;PAGE_TYPE=BlastDocs&amp;DOC_TYPE=BlastHelp#psiblast\" target=\"helpWin\" title=\"Additional PSI BLAST formatting help\">more...</a>\n",
      "</li>  \n",
      "<li>Inclusion Threshold: This sets the statistical significance threshold for including a sequence in the model used \n",
      "by PSI-BLAST to create the PSSM on the next iteration.</li> \n",
      "</ul>\n",
      "</div><!-- ARIA -->\n",
      "</div>\n",
      "</td>\n",
      "</tr>\n",
      "</table>\n",
      "</dd>\n",
      "</dl>\n",
      "\n",
      "<input name=\"RID\" value=\"RES2M90F013\" type=\"hidden\" />\n",
      "<input name=\"CDD_RID\" value=\"\" type=\"hidden\" />\n",
      "<input name=\"CDD_SEARCH_STATE\" type=\"hidden\" value=\"\" />\n",
      "\n",
      "<input name=\"STEP_NUMBER\" value=\"\" id=\"stepNumber\" type=\"hidden\" />\n",
      "<input name=\"CMD\" value=\"Get\" type=\"hidden\" />\n",
      "<input name=\"FORMAT_EQ_OP\" value=\"AND\" type=\"hidden\" />\n",
      "<input name=\"RESULTS_PAGE_TARGET\" type=\"hidden\" id=\"resPageTarget\" value=\"Blast_Results_for_1458582564\" />\n",
      "<input name=\"QUERY_INFO\" type=\"hidden\" value=\"Protein Sequence\" />                   \t\t\n",
      "<input name=\"ENTREZ_QUERY\" type=\"hidden\" value=\"\" />\n",
      "<input name=\"QUERY_INDEX\" type=\"hidden\" value=\"0\"/>\n",
      "<input name=\"NUM_QUERIES\" type=\"hidden\" value=\"1\"/>\n",
      "<input name=\"CONFIG_DESCR\" type=\"hidden\" value=\"2,3,6,7,8,9,10,11,12\" />\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "<!-- Those params are set in the template (blastn.dat, blastp.dat etc. -->\n",
      "<input name=\"BLAST_PROGRAMS\" type=\"hidden\" value=\"blastp\"/>\n",
      "<input name=\"PAGE\" type=\"hidden\" value=\"Proteins\"/>\n",
      "<input name=\"PROGRAM\" type=\"hidden\" value=\"blastp\"/>\n",
      "<input name=\"MEGABLAST\" type=\"hidden\" value=\"\" />\n",
      "<input name=\"RUN_PSIBLAST\" type=\"hidden\" value=\"\" />\n",
      "<input name=\"BLAST_SPEC\" id=\"blastSpec\" type=\"hidden\" value=\"\"/>\n",
      "\n",
      "\n",
      "<input name=\"QUERY\" type=\"hidden\" value=\"\"/>\n",
      "<input name=\"JOB_TITLE\" type=\"hidden\" value=\"Protein Sequence\"/>\n",
      "<input name=\"QUERY_TO\" type=\"hidden\" value=\"\"/>\n",
      "<input name=\"QUERY_FROM\" type=\"hidden\" value=\"\"/>\n",
      "<input name=\"SUBJECTS_FROM\" type=\"hidden\" value=\"\"/>\n",
      "<input name=\"SUBJECTS_TO\" type=\"hidden\" value=\"\"/>\n",
      "<input name=\"EQ_TEXT\" type=\"hidden\" value=\"\"/>\n",
      "<input name=\"ORGN\" type=\"hidden\" value=\"\"/>\n",
      "<input name=\"EQ_MENU\" type=\"hidden\" value=\"\"/>\n",
      "<input name=\"ORG_EXCLUDE\" type=\"hidden\" value=\"\"/>\n",
      "<input name=\"PHI_PATTERN\" type=\"hidden\" value=\"\"/>\n",
      "<input name=\"EXPECT\" type=\"hidden\" value=\"\"/>\t\t\t\t\t\t\t\t\t\n",
      "<input name=\"DATABASE\" type=\"hidden\" value=\"nr\"/>\n",
      "<input name=\"DB_GROUP\" type=\"hidden\" value=\"\"/>\n",
      "<input name=\"SUBGROUP_NAME\" type=\"hidden\" value=\"\"/>\n",
      "\n",
      "<input name=\"GENETIC_CODE\" type=\"hidden\" value=\"\"/>\n",
      "<input name=\"WORD_SIZE\" type=\"hidden\" value=\"\"/>\n",
      "<input name=\"MATCH_SCORES\" type=\"hidden\" value=\"\"/>\t\t\t\n",
      "<input name=\"MATRIX_NAME\" type=\"hidden\" value=\"\"/>\t\t\t\t\n",
      "<input name=\"GAPCOSTS\" type=\"hidden\" value=\"\"/>\n",
      "<input name=\"MAX_NUM_SEQ\" id=\"maxNumSeq\" type=\"hidden\" value=\"\"/>\t\t\t\t\t\n",
      "<input name=\"COMPOSITION_BASED_STATISTICS\" type=\"hidden\" value=\"\"/>\t\t\t\n",
      "<input name=\"NEWWIN\" type=\"hidden\" value=\"\"/>\n",
      "<input name=\"SHORT_QUERY_ADJUST\" type=\"hidden\" value=\"\"/>\n",
      "<input name=\"FILTER\" type=\"hidden\" value=\"\"/>\n",
      "<input name=\"REPEATS\" type=\"hidden\" value=\"\"/>\n",
      "<input name=\"ID_FOR_PSSM\" type=\"hidden\" value=\"\"/>\n",
      "<input name=\"EXCLUDE_MODELS\" type=\"hidden\" value=\"\"/>\n",
      "<input name=\"EXCLUDE_SEQ_UNCULT\" type=\"hidden\" value=\"\"/>\n",
      "<input name=\"WP_PROTEINS\" type=\"hidden\" value=\"\"/>\n",
      "<input name=\"SEQ_FROM_TYPE\" type=\"hidden\" value=\"\"/>\n",
      "<input name=\"ENTREZ_QUERY\" type=\"hidden\" value=\"\"/>\n",
      "<input name=\"ENTREZ_QUERY_PRESET\" type=\"hidden\" value=\"\"/>\n",
      "<input name=\"ENTREZ_QUERY_PRESET_EXCL\" type=\"hidden\" value=\"\"/>\n",
      "<input name=\"NUM_ORG\" type=\"hidden\" value = \"1\" />\n",
      "\n",
      "<!-- PSSM -->\n",
      "<input name=\"LCASE_MASK\" type=\"hidden\" value=\"\"/>\n",
      "<input name=\"TEMPLATE_TYPE\" type=\"hidden\" value=\"\"/>\n",
      "<input name=\"TEMPLATE_LENGTH\" type=\"hidden\" value=\"\"/>\n",
      "<input name=\"I_THRESH\" type=\"hidden\" value=\"\"/>\n",
      "<input name=\"PSI_PSEUDOCOUNT\" type=\"hidden\" value=\"\"/>\n",
      "<input name=\"DI_THRESH\" type=\"hidden\" id=\"diThresh\" value=\"\"/>\n",
      "<input name=\"HSP_RANGE_MAX\" type=\"hidden\" value=\"\"/>\n",
      "\n",
      "\n",
      "\n",
      "<input name=\"ADJUSTED_FOR_SHORT_QUERY\" type=\"hidden\" value=\"\"/>\n",
      "<input name=\"MIXED_QUERIES\" type=\"hidden\" value=\"\"/>\n",
      "<input name=\"MIXED_DATABASE\" id=\"mixedDb\" type=\"hidden\" value=\"\"/>\n",
      "<input name=\"BUILD_NAME\"  type=\"hidden\" value=\"\"/>\n",
      "<input name=\"ORG_DBS\"  type=\"hidden\" value=\"giless_dbvers5\"/>\n",
      "<input name=\"WWW_BLAST_TYPE\" type=\"hidden\" value=\"\"/>\n",
      "\n",
      "<!--QBlastInfoBegin\n",
      "    RID = RES2M90F013\n",
      "    RTOE = 57\n",
      "QBlastInfoEnd\n",
      "-->\n",
      "</form>\t\t                              \t\t\n",
      "\t\t\t\t\n",
      "\t\t\t\t</div><!-- /#content -->\n",
      "\n",
      "        </div><!-- /#content-wrap -->\n",
      "\n",
      "\t\t <footer id=\"ncbi-footer\" class=\"usa-footer usa-footer-medium usa-sans ncbi-footer\" role=\"contentinfo\" data-section=\"Footer\">\n",
      "  <div class=\"icon-section\">\n",
      "    <div class=\"usa-grid\">\n",
      "\t\t<div class=\"usa-width-one-whole\">\n",
      "\t\t<h3 class=\"usa-sr-only\">Connect</h3>\n",
      "\t\t<ul class=\"ncbi-inline-list icon-list\">\n",
      "\t\t\t<li><a id=\"footer_twitter\" href=\"https://twitter.com/ncbi\">Twitter</a></li>\n",
      "\t\t\t<li><a id=\"footer_facebook\" href=\"https://www.facebook.com/ncbi.nlm\">Facebook</a></li>\n",
      "\t\t\t<li><a id=\"footer_youtube\" href=\"https://www.youtube.com/user/NCBINLM\">YouTube</a></li>\n",
      "\t\t\t<li><a id=\"footer_linkedin\" href=\"https://www.linkedin.com/company/national-center-for-biotechnology-information-ncbi-\">LinkedIn</a></li>\n",
      "\t\t\t<li><a id=\"footer_github\" href=\"https://github.com/ncbi\">GitHub</a></li>\n",
      "\t\t</ul>\n",
      "\t\t<ul class=\"ncbi-inline-list icon-list icon-list-elastic\">\t\n",
      "\t\t\t<li class=\"li-blog\"><a id=\"footer_blog\" href=\"https://ncbiinsights.ncbi.nlm.nih.gov/\">Blog</a></li>\n",
      "\t\t\t<li class=\"li-support\"><a id=\"footer_support\" href=\"https://support.ncbi.nlm.nih.gov/\">Support Center</a></li>\n",
      "\t\t</ul>\n",
      "\t\t</div>\n",
      "\t\t</div>\n",
      "  </div>\n",
      "  <div class=\"usa-footer-secondary_section ncbi-section\">\n",
      "    <div class=\"usa-grid\">\n",
      "      <div class=\"usa-width-one-half ncbi\">\n",
      "        <h3 class=\"ncbi-full-name\">National Center for Biotechnology Information</h3>\n",
      "\t\t<address>\n",
      "        <a id=\"google_maps\" href=\"https://www.google.com/maps/place/8600+Rockville+Pike,+Bethesda,+MD+20894/@38.9959508,-77.101021,17z/data=!3m1!4b1!4m5!3m4!1s0x89b7c95e25765ddb:0x19156f88b27635b8!8m2!3d38.9959508!4d-77.0988323\">\n",
      "        <span class=\"h-adr\"> <span class=\"p-street-address break\">8600 Rockville Pike</span>\n",
      "        <span class=\"p-locality\">Bethesda</span> <span class=\"p-region\"> MD</span>, <span class=\"p-postal-code\">20894</span> <span class=\"p-country-name\">USA</span> </span>\n",
      "\t\t</a>\n",
      "\t\t</address>\n",
      "        <ul class=\"ncbi-inline-list\">\n",
      "\t\t\t<li><a id=\"footer_about\" href=\"https://www.ncbi.nlm.nih.gov/home/about/\">About us</a></li>\n",
      "          <li><a id=\"footer_contact\" href=\"https://www.ncbi.nlm.nih.gov/home/about/contact.shtml\">Contact us</a></li>\n",
      "          <li><a id=\"footer_policies\" href=\"https://www.ncbi.nlm.nih.gov/home/about/policies.shtml\">Policies</a></li>\n",
      "          <li><a id=\"footer_foia\" href=\"https://www.nih.gov/institutes-nih/nih-office-director/office-communications-public-liaison/freedom-information-act-office\">FOIA</a></li>\n",
      "        </ul>\n",
      "      </div>\n",
      "      <div class=\"usa-width-one-half\">\n",
      "      <div class=\"usa-grid-full\">\n",
      "\t\t<div class=\"usa-width-one-third\">\n",
      "\t\t<h4>Popular</h4>\n",
      "\n",
      "        <ul class=\"ncbi-unstyled-list\">\n",
      "\t\t\t\t<li><a id=\"footer_pubmed\" href=\"https://www.ncbi.nlm.nih.gov/pubmed/\">PubMed</a></li>\n",
      "\t\t\t\t<li><a id=\"footer_pmc\" href=\"https://www.ncbi.nlm.nih.gov/pmc/\">PubMed Central</a></li>\n",
      "\t\t\t\t<li><a id=\"footer_bookshelf\" href=\"https://www.ncbi.nlm.nih.gov/books/\">Bookshelf</a></li>\n",
      "\t\t\t\t<li><a id=\"footer_pubchem\" href=\"https://pubchem.ncbi.nlm.nih.gov\">PubChem</a></li>\n",
      "\t\t\t\t<li><a id=\"footer_gene\" href=\"https://www.ncbi.nlm.nih.gov/gene/\">Gene</a></li>\n",
      "\t\t\t\t<li><a id=\"footer_blast\" href=\"https://blast.ncbi.nlm.nih.gov\">BLAST</a></li>\n",
      "\t\t\t\t<li><a id=\"footer_nucleotide\" href=\"https://www.ncbi.nlm.nih.gov/nucleotide/\">Nucleotide</a></li>\n",
      "\t\t\t\t<li><a id=\"footer_protein\" href=\"https://www.ncbi.nlm.nih.gov/protein/\">Protein</a></li>\n",
      "\t\t\t\t<li><a id=\"footer_geo\" href=\"https://www.ncbi.nlm.nih.gov/geo/\">GEO</a></li>\n",
      "        </ul>\n",
      "\n",
      "\t\t\t</div>\n",
      "\t\t<div class=\"usa-width-one-third\">\n",
      "\t\t\t<h4>Resources</h4>\n",
      "\t\t\t<ul class=\"ncbi-unstyled-list ncbi-undecorated-links\">\n",
      "\t\t\t\t  <li><a id=\"footer_literature\" href=\"https://www.ncbi.nlm.nih.gov/home/literature/\">Literature</a></li>\n",
      "\t\t\t\t  <li><a id=\"footer_health\" href=\"https://www.ncbi.nlm.nih.gov/home/health/\">Health</a></li>\n",
      "\t\t\t\t  <li><a id=\"footer_genomes\" href=\"https://www.ncbi.nlm.nih.gov/home/genomes/\">Genomes</a></li>\n",
      "\t\t\t\t  <li><a id=\"footer_genes\" href=\"https://www.ncbi.nlm.nih.gov/home/genes/\">Genes</a></li>\n",
      "\t\t\t\t  <li><a id=\"footer_proteins\" href=\"https://www.ncbi.nlm.nih.gov/home/proteins/\">Proteins</a></li>\n",
      "\t\t\t\t  <li><a id=\"footer_chemicals\" href=\"https://www.ncbi.nlm.nih.gov/home/chemicals/\">Chemicals</a></li>\n",
      "\t\t\t</ul>\n",
      "\t\t</div>\n",
      "\t\t<div class=\"usa-width-one-third\">\n",
      "   <h4>Actions</h4>\n",
      "   <ul class=\"ncbi-unstyled-list\">\n",
      "              <li><a id=\"footer_submit\" href=\"https://www.ncbi.nlm.nih.gov/home/submit/\">Submit</a></li>\n",
      "              <li><a id=\"footer_download\" href=\"https://www.ncbi.nlm.nih.gov/home/download/\">Download</a></li>\n",
      "              <li><a id=\"footer_learn\" href=\"https://www.ncbi.nlm.nih.gov/home/learn/\">Learn</a></li>\n",
      "              <li><a id=\"footer_develop\" href=\"https://www.ncbi.nlm.nih.gov/home/develop/\">Develop</a></li>\n",
      "              <li><a id=\"footer_analyze\" href=\"https://www.ncbi.nlm.nih.gov/home/analyze/\">Analyze</a></li>\n",
      "              <li><a id=\"footer_research\" href=\"https://www.ncbi.nlm.nih.gov/research/\">Research</a></li>\n",
      "   </ul>\n",
      "\t\t  </div>\n",
      "\t\t  </div>\n",
      "\t\t  </div>\n",
      "    </div>\n",
      "\t</div>\n",
      "     <div class=\"usa-footer-secondary_section gov-agencies\"> \n",
      "    <div class=\"usa-grid\">\n",
      "    <div class=\"usa-width-one-whole\">\n",
      "\t\t<p>\n",
      "\t\t\t<a id=\"footer_nlm\" href=\"https://www.nlm.nih.gov/\">NLM</a>\n",
      "\t\t\t&ensp;|&ensp;\n",
      "\t\t\t<a id=\"footer_nih\" href=\"https://www.nih.gov/\">NIH</a>\n",
      "\t\t\t&ensp;|&ensp;\n",
      "\t\t\t<a id=\"footer_hhs\" href=\"https://www.hhs.gov/\">HHS</a>\n",
      "\t\t\t&ensp;|&ensp;\n",
      "\t\t\t<a id=\"footer_usagov\" href=\"https://www.usa.gov/\">USA.gov</a>\n",
      "\t\t</p>\n",
      "\t </div>\n",
      "\t </div>\n",
      "\t</div>\n",
      "\n",
      " </footer>\n",
      "<script type=\"text/javascript\" src=\"js/nwds.js\"></script>\n",
      "<script type=\"text/javascript\" src=\"js/ncbipopup.js\"></script>\n",
      "<script type=\"text/javascript\" src=\"js/headerNew.js\"></script>\n",
      "<script src=\"js/uswds.min.js\"></script>\n",
      "   </div><!--/#wrap-->\n",
      "\n",
      "<script type=\"text/javascript\" src=\"/portal/portal3rc.fcgi/rlib/js/InstrumentOmnitureBaseJS/InstrumentNCBIBaseJS/InstrumentPageStarterJS.js\"></script>\n",
      "<!--\n",
      "<script type=\"text/javascript\" src=\"://www.ncbi.nlm.nih.gov/portal/portal3rc.fcgi/supportedbrowsers/js/nonportalbc_min.js\"></script>\n",
      "<script type=\"text/javascript\">$(\"#browsers_ajax\").browserCheck();</script>\n",
      "-->\n",
      "</body>\n",
      "\n",
      "</html>\n",
      "\n",
      " \n",
      "\n"
     ]
    }
   ],
   "source": [
    "print(response.text)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "slideshow": {
     "slide_type": "slide"
    }
   },
   "source": [
    "# BLAST GET"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Using the request id parsed from the result of the PUT, get the results of the search.  Common attributes:\n",
    "\n",
    "* RID - mandatory\n",
    "* FORMAT_TYPE - HTML, Text, ASN.1, XML\n",
    "* ALIGNMENTS - number of alignments (default 500)\n",
    "* ALIGNMENT_VIEW - Pairwise, QueryAnchored, QueryAnchoredNoIdentities, FlatQueryAnchored, FlatQueryAnchoredNoIdentities, Tabular"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "RES2M90F013\n"
     ]
    }
   ],
   "source": [
    "rid = re.search(r'RID = (\\S+)',response.text).group(1)\n",
    "print(rid)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {
    "slideshow": {
     "slide_type": "slide"
    }
   },
   "outputs": [],
   "source": [
    "result = requests.get('http://www.ncbi.nlm.nih.gov/blast/Blast.cgi?CMD=GET&RID=%s&FORMAT_TYPE=XML' % rid).text"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<!DOCTYPE html PUBLIC \"-//W3C//DTD XHTML 1.0 Transitional//EN\" \"http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd\">\n",
      "<html xmlns=\"http://www.w3.org/1999/xhtml\" xml:lang=\"en\" lang=\"en\">\n",
      "<head>\n",
      "<meta http-equiv=\"Content-Type\" content=\"text/html; charset=UTF-8\" />\n",
      "<meta name=\"jig\" content=\"ncbitoggler\"/>\n",
      "<meta name=\"ncbitoggler\" content=\"animation:'none'\"/>\n",
      "<title>NCBI Blast:</title>\n",
      "<script type=\"text/javascript\" src=\"/core/jig/1.15.2/js/jig.min.js             \"></script>\n",
      "<script type=\"text/javascript\">    jQuery.getScript(\"/core/alerts/alerts.js\", function() {\n",
      "        galert(['div#header', 'body > *:nth-child(1)'])\n",
      "    });</script>\n",
      "<meta http-equiv=\"Pragma\" content=\"no-cache\">\n",
      "<link rel=\"stylesheet\" type=\"text/css\" href=\"css/uswds.min.css\" media=\"screen\" />\n",
      "<link rel=\"stylesheet\"  type=\"text/css\" href=\"https://www.ncbi.nlm.nih.gov/style-guide/static/nwds/css/nwds.css\"/>\n",
      "<link rel=\"stylesheet\" href=\"css/headerNew.css\"/>\n",
      "<link rel=\"stylesheet\" href=\"https://use.fontawesome.com/releases/v5.5.0/css/all.css\" crossorigin=\"anonymous\"> <!-- Font Awesome icons -->\n",
      "<link rel=\"stylesheet\" type=\"text/css\" href=\"css/footerNew.css\" media=\"screen\" />\n",
      "<link rel=\"stylesheet\" type=\"text/css\" href=\"css/main.css\" media=\"screen\" />\n",
      "<link rel=\"stylesheet\" type=\"text/css\" href=\"css/blastRes.css\" media=\"screen\" />\n",
      "<link rel=\"stylesheet\" type=\"text/css\" href=\"css/print.css\" media=\"print\" />\n",
      "<!--[if lte IE 6]>\n",
      "<link rel=\"stylesheet\" type=\"text/css\" href=\"css/ie6_or_less.css\" />\n",
      "<![endif]-->\n",
      "<script type=\"text/javascript\" src=\"js/utils.js\"></script>\n",
      "<script type=\"text/javascript\" src=\"js/results.js\"></script>\n",
      "</head>\n",
      "\n",
      "<body id=\"type-a\" class=\"noToggleCheck\" >\n",
      "<div id=\"wrap\">\n",
      "\t\t    \t\n",
      "<header class=\"ncbi-header\" role=\"banner\" data-section=\"Header\">\n",
      "<ul class=\"skip_nav\">\n",
      "    <li><a accesskey=\"1\" title=\"NCBI Homepage\" href=\"https://www.ncbi.nlm.nih.gov/\">NCBI Home</a></li>\n",
      "    <li><a accesskey=\"2\" title=\"MyNCBI\" href=\"https://www.ncbi.nlm.nih.gov/myncbi\">Sign in to NCBI</a></li>\n",
      "    <li><a accesskey=\"3\" title=\"Skip to main content\" href=\"#mainCont\">Skip to Main Content</a></li>\n",
      "    <li><a accesskey=\"4\" title=\"Skip to navigation\" href=\"#navcontent\">Skip to Navigation</a></li>\n",
      "    <li><a accesskey=\"0\" title=\"About NCBI Accesskeys\" href=\"https://www.ncbi.nlm.nih.gov/guide/browsers/#accesskeys\">About NCBI Accesskeys</a></li>\n",
      "</ul>\n",
      "<div class=\"usa-grid\">\n",
      "    <div class=\"usa-width-one-whole\">\n",
      "        <div class=\"agencies\">\n",
      "            <div class=\"nih\"><span class=\"usa-sr-only\">National Institutes of Health</span></div>\n",
      "            <div class=\"nlm\">U.S. National Library of Medicine</div>\n",
      "            <div class=\"ncbi\">National Center for Biotechnology Information</div>\n",
      "            <a href=\"https://www.ncbi.nlm.nih.gov/\" class=\"homepage\"><span class=\"usa-sr-only\" id=\"header_ncbi\">NCBI homepage</span></a>\n",
      "        </div>\n",
      "\n",
      "        <div class=\"account\">\n",
      "            <a id=\"account_login\" href=\"https://www.ncbi.nlm.nih.gov/account/?back_url=https%3A%2F%2Fblast%2Encbi%2Enlm%2Enih%2Egov%2FBlast%2Ecgi%3FCMD%3DGET%26FORMAT%5FTYPE%3DXML%26RID%3DRES2M90F013\" class=\"usa-button header-button\">Log in</a>\n",
      "            <button id=\"account_info\" class=\"header-button\" aria-controls=\"account_popup\">\n",
      "                <span class=\"fa fa-user\" aria-hidden=\"true\"></span>\n",
      "                <span class=\"username desktop-only\" aria-hidden=\"true\" id=\"uname_short\"></span>\n",
      "                <span class=\"sr-only\">Show account info</span>\n",
      "            </button>\n",
      "        </div>\n",
      "\n",
      "        <div class=\"ncbi-popup-anchor\">\n",
      "            <div class=\"ncbi-popup account-popup\" id=\"account_popup\" aria-hidden=\"true\" role=\"dialog\" aria-labelledby=\"account-popup-header\">\n",
      "                <div class=\"ncbi-popup-head\">\n",
      "                    <button class=\"ncbi-close-button\"><span class=\"fa fa-window-close\"></span><span class=\"usa-sr-only\">Close</span></button>\n",
      "                    <h4>Account</h4>\n",
      "                </div>\n",
      "                <div class=\"account-user-info\">\n",
      "                    Logged in as:<br>\n",
      "                    <b><span class=\"username\" id=\"uname_long\">username</span></b>\n",
      "                </div>\n",
      "                <div class=\"account-links\">\n",
      "                    <ul class=\"usa-unstyled-list\">\n",
      "                        <li><a id=\"account_myncbi\" href=\"https://www.ncbi.nlm.nih.gov/myncbi/\">Dashboard</a> <span class=\"ncbi-text-small-light\">(My NCBI)</span></li>\n",
      "                        <li><a id=\"account_pubs\" href=\"https://www.ncbi.nlm.nih.gov/myncbi/collections/bibliography/\">Publications</a> <span class=\"ncbi-text-small-light\">(My Bibliography)</span></li>\n",
      "                        <li><a id=\"account_settings\" href=\"https://www.ncbi.nlm.nih.gov/account/settings/\">Account settings</a></li>\n",
      "                        <li><a id=\"account_logout\" href=\"https://www.ncbi.nlm.nih.gov/account/signout/?back_url=https%3A%2F%2Fblast%2Encbi%2Enlm%2Enih%2Egov%2FBlast%2Ecgi%3FCMD%3DGET%26FORMAT%5FTYPE%3DXML%26RID%3DRES2M90F013\">Log out</a></li>\n",
      "                    </ul>\n",
      "                </div>\n",
      "            </div>\n",
      "        </div>\n",
      "\n",
      "    </div>\n",
      "</div>\n",
      "</header>\n",
      "<nav class=\"ncbi-topnav\" id=\"navcontent\">\n",
      "    <div class=\"usa-grid\">\n",
      "        <a class=\"ncbi-topnav-root\" href=\"Blast.cgi\">BLAST <sup>&reg;</sup></a> <span id=\"brc\"><span class=\"brcrmbsign\">&raquo;</span> <a href=\"Blast.cgi?PAGE=Nucleotides&amp;PROGRAM=blastn&amp;PAGE_TYPE=BlastSearch&amp;BLAST_SPEC=\">blastn suite</a> <span class=\"brcrmbsign\">&raquo;</span> RID-RES2M90F013</span>\n",
      "        <ul class=\"rf ncbi-topnav-list\" id=\"topnav-list\">\n",
      "            <li class=\"first \"><a href=\"Blast.cgi?CMD=Web&amp;PAGE_TYPE=BlastHome\" title=\"BLAST Home\">Home</a></li>\n",
      "            <li class=\"recent \"><a href=\"Blast.cgi?CMD=GetSaved&amp;RECENT_RESULTS=on\" title=\"Unexpired BLAST jobs\">Recent Results</a></li>                \n",
      "            <li class=\"saved \"><a href=\"Blast.cgi?CMD=GetSaved\" title=\"Saved sets of BLAST search parameters\">Saved Strategies</a></li>\n",
      "            <li  class= \"last documentation \"> <a href=\"Blast.cgi?CMD=Web&amp;PAGE_TYPE=BlastDocs\" title=\"BLAST documentation\">Help</a></li>                            \n",
      "        </ul>\n",
      "    </div>\n",
      "</nav>\n",
      "\n",
      "\n",
      "\n",
      "        <div id=\"content-wrap\">\n",
      "\n",
      "                <div class=\"pageTitle\">                            \n",
      "                   Format Request Status                   \n",
      "                </div>\n",
      "\t\t<div class=\"inlineDiv resHeader\">\t\t\t\t   \n",
      "\t\t\t\t   <a  id=\"frmPage\"  class=\"WAITING\" href=\"#\" submitForm=\"reformat\">[Formatting options] </a>                   \n",
      "                </div>\n",
      "                <h2 id=\"jtitle\" >Job Title: </h2>\n",
      "\t\t\t\t\t\t\t\t\n",
      "                <div id=\"content\">                \n",
      "                <!--<ul id=\"msg\" class=\"msg\"><li class=\"\"><p class=\"\"></p><p class=\"\"></p><p class=\"\"></p></ul> -->\n",
      "                <ul id=\"msg\" class=\"msg\"><li class=\"\"></li></ul>\n",
      "                <p><!--\n",
      "                QBlastInfoBegin\n",
      "\t                Status=WAITING\n",
      "                QBlastInfoEnd\n",
      "                --></p> \n",
      "                               \n",
      "                <SCRIPT LANGUAGE=\"JavaScript\"><!--\n",
      "                    var tm = \"2000\";\n",
      "                    if (tm != \"\") {                    \n",
      "                        setTimeout('document.forms[0].submit();',tm);\n",
      "                    }\n",
      "                //--></SCRIPT>                                \n",
      "                <table id=\"statInfo\" class=\"WAITING\">\n",
      "                <tr><td>Request ID</td><td> <b>RES2M90F013</b></td></tr>\n",
      "                <tr class=\"odd\"><td>Status</td><td>Searching</td></tr>\n",
      "                <tr><td>Submitted at</td><td>Mon Oct 25 21:56:36 2021</td></tr>\n",
      "                <tr class=\"odd\"><td>Current time</td><td>Mon Oct 25 21:56:39 2021</td></tr>\n",
      "                <tr><td>Time since submission</td><td>00:00:02</td></tr>\n",
      "                </table>\n",
      "                <p class=\"WAITING\">This page will be automatically updated in <b>2</b> seconds</p>       \n",
      "                <form action=\"Blast.cgi\" enctype=\"application/x-www-form-urlencoded\" method=\"POST\" id=\"results\">\n",
      "                <input name=\"FORMAT_TYPE\" type=\"hidden\" value=\"XML\"><input name=\"RID\" type=\"hidden\" value=\"RES2M90F013\"><input name=\"SEARCH_DB_STATUS\" type=\"hidden\" value=\"21\"><input name=\"USER_TYPE\" type=\"hidden\" value=\"2\"><input name=\"_PGR\" type=\"hidden\" value=\"0\">                \n",
      "                <input name=\"_PGR\" type=\"hidden\" value=\"0\" >\n",
      "                <input name=\"CMD\" type=\"hidden\" value=\"Get\">\n",
      "               \n",
      "                </form>\n",
      "                                \n",
      "\t\t\t\t</div><!-- /#content -->\n",
      "\t\t\t\t<form action=\"Blast.cgi\" enctype=\"application/x-www-form-urlencoded\"  method=\"post\" name=\"reformat\" id=\"reformat\">\t\t\t\t\n",
      "\t\t\t\t   <input name=\"QUERY_INFO\" type=\"hidden\" value=\"\" />    \t\t\t\t\n",
      "\t\t\t\t   <input name=\"ENTREZ_QUERY\" type=\"hidden\" value=\"\" />\n",
      "                   <input name=\"CDD_RID\" type=\"hidden\" value=\"\" />\n",
      "                   <input name=\"CDD_SEARCH_STATE\" type=\"hidden\" value=\"\" />\n",
      "                   <input name=\"RID\" type=\"hidden\" value=\"RES2M90F013\" />\n",
      "\t\t\t\t   <input name=\"STEP_NUMBER\" type=\"hidden\" value=\"\" />\n",
      "\t\t\t\t   <input name=\"CMD\" type=\"hidden\" value=\"Web\"/>\t\t\t\t\n",
      "\t\t\t\t   <input NAME=\"PAGE_TYPE\" type=\"hidden\"  value=\"BlastFormatting\"/>\n",
      "\t\t\t\t\n",
      "\t\t\t\t   <!-- TO DO: test all of those changes -->\t\t\t\t   \t\n",
      "\t\t\t\t   <!-- Psi blast params  PSI_BLAST_PARAMS - commented- using forms[0] from fromatter> -->\n",
      "\t\t\t\t   <!-- Current Formatting options FORMATTING_OPTIONS- commented- using forms[0] from fromatter> -->\n",
      "\t\t\t\t   <!-- Current Search options CURR_SAVED_OPTIONS - commented- using forms[0] from fromatter> -->\n",
      "                 </form>\t\t\t\t\n",
      "        </div><!-- /#content-wrap -->\n",
      "\n",
      "         <footer id=\"ncbi-footer\" class=\"usa-footer usa-footer-medium usa-sans ncbi-footer\" role=\"contentinfo\" data-section=\"Footer\">\n",
      "  <div class=\"icon-section\">\n",
      "    <div class=\"usa-grid\">\n",
      "\t\t<div class=\"usa-width-one-whole\">\n",
      "\t\t<h3 class=\"usa-sr-only\">Connect</h3>\n",
      "\t\t<ul class=\"ncbi-inline-list icon-list\">\n",
      "\t\t\t<li><a id=\"footer_twitter\" href=\"https://twitter.com/ncbi\">Twitter</a></li>\n",
      "\t\t\t<li><a id=\"footer_facebook\" href=\"https://www.facebook.com/ncbi.nlm\">Facebook</a></li>\n",
      "\t\t\t<li><a id=\"footer_youtube\" href=\"https://www.youtube.com/user/NCBINLM\">YouTube</a></li>\n",
      "\t\t\t<li><a id=\"footer_linkedin\" href=\"https://www.linkedin.com/company/national-center-for-biotechnology-information-ncbi-\">LinkedIn</a></li>\n",
      "\t\t\t<li><a id=\"footer_github\" href=\"https://github.com/ncbi\">GitHub</a></li>\n",
      "\t\t</ul>\n",
      "\t\t<ul class=\"ncbi-inline-list icon-list icon-list-elastic\">\t\n",
      "\t\t\t<li class=\"li-blog\"><a id=\"footer_blog\" href=\"https://ncbiinsights.ncbi.nlm.nih.gov/\">Blog</a></li>\n",
      "\t\t\t<li class=\"li-support\"><a id=\"footer_support\" href=\"https://support.ncbi.nlm.nih.gov/\">Support Center</a></li>\n",
      "\t\t</ul>\n",
      "\t\t</div>\n",
      "\t\t</div>\n",
      "  </div>\n",
      "  <div class=\"usa-footer-secondary_section ncbi-section\">\n",
      "    <div class=\"usa-grid\">\n",
      "      <div class=\"usa-width-one-half ncbi\">\n",
      "        <h3 class=\"ncbi-full-name\">National Center for Biotechnology Information</h3>\n",
      "\t\t<address>\n",
      "        <a id=\"google_maps\" href=\"https://www.google.com/maps/place/8600+Rockville+Pike,+Bethesda,+MD+20894/@38.9959508,-77.101021,17z/data=!3m1!4b1!4m5!3m4!1s0x89b7c95e25765ddb:0x19156f88b27635b8!8m2!3d38.9959508!4d-77.0988323\">\n",
      "        <span class=\"h-adr\"> <span class=\"p-street-address break\">8600 Rockville Pike</span>\n",
      "        <span class=\"p-locality\">Bethesda</span> <span class=\"p-region\"> MD</span>, <span class=\"p-postal-code\">20894</span> <span class=\"p-country-name\">USA</span> </span>\n",
      "\t\t</a>\n",
      "\t\t</address>\n",
      "        <ul class=\"ncbi-inline-list\">\n",
      "\t\t\t<li><a id=\"footer_about\" href=\"https://www.ncbi.nlm.nih.gov/home/about/\">About us</a></li>\n",
      "          <li><a id=\"footer_contact\" href=\"https://www.ncbi.nlm.nih.gov/home/about/contact.shtml\">Contact us</a></li>\n",
      "          <li><a id=\"footer_policies\" href=\"https://www.ncbi.nlm.nih.gov/home/about/policies.shtml\">Policies</a></li>\n",
      "          <li><a id=\"footer_foia\" href=\"https://www.nih.gov/institutes-nih/nih-office-director/office-communications-public-liaison/freedom-information-act-office\">FOIA</a></li>\n",
      "        </ul>\n",
      "      </div>\n",
      "      <div class=\"usa-width-one-half\">\n",
      "      <div class=\"usa-grid-full\">\n",
      "\t\t<div class=\"usa-width-one-third\">\n",
      "\t\t<h4>Popular</h4>\n",
      "\n",
      "        <ul class=\"ncbi-unstyled-list\">\n",
      "\t\t\t\t<li><a id=\"footer_pubmed\" href=\"https://www.ncbi.nlm.nih.gov/pubmed/\">PubMed</a></li>\n",
      "\t\t\t\t<li><a id=\"footer_pmc\" href=\"https://www.ncbi.nlm.nih.gov/pmc/\">PubMed Central</a></li>\n",
      "\t\t\t\t<li><a id=\"footer_bookshelf\" href=\"https://www.ncbi.nlm.nih.gov/books/\">Bookshelf</a></li>\n",
      "\t\t\t\t<li><a id=\"footer_pubchem\" href=\"https://pubchem.ncbi.nlm.nih.gov\">PubChem</a></li>\n",
      "\t\t\t\t<li><a id=\"footer_gene\" href=\"https://www.ncbi.nlm.nih.gov/gene/\">Gene</a></li>\n",
      "\t\t\t\t<li><a id=\"footer_blast\" href=\"https://blast.ncbi.nlm.nih.gov\">BLAST</a></li>\n",
      "\t\t\t\t<li><a id=\"footer_nucleotide\" href=\"https://www.ncbi.nlm.nih.gov/nucleotide/\">Nucleotide</a></li>\n",
      "\t\t\t\t<li><a id=\"footer_protein\" href=\"https://www.ncbi.nlm.nih.gov/protein/\">Protein</a></li>\n",
      "\t\t\t\t<li><a id=\"footer_geo\" href=\"https://www.ncbi.nlm.nih.gov/geo/\">GEO</a></li>\n",
      "        </ul>\n",
      "\n",
      "\t\t\t</div>\n",
      "\t\t<div class=\"usa-width-one-third\">\n",
      "\t\t\t<h4>Resources</h4>\n",
      "\t\t\t<ul class=\"ncbi-unstyled-list ncbi-undecorated-links\">\n",
      "\t\t\t\t  <li><a id=\"footer_literature\" href=\"https://www.ncbi.nlm.nih.gov/home/literature/\">Literature</a></li>\n",
      "\t\t\t\t  <li><a id=\"footer_health\" href=\"https://www.ncbi.nlm.nih.gov/home/health/\">Health</a></li>\n",
      "\t\t\t\t  <li><a id=\"footer_genomes\" href=\"https://www.ncbi.nlm.nih.gov/home/genomes/\">Genomes</a></li>\n",
      "\t\t\t\t  <li><a id=\"footer_genes\" href=\"https://www.ncbi.nlm.nih.gov/home/genes/\">Genes</a></li>\n",
      "\t\t\t\t  <li><a id=\"footer_proteins\" href=\"https://www.ncbi.nlm.nih.gov/home/proteins/\">Proteins</a></li>\n",
      "\t\t\t\t  <li><a id=\"footer_chemicals\" href=\"https://www.ncbi.nlm.nih.gov/home/chemicals/\">Chemicals</a></li>\n",
      "\t\t\t</ul>\n",
      "\t\t</div>\n",
      "\t\t<div class=\"usa-width-one-third\">\n",
      "   <h4>Actions</h4>\n",
      "   <ul class=\"ncbi-unstyled-list\">\n",
      "              <li><a id=\"footer_submit\" href=\"https://www.ncbi.nlm.nih.gov/home/submit/\">Submit</a></li>\n",
      "              <li><a id=\"footer_download\" href=\"https://www.ncbi.nlm.nih.gov/home/download/\">Download</a></li>\n",
      "              <li><a id=\"footer_learn\" href=\"https://www.ncbi.nlm.nih.gov/home/learn/\">Learn</a></li>\n",
      "              <li><a id=\"footer_develop\" href=\"https://www.ncbi.nlm.nih.gov/home/develop/\">Develop</a></li>\n",
      "              <li><a id=\"footer_analyze\" href=\"https://www.ncbi.nlm.nih.gov/home/analyze/\">Analyze</a></li>\n",
      "              <li><a id=\"footer_research\" href=\"https://www.ncbi.nlm.nih.gov/research/\">Research</a></li>\n",
      "   </ul>\n",
      "\t\t  </div>\n",
      "\t\t  </div>\n",
      "\t\t  </div>\n",
      "    </div>\n",
      "\t</div>\n",
      "     <div class=\"usa-footer-secondary_section gov-agencies\"> \n",
      "    <div class=\"usa-grid\">\n",
      "    <div class=\"usa-width-one-whole\">\n",
      "\t\t<p>\n",
      "\t\t\t<a id=\"footer_nlm\" href=\"https://www.nlm.nih.gov/\">NLM</a>\n",
      "\t\t\t&ensp;|&ensp;\n",
      "\t\t\t<a id=\"footer_nih\" href=\"https://www.nih.gov/\">NIH</a>\n",
      "\t\t\t&ensp;|&ensp;\n",
      "\t\t\t<a id=\"footer_hhs\" href=\"https://www.hhs.gov/\">HHS</a>\n",
      "\t\t\t&ensp;|&ensp;\n",
      "\t\t\t<a id=\"footer_usagov\" href=\"https://www.usa.gov/\">USA.gov</a>\n",
      "\t\t</p>\n",
      "\t </div>\n",
      "\t </div>\n",
      "\t</div>\n",
      "\n",
      " </footer>\n",
      "<script type=\"text/javascript\" src=\"js/nwds.js\"></script>\n",
      "<script type=\"text/javascript\" src=\"js/ncbipopup.js\"></script>\n",
      "<script type=\"text/javascript\" src=\"js/headerNew.js\"></script>\n",
      "<script src=\"js/uswds.min.js\"></script>\n",
      "        <script type=\"text/javascript\" src=\"/portal/portal3rc.fcgi/rlib/js/InstrumentOmnitureBaseJS/InstrumentNCBIBaseJS/InstrumentPageStarterJS.js\"></script>\n",
      "<!--\n",
      "<script type=\"text/javascript\" src=\"://www.ncbi.nlm.nih.gov/portal/portal3rc.fcgi/supportedbrowsers/js/nonportalbc_min.js\"></script>\n",
      "<script type=\"text/javascript\">$(\"#browsers_ajax\").browserCheck();</script>\n",
      "-->\n",
      "   </div><!--/#wrap-->\n",
      "</body>\n",
      "\n",
      "</html>\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "print(result)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {
    "slideshow": {
     "slide_type": "slide"
    }
   },
   "outputs": [],
   "source": [
    "result = requests.get('http://www.ncbi.nlm.nih.gov/blast/Blast.cgi?CMD=GET&RID=%s&FORMAT_TYPE=XML' % rid).text"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<?xml version=\"1.0\"?>\n",
      "<!DOCTYPE BlastOutput PUBLIC \"-//NCBI//NCBI BlastOutput/EN\" \"http://www.ncbi.nlm.nih.gov/dtd/NCBI_BlastOutput.dtd\">\n",
      "<BlastOutput>\n",
      "  <BlastOutput_program>blastp</BlastOutput_program>\n",
      "  <BlastOutput_version>BLASTP 2.12.0+</BlastOutput_version>\n",
      "  <BlastOutput_reference>Stephen F. Altschul, Thomas L. Madden, Alejandro A. Sch&amp;auml;ffer, Jinghui Zhang, Zheng Zhang, Webb Miller, and David J. Lipman (1997), &quot;Gapped BLAST and PSI-BLAST: a new generation of protein database search programs&quot;, Nucleic Acids Res. 25:3389-3402.</BlastOutput_reference>\n",
      "  <BlastOutput_db>nr</BlastOutput_db>\n",
      "  <BlastOutput_query-ID>Query_14089</BlastOutput_query-ID>\n",
      "  <BlastOutput_query-def>unnamed protein product</BlastOutput_query-def>\n",
      "  <BlastOutput_query-len>15</BlastOutput_query-len>\n",
      "  <BlastOutput_param>\n",
      "    <Parameters>\n",
      "      <Parameters_matrix>BLOSUM62</Parameters_matrix>\n",
      "      <Parameters_expect>10</Parameters_expect>\n",
      "      <Parameters_gap-open>11</Parameters_gap-open>\n",
      "      <Parameters_gap-extend>1</Parameters_gap-extend>\n",
      "      <Parameters_filter>F</Parameters_filter>\n",
      "    </Parameters>\n",
      "  </BlastOutput_param>\n",
      "<BlastOutput_iterations>\n",
      "<Iteration>\n",
      "  <Iteration_iter-num>1</Iteration_iter-num>\n",
      "  <Iteration_query-ID>Query_14089</Iteration_query-ID>\n",
      "  <Iteration_query-def>unnamed protein product</Iteration_query-def>\n",
      "  <Iteration_query-len>15</Iteration_query-len>\n",
      "<Iteration_hits>\n",
      "<Hit>\n",
      "  <Hit_num>1</Hit_num>\n",
      "  <Hit_id>dbj|BBA16878.1|</Hit_id>\n",
      "  <Hit_def>PB1-p53-T2A-Mmj-MDM2 [synthetic construct]</Hit_def>\n",
      "  <Hit_accession>BBA16878</Hit_accession>\n",
      "  <Hit_len>617</Hit_len>\n",
      "  <Hit_hsps>\n",
      "    <Hsp>\n",
      "      <Hsp_num>1</Hsp_num>\n",
      "      <Hsp_bit-score>36.965</Hsp_bit-score>\n",
      "      <Hsp_score>84</Hsp_score>\n",
      "      <Hsp_evalue>0.242393</Hsp_evalue>\n",
      "      <Hsp_query-from>1</Hsp_query-from>\n",
      "      <Hsp_query-to>15</Hsp_query-to>\n",
      "      <Hsp_hit-from>155</Hsp_hit-from>\n",
      "      <Hsp_hit-to>169</Hsp_hit-to>\n",
      "      <Hsp_query-frame>0</Hsp_query-frame>\n",
      "      <Hsp_hit-frame>0</Hsp_hit-frame>\n",
      "      <Hsp_identity>15</Hsp_identity>\n",
      "      <Hsp_positive>15</Hsp_positive>\n",
      "      <Hsp_gaps>0</Hsp_gaps>\n",
      "      <Hsp_align-len>15</Hsp_align-len>\n",
      "      <Hsp_qseq>SQETFSDLWKLLPEN</Hsp_qseq>\n",
      "      <Hsp_hseq>SQETFSDLWKLLPEN</Hsp_hseq>\n",
      "      <Hsp_midline>SQETFSDLWKLLPEN</Hsp_midline>\n",
      "    </Hsp>\n",
      "  </Hit_hsps>\n",
      "</Hit>\n",
      "<Hit>\n",
      "  <Hit_num>2</Hit_num>\n",
      "  <Hit_id>gb|AAT40418.1|</Hit_id>\n",
      "  <Hit_def>p53 tumor suppressor, partial [Homo sapiens] &gt;pdb|2K8F|B Structural Basis for the Regulation of p53 Function by p300 [Homo sapiens]</Hit_def>\n",
      "  <Hit_accession>AAT40418</Hit_accession>\n",
      "  <Hit_len>39</Hit_len>\n",
      "  <Hit_hsps>\n",
      "    <Hsp>\n",
      "      <Hsp_num>1</Hsp_num>\n",
      "      <Hsp_bit-score>34.2686</Hsp_bit-score>\n",
      "      <Hsp_score>77</Hsp_score>\n",
      "      <Hsp_evalue>0.511594</Hsp_evalue>\n",
      "      <Hsp_query-from>1</Hsp_query-from>\n",
      "      <Hsp_query-to>15</Hsp_query-to>\n",
      "      <Hsp_hit-from>15</Hsp_hit-from>\n",
      "      <Hsp_hit-to>29</Hsp_hit-to>\n",
      "      <Hsp_query-frame>0</Hsp_query-frame>\n",
      "      <Hsp_hit-frame>0</Hsp_hit-frame>\n",
      "      <Hsp_identity>15</Hsp_identity>\n",
      "      <Hsp_positive>15</Hsp_positive>\n",
      "      <Hsp_gaps>0</Hsp_gaps>\n",
      "      <Hsp_align-len>15</Hsp_align-len>\n",
      "      <Hsp_qseq>SQETFSDLWKLLPEN</Hsp_qseq>\n",
      "      <Hsp_hseq>SQETFSDLWKLLPEN</Hsp_hseq>\n",
      "      <Hsp_midline>SQETFSDLWKLLPEN</Hsp_midline>\n",
      "    </Hsp>\n",
      "  </Hit_hsps>\n",
      "</Hit>\n",
      "<Hit>\n",
      "  <Hit_num>3</Hit_num>\n",
      "  <Hit_id>gb|AAF63442.1|</Hit_id>\n",
      "  <Hit_def>p53 tumor suppressor protein, partial [Homo sapiens]</Hit_def>\n",
      "  <Hit_accession>AAF63442</Hit_accession>\n",
      "  <Hit_len>32</Hit_len>\n",
      "  <Hit_hsps>\n",
      "    <Hsp>\n",
      "      <Hsp_num>1</Hsp_num>\n",
      "      <Hsp_bit-score>33.8834</Hsp_bit-score>\n",
      "      <Hsp_score>76</Hsp_score>\n",
      "      <Hsp_evalue>0.613213</Hsp_evalue>\n",
      "      <Hsp_query-from>1</Hsp_query-from>\n",
      "      <Hsp_query-to>15</Hsp_query-to>\n",
      "      <Hsp_hit-from>15</Hsp_hit-from>\n",
      "      <Hsp_hit-to>29</Hsp_hit-to>\n",
      "      <Hsp_query-frame>0</Hsp_query-frame>\n",
      "      <Hsp_hit-frame>0</Hsp_hit-frame>\n",
      "      <Hsp_identity>15</Hsp_identity>\n",
      "      <Hsp_positive>15</Hsp_positive>\n",
      "      <Hsp_gaps>0</Hsp_gaps>\n",
      "      <Hsp_align-len>15</Hsp_align-len>\n",
      "      <Hsp_qseq>SQETFSDLWKLLPEN</Hsp_qseq>\n",
      "      <Hsp_hseq>SQETFSDLWKLLPEN</Hsp_hseq>\n",
      "      <Hsp_midline>SQETFSDLWKLLPEN</Hsp_midline>\n",
      "    </Hsp>\n",
      "  </Hit_hsps>\n",
      "</Hit>\n",
      "<Hit>\n",
      "  <Hit_num>4</Hit_num>\n",
      "  <Hit_id>pdb|1YCR|B</Hit_id>\n",
      "  <Hit_def>MDM2 BOUND TO THE TRANSACTIVATION DOMAIN OF P53 [unidentified] &gt;pdb|2MWY|B Mdmx-p53 [Homo sapiens] &gt;pdb|2Z5S|P Molecular basis for the inhibition of p53 by Mdmx [synthetic construct] &gt;pdb|2Z5S|Q Molecular basis for the inhibition of p53 by Mdmx [synthetic construct] &gt;pdb|2Z5S|R Molecular basis for the inhibition of p53 by Mdmx [synthetic construct] &gt;pdb|2Z5T|P Molecular basis for the inhibition of p53 by Mdmx [synthetic construct] &gt;pdb|2Z5T|Q Molecular basis for the inhibition of p53 by Mdmx [synthetic construct] &gt;pdb|2Z5T|R Molecular basis for the inhibition of p53 by Mdmx [synthetic construct] &gt;pdb|3DAB|B Structure of the human Mdmx protein bound to the p53 tumor suppressor transactivation domain [synthetic construct] &gt;pdb|3DAB|D Structure of the human Mdmx protein bound to the p53 tumor suppressor transactivation domain [synthetic construct] &gt;pdb|3DAB|F Structure of the human Mdmx protein bound to the p53 tumor suppressor transactivation domain [synthetic construct] &gt;pdb|3DAB|H Structure of the human Mdmx protein bound to the p53 tumor suppressor transactivation domain [synthetic construct] &gt;pdb|4HFZ|B Crystal Structure of an MDM2/P53 Peptide Complex [Homo sapiens] &gt;pdb|4HFZ|D Crystal Structure of an MDM2/P53 Peptide Complex [Homo sapiens]</Hit_def>\n",
      "  <Hit_accession>1YCR_B</Hit_accession>\n",
      "  <Hit_len>15</Hit_len>\n",
      "  <Hit_hsps>\n",
      "    <Hsp>\n",
      "      <Hsp_num>1</Hsp_num>\n",
      "      <Hsp_bit-score>33.4982</Hsp_bit-score>\n",
      "      <Hsp_score>75</Hsp_score>\n",
      "      <Hsp_evalue>0.662402</Hsp_evalue>\n",
      "      <Hsp_query-from>1</Hsp_query-from>\n",
      "      <Hsp_query-to>15</Hsp_query-to>\n",
      "      <Hsp_hit-from>1</Hsp_hit-from>\n",
      "      <Hsp_hit-to>15</Hsp_hit-to>\n",
      "      <Hsp_query-frame>0</Hsp_query-frame>\n",
      "      <Hsp_hit-frame>0</Hsp_hit-frame>\n",
      "      <Hsp_identity>15</Hsp_identity>\n",
      "      <Hsp_positive>15</Hsp_positive>\n",
      "      <Hsp_gaps>0</Hsp_gaps>\n",
      "      <Hsp_align-len>15</Hsp_align-len>\n",
      "      <Hsp_qseq>SQETFSDLWKLLPEN</Hsp_qseq>\n",
      "      <Hsp_hseq>SQETFSDLWKLLPEN</Hsp_hseq>\n",
      "      <Hsp_midline>SQETFSDLWKLLPEN</Hsp_midline>\n",
      "    </Hsp>\n",
      "  </Hit_hsps>\n",
      "</Hit>\n",
      "<Hit>\n",
      "  <Hit_num>5</Hit_num>\n",
      "  <Hit_id>pdb|2LY4|B</Hit_id>\n",
      "  <Hit_def>HMGB1-facilitated p53 DNA binding occurs via HMG-box/p53 transactivation domain interaction and is regulated by the acidic tail [Homo sapiens]</Hit_def>\n",
      "  <Hit_accession>2LY4_B</Hit_accession>\n",
      "  <Hit_len>93</Hit_len>\n",
      "  <Hit_hsps>\n",
      "    <Hsp>\n",
      "      <Hsp_num>1</Hsp_num>\n",
      "      <Hsp_bit-score>35.039</Hsp_bit-score>\n",
      "      <Hsp_score>79</Hsp_score>\n",
      "      <Hsp_evalue>0.670036</Hsp_evalue>\n",
      "      <Hsp_query-from>1</Hsp_query-from>\n",
      "      <Hsp_query-to>15</Hsp_query-to>\n",
      "      <Hsp_hit-from>15</Hsp_hit-from>\n",
      "      <Hsp_hit-to>29</Hsp_hit-to>\n",
      "      <Hsp_query-frame>0</Hsp_query-frame>\n",
      "      <Hsp_hit-frame>0</Hsp_hit-frame>\n",
      "      <Hsp_identity>15</Hsp_identity>\n",
      "      <Hsp_positive>15</Hsp_positive>\n",
      "      <Hsp_gaps>0</Hsp_gaps>\n",
      "      <Hsp_align-len>15</Hsp_align-len>\n",
      "      <Hsp_qseq>SQETFSDLWKLLPEN</Hsp_qseq>\n",
      "      <Hsp_hseq>SQETFSDLWKLLPEN</Hsp_hseq>\n",
      "      <Hsp_midline>SQETFSDLWKLLPEN</Hsp_midline>\n",
      "    </Hsp>\n",
      "  </Hit_hsps>\n",
      "</Hit>\n",
      "<Hit>\n",
      "  <Hit_num>6</Hit_num>\n",
      "  <Hit_id>gb|AXC07658.1|</Hit_id>\n",
      "  <Hit_def>human P53 transactivation domain protein [Cloning vector pDSG288] &gt;gb|AXC07704.1| human P53 transactivation domain protein [Cloning vector pDSG319] &gt;gb|AXC07918.1| A8_P53TA [Cloning vector pDSG360]</Hit_def>\n",
      "  <Hit_accession>AXC07658</Hit_accession>\n",
      "  <Hit_len>102</Hit_len>\n",
      "  <Hit_hsps>\n",
      "    <Hsp>\n",
      "      <Hsp_num>1</Hsp_num>\n",
      "      <Hsp_bit-score>35.039</Hsp_bit-score>\n",
      "      <Hsp_score>79</Hsp_score>\n",
      "      <Hsp_evalue>0.693737</Hsp_evalue>\n",
      "      <Hsp_query-from>1</Hsp_query-from>\n",
      "      <Hsp_query-to>15</Hsp_query-to>\n",
      "      <Hsp_hit-from>15</Hsp_hit-from>\n",
      "      <Hsp_hit-to>29</Hsp_hit-to>\n",
      "      <Hsp_query-frame>0</Hsp_query-frame>\n",
      "      <Hsp_hit-frame>0</Hsp_hit-frame>\n",
      "      <Hsp_identity>15</Hsp_identity>\n",
      "      <Hsp_positive>15</Hsp_positive>\n",
      "      <Hsp_gaps>0</Hsp_gaps>\n",
      "      <Hsp_align-len>15</Hsp_align-len>\n",
      "      <Hsp_qseq>SQETFSDLWKLLPEN</Hsp_qseq>\n",
      "      <Hsp_hseq>SQETFSDLWKLLPEN</Hsp_hseq>\n",
      "      <Hsp_midline>SQETFSDLWKLLPEN</Hsp_midline>\n",
      "    </Hsp>\n",
      "  </Hit_hsps>\n",
      "</Hit>\n",
      "<Hit>\n",
      "  <Hit_num>7</Hit_num>\n",
      "  <Hit_id>pdb|1YCQ|B</Hit_id>\n",
      "  <Hit_def>XENOPUS LAEVIS MDM2 BOUND TO THE TRANSACTIVATION DOMAIN OF HUMAN P53 [unidentified]</Hit_def>\n",
      "  <Hit_accession>1YCQ_B</Hit_accession>\n",
      "  <Hit_len>17</Hit_len>\n",
      "  <Hit_hsps>\n",
      "    <Hsp>\n",
      "      <Hsp_num>1</Hsp_num>\n",
      "      <Hsp_bit-score>33.4982</Hsp_bit-score>\n",
      "      <Hsp_score>75</Hsp_score>\n",
      "      <Hsp_evalue>0.715274</Hsp_evalue>\n",
      "      <Hsp_query-from>1</Hsp_query-from>\n",
      "      <Hsp_query-to>15</Hsp_query-to>\n",
      "      <Hsp_hit-from>3</Hsp_hit-from>\n",
      "      <Hsp_hit-to>17</Hsp_hit-to>\n",
      "      <Hsp_query-frame>0</Hsp_query-frame>\n",
      "      <Hsp_hit-frame>0</Hsp_hit-frame>\n",
      "      <Hsp_identity>15</Hsp_identity>\n",
      "      <Hsp_positive>15</Hsp_positive>\n",
      "      <Hsp_gaps>0</Hsp_gaps>\n",
      "      <Hsp_align-len>15</Hsp_align-len>\n",
      "      <Hsp_qseq>SQETFSDLWKLLPEN</Hsp_qseq>\n",
      "      <Hsp_hseq>SQETFSDLWKLLPEN</Hsp_hseq>\n",
      "      <Hsp_midline>SQETFSDLWKLLPEN</Hsp_midline>\n",
      "    </Hsp>\n",
      "  </Hit_hsps>\n",
      "</Hit>\n",
      "<Hit>\n",
      "  <Hit_num>8</Hit_num>\n",
      "  <Hit_id>dbj|BBA16895.1|</Hit_id>\n",
      "  <Hit_def>AG-p53 [synthetic construct]</Hit_def>\n",
      "  <Hit_accession>BBA16895</Hit_accession>\n",
      "  <Hit_len>334</Hit_len>\n",
      "  <Hit_hsps>\n",
      "    <Hsp>\n",
      "      <Hsp_num>1</Hsp_num>\n",
      "      <Hsp_bit-score>35.4242</Hsp_bit-score>\n",
      "      <Hsp_score>80</Hsp_score>\n",
      "      <Hsp_evalue>0.837231</Hsp_evalue>\n",
      "      <Hsp_query-from>1</Hsp_query-from>\n",
      "      <Hsp_query-to>15</Hsp_query-to>\n",
      "      <Hsp_hit-from>279</Hsp_hit-from>\n",
      "      <Hsp_hit-to>293</Hsp_hit-to>\n",
      "      <Hsp_query-frame>0</Hsp_query-frame>\n",
      "      <Hsp_hit-frame>0</Hsp_hit-frame>\n",
      "      <Hsp_identity>15</Hsp_identity>\n",
      "      <Hsp_positive>15</Hsp_positive>\n",
      "      <Hsp_gaps>0</Hsp_gaps>\n",
      "      <Hsp_align-len>15</Hsp_align-len>\n",
      "      <Hsp_qseq>SQETFSDLWKLLPEN</Hsp_qseq>\n",
      "      <Hsp_hseq>SQETFSDLWKLLPEN</Hsp_hseq>\n",
      "      <Hsp_midline>SQETFSDLWKLLPEN</Hsp_midline>\n",
      "    </Hsp>\n",
      "  </Hit_hsps>\n",
      "</Hit>\n",
      "<Hit>\n",
      "  <Hit_num>9</Hit_num>\n",
      "  <Hit_id>gb|ABD62752.1|</Hit_id>\n",
      "  <Hit_def>p53deltaEx4, partial [Homo sapiens]</Hit_def>\n",
      "  <Hit_accession>ABD62752</Hit_accession>\n",
      "  <Hit_len>112</Hit_len>\n",
      "  <Hit_hsps>\n",
      "    <Hsp>\n",
      "      <Hsp_num>1</Hsp_num>\n",
      "      <Hsp_bit-score>34.6538</Hsp_bit-score>\n",
      "      <Hsp_score>78</Hsp_score>\n",
      "      <Hsp_evalue>0.953937</Hsp_evalue>\n",
      "      <Hsp_query-from>1</Hsp_query-from>\n",
      "      <Hsp_query-to>15</Hsp_query-to>\n",
      "      <Hsp_hit-from>20</Hsp_hit-from>\n",
      "      <Hsp_hit-to>34</Hsp_hit-to>\n",
      "      <Hsp_query-frame>0</Hsp_query-frame>\n",
      "      <Hsp_hit-frame>0</Hsp_hit-frame>\n",
      "      <Hsp_identity>15</Hsp_identity>\n",
      "      <Hsp_positive>15</Hsp_positive>\n",
      "      <Hsp_gaps>0</Hsp_gaps>\n",
      "      <Hsp_align-len>15</Hsp_align-len>\n",
      "      <Hsp_qseq>SQETFSDLWKLLPEN</Hsp_qseq>\n",
      "      <Hsp_hseq>SQETFSDLWKLLPEN</Hsp_hseq>\n",
      "      <Hsp_midline>SQETFSDLWKLLPEN</Hsp_midline>\n",
      "    </Hsp>\n",
      "  </Hit_hsps>\n",
      "</Hit>\n",
      "<Hit>\n",
      "  <Hit_num>10</Hit_num>\n",
      "  <Hit_id>gb|AXC07656.1|</Hit_id>\n",
      "  <Hit_def>mutated Neae [Cloning vector pDSG288] &gt;gb|AXC07703.1| mutated Neae [Cloning vector pDSG319] &gt;gb|AXC07920.1| mutated Neae [Cloning vector pDSG360]</Hit_def>\n",
      "  <Hit_accession>AXC07656</Hit_accession>\n",
      "  <Hit_len>760</Hit_len>\n",
      "  <Hit_hsps>\n",
      "    <Hsp>\n",
      "      <Hsp_num>1</Hsp_num>\n",
      "      <Hsp_bit-score>35.4242</Hsp_bit-score>\n",
      "      <Hsp_score>80</Hsp_score>\n",
      "      <Hsp_evalue>0.956563</Hsp_evalue>\n",
      "      <Hsp_query-from>1</Hsp_query-from>\n",
      "      <Hsp_query-to>15</Hsp_query-to>\n",
      "      <Hsp_hit-from>673</Hsp_hit-from>\n",
      "      <Hsp_hit-to>687</Hsp_hit-to>\n",
      "      <Hsp_query-frame>0</Hsp_query-frame>\n",
      "      <Hsp_hit-frame>0</Hsp_hit-frame>\n",
      "      <Hsp_identity>15</Hsp_identity>\n",
      "      <Hsp_positive>15</Hsp_positive>\n",
      "      <Hsp_gaps>0</Hsp_gaps>\n",
      "      <Hsp_align-len>15</Hsp_align-len>\n",
      "      <Hsp_qseq>SQETFSDLWKLLPEN</Hsp_qseq>\n",
      "      <Hsp_hseq>SQETFSDLWKLLPEN</Hsp_hseq>\n",
      "      <Hsp_midline>SQETFSDLWKLLPEN</Hsp_midline>\n",
      "    </Hsp>\n",
      "  </Hit_hsps>\n",
      "</Hit>\n",
      "<Hit>\n",
      "  <Hit_num>11</Hit_num>\n",
      "  <Hit_id>gb|PNI36855.1|</Hit_id>\n",
      "  <Hit_def>TP53 isoform 13, partial [Pan troglodytes]</Hit_def>\n",
      "  <Hit_accession>PNI36855</Hit_accession>\n",
      "  <Hit_len>128</Hit_len>\n",
      "  <Hit_hsps>\n",
      "    <Hsp>\n",
      "      <Hsp_num>1</Hsp_num>\n",
      "      <Hsp_bit-score>35.039</Hsp_bit-score>\n",
      "      <Hsp_score>79</Hsp_score>\n",
      "      <Hsp_evalue>1.05819</Hsp_evalue>\n",
      "      <Hsp_query-from>1</Hsp_query-from>\n",
      "      <Hsp_query-to>15</Hsp_query-to>\n",
      "      <Hsp_hit-from>15</Hsp_hit-from>\n",
      "      <Hsp_hit-to>29</Hsp_hit-to>\n",
      "      <Hsp_query-frame>0</Hsp_query-frame>\n",
      "      <Hsp_hit-frame>0</Hsp_hit-frame>\n",
      "      <Hsp_identity>15</Hsp_identity>\n",
      "      <Hsp_positive>15</Hsp_positive>\n",
      "      <Hsp_gaps>0</Hsp_gaps>\n",
      "      <Hsp_align-len>15</Hsp_align-len>\n",
      "      <Hsp_qseq>SQETFSDLWKLLPEN</Hsp_qseq>\n",
      "      <Hsp_hseq>SQETFSDLWKLLPEN</Hsp_hseq>\n",
      "      <Hsp_midline>SQETFSDLWKLLPEN</Hsp_midline>\n",
      "    </Hsp>\n",
      "  </Hit_hsps>\n",
      "</Hit>\n",
      "<Hit>\n",
      "  <Hit_num>12</Hit_num>\n",
      "  <Hit_id>pdb|2L14|B</Hit_id>\n",
      "  <Hit_def>Structure of CBP nuclear coactivator binding domain in complex with p53 TAD [Homo sapiens]</Hit_def>\n",
      "  <Hit_accession>2L14_B</Hit_accession>\n",
      "  <Hit_len>49</Hit_len>\n",
      "  <Hit_hsps>\n",
      "    <Hsp>\n",
      "      <Hsp_num>1</Hsp_num>\n",
      "      <Hsp_bit-score>33.4982</Hsp_bit-score>\n",
      "      <Hsp_score>75</Hsp_score>\n",
      "      <Hsp_evalue>1.10104</Hsp_evalue>\n",
      "      <Hsp_query-from>1</Hsp_query-from>\n",
      "      <Hsp_query-to>15</Hsp_query-to>\n",
      "      <Hsp_hit-from>3</Hsp_hit-from>\n",
      "      <Hsp_hit-to>17</Hsp_hit-to>\n",
      "      <Hsp_query-frame>0</Hsp_query-frame>\n",
      "      <Hsp_hit-frame>0</Hsp_hit-frame>\n",
      "      <Hsp_identity>15</Hsp_identity>\n",
      "      <Hsp_positive>15</Hsp_positive>\n",
      "      <Hsp_gaps>0</Hsp_gaps>\n",
      "      <Hsp_align-len>15</Hsp_align-len>\n",
      "      <Hsp_qseq>SQETFSDLWKLLPEN</Hsp_qseq>\n",
      "      <Hsp_hseq>SQETFSDLWKLLPEN</Hsp_hseq>\n",
      "      <Hsp_midline>SQETFSDLWKLLPEN</Hsp_midline>\n",
      "    </Hsp>\n",
      "  </Hit_hsps>\n",
      "</Hit>\n",
      "<Hit>\n",
      "  <Hit_num>13</Hit_num>\n",
      "  <Hit_id>ref|XP_036996610.1|</Hit_id>\n",
      "  <Hit_def>cellular tumor antigen p53, partial [Artibeus jamaicensis]</Hit_def>\n",
      "  <Hit_accession>XP_036996610</Hit_accession>\n",
      "  <Hit_len>125</Hit_len>\n",
      "  <Hit_hsps>\n",
      "    <Hsp>\n",
      "      <Hsp_num>1</Hsp_num>\n",
      "      <Hsp_bit-score>34.6538</Hsp_bit-score>\n",
      "      <Hsp_score>78</Hsp_score>\n",
      "      <Hsp_evalue>1.1186</Hsp_evalue>\n",
      "      <Hsp_query-from>1</Hsp_query-from>\n",
      "      <Hsp_query-to>15</Hsp_query-to>\n",
      "      <Hsp_hit-from>15</Hsp_hit-from>\n",
      "      <Hsp_hit-to>29</Hsp_hit-to>\n",
      "      <Hsp_query-frame>0</Hsp_query-frame>\n",
      "      <Hsp_hit-frame>0</Hsp_hit-frame>\n",
      "      <Hsp_identity>15</Hsp_identity>\n",
      "      <Hsp_positive>15</Hsp_positive>\n",
      "      <Hsp_gaps>0</Hsp_gaps>\n",
      "      <Hsp_align-len>15</Hsp_align-len>\n",
      "      <Hsp_qseq>SQETFSDLWKLLPEN</Hsp_qseq>\n",
      "      <Hsp_hseq>SQETFSDLWKLLPEN</Hsp_hseq>\n",
      "      <Hsp_midline>SQETFSDLWKLLPEN</Hsp_midline>\n",
      "    </Hsp>\n",
      "  </Hit_hsps>\n",
      "</Hit>\n",
      "<Hit>\n",
      "  <Hit_num>14</Hit_num>\n",
      "  <Hit_id>pdb|5HPD|A</Hit_id>\n",
      "  <Hit_def>Solution Structure of TAZ2-p53TAD [Mus musculus]</Hit_def>\n",
      "  <Hit_accession>5HPD_A</Hit_accession>\n",
      "  <Hit_len>158</Hit_len>\n",
      "  <Hit_hsps>\n",
      "    <Hsp>\n",
      "      <Hsp_num>1</Hsp_num>\n",
      "      <Hsp_bit-score>35.039</Hsp_bit-score>\n",
      "      <Hsp_score>79</Hsp_score>\n",
      "      <Hsp_evalue>1.13439</Hsp_evalue>\n",
      "      <Hsp_query-from>1</Hsp_query-from>\n",
      "      <Hsp_query-to>15</Hsp_query-to>\n",
      "      <Hsp_hit-from>112</Hsp_hit-from>\n",
      "      <Hsp_hit-to>126</Hsp_hit-to>\n",
      "      <Hsp_query-frame>0</Hsp_query-frame>\n",
      "      <Hsp_hit-frame>0</Hsp_hit-frame>\n",
      "      <Hsp_identity>15</Hsp_identity>\n",
      "      <Hsp_positive>15</Hsp_positive>\n",
      "      <Hsp_gaps>0</Hsp_gaps>\n",
      "      <Hsp_align-len>15</Hsp_align-len>\n",
      "      <Hsp_qseq>SQETFSDLWKLLPEN</Hsp_qseq>\n",
      "      <Hsp_hseq>SQETFSDLWKLLPEN</Hsp_hseq>\n",
      "      <Hsp_midline>SQETFSDLWKLLPEN</Hsp_midline>\n",
      "    </Hsp>\n",
      "  </Hit_hsps>\n",
      "</Hit>\n",
      "<Hit>\n",
      "  <Hit_num>15</Hit_num>\n",
      "  <Hit_id>gb|PNJ12864.1|</Hit_id>\n",
      "  <Hit_def>TP53 isoform 12, partial [Pongo abelii]</Hit_def>\n",
      "  <Hit_accession>PNJ12864</Hit_accession>\n",
      "  <Hit_len>128</Hit_len>\n",
      "  <Hit_hsps>\n",
      "    <Hsp>\n",
      "      <Hsp_num>1</Hsp_num>\n",
      "      <Hsp_bit-score>34.6538</Hsp_bit-score>\n",
      "      <Hsp_score>78</Hsp_score>\n",
      "      <Hsp_evalue>1.16203</Hsp_evalue>\n",
      "      <Hsp_query-from>1</Hsp_query-from>\n",
      "      <Hsp_query-to>15</Hsp_query-to>\n",
      "      <Hsp_hit-from>15</Hsp_hit-from>\n",
      "      <Hsp_hit-to>29</Hsp_hit-to>\n",
      "      <Hsp_query-frame>0</Hsp_query-frame>\n",
      "      <Hsp_hit-frame>0</Hsp_hit-frame>\n",
      "      <Hsp_identity>15</Hsp_identity>\n",
      "      <Hsp_positive>15</Hsp_positive>\n",
      "      <Hsp_gaps>0</Hsp_gaps>\n",
      "      <Hsp_align-len>15</Hsp_align-len>\n",
      "      <Hsp_qseq>SQETFSDLWKLLPEN</Hsp_qseq>\n",
      "      <Hsp_hseq>SQETFSDLWKLLPEN</Hsp_hseq>\n",
      "      <Hsp_midline>SQETFSDLWKLLPEN</Hsp_midline>\n",
      "    </Hsp>\n",
      "  </Hit_hsps>\n",
      "</Hit>\n",
      "<Hit>\n",
      "  <Hit_num>16</Hit_num>\n",
      "  <Hit_id>ref|XP_008268882.1|</Hit_id>\n",
      "  <Hit_def>PREDICTED: cellular tumor antigen p53 isoform X1 [Oryctolagus cuniculus]</Hit_def>\n",
      "  <Hit_accession>XP_008268882</Hit_accession>\n",
      "  <Hit_len>412</Hit_len>\n",
      "  <Hit_hsps>\n",
      "    <Hsp>\n",
      "      <Hsp_num>1</Hsp_num>\n",
      "      <Hsp_bit-score>35.039</Hsp_bit-score>\n",
      "      <Hsp_score>79</Hsp_score>\n",
      "      <Hsp_evalue>1.1659</Hsp_evalue>\n",
      "      <Hsp_query-from>1</Hsp_query-from>\n",
      "      <Hsp_query-to>15</Hsp_query-to>\n",
      "      <Hsp_hit-from>36</Hsp_hit-from>\n",
      "      <Hsp_hit-to>50</Hsp_hit-to>\n",
      "      <Hsp_query-frame>0</Hsp_query-frame>\n",
      "      <Hsp_hit-frame>0</Hsp_hit-frame>\n",
      "      <Hsp_identity>15</Hsp_identity>\n",
      "      <Hsp_positive>15</Hsp_positive>\n",
      "      <Hsp_gaps>0</Hsp_gaps>\n",
      "      <Hsp_align-len>15</Hsp_align-len>\n",
      "      <Hsp_qseq>SQETFSDLWKLLPEN</Hsp_qseq>\n",
      "      <Hsp_hseq>SQETFSDLWKLLPEN</Hsp_hseq>\n",
      "      <Hsp_midline>SQETFSDLWKLLPEN</Hsp_midline>\n",
      "    </Hsp>\n",
      "  </Hit_hsps>\n",
      "</Hit>\n",
      "<Hit>\n",
      "  <Hit_num>17</Hit_num>\n",
      "  <Hit_id>ref|XP_017706033.1|</Hit_id>\n",
      "  <Hit_def>PREDICTED: cellular tumor antigen p53 isoform X1 [Rhinopithecus bieti]</Hit_def>\n",
      "  <Hit_accession>XP_017706033</Hit_accession>\n",
      "  <Hit_len>412</Hit_len>\n",
      "  <Hit_hsps>\n",
      "    <Hsp>\n",
      "      <Hsp_num>1</Hsp_num>\n",
      "      <Hsp_bit-score>35.039</Hsp_bit-score>\n",
      "      <Hsp_score>79</Hsp_score>\n",
      "      <Hsp_evalue>1.18892</Hsp_evalue>\n",
      "      <Hsp_query-from>1</Hsp_query-from>\n",
      "      <Hsp_query-to>15</Hsp_query-to>\n",
      "      <Hsp_hit-from>34</Hsp_hit-from>\n",
      "      <Hsp_hit-to>48</Hsp_hit-to>\n",
      "      <Hsp_query-frame>0</Hsp_query-frame>\n",
      "      <Hsp_hit-frame>0</Hsp_hit-frame>\n",
      "      <Hsp_identity>15</Hsp_identity>\n",
      "      <Hsp_positive>15</Hsp_positive>\n",
      "      <Hsp_gaps>0</Hsp_gaps>\n",
      "      <Hsp_align-len>15</Hsp_align-len>\n",
      "      <Hsp_qseq>SQETFSDLWKLLPEN</Hsp_qseq>\n",
      "      <Hsp_hseq>SQETFSDLWKLLPEN</Hsp_hseq>\n",
      "      <Hsp_midline>SQETFSDLWKLLPEN</Hsp_midline>\n",
      "    </Hsp>\n",
      "  </Hit_hsps>\n",
      "</Hit>\n",
      "<Hit>\n",
      "  <Hit_num>18</Hit_num>\n",
      "  <Hit_id>ref|XP_030717425.1|</Hit_id>\n",
      "  <Hit_def>cellular tumor antigen p53 isoform X1 [Globicephala melas]</Hit_def>\n",
      "  <Hit_accession>XP_030717425</Hit_accession>\n",
      "  <Hit_len>406</Hit_len>\n",
      "  <Hit_hsps>\n",
      "    <Hsp>\n",
      "      <Hsp_num>1</Hsp_num>\n",
      "      <Hsp_bit-score>35.039</Hsp_bit-score>\n",
      "      <Hsp_score>79</Hsp_score>\n",
      "      <Hsp_evalue>1.22516</Hsp_evalue>\n",
      "      <Hsp_query-from>1</Hsp_query-from>\n",
      "      <Hsp_query-to>15</Hsp_query-to>\n",
      "      <Hsp_hit-from>28</Hsp_hit-from>\n",
      "      <Hsp_hit-to>42</Hsp_hit-to>\n",
      "      <Hsp_query-frame>0</Hsp_query-frame>\n",
      "      <Hsp_hit-frame>0</Hsp_hit-frame>\n",
      "      <Hsp_identity>15</Hsp_identity>\n",
      "      <Hsp_positive>15</Hsp_positive>\n",
      "      <Hsp_gaps>0</Hsp_gaps>\n",
      "      <Hsp_align-len>15</Hsp_align-len>\n",
      "      <Hsp_qseq>SQETFSDLWKLLPEN</Hsp_qseq>\n",
      "      <Hsp_hseq>SQETFSDLWKLLPEN</Hsp_hseq>\n",
      "      <Hsp_midline>SQETFSDLWKLLPEN</Hsp_midline>\n",
      "    </Hsp>\n",
      "  </Hit_hsps>\n",
      "</Hit>\n",
      "<Hit>\n",
      "  <Hit_num>19</Hit_num>\n",
      "  <Hit_id>ref|XP_018868682.2|</Hit_id>\n",
      "  <Hit_def>cellular tumor antigen p53 isoform X2 [Gorilla gorilla gorilla]</Hit_def>\n",
      "  <Hit_accession>XP_018868682</Hit_accession>\n",
      "  <Hit_len>405</Hit_len>\n",
      "  <Hit_hsps>\n",
      "    <Hsp>\n",
      "      <Hsp_num>1</Hsp_num>\n",
      "      <Hsp_bit-score>35.039</Hsp_bit-score>\n",
      "      <Hsp_score>79</Hsp_score>\n",
      "      <Hsp_evalue>1.2253</Hsp_evalue>\n",
      "      <Hsp_query-from>1</Hsp_query-from>\n",
      "      <Hsp_query-to>15</Hsp_query-to>\n",
      "      <Hsp_hit-from>27</Hsp_hit-from>\n",
      "      <Hsp_hit-to>41</Hsp_hit-to>\n",
      "      <Hsp_query-frame>0</Hsp_query-frame>\n",
      "      <Hsp_hit-frame>0</Hsp_hit-frame>\n",
      "      <Hsp_identity>15</Hsp_identity>\n",
      "      <Hsp_positive>15</Hsp_positive>\n",
      "      <Hsp_gaps>0</Hsp_gaps>\n",
      "      <Hsp_align-len>15</Hsp_align-len>\n",
      "      <Hsp_qseq>SQETFSDLWKLLPEN</Hsp_qseq>\n",
      "      <Hsp_hseq>SQETFSDLWKLLPEN</Hsp_hseq>\n",
      "      <Hsp_midline>SQETFSDLWKLLPEN</Hsp_midline>\n",
      "    </Hsp>\n",
      "  </Hit_hsps>\n",
      "</Hit>\n",
      "<Hit>\n",
      "  <Hit_num>20</Hit_num>\n",
      "  <Hit_id>ref|XP_010360689.1|</Hit_id>\n",
      "  <Hit_def>cellular tumor antigen p53 [Rhinopithecus roxellana] &gt;ref|XP_017706034.1| PREDICTED: cellular tumor antigen p53 isoform X2 [Rhinopithecus bieti]</Hit_def>\n",
      "  <Hit_accession>XP_010360689</Hit_accession>\n",
      "  <Hit_len>393</Hit_len>\n",
      "  <Hit_hsps>\n",
      "    <Hsp>\n",
      "      <Hsp_num>1</Hsp_num>\n",
      "      <Hsp_bit-score>35.039</Hsp_bit-score>\n",
      "      <Hsp_score>79</Hsp_score>\n",
      "      <Hsp_evalue>1.22709</Hsp_evalue>\n",
      "      <Hsp_query-from>1</Hsp_query-from>\n",
      "      <Hsp_query-to>15</Hsp_query-to>\n",
      "      <Hsp_hit-from>15</Hsp_hit-from>\n",
      "      <Hsp_hit-to>29</Hsp_hit-to>\n",
      "      <Hsp_query-frame>0</Hsp_query-frame>\n",
      "      <Hsp_hit-frame>0</Hsp_hit-frame>\n",
      "      <Hsp_identity>15</Hsp_identity>\n",
      "      <Hsp_positive>15</Hsp_positive>\n",
      "      <Hsp_gaps>0</Hsp_gaps>\n",
      "      <Hsp_align-len>15</Hsp_align-len>\n",
      "      <Hsp_qseq>SQETFSDLWKLLPEN</Hsp_qseq>\n",
      "      <Hsp_hseq>SQETFSDLWKLLPEN</Hsp_hseq>\n",
      "      <Hsp_midline>SQETFSDLWKLLPEN</Hsp_midline>\n",
      "    </Hsp>\n",
      "  </Hit_hsps>\n",
      "</Hit>\n",
      "<Hit>\n",
      "  <Hit_num>21</Hit_num>\n",
      "  <Hit_id>gb|AAH03596.1|</Hit_id>\n",
      "  <Hit_def>Tumor protein p53 [Homo sapiens] &gt;gb|ABM83418.1| tumor protein p53 (Li-Fraumeni syndrome) [synthetic construct] &gt;gb|AIC55259.1| TP53, partial [synthetic construct] &gt;emb|SJX37454.1| unnamed protein product, partial [Human ORFeome Gateway entry vector]</Hit_def>\n",
      "  <Hit_accession>AAH03596</Hit_accession>\n",
      "  <Hit_len>393</Hit_len>\n",
      "  <Hit_hsps>\n",
      "    <Hsp>\n",
      "      <Hsp_num>1</Hsp_num>\n",
      "      <Hsp_bit-score>35.039</Hsp_bit-score>\n",
      "      <Hsp_score>79</Hsp_score>\n",
      "      <Hsp_evalue>1.22709</Hsp_evalue>\n",
      "      <Hsp_query-from>1</Hsp_query-from>\n",
      "      <Hsp_query-to>15</Hsp_query-to>\n",
      "      <Hsp_hit-from>15</Hsp_hit-from>\n",
      "      <Hsp_hit-to>29</Hsp_hit-to>\n",
      "      <Hsp_query-frame>0</Hsp_query-frame>\n",
      "      <Hsp_hit-frame>0</Hsp_hit-frame>\n",
      "      <Hsp_identity>15</Hsp_identity>\n",
      "      <Hsp_positive>15</Hsp_positive>\n",
      "      <Hsp_gaps>0</Hsp_gaps>\n",
      "      <Hsp_align-len>15</Hsp_align-len>\n",
      "      <Hsp_qseq>SQETFSDLWKLLPEN</Hsp_qseq>\n",
      "      <Hsp_hseq>SQETFSDLWKLLPEN</Hsp_hseq>\n",
      "      <Hsp_midline>SQETFSDLWKLLPEN</Hsp_midline>\n",
      "    </Hsp>\n",
      "  </Hit_hsps>\n",
      "</Hit>\n",
      "<Hit>\n",
      "  <Hit_num>22</Hit_num>\n",
      "  <Hit_id>ref|NP_001274298.1|</Hit_id>\n",
      "  <Hit_def>cellular tumor antigen p53 [Tupaia chinensis] &gt;sp|Q9TTA1.1| RecName: Full=Cellular tumor antigen p53; AltName: Full=Tumor suppressor p53 [Tupaia belangeri] &gt;gb|AAF22640.1| p53 tumor suppressor protein [Tupaia chinensis]</Hit_def>\n",
      "  <Hit_accession>NP_001274298</Hit_accession>\n",
      "  <Hit_len>393</Hit_len>\n",
      "  <Hit_hsps>\n",
      "    <Hsp>\n",
      "      <Hsp_num>1</Hsp_num>\n",
      "      <Hsp_bit-score>35.039</Hsp_bit-score>\n",
      "      <Hsp_score>79</Hsp_score>\n",
      "      <Hsp_evalue>1.22709</Hsp_evalue>\n",
      "      <Hsp_query-from>1</Hsp_query-from>\n",
      "      <Hsp_query-to>15</Hsp_query-to>\n",
      "      <Hsp_hit-from>15</Hsp_hit-from>\n",
      "      <Hsp_hit-to>29</Hsp_hit-to>\n",
      "      <Hsp_query-frame>0</Hsp_query-frame>\n",
      "      <Hsp_hit-frame>0</Hsp_hit-frame>\n",
      "      <Hsp_identity>15</Hsp_identity>\n",
      "      <Hsp_positive>15</Hsp_positive>\n",
      "      <Hsp_gaps>0</Hsp_gaps>\n",
      "      <Hsp_align-len>15</Hsp_align-len>\n",
      "      <Hsp_qseq>SQETFSDLWKLLPEN</Hsp_qseq>\n",
      "      <Hsp_hseq>SQETFSDLWKLLPEN</Hsp_hseq>\n",
      "      <Hsp_midline>SQETFSDLWKLLPEN</Hsp_midline>\n",
      "    </Hsp>\n",
      "  </Hit_hsps>\n",
      "</Hit>\n",
      "<Hit>\n",
      "  <Hit_num>23</Hit_num>\n",
      "  <Hit_id>ref|XP_017204298.1|</Hit_id>\n",
      "  <Hit_def>PREDICTED: cellular tumor antigen p53 isoform X2 [Oryctolagus cuniculus]</Hit_def>\n",
      "  <Hit_accession>XP_017204298</Hit_accession>\n",
      "  <Hit_len>410</Hit_len>\n",
      "  <Hit_hsps>\n",
      "    <Hsp>\n",
      "      <Hsp_num>1</Hsp_num>\n",
      "      <Hsp_bit-score>35.039</Hsp_bit-score>\n",
      "      <Hsp_score>79</Hsp_score>\n",
      "      <Hsp_evalue>1.23662</Hsp_evalue>\n",
      "      <Hsp_query-from>1</Hsp_query-from>\n",
      "      <Hsp_query-to>15</Hsp_query-to>\n",
      "      <Hsp_hit-from>34</Hsp_hit-from>\n",
      "      <Hsp_hit-to>48</Hsp_hit-to>\n",
      "      <Hsp_query-frame>0</Hsp_query-frame>\n",
      "      <Hsp_hit-frame>0</Hsp_hit-frame>\n",
      "      <Hsp_identity>15</Hsp_identity>\n",
      "      <Hsp_positive>15</Hsp_positive>\n",
      "      <Hsp_gaps>0</Hsp_gaps>\n",
      "      <Hsp_align-len>15</Hsp_align-len>\n",
      "      <Hsp_qseq>SQETFSDLWKLLPEN</Hsp_qseq>\n",
      "      <Hsp_hseq>SQETFSDLWKLLPEN</Hsp_hseq>\n",
      "      <Hsp_midline>SQETFSDLWKLLPEN</Hsp_midline>\n",
      "    </Hsp>\n",
      "  </Hit_hsps>\n",
      "</Hit>\n",
      "<Hit>\n",
      "  <Hit_num>24</Hit_num>\n",
      "  <Hit_id>ref|XP_018868681.2|</Hit_id>\n",
      "  <Hit_def>cellular tumor antigen p53 isoform X1 [Gorilla gorilla gorilla]</Hit_def>\n",
      "  <Hit_accession>XP_018868681</Hit_accession>\n",
      "  <Hit_len>408</Hit_len>\n",
      "  <Hit_hsps>\n",
      "    <Hsp>\n",
      "      <Hsp_num>1</Hsp_num>\n",
      "      <Hsp_bit-score>35.039</Hsp_bit-score>\n",
      "      <Hsp_score>79</Hsp_score>\n",
      "      <Hsp_evalue>1.23691</Hsp_evalue>\n",
      "      <Hsp_query-from>1</Hsp_query-from>\n",
      "      <Hsp_query-to>15</Hsp_query-to>\n",
      "      <Hsp_hit-from>30</Hsp_hit-from>\n",
      "      <Hsp_hit-to>44</Hsp_hit-to>\n",
      "      <Hsp_query-frame>0</Hsp_query-frame>\n",
      "      <Hsp_hit-frame>0</Hsp_hit-frame>\n",
      "      <Hsp_identity>15</Hsp_identity>\n",
      "      <Hsp_positive>15</Hsp_positive>\n",
      "      <Hsp_gaps>0</Hsp_gaps>\n",
      "      <Hsp_align-len>15</Hsp_align-len>\n",
      "      <Hsp_qseq>SQETFSDLWKLLPEN</Hsp_qseq>\n",
      "      <Hsp_hseq>SQETFSDLWKLLPEN</Hsp_hseq>\n",
      "      <Hsp_midline>SQETFSDLWKLLPEN</Hsp_midline>\n",
      "    </Hsp>\n",
      "  </Hit_hsps>\n",
      "</Hit>\n",
      "<Hit>\n",
      "  <Hit_num>25</Hit_num>\n",
      "  <Hit_id>gb|EAW90142.1|</Hit_id>\n",
      "  <Hit_def>tumor protein p53 (Li-Fraumeni syndrome), isoform CRA_c [Homo sapiens]</Hit_def>\n",
      "  <Hit_accession>EAW90142</Hit_accession>\n",
      "  <Hit_len>408</Hit_len>\n",
      "  <Hit_hsps>\n",
      "    <Hsp>\n",
      "      <Hsp_num>1</Hsp_num>\n",
      "      <Hsp_bit-score>35.039</Hsp_bit-score>\n",
      "      <Hsp_score>79</Hsp_score>\n",
      "      <Hsp_evalue>1.23691</Hsp_evalue>\n",
      "      <Hsp_query-from>1</Hsp_query-from>\n",
      "      <Hsp_query-to>15</Hsp_query-to>\n",
      "      <Hsp_hit-from>30</Hsp_hit-from>\n",
      "      <Hsp_hit-to>44</Hsp_hit-to>\n",
      "      <Hsp_query-frame>0</Hsp_query-frame>\n",
      "      <Hsp_hit-frame>0</Hsp_hit-frame>\n",
      "      <Hsp_identity>15</Hsp_identity>\n",
      "      <Hsp_positive>15</Hsp_positive>\n",
      "      <Hsp_gaps>0</Hsp_gaps>\n",
      "      <Hsp_align-len>15</Hsp_align-len>\n",
      "      <Hsp_qseq>SQETFSDLWKLLPEN</Hsp_qseq>\n",
      "      <Hsp_hseq>SQETFSDLWKLLPEN</Hsp_hseq>\n",
      "      <Hsp_midline>SQETFSDLWKLLPEN</Hsp_midline>\n",
      "    </Hsp>\n",
      "  </Hit_hsps>\n",
      "</Hit>\n",
      "<Hit>\n",
      "  <Hit_num>26</Hit_num>\n",
      "  <Hit_id>emb|CAA42634.1|</Hit_id>\n",
      "  <Hit_def>p53 transformation suppressor, partial [Homo sapiens]</Hit_def>\n",
      "  <Hit_accession>CAA42634</Hit_accession>\n",
      "  <Hit_len>393</Hit_len>\n",
      "  <Hit_hsps>\n",
      "    <Hsp>\n",
      "      <Hsp_num>1</Hsp_num>\n",
      "      <Hsp_bit-score>35.039</Hsp_bit-score>\n",
      "      <Hsp_score>79</Hsp_score>\n",
      "      <Hsp_evalue>1.23915</Hsp_evalue>\n",
      "      <Hsp_query-from>1</Hsp_query-from>\n",
      "      <Hsp_query-to>15</Hsp_query-to>\n",
      "      <Hsp_hit-from>15</Hsp_hit-from>\n",
      "      <Hsp_hit-to>29</Hsp_hit-to>\n",
      "      <Hsp_query-frame>0</Hsp_query-frame>\n",
      "      <Hsp_hit-frame>0</Hsp_hit-frame>\n",
      "      <Hsp_identity>15</Hsp_identity>\n",
      "      <Hsp_positive>15</Hsp_positive>\n",
      "      <Hsp_gaps>0</Hsp_gaps>\n",
      "      <Hsp_align-len>15</Hsp_align-len>\n",
      "      <Hsp_qseq>SQETFSDLWKLLPEN</Hsp_qseq>\n",
      "      <Hsp_hseq>SQETFSDLWKLLPEN</Hsp_hseq>\n",
      "      <Hsp_midline>SQETFSDLWKLLPEN</Hsp_midline>\n",
      "    </Hsp>\n",
      "  </Hit_hsps>\n",
      "</Hit>\n",
      "<Hit>\n",
      "  <Hit_num>27</Hit_num>\n",
      "  <Hit_id>gb|ACI25593.1|</Hit_id>\n",
      "  <Hit_def>mutant p53 [Homo sapiens]</Hit_def>\n",
      "  <Hit_accession>ACI25593</Hit_accession>\n",
      "  <Hit_len>393</Hit_len>\n",
      "  <Hit_hsps>\n",
      "    <Hsp>\n",
      "      <Hsp_num>1</Hsp_num>\n",
      "      <Hsp_bit-score>35.039</Hsp_bit-score>\n",
      "      <Hsp_score>79</Hsp_score>\n",
      "      <Hsp_evalue>1.23915</Hsp_evalue>\n",
      "      <Hsp_query-from>1</Hsp_query-from>\n",
      "      <Hsp_query-to>15</Hsp_query-to>\n",
      "      <Hsp_hit-from>15</Hsp_hit-from>\n",
      "      <Hsp_hit-to>29</Hsp_hit-to>\n",
      "      <Hsp_query-frame>0</Hsp_query-frame>\n",
      "      <Hsp_hit-frame>0</Hsp_hit-frame>\n",
      "      <Hsp_identity>15</Hsp_identity>\n",
      "      <Hsp_positive>15</Hsp_positive>\n",
      "      <Hsp_gaps>0</Hsp_gaps>\n",
      "      <Hsp_align-len>15</Hsp_align-len>\n",
      "      <Hsp_qseq>SQETFSDLWKLLPEN</Hsp_qseq>\n",
      "      <Hsp_hseq>SQETFSDLWKLLPEN</Hsp_hseq>\n",
      "      <Hsp_midline>SQETFSDLWKLLPEN</Hsp_midline>\n",
      "    </Hsp>\n",
      "  </Hit_hsps>\n",
      "</Hit>\n",
      "<Hit>\n",
      "  <Hit_num>28</Hit_num>\n",
      "  <Hit_id>gb|AEY81367.1|</Hit_id>\n",
      "  <Hit_def>mutant p53 protein, partial [Homo sapiens] &gt;emb|CAA42630.1| p53 transformation suppressor, partial [Homo sapiens] &gt;emb|CAA42632.1| p53 transformation suppressor, partial [Homo sapiens]</Hit_def>\n",
      "  <Hit_accession>AEY81367</Hit_accession>\n",
      "  <Hit_len>393</Hit_len>\n",
      "  <Hit_hsps>\n",
      "    <Hsp>\n",
      "      <Hsp_num>1</Hsp_num>\n",
      "      <Hsp_bit-score>35.039</Hsp_bit-score>\n",
      "      <Hsp_score>79</Hsp_score>\n",
      "      <Hsp_evalue>1.23915</Hsp_evalue>\n",
      "      <Hsp_query-from>1</Hsp_query-from>\n",
      "      <Hsp_query-to>15</Hsp_query-to>\n",
      "      <Hsp_hit-from>15</Hsp_hit-from>\n",
      "      <Hsp_hit-to>29</Hsp_hit-to>\n",
      "      <Hsp_query-frame>0</Hsp_query-frame>\n",
      "      <Hsp_hit-frame>0</Hsp_hit-frame>\n",
      "      <Hsp_identity>15</Hsp_identity>\n",
      "      <Hsp_positive>15</Hsp_positive>\n",
      "      <Hsp_gaps>0</Hsp_gaps>\n",
      "      <Hsp_align-len>15</Hsp_align-len>\n",
      "      <Hsp_qseq>SQETFSDLWKLLPEN</Hsp_qseq>\n",
      "      <Hsp_hseq>SQETFSDLWKLLPEN</Hsp_hseq>\n",
      "      <Hsp_midline>SQETFSDLWKLLPEN</Hsp_midline>\n",
      "    </Hsp>\n",
      "  </Hit_hsps>\n",
      "</Hit>\n",
      "<Hit>\n",
      "  <Hit_num>29</Hit_num>\n",
      "  <Hit_id>gb|AKI70249.1|</Hit_id>\n",
      "  <Hit_def>TP53, partial [synthetic construct]</Hit_def>\n",
      "  <Hit_accession>AKI70249</Hit_accession>\n",
      "  <Hit_len>393</Hit_len>\n",
      "  <Hit_hsps>\n",
      "    <Hsp>\n",
      "      <Hsp_num>1</Hsp_num>\n",
      "      <Hsp_bit-score>35.039</Hsp_bit-score>\n",
      "      <Hsp_score>79</Hsp_score>\n",
      "      <Hsp_evalue>1.23915</Hsp_evalue>\n",
      "      <Hsp_query-from>1</Hsp_query-from>\n",
      "      <Hsp_query-to>15</Hsp_query-to>\n",
      "      <Hsp_hit-from>15</Hsp_hit-from>\n",
      "      <Hsp_hit-to>29</Hsp_hit-to>\n",
      "      <Hsp_query-frame>0</Hsp_query-frame>\n",
      "      <Hsp_hit-frame>0</Hsp_hit-frame>\n",
      "      <Hsp_identity>15</Hsp_identity>\n",
      "      <Hsp_positive>15</Hsp_positive>\n",
      "      <Hsp_gaps>0</Hsp_gaps>\n",
      "      <Hsp_align-len>15</Hsp_align-len>\n",
      "      <Hsp_qseq>SQETFSDLWKLLPEN</Hsp_qseq>\n",
      "      <Hsp_hseq>SQETFSDLWKLLPEN</Hsp_hseq>\n",
      "      <Hsp_midline>SQETFSDLWKLLPEN</Hsp_midline>\n",
      "    </Hsp>\n",
      "  </Hit_hsps>\n",
      "</Hit>\n",
      "<Hit>\n",
      "  <Hit_num>30</Hit_num>\n",
      "  <Hit_id>gb|AAV38428.1|</Hit_id>\n",
      "  <Hit_def>tumor protein p53 (Li-Fraumeni syndrome) [Homo sapiens] &gt;gb|AAX41259.1| tumor protein p53 [synthetic construct]</Hit_def>\n",
      "  <Hit_accession>AAV38428</Hit_accession>\n",
      "  <Hit_len>393</Hit_len>\n",
      "  <Hit_hsps>\n",
      "    <Hsp>\n",
      "      <Hsp_num>1</Hsp_num>\n",
      "      <Hsp_bit-score>35.039</Hsp_bit-score>\n",
      "      <Hsp_score>79</Hsp_score>\n",
      "      <Hsp_evalue>1.23915</Hsp_evalue>\n",
      "      <Hsp_query-from>1</Hsp_query-from>\n",
      "      <Hsp_query-to>15</Hsp_query-to>\n",
      "      <Hsp_hit-from>15</Hsp_hit-from>\n",
      "      <Hsp_hit-to>29</Hsp_hit-to>\n",
      "      <Hsp_query-frame>0</Hsp_query-frame>\n",
      "      <Hsp_hit-frame>0</Hsp_hit-frame>\n",
      "      <Hsp_identity>15</Hsp_identity>\n",
      "      <Hsp_positive>15</Hsp_positive>\n",
      "      <Hsp_gaps>0</Hsp_gaps>\n",
      "      <Hsp_align-len>15</Hsp_align-len>\n",
      "      <Hsp_qseq>SQETFSDLWKLLPEN</Hsp_qseq>\n",
      "      <Hsp_hseq>SQETFSDLWKLLPEN</Hsp_hseq>\n",
      "      <Hsp_midline>SQETFSDLWKLLPEN</Hsp_midline>\n",
      "    </Hsp>\n",
      "  </Hit_hsps>\n",
      "</Hit>\n",
      "<Hit>\n",
      "  <Hit_num>31</Hit_num>\n",
      "  <Hit_id>ref|XP_003810114.2|</Hit_id>\n",
      "  <Hit_def>cellular tumor antigen p53 [Pan paniscus] &gt;gb|AAC12971.1| p53 [Homo sapiens] &gt;dbj|BAI45431.1| tumor protein p53, partial [synthetic construct] &gt;gb|AAD28628.1| tumor suppressor protein p53 [Homo sapiens] &gt;gb|AAG28785.1| p53 protein [Homo sapiens] &gt;gb|AAV80424.1| tumor protein p53 (Li-Fraumeni syndrome) [Homo sapiens]</Hit_def>\n",
      "  <Hit_accession>XP_003810114</Hit_accession>\n",
      "  <Hit_len>393</Hit_len>\n",
      "  <Hit_hsps>\n",
      "    <Hsp>\n",
      "      <Hsp_num>1</Hsp_num>\n",
      "      <Hsp_bit-score>35.039</Hsp_bit-score>\n",
      "      <Hsp_score>79</Hsp_score>\n",
      "      <Hsp_evalue>1.23915</Hsp_evalue>\n",
      "      <Hsp_query-from>1</Hsp_query-from>\n",
      "      <Hsp_query-to>15</Hsp_query-to>\n",
      "      <Hsp_hit-from>15</Hsp_hit-from>\n",
      "      <Hsp_hit-to>29</Hsp_hit-to>\n",
      "      <Hsp_query-frame>0</Hsp_query-frame>\n",
      "      <Hsp_hit-frame>0</Hsp_hit-frame>\n",
      "      <Hsp_identity>15</Hsp_identity>\n",
      "      <Hsp_positive>15</Hsp_positive>\n",
      "      <Hsp_gaps>0</Hsp_gaps>\n",
      "      <Hsp_align-len>15</Hsp_align-len>\n",
      "      <Hsp_qseq>SQETFSDLWKLLPEN</Hsp_qseq>\n",
      "      <Hsp_hseq>SQETFSDLWKLLPEN</Hsp_hseq>\n",
      "      <Hsp_midline>SQETFSDLWKLLPEN</Hsp_midline>\n",
      "    </Hsp>\n",
      "  </Hit_hsps>\n",
      "</Hit>\n",
      "<Hit>\n",
      "  <Hit_num>32</Hit_num>\n",
      "  <Hit_id>gb|AKI70246.1|</Hit_id>\n",
      "  <Hit_def>TP53, partial [synthetic construct]</Hit_def>\n",
      "  <Hit_accession>AKI70246</Hit_accession>\n",
      "  <Hit_len>393</Hit_len>\n",
      "  <Hit_hsps>\n",
      "    <Hsp>\n",
      "      <Hsp_num>1</Hsp_num>\n",
      "      <Hsp_bit-score>35.039</Hsp_bit-score>\n",
      "      <Hsp_score>79</Hsp_score>\n",
      "      <Hsp_evalue>1.23915</Hsp_evalue>\n",
      "      <Hsp_query-from>1</Hsp_query-from>\n",
      "      <Hsp_query-to>15</Hsp_query-to>\n",
      "      <Hsp_hit-from>15</Hsp_hit-from>\n",
      "      <Hsp_hit-to>29</Hsp_hit-to>\n",
      "      <Hsp_query-frame>0</Hsp_query-frame>\n",
      "      <Hsp_hit-frame>0</Hsp_hit-frame>\n",
      "      <Hsp_identity>15</Hsp_identity>\n",
      "      <Hsp_positive>15</Hsp_positive>\n",
      "      <Hsp_gaps>0</Hsp_gaps>\n",
      "      <Hsp_align-len>15</Hsp_align-len>\n",
      "      <Hsp_qseq>SQETFSDLWKLLPEN</Hsp_qseq>\n",
      "      <Hsp_hseq>SQETFSDLWKLLPEN</Hsp_hseq>\n",
      "      <Hsp_midline>SQETFSDLWKLLPEN</Hsp_midline>\n",
      "    </Hsp>\n",
      "  </Hit_hsps>\n",
      "</Hit>\n",
      "<Hit>\n",
      "  <Hit_num>33</Hit_num>\n",
      "  <Hit_id>gb|ABM86630.1|</Hit_id>\n",
      "  <Hit_def>tumor protein p53 (Li-Fraumeni syndrome), partial [synthetic construct]</Hit_def>\n",
      "  <Hit_accession>ABM86630</Hit_accession>\n",
      "  <Hit_len>393</Hit_len>\n",
      "  <Hit_hsps>\n",
      "    <Hsp>\n",
      "      <Hsp_num>1</Hsp_num>\n",
      "      <Hsp_bit-score>35.039</Hsp_bit-score>\n",
      "      <Hsp_score>79</Hsp_score>\n",
      "      <Hsp_evalue>1.23915</Hsp_evalue>\n",
      "      <Hsp_query-from>1</Hsp_query-from>\n",
      "      <Hsp_query-to>15</Hsp_query-to>\n",
      "      <Hsp_hit-from>15</Hsp_hit-from>\n",
      "      <Hsp_hit-to>29</Hsp_hit-to>\n",
      "      <Hsp_query-frame>0</Hsp_query-frame>\n",
      "      <Hsp_hit-frame>0</Hsp_hit-frame>\n",
      "      <Hsp_identity>15</Hsp_identity>\n",
      "      <Hsp_positive>15</Hsp_positive>\n",
      "      <Hsp_gaps>0</Hsp_gaps>\n",
      "      <Hsp_align-len>15</Hsp_align-len>\n",
      "      <Hsp_qseq>SQETFSDLWKLLPEN</Hsp_qseq>\n",
      "      <Hsp_hseq>SQETFSDLWKLLPEN</Hsp_hseq>\n",
      "      <Hsp_midline>SQETFSDLWKLLPEN</Hsp_midline>\n",
      "    </Hsp>\n",
      "  </Hit_hsps>\n",
      "</Hit>\n",
      "<Hit>\n",
      "  <Hit_num>34</Hit_num>\n",
      "  <Hit_id>gb|AKI70248.1|</Hit_id>\n",
      "  <Hit_def>TP53, partial [synthetic construct]</Hit_def>\n",
      "  <Hit_accession>AKI70248</Hit_accession>\n",
      "  <Hit_len>393</Hit_len>\n",
      "  <Hit_hsps>\n",
      "    <Hsp>\n",
      "      <Hsp_num>1</Hsp_num>\n",
      "      <Hsp_bit-score>35.039</Hsp_bit-score>\n",
      "      <Hsp_score>79</Hsp_score>\n",
      "      <Hsp_evalue>1.23915</Hsp_evalue>\n",
      "      <Hsp_query-from>1</Hsp_query-from>\n",
      "      <Hsp_query-to>15</Hsp_query-to>\n",
      "      <Hsp_hit-from>15</Hsp_hit-from>\n",
      "      <Hsp_hit-to>29</Hsp_hit-to>\n",
      "      <Hsp_query-frame>0</Hsp_query-frame>\n",
      "      <Hsp_hit-frame>0</Hsp_hit-frame>\n",
      "      <Hsp_identity>15</Hsp_identity>\n",
      "      <Hsp_positive>15</Hsp_positive>\n",
      "      <Hsp_gaps>0</Hsp_gaps>\n",
      "      <Hsp_align-len>15</Hsp_align-len>\n",
      "      <Hsp_qseq>SQETFSDLWKLLPEN</Hsp_qseq>\n",
      "      <Hsp_hseq>SQETFSDLWKLLPEN</Hsp_hseq>\n",
      "      <Hsp_midline>SQETFSDLWKLLPEN</Hsp_midline>\n",
      "    </Hsp>\n",
      "  </Hit_hsps>\n",
      "</Hit>\n",
      "<Hit>\n",
      "  <Hit_num>35</Hit_num>\n",
      "  <Hit_id>gb|AAD28535.1|</Hit_id>\n",
      "  <Hit_def>tumor suppressor protein p53 [Homo sapiens]</Hit_def>\n",
      "  <Hit_accession>AAD28535</Hit_accession>\n",
      "  <Hit_len>393</Hit_len>\n",
      "  <Hit_hsps>\n",
      "    <Hsp>\n",
      "      <Hsp_num>1</Hsp_num>\n",
      "      <Hsp_bit-score>35.039</Hsp_bit-score>\n",
      "      <Hsp_score>79</Hsp_score>\n",
      "      <Hsp_evalue>1.23915</Hsp_evalue>\n",
      "      <Hsp_query-from>1</Hsp_query-from>\n",
      "      <Hsp_query-to>15</Hsp_query-to>\n",
      "      <Hsp_hit-from>15</Hsp_hit-from>\n",
      "      <Hsp_hit-to>29</Hsp_hit-to>\n",
      "      <Hsp_query-frame>0</Hsp_query-frame>\n",
      "      <Hsp_hit-frame>0</Hsp_hit-frame>\n",
      "      <Hsp_identity>15</Hsp_identity>\n",
      "      <Hsp_positive>15</Hsp_positive>\n",
      "      <Hsp_gaps>0</Hsp_gaps>\n",
      "      <Hsp_align-len>15</Hsp_align-len>\n",
      "      <Hsp_qseq>SQETFSDLWKLLPEN</Hsp_qseq>\n",
      "      <Hsp_hseq>SQETFSDLWKLLPEN</Hsp_hseq>\n",
      "      <Hsp_midline>SQETFSDLWKLLPEN</Hsp_midline>\n",
      "    </Hsp>\n",
      "  </Hit_hsps>\n",
      "</Hit>\n",
      "<Hit>\n",
      "  <Hit_num>36</Hit_num>\n",
      "  <Hit_id>emb|CAA42629.1|</Hit_id>\n",
      "  <Hit_def>p53 transformation suppressor, partial [Homo sapiens]</Hit_def>\n",
      "  <Hit_accession>CAA42629</Hit_accession>\n",
      "  <Hit_len>393</Hit_len>\n",
      "  <Hit_hsps>\n",
      "    <Hsp>\n",
      "      <Hsp_num>1</Hsp_num>\n",
      "      <Hsp_bit-score>35.039</Hsp_bit-score>\n",
      "      <Hsp_score>79</Hsp_score>\n",
      "      <Hsp_evalue>1.23915</Hsp_evalue>\n",
      "      <Hsp_query-from>1</Hsp_query-from>\n",
      "      <Hsp_query-to>15</Hsp_query-to>\n",
      "      <Hsp_hit-from>15</Hsp_hit-from>\n",
      "      <Hsp_hit-to>29</Hsp_hit-to>\n",
      "      <Hsp_query-frame>0</Hsp_query-frame>\n",
      "      <Hsp_hit-frame>0</Hsp_hit-frame>\n",
      "      <Hsp_identity>15</Hsp_identity>\n",
      "      <Hsp_positive>15</Hsp_positive>\n",
      "      <Hsp_gaps>0</Hsp_gaps>\n",
      "      <Hsp_align-len>15</Hsp_align-len>\n",
      "      <Hsp_qseq>SQETFSDLWKLLPEN</Hsp_qseq>\n",
      "      <Hsp_hseq>SQETFSDLWKLLPEN</Hsp_hseq>\n",
      "      <Hsp_midline>SQETFSDLWKLLPEN</Hsp_midline>\n",
      "    </Hsp>\n",
      "  </Hit_hsps>\n",
      "</Hit>\n",
      "<Hit>\n",
      "  <Hit_num>37</Hit_num>\n",
      "  <Hit_id>ref|XP_033704460.1|</Hit_id>\n",
      "  <Hit_def>cellular tumor antigen p53 isoform X1 [Tursiops truncatus]</Hit_def>\n",
      "  <Hit_accession>XP_033704460</Hit_accession>\n",
      "  <Hit_len>400</Hit_len>\n",
      "  <Hit_hsps>\n",
      "    <Hsp>\n",
      "      <Hsp_num>1</Hsp_num>\n",
      "      <Hsp_bit-score>35.039</Hsp_bit-score>\n",
      "      <Hsp_score>79</Hsp_score>\n",
      "      <Hsp_evalue>1.25025</Hsp_evalue>\n",
      "      <Hsp_query-from>1</Hsp_query-from>\n",
      "      <Hsp_query-to>15</Hsp_query-to>\n",
      "      <Hsp_hit-from>28</Hsp_hit-from>\n",
      "      <Hsp_hit-to>42</Hsp_hit-to>\n",
      "      <Hsp_query-frame>0</Hsp_query-frame>\n",
      "      <Hsp_hit-frame>0</Hsp_hit-frame>\n",
      "      <Hsp_identity>15</Hsp_identity>\n",
      "      <Hsp_positive>15</Hsp_positive>\n",
      "      <Hsp_gaps>0</Hsp_gaps>\n",
      "      <Hsp_align-len>15</Hsp_align-len>\n",
      "      <Hsp_qseq>SQETFSDLWKLLPEN</Hsp_qseq>\n",
      "      <Hsp_hseq>SQETFSDLWKLLPEN</Hsp_hseq>\n",
      "      <Hsp_midline>SQETFSDLWKLLPEN</Hsp_midline>\n",
      "    </Hsp>\n",
      "  </Hit_hsps>\n",
      "</Hit>\n",
      "<Hit>\n",
      "  <Hit_num>38</Hit_num>\n",
      "  <Hit_id>ref|XP_026986728.1|</Hit_id>\n",
      "  <Hit_def>cellular tumor antigen p53 isoform X1 [Lagenorhynchus obliquidens]</Hit_def>\n",
      "  <Hit_accession>XP_026986728</Hit_accession>\n",
      "  <Hit_len>400</Hit_len>\n",
      "  <Hit_hsps>\n",
      "    <Hsp>\n",
      "      <Hsp_num>1</Hsp_num>\n",
      "      <Hsp_bit-score>35.039</Hsp_bit-score>\n",
      "      <Hsp_score>79</Hsp_score>\n",
      "      <Hsp_evalue>1.25025</Hsp_evalue>\n",
      "      <Hsp_query-from>1</Hsp_query-from>\n",
      "      <Hsp_query-to>15</Hsp_query-to>\n",
      "      <Hsp_hit-from>28</Hsp_hit-from>\n",
      "      <Hsp_hit-to>42</Hsp_hit-to>\n",
      "      <Hsp_query-frame>0</Hsp_query-frame>\n",
      "      <Hsp_hit-frame>0</Hsp_hit-frame>\n",
      "      <Hsp_identity>15</Hsp_identity>\n",
      "      <Hsp_positive>15</Hsp_positive>\n",
      "      <Hsp_gaps>0</Hsp_gaps>\n",
      "      <Hsp_align-len>15</Hsp_align-len>\n",
      "      <Hsp_qseq>SQETFSDLWKLLPEN</Hsp_qseq>\n",
      "      <Hsp_hseq>SQETFSDLWKLLPEN</Hsp_hseq>\n",
      "      <Hsp_midline>SQETFSDLWKLLPEN</Hsp_midline>\n",
      "    </Hsp>\n",
      "  </Hit_hsps>\n",
      "</Hit>\n",
      "<Hit>\n",
      "  <Hit_num>39</Hit_num>\n",
      "  <Hit_id>gb|AAA61211.1|</Hit_id>\n",
      "  <Hit_def>p53 antigen [Homo sapiens]</Hit_def>\n",
      "  <Hit_accession>AAA61211</Hit_accession>\n",
      "  <Hit_len>393</Hit_len>\n",
      "  <Hit_hsps>\n",
      "    <Hsp>\n",
      "      <Hsp_num>1</Hsp_num>\n",
      "      <Hsp_bit-score>35.039</Hsp_bit-score>\n",
      "      <Hsp_score>79</Hsp_score>\n",
      "      <Hsp_evalue>1.25132</Hsp_evalue>\n",
      "      <Hsp_query-from>1</Hsp_query-from>\n",
      "      <Hsp_query-to>15</Hsp_query-to>\n",
      "      <Hsp_hit-from>15</Hsp_hit-from>\n",
      "      <Hsp_hit-to>29</Hsp_hit-to>\n",
      "      <Hsp_query-frame>0</Hsp_query-frame>\n",
      "      <Hsp_hit-frame>0</Hsp_hit-frame>\n",
      "      <Hsp_identity>15</Hsp_identity>\n",
      "      <Hsp_positive>15</Hsp_positive>\n",
      "      <Hsp_gaps>0</Hsp_gaps>\n",
      "      <Hsp_align-len>15</Hsp_align-len>\n",
      "      <Hsp_qseq>SQETFSDLWKLLPEN</Hsp_qseq>\n",
      "      <Hsp_hseq>SQETFSDLWKLLPEN</Hsp_hseq>\n",
      "      <Hsp_midline>SQETFSDLWKLLPEN</Hsp_midline>\n",
      "    </Hsp>\n",
      "  </Hit_hsps>\n",
      "</Hit>\n",
      "<Hit>\n",
      "  <Hit_num>40</Hit_num>\n",
      "  <Hit_id>emb|CAA42626.1|</Hit_id>\n",
      "  <Hit_def>p53 transformation suppressor, partial [Homo sapiens]</Hit_def>\n",
      "  <Hit_accession>CAA42626</Hit_accession>\n",
      "  <Hit_len>393</Hit_len>\n",
      "  <Hit_hsps>\n",
      "    <Hsp>\n",
      "      <Hsp_num>1</Hsp_num>\n",
      "      <Hsp_bit-score>35.039</Hsp_bit-score>\n",
      "      <Hsp_score>79</Hsp_score>\n",
      "      <Hsp_evalue>1.25132</Hsp_evalue>\n",
      "      <Hsp_query-from>1</Hsp_query-from>\n",
      "      <Hsp_query-to>15</Hsp_query-to>\n",
      "      <Hsp_hit-from>15</Hsp_hit-from>\n",
      "      <Hsp_hit-to>29</Hsp_hit-to>\n",
      "      <Hsp_query-frame>0</Hsp_query-frame>\n",
      "      <Hsp_hit-frame>0</Hsp_hit-frame>\n",
      "      <Hsp_identity>15</Hsp_identity>\n",
      "      <Hsp_positive>15</Hsp_positive>\n",
      "      <Hsp_gaps>0</Hsp_gaps>\n",
      "      <Hsp_align-len>15</Hsp_align-len>\n",
      "      <Hsp_qseq>SQETFSDLWKLLPEN</Hsp_qseq>\n",
      "      <Hsp_hseq>SQETFSDLWKLLPEN</Hsp_hseq>\n",
      "      <Hsp_midline>SQETFSDLWKLLPEN</Hsp_midline>\n",
      "    </Hsp>\n",
      "  </Hit_hsps>\n",
      "</Hit>\n",
      "<Hit>\n",
      "  <Hit_num>41</Hit_num>\n",
      "  <Hit_id>gb|AAA59987.1|</Hit_id>\n",
      "  <Hit_def>phosphoprotein p53 [Homo sapiens]</Hit_def>\n",
      "  <Hit_accession>AAA59987</Hit_accession>\n",
      "  <Hit_len>393</Hit_len>\n",
      "  <Hit_hsps>\n",
      "    <Hsp>\n",
      "      <Hsp_num>1</Hsp_num>\n",
      "      <Hsp_bit-score>35.039</Hsp_bit-score>\n",
      "      <Hsp_score>79</Hsp_score>\n",
      "      <Hsp_evalue>1.25132</Hsp_evalue>\n",
      "      <Hsp_query-from>1</Hsp_query-from>\n",
      "      <Hsp_query-to>15</Hsp_query-to>\n",
      "      <Hsp_hit-from>15</Hsp_hit-from>\n",
      "      <Hsp_hit-to>29</Hsp_hit-to>\n",
      "      <Hsp_query-frame>0</Hsp_query-frame>\n",
      "      <Hsp_hit-frame>0</Hsp_hit-frame>\n",
      "      <Hsp_identity>15</Hsp_identity>\n",
      "      <Hsp_positive>15</Hsp_positive>\n",
      "      <Hsp_gaps>0</Hsp_gaps>\n",
      "      <Hsp_align-len>15</Hsp_align-len>\n",
      "      <Hsp_qseq>SQETFSDLWKLLPEN</Hsp_qseq>\n",
      "      <Hsp_hseq>SQETFSDLWKLLPEN</Hsp_hseq>\n",
      "      <Hsp_midline>SQETFSDLWKLLPEN</Hsp_midline>\n",
      "    </Hsp>\n",
      "  </Hit_hsps>\n",
      "</Hit>\n",
      "<Hit>\n",
      "  <Hit_num>42</Hit_num>\n",
      "  <Hit_id>gb|AYE20616.1|</Hit_id>\n",
      "  <Hit_def>mutant tumor protein p53 [Homo sapiens]</Hit_def>\n",
      "  <Hit_accession>AYE20616</Hit_accession>\n",
      "  <Hit_len>393</Hit_len>\n",
      "  <Hit_hsps>\n",
      "    <Hsp>\n",
      "      <Hsp_num>1</Hsp_num>\n",
      "      <Hsp_bit-score>35.039</Hsp_bit-score>\n",
      "      <Hsp_score>79</Hsp_score>\n",
      "      <Hsp_evalue>1.25132</Hsp_evalue>\n",
      "      <Hsp_query-from>1</Hsp_query-from>\n",
      "      <Hsp_query-to>15</Hsp_query-to>\n",
      "      <Hsp_hit-from>15</Hsp_hit-from>\n",
      "      <Hsp_hit-to>29</Hsp_hit-to>\n",
      "      <Hsp_query-frame>0</Hsp_query-frame>\n",
      "      <Hsp_hit-frame>0</Hsp_hit-frame>\n",
      "      <Hsp_identity>15</Hsp_identity>\n",
      "      <Hsp_positive>15</Hsp_positive>\n",
      "      <Hsp_gaps>0</Hsp_gaps>\n",
      "      <Hsp_align-len>15</Hsp_align-len>\n",
      "      <Hsp_qseq>SQETFSDLWKLLPEN</Hsp_qseq>\n",
      "      <Hsp_hseq>SQETFSDLWKLLPEN</Hsp_hseq>\n",
      "      <Hsp_midline>SQETFSDLWKLLPEN</Hsp_midline>\n",
      "    </Hsp>\n",
      "  </Hit_hsps>\n",
      "</Hit>\n",
      "<Hit>\n",
      "  <Hit_num>43</Hit_num>\n",
      "  <Hit_id>emb|CAA42635.1|</Hit_id>\n",
      "  <Hit_def>p53 transformation suppressor, partial [Homo sapiens]</Hit_def>\n",
      "  <Hit_accession>CAA42635</Hit_accession>\n",
      "  <Hit_len>393</Hit_len>\n",
      "  <Hit_hsps>\n",
      "    <Hsp>\n",
      "      <Hsp_num>1</Hsp_num>\n",
      "      <Hsp_bit-score>35.039</Hsp_bit-score>\n",
      "      <Hsp_score>79</Hsp_score>\n",
      "      <Hsp_evalue>1.25132</Hsp_evalue>\n",
      "      <Hsp_query-from>1</Hsp_query-from>\n",
      "      <Hsp_query-to>15</Hsp_query-to>\n",
      "      <Hsp_hit-from>15</Hsp_hit-from>\n",
      "      <Hsp_hit-to>29</Hsp_hit-to>\n",
      "      <Hsp_query-frame>0</Hsp_query-frame>\n",
      "      <Hsp_hit-frame>0</Hsp_hit-frame>\n",
      "      <Hsp_identity>15</Hsp_identity>\n",
      "      <Hsp_positive>15</Hsp_positive>\n",
      "      <Hsp_gaps>0</Hsp_gaps>\n",
      "      <Hsp_align-len>15</Hsp_align-len>\n",
      "      <Hsp_qseq>SQETFSDLWKLLPEN</Hsp_qseq>\n",
      "      <Hsp_hseq>SQETFSDLWKLLPEN</Hsp_hseq>\n",
      "      <Hsp_midline>SQETFSDLWKLLPEN</Hsp_midline>\n",
      "    </Hsp>\n",
      "  </Hit_hsps>\n",
      "</Hit>\n",
      "<Hit>\n",
      "  <Hit_num>44</Hit_num>\n",
      "  <Hit_id>gb|AYE20615.1|</Hit_id>\n",
      "  <Hit_def>mutant tumor protein p53 [Homo sapiens]</Hit_def>\n",
      "  <Hit_accession>AYE20615</Hit_accession>\n",
      "  <Hit_len>393</Hit_len>\n",
      "  <Hit_hsps>\n",
      "    <Hsp>\n",
      "      <Hsp_num>1</Hsp_num>\n",
      "      <Hsp_bit-score>35.039</Hsp_bit-score>\n",
      "      <Hsp_score>79</Hsp_score>\n",
      "      <Hsp_evalue>1.25132</Hsp_evalue>\n",
      "      <Hsp_query-from>1</Hsp_query-from>\n",
      "      <Hsp_query-to>15</Hsp_query-to>\n",
      "      <Hsp_hit-from>15</Hsp_hit-from>\n",
      "      <Hsp_hit-to>29</Hsp_hit-to>\n",
      "      <Hsp_query-frame>0</Hsp_query-frame>\n",
      "      <Hsp_hit-frame>0</Hsp_hit-frame>\n",
      "      <Hsp_identity>15</Hsp_identity>\n",
      "      <Hsp_positive>15</Hsp_positive>\n",
      "      <Hsp_gaps>0</Hsp_gaps>\n",
      "      <Hsp_align-len>15</Hsp_align-len>\n",
      "      <Hsp_qseq>SQETFSDLWKLLPEN</Hsp_qseq>\n",
      "      <Hsp_hseq>SQETFSDLWKLLPEN</Hsp_hseq>\n",
      "      <Hsp_midline>SQETFSDLWKLLPEN</Hsp_midline>\n",
      "    </Hsp>\n",
      "  </Hit_hsps>\n",
      "</Hit>\n",
      "<Hit>\n",
      "  <Hit_num>45</Hit_num>\n",
      "  <Hit_id>gb|AAX36369.1|</Hit_id>\n",
      "  <Hit_def>tumor protein p53 [synthetic construct]</Hit_def>\n",
      "  <Hit_accession>AAX36369</Hit_accession>\n",
      "  <Hit_len>393</Hit_len>\n",
      "  <Hit_hsps>\n",
      "    <Hsp>\n",
      "      <Hsp_num>1</Hsp_num>\n",
      "      <Hsp_bit-score>35.039</Hsp_bit-score>\n",
      "      <Hsp_score>79</Hsp_score>\n",
      "      <Hsp_evalue>1.25132</Hsp_evalue>\n",
      "      <Hsp_query-from>1</Hsp_query-from>\n",
      "      <Hsp_query-to>15</Hsp_query-to>\n",
      "      <Hsp_hit-from>15</Hsp_hit-from>\n",
      "      <Hsp_hit-to>29</Hsp_hit-to>\n",
      "      <Hsp_query-frame>0</Hsp_query-frame>\n",
      "      <Hsp_hit-frame>0</Hsp_hit-frame>\n",
      "      <Hsp_identity>15</Hsp_identity>\n",
      "      <Hsp_positive>15</Hsp_positive>\n",
      "      <Hsp_gaps>0</Hsp_gaps>\n",
      "      <Hsp_align-len>15</Hsp_align-len>\n",
      "      <Hsp_qseq>SQETFSDLWKLLPEN</Hsp_qseq>\n",
      "      <Hsp_hseq>SQETFSDLWKLLPEN</Hsp_hseq>\n",
      "      <Hsp_midline>SQETFSDLWKLLPEN</Hsp_midline>\n",
      "    </Hsp>\n",
      "  </Hit_hsps>\n",
      "</Hit>\n",
      "<Hit>\n",
      "  <Hit_num>46</Hit_num>\n",
      "  <Hit_id>dbj|BAJ52715.1|</Hit_id>\n",
      "  <Hit_def>tumor suppressor gene p53 [Homo sapiens]</Hit_def>\n",
      "  <Hit_accession>BAJ52715</Hit_accession>\n",
      "  <Hit_len>393</Hit_len>\n",
      "  <Hit_hsps>\n",
      "    <Hsp>\n",
      "      <Hsp_num>1</Hsp_num>\n",
      "      <Hsp_bit-score>35.039</Hsp_bit-score>\n",
      "      <Hsp_score>79</Hsp_score>\n",
      "      <Hsp_evalue>1.25132</Hsp_evalue>\n",
      "      <Hsp_query-from>1</Hsp_query-from>\n",
      "      <Hsp_query-to>15</Hsp_query-to>\n",
      "      <Hsp_hit-from>15</Hsp_hit-from>\n",
      "      <Hsp_hit-to>29</Hsp_hit-to>\n",
      "      <Hsp_query-frame>0</Hsp_query-frame>\n",
      "      <Hsp_hit-frame>0</Hsp_hit-frame>\n",
      "      <Hsp_identity>15</Hsp_identity>\n",
      "      <Hsp_positive>15</Hsp_positive>\n",
      "      <Hsp_gaps>0</Hsp_gaps>\n",
      "      <Hsp_align-len>15</Hsp_align-len>\n",
      "      <Hsp_qseq>SQETFSDLWKLLPEN</Hsp_qseq>\n",
      "      <Hsp_hseq>SQETFSDLWKLLPEN</Hsp_hseq>\n",
      "      <Hsp_midline>SQETFSDLWKLLPEN</Hsp_midline>\n",
      "    </Hsp>\n",
      "  </Hit_hsps>\n",
      "</Hit>\n",
      "<Hit>\n",
      "  <Hit_num>47</Hit_num>\n",
      "  <Hit_id>ref|XP_030656345.1|</Hit_id>\n",
      "  <Hit_def>cellular tumor antigen p53 [Nomascus leucogenys]</Hit_def>\n",
      "  <Hit_accession>XP_030656345</Hit_accession>\n",
      "  <Hit_len>393</Hit_len>\n",
      "  <Hit_hsps>\n",
      "    <Hsp>\n",
      "      <Hsp_num>1</Hsp_num>\n",
      "      <Hsp_bit-score>35.039</Hsp_bit-score>\n",
      "      <Hsp_score>79</Hsp_score>\n",
      "      <Hsp_evalue>1.25132</Hsp_evalue>\n",
      "      <Hsp_query-from>1</Hsp_query-from>\n",
      "      <Hsp_query-to>15</Hsp_query-to>\n",
      "      <Hsp_hit-from>15</Hsp_hit-from>\n",
      "      <Hsp_hit-to>29</Hsp_hit-to>\n",
      "      <Hsp_query-frame>0</Hsp_query-frame>\n",
      "      <Hsp_hit-frame>0</Hsp_hit-frame>\n",
      "      <Hsp_identity>15</Hsp_identity>\n",
      "      <Hsp_positive>15</Hsp_positive>\n",
      "      <Hsp_gaps>0</Hsp_gaps>\n",
      "      <Hsp_align-len>15</Hsp_align-len>\n",
      "      <Hsp_qseq>SQETFSDLWKLLPEN</Hsp_qseq>\n",
      "      <Hsp_hseq>SQETFSDLWKLLPEN</Hsp_hseq>\n",
      "      <Hsp_midline>SQETFSDLWKLLPEN</Hsp_midline>\n",
      "    </Hsp>\n",
      "  </Hit_hsps>\n",
      "</Hit>\n",
      "<Hit>\n",
      "  <Hit_num>48</Hit_num>\n",
      "  <Hit_id>emb|CAA42633.1|</Hit_id>\n",
      "  <Hit_def>p53 transformation suppressor, partial [Homo sapiens]</Hit_def>\n",
      "  <Hit_accession>CAA42633</Hit_accession>\n",
      "  <Hit_len>393</Hit_len>\n",
      "  <Hit_hsps>\n",
      "    <Hsp>\n",
      "      <Hsp_num>1</Hsp_num>\n",
      "      <Hsp_bit-score>35.039</Hsp_bit-score>\n",
      "      <Hsp_score>79</Hsp_score>\n",
      "      <Hsp_evalue>1.25132</Hsp_evalue>\n",
      "      <Hsp_query-from>1</Hsp_query-from>\n",
      "      <Hsp_query-to>15</Hsp_query-to>\n",
      "      <Hsp_hit-from>15</Hsp_hit-from>\n",
      "      <Hsp_hit-to>29</Hsp_hit-to>\n",
      "      <Hsp_query-frame>0</Hsp_query-frame>\n",
      "      <Hsp_hit-frame>0</Hsp_hit-frame>\n",
      "      <Hsp_identity>15</Hsp_identity>\n",
      "      <Hsp_positive>15</Hsp_positive>\n",
      "      <Hsp_gaps>0</Hsp_gaps>\n",
      "      <Hsp_align-len>15</Hsp_align-len>\n",
      "      <Hsp_qseq>SQETFSDLWKLLPEN</Hsp_qseq>\n",
      "      <Hsp_hseq>SQETFSDLWKLLPEN</Hsp_hseq>\n",
      "      <Hsp_midline>SQETFSDLWKLLPEN</Hsp_midline>\n",
      "    </Hsp>\n",
      "  </Hit_hsps>\n",
      "</Hit>\n",
      "<Hit>\n",
      "  <Hit_num>49</Hit_num>\n",
      "  <Hit_id>emb|CAA42628.1|</Hit_id>\n",
      "  <Hit_def>p53 transformation suppressor, partial [Homo sapiens]</Hit_def>\n",
      "  <Hit_accession>CAA42628</Hit_accession>\n",
      "  <Hit_len>393</Hit_len>\n",
      "  <Hit_hsps>\n",
      "    <Hsp>\n",
      "      <Hsp_num>1</Hsp_num>\n",
      "      <Hsp_bit-score>35.039</Hsp_bit-score>\n",
      "      <Hsp_score>79</Hsp_score>\n",
      "      <Hsp_evalue>1.25132</Hsp_evalue>\n",
      "      <Hsp_query-from>1</Hsp_query-from>\n",
      "      <Hsp_query-to>15</Hsp_query-to>\n",
      "      <Hsp_hit-from>15</Hsp_hit-from>\n",
      "      <Hsp_hit-to>29</Hsp_hit-to>\n",
      "      <Hsp_query-frame>0</Hsp_query-frame>\n",
      "      <Hsp_hit-frame>0</Hsp_hit-frame>\n",
      "      <Hsp_identity>15</Hsp_identity>\n",
      "      <Hsp_positive>15</Hsp_positive>\n",
      "      <Hsp_gaps>0</Hsp_gaps>\n",
      "      <Hsp_align-len>15</Hsp_align-len>\n",
      "      <Hsp_qseq>SQETFSDLWKLLPEN</Hsp_qseq>\n",
      "      <Hsp_hseq>SQETFSDLWKLLPEN</Hsp_hseq>\n",
      "      <Hsp_midline>SQETFSDLWKLLPEN</Hsp_midline>\n",
      "    </Hsp>\n",
      "  </Hit_hsps>\n",
      "</Hit>\n",
      "<Hit>\n",
      "  <Hit_num>50</Hit_num>\n",
      "  <Hit_id>gb|AKI70247.1|</Hit_id>\n",
      "  <Hit_def>TP53, partial [synthetic construct]</Hit_def>\n",
      "  <Hit_accession>AKI70247</Hit_accession>\n",
      "  <Hit_len>393</Hit_len>\n",
      "  <Hit_hsps>\n",
      "    <Hsp>\n",
      "      <Hsp_num>1</Hsp_num>\n",
      "      <Hsp_bit-score>35.039</Hsp_bit-score>\n",
      "      <Hsp_score>79</Hsp_score>\n",
      "      <Hsp_evalue>1.25132</Hsp_evalue>\n",
      "      <Hsp_query-from>1</Hsp_query-from>\n",
      "      <Hsp_query-to>15</Hsp_query-to>\n",
      "      <Hsp_hit-from>15</Hsp_hit-from>\n",
      "      <Hsp_hit-to>29</Hsp_hit-to>\n",
      "      <Hsp_query-frame>0</Hsp_query-frame>\n",
      "      <Hsp_hit-frame>0</Hsp_hit-frame>\n",
      "      <Hsp_identity>15</Hsp_identity>\n",
      "      <Hsp_positive>15</Hsp_positive>\n",
      "      <Hsp_gaps>0</Hsp_gaps>\n",
      "      <Hsp_align-len>15</Hsp_align-len>\n",
      "      <Hsp_qseq>SQETFSDLWKLLPEN</Hsp_qseq>\n",
      "      <Hsp_hseq>SQETFSDLWKLLPEN</Hsp_hseq>\n",
      "      <Hsp_midline>SQETFSDLWKLLPEN</Hsp_midline>\n",
      "    </Hsp>\n",
      "  </Hit_hsps>\n",
      "</Hit>\n",
      "<Hit>\n",
      "  <Hit_num>51</Hit_num>\n",
      "  <Hit_id>gb|AAB91535.1|</Hit_id>\n",
      "  <Hit_def>P53 [Macaca fascicularis]</Hit_def>\n",
      "  <Hit_accession>AAB91535</Hit_accession>\n",
      "  <Hit_len>393</Hit_len>\n",
      "  <Hit_hsps>\n",
      "    <Hsp>\n",
      "      <Hsp_num>1</Hsp_num>\n",
      "      <Hsp_bit-score>35.039</Hsp_bit-score>\n",
      "      <Hsp_score>79</Hsp_score>\n",
      "      <Hsp_evalue>1.25132</Hsp_evalue>\n",
      "      <Hsp_query-from>1</Hsp_query-from>\n",
      "      <Hsp_query-to>15</Hsp_query-to>\n",
      "      <Hsp_hit-from>15</Hsp_hit-from>\n",
      "      <Hsp_hit-to>29</Hsp_hit-to>\n",
      "      <Hsp_query-frame>0</Hsp_query-frame>\n",
      "      <Hsp_hit-frame>0</Hsp_hit-frame>\n",
      "      <Hsp_identity>15</Hsp_identity>\n",
      "      <Hsp_positive>15</Hsp_positive>\n",
      "      <Hsp_gaps>0</Hsp_gaps>\n",
      "      <Hsp_align-len>15</Hsp_align-len>\n",
      "      <Hsp_qseq>SQETFSDLWKLLPEN</Hsp_qseq>\n",
      "      <Hsp_hseq>SQETFSDLWKLLPEN</Hsp_hseq>\n",
      "      <Hsp_midline>SQETFSDLWKLLPEN</Hsp_midline>\n",
      "    </Hsp>\n",
      "  </Hit_hsps>\n",
      "</Hit>\n",
      "<Hit>\n",
      "  <Hit_num>52</Hit_num>\n",
      "  <Hit_id>ref|XP_031995228.1|</Hit_id>\n",
      "  <Hit_def>cellular tumor antigen p53 [Hylobates moloch]</Hit_def>\n",
      "  <Hit_accession>XP_031995228</Hit_accession>\n",
      "  <Hit_len>393</Hit_len>\n",
      "  <Hit_hsps>\n",
      "    <Hsp>\n",
      "      <Hsp_num>1</Hsp_num>\n",
      "      <Hsp_bit-score>35.039</Hsp_bit-score>\n",
      "      <Hsp_score>79</Hsp_score>\n",
      "      <Hsp_evalue>1.25132</Hsp_evalue>\n",
      "      <Hsp_query-from>1</Hsp_query-from>\n",
      "      <Hsp_query-to>15</Hsp_query-to>\n",
      "      <Hsp_hit-from>15</Hsp_hit-from>\n",
      "      <Hsp_hit-to>29</Hsp_hit-to>\n",
      "      <Hsp_query-frame>0</Hsp_query-frame>\n",
      "      <Hsp_hit-frame>0</Hsp_hit-frame>\n",
      "      <Hsp_identity>15</Hsp_identity>\n",
      "      <Hsp_positive>15</Hsp_positive>\n",
      "      <Hsp_gaps>0</Hsp_gaps>\n",
      "      <Hsp_align-len>15</Hsp_align-len>\n",
      "      <Hsp_qseq>SQETFSDLWKLLPEN</Hsp_qseq>\n",
      "      <Hsp_hseq>SQETFSDLWKLLPEN</Hsp_hseq>\n",
      "      <Hsp_midline>SQETFSDLWKLLPEN</Hsp_midline>\n",
      "    </Hsp>\n",
      "  </Hit_hsps>\n",
      "</Hit>\n",
      "<Hit>\n",
      "  <Hit_num>53</Hit_num>\n",
      "  <Hit_id>gb|AAR10356.1|</Hit_id>\n",
      "  <Hit_def>tumor suppressor p53, partial [Homo sapiens]</Hit_def>\n",
      "  <Hit_accession>AAR10356</Hit_accession>\n",
      "  <Hit_len>379</Hit_len>\n",
      "  <Hit_hsps>\n",
      "    <Hsp>\n",
      "      <Hsp_num>1</Hsp_num>\n",
      "      <Hsp_bit-score>35.039</Hsp_bit-score>\n",
      "      <Hsp_score>79</Hsp_score>\n",
      "      <Hsp_evalue>1.2536</Hsp_evalue>\n",
      "      <Hsp_query-from>1</Hsp_query-from>\n",
      "      <Hsp_query-to>15</Hsp_query-to>\n",
      "      <Hsp_hit-from>15</Hsp_hit-from>\n",
      "      <Hsp_hit-to>29</Hsp_hit-to>\n",
      "      <Hsp_query-frame>0</Hsp_query-frame>\n",
      "      <Hsp_hit-frame>0</Hsp_hit-frame>\n",
      "      <Hsp_identity>15</Hsp_identity>\n",
      "      <Hsp_positive>15</Hsp_positive>\n",
      "      <Hsp_gaps>0</Hsp_gaps>\n",
      "      <Hsp_align-len>15</Hsp_align-len>\n",
      "      <Hsp_qseq>SQETFSDLWKLLPEN</Hsp_qseq>\n",
      "      <Hsp_hseq>SQETFSDLWKLLPEN</Hsp_hseq>\n",
      "      <Hsp_midline>SQETFSDLWKLLPEN</Hsp_midline>\n",
      "    </Hsp>\n",
      "  </Hit_hsps>\n",
      "</Hit>\n",
      "<Hit>\n",
      "  <Hit_num>54</Hit_num>\n",
      "  <Hit_id>ref|XP_030717426.1|</Hit_id>\n",
      "  <Hit_def>cellular tumor antigen p53 isoform X2 [Globicephala melas]</Hit_def>\n",
      "  <Hit_accession>XP_030717426</Hit_accession>\n",
      "  <Hit_len>400</Hit_len>\n",
      "  <Hit_hsps>\n",
      "    <Hsp>\n",
      "      <Hsp_num>1</Hsp_num>\n",
      "      <Hsp_bit-score>35.039</Hsp_bit-score>\n",
      "      <Hsp_score>79</Hsp_score>\n",
      "      <Hsp_evalue>1.26253</Hsp_evalue>\n",
      "      <Hsp_query-from>1</Hsp_query-from>\n",
      "      <Hsp_query-to>15</Hsp_query-to>\n",
      "      <Hsp_hit-from>28</Hsp_hit-from>\n",
      "      <Hsp_hit-to>42</Hsp_hit-to>\n",
      "      <Hsp_query-frame>0</Hsp_query-frame>\n",
      "      <Hsp_hit-frame>0</Hsp_hit-frame>\n",
      "      <Hsp_identity>15</Hsp_identity>\n",
      "      <Hsp_positive>15</Hsp_positive>\n",
      "      <Hsp_gaps>0</Hsp_gaps>\n",
      "      <Hsp_align-len>15</Hsp_align-len>\n",
      "      <Hsp_qseq>SQETFSDLWKLLPEN</Hsp_qseq>\n",
      "      <Hsp_hseq>SQETFSDLWKLLPEN</Hsp_hseq>\n",
      "      <Hsp_midline>SQETFSDLWKLLPEN</Hsp_midline>\n",
      "    </Hsp>\n",
      "  </Hit_hsps>\n",
      "</Hit>\n",
      "<Hit>\n",
      "  <Hit_num>55</Hit_num>\n",
      "  <Hit_id>gb|PNI36854.1|</Hit_id>\n",
      "  <Hit_def>TP53 isoform 12, partial [Pan troglodytes] &gt;gb|PNJ12863.1| TP53 isoform 11, partial [Pongo abelii]</Hit_def>\n",
      "  <Hit_accession>PNI36854</Hit_accession>\n",
      "  <Hit_len>155</Hit_len>\n",
      "  <Hit_hsps>\n",
      "    <Hsp>\n",
      "      <Hsp_num>1</Hsp_num>\n",
      "      <Hsp_bit-score>35.039</Hsp_bit-score>\n",
      "      <Hsp_score>79</Hsp_score>\n",
      "      <Hsp_evalue>1.26292</Hsp_evalue>\n",
      "      <Hsp_query-from>1</Hsp_query-from>\n",
      "      <Hsp_query-to>15</Hsp_query-to>\n",
      "      <Hsp_hit-from>15</Hsp_hit-from>\n",
      "      <Hsp_hit-to>29</Hsp_hit-to>\n",
      "      <Hsp_query-frame>0</Hsp_query-frame>\n",
      "      <Hsp_hit-frame>0</Hsp_hit-frame>\n",
      "      <Hsp_identity>15</Hsp_identity>\n",
      "      <Hsp_positive>15</Hsp_positive>\n",
      "      <Hsp_gaps>0</Hsp_gaps>\n",
      "      <Hsp_align-len>15</Hsp_align-len>\n",
      "      <Hsp_qseq>SQETFSDLWKLLPEN</Hsp_qseq>\n",
      "      <Hsp_hseq>SQETFSDLWKLLPEN</Hsp_hseq>\n",
      "      <Hsp_midline>SQETFSDLWKLLPEN</Hsp_midline>\n",
      "    </Hsp>\n",
      "  </Hit_hsps>\n",
      "</Hit>\n",
      "<Hit>\n",
      "  <Hit_num>56</Hit_num>\n",
      "  <Hit_id>gb|EAW90140.1|</Hit_id>\n",
      "  <Hit_def>tumor protein p53 (Li-Fraumeni syndrome), isoform CRA_a [Homo sapiens]</Hit_def>\n",
      "  <Hit_accession>EAW90140</Hit_accession>\n",
      "  <Hit_len>394</Hit_len>\n",
      "  <Hit_hsps>\n",
      "    <Hsp>\n",
      "      <Hsp_num>1</Hsp_num>\n",
      "      <Hsp_bit-score>35.039</Hsp_bit-score>\n",
      "      <Hsp_score>79</Hsp_score>\n",
      "      <Hsp_evalue>1.26346</Hsp_evalue>\n",
      "      <Hsp_query-from>1</Hsp_query-from>\n",
      "      <Hsp_query-to>15</Hsp_query-to>\n",
      "      <Hsp_hit-from>15</Hsp_hit-from>\n",
      "      <Hsp_hit-to>29</Hsp_hit-to>\n",
      "      <Hsp_query-frame>0</Hsp_query-frame>\n",
      "      <Hsp_hit-frame>0</Hsp_hit-frame>\n",
      "      <Hsp_identity>15</Hsp_identity>\n",
      "      <Hsp_positive>15</Hsp_positive>\n",
      "      <Hsp_gaps>0</Hsp_gaps>\n",
      "      <Hsp_align-len>15</Hsp_align-len>\n",
      "      <Hsp_qseq>SQETFSDLWKLLPEN</Hsp_qseq>\n",
      "      <Hsp_hseq>SQETFSDLWKLLPEN</Hsp_hseq>\n",
      "      <Hsp_midline>SQETFSDLWKLLPEN</Hsp_midline>\n",
      "    </Hsp>\n",
      "  </Hit_hsps>\n",
      "</Hit>\n",
      "<Hit>\n",
      "  <Hit_num>57</Hit_num>\n",
      "  <Hit_id>gb|AYE20613.1|</Hit_id>\n",
      "  <Hit_def>mutant tumor protein p53 [Homo sapiens]</Hit_def>\n",
      "  <Hit_accession>AYE20613</Hit_accession>\n",
      "  <Hit_len>393</Hit_len>\n",
      "  <Hit_hsps>\n",
      "    <Hsp>\n",
      "      <Hsp_num>1</Hsp_num>\n",
      "      <Hsp_bit-score>35.039</Hsp_bit-score>\n",
      "      <Hsp_score>79</Hsp_score>\n",
      "      <Hsp_evalue>1.26362</Hsp_evalue>\n",
      "      <Hsp_query-from>1</Hsp_query-from>\n",
      "      <Hsp_query-to>15</Hsp_query-to>\n",
      "      <Hsp_hit-from>15</Hsp_hit-from>\n",
      "      <Hsp_hit-to>29</Hsp_hit-to>\n",
      "      <Hsp_query-frame>0</Hsp_query-frame>\n",
      "      <Hsp_hit-frame>0</Hsp_hit-frame>\n",
      "      <Hsp_identity>15</Hsp_identity>\n",
      "      <Hsp_positive>15</Hsp_positive>\n",
      "      <Hsp_gaps>0</Hsp_gaps>\n",
      "      <Hsp_align-len>15</Hsp_align-len>\n",
      "      <Hsp_qseq>SQETFSDLWKLLPEN</Hsp_qseq>\n",
      "      <Hsp_hseq>SQETFSDLWKLLPEN</Hsp_hseq>\n",
      "      <Hsp_midline>SQETFSDLWKLLPEN</Hsp_midline>\n",
      "    </Hsp>\n",
      "  </Hit_hsps>\n",
      "</Hit>\n",
      "<Hit>\n",
      "  <Hit_num>58</Hit_num>\n",
      "  <Hit_id>gb|AYE20617.1|</Hit_id>\n",
      "  <Hit_def>mutant tumor protein p53 [Homo sapiens]</Hit_def>\n",
      "  <Hit_accession>AYE20617</Hit_accession>\n",
      "  <Hit_len>393</Hit_len>\n",
      "  <Hit_hsps>\n",
      "    <Hsp>\n",
      "      <Hsp_num>1</Hsp_num>\n",
      "      <Hsp_bit-score>35.039</Hsp_bit-score>\n",
      "      <Hsp_score>79</Hsp_score>\n",
      "      <Hsp_evalue>1.26362</Hsp_evalue>\n",
      "      <Hsp_query-from>1</Hsp_query-from>\n",
      "      <Hsp_query-to>15</Hsp_query-to>\n",
      "      <Hsp_hit-from>15</Hsp_hit-from>\n",
      "      <Hsp_hit-to>29</Hsp_hit-to>\n",
      "      <Hsp_query-frame>0</Hsp_query-frame>\n",
      "      <Hsp_hit-frame>0</Hsp_hit-frame>\n",
      "      <Hsp_identity>15</Hsp_identity>\n",
      "      <Hsp_positive>15</Hsp_positive>\n",
      "      <Hsp_gaps>0</Hsp_gaps>\n",
      "      <Hsp_align-len>15</Hsp_align-len>\n",
      "      <Hsp_qseq>SQETFSDLWKLLPEN</Hsp_qseq>\n",
      "      <Hsp_hseq>SQETFSDLWKLLPEN</Hsp_hseq>\n",
      "      <Hsp_midline>SQETFSDLWKLLPEN</Hsp_midline>\n",
      "    </Hsp>\n",
      "  </Hit_hsps>\n",
      "</Hit>\n",
      "<Hit>\n",
      "  <Hit_num>59</Hit_num>\n",
      "  <Hit_id>gb|AYE20612.1|</Hit_id>\n",
      "  <Hit_def>mutant tumor protein p53 [Homo sapiens]</Hit_def>\n",
      "  <Hit_accession>AYE20612</Hit_accession>\n",
      "  <Hit_len>393</Hit_len>\n",
      "  <Hit_hsps>\n",
      "    <Hsp>\n",
      "      <Hsp_num>1</Hsp_num>\n",
      "      <Hsp_bit-score>35.039</Hsp_bit-score>\n",
      "      <Hsp_score>79</Hsp_score>\n",
      "      <Hsp_evalue>1.26362</Hsp_evalue>\n",
      "      <Hsp_query-from>1</Hsp_query-from>\n",
      "      <Hsp_query-to>15</Hsp_query-to>\n",
      "      <Hsp_hit-from>15</Hsp_hit-from>\n",
      "      <Hsp_hit-to>29</Hsp_hit-to>\n",
      "      <Hsp_query-frame>0</Hsp_query-frame>\n",
      "      <Hsp_hit-frame>0</Hsp_hit-frame>\n",
      "      <Hsp_identity>15</Hsp_identity>\n",
      "      <Hsp_positive>15</Hsp_positive>\n",
      "      <Hsp_gaps>0</Hsp_gaps>\n",
      "      <Hsp_align-len>15</Hsp_align-len>\n",
      "      <Hsp_qseq>SQETFSDLWKLLPEN</Hsp_qseq>\n",
      "      <Hsp_hseq>SQETFSDLWKLLPEN</Hsp_hseq>\n",
      "      <Hsp_midline>SQETFSDLWKLLPEN</Hsp_midline>\n",
      "    </Hsp>\n",
      "  </Hit_hsps>\n",
      "</Hit>\n",
      "<Hit>\n",
      "  <Hit_num>60</Hit_num>\n",
      "  <Hit_id>gb|AYE20610.1|</Hit_id>\n",
      "  <Hit_def>mutant tumor protein p53 [Homo sapiens]</Hit_def>\n",
      "  <Hit_accession>AYE20610</Hit_accession>\n",
      "  <Hit_len>393</Hit_len>\n",
      "  <Hit_hsps>\n",
      "    <Hsp>\n",
      "      <Hsp_num>1</Hsp_num>\n",
      "      <Hsp_bit-score>35.039</Hsp_bit-score>\n",
      "      <Hsp_score>79</Hsp_score>\n",
      "      <Hsp_evalue>1.26362</Hsp_evalue>\n",
      "      <Hsp_query-from>1</Hsp_query-from>\n",
      "      <Hsp_query-to>15</Hsp_query-to>\n",
      "      <Hsp_hit-from>15</Hsp_hit-from>\n",
      "      <Hsp_hit-to>29</Hsp_hit-to>\n",
      "      <Hsp_query-frame>0</Hsp_query-frame>\n",
      "      <Hsp_hit-frame>0</Hsp_hit-frame>\n",
      "      <Hsp_identity>15</Hsp_identity>\n",
      "      <Hsp_positive>15</Hsp_positive>\n",
      "      <Hsp_gaps>0</Hsp_gaps>\n",
      "      <Hsp_align-len>15</Hsp_align-len>\n",
      "      <Hsp_qseq>SQETFSDLWKLLPEN</Hsp_qseq>\n",
      "      <Hsp_hseq>SQETFSDLWKLLPEN</Hsp_hseq>\n",
      "      <Hsp_midline>SQETFSDLWKLLPEN</Hsp_midline>\n",
      "    </Hsp>\n",
      "  </Hit_hsps>\n",
      "</Hit>\n",
      "<Hit>\n",
      "  <Hit_num>61</Hit_num>\n",
      "  <Hit_id>dbj|BAC16799.1|</Hit_id>\n",
      "  <Hit_def>P53 [Homo sapiens]</Hit_def>\n",
      "  <Hit_accession>BAC16799</Hit_accession>\n",
      "  <Hit_len>393</Hit_len>\n",
      "  <Hit_hsps>\n",
      "    <Hsp>\n",
      "      <Hsp_num>1</Hsp_num>\n",
      "      <Hsp_bit-score>35.039</Hsp_bit-score>\n",
      "      <Hsp_score>79</Hsp_score>\n",
      "      <Hsp_evalue>1.26362</Hsp_evalue>\n",
      "      <Hsp_query-from>1</Hsp_query-from>\n",
      "      <Hsp_query-to>15</Hsp_query-to>\n",
      "      <Hsp_hit-from>15</Hsp_hit-from>\n",
      "      <Hsp_hit-to>29</Hsp_hit-to>\n",
      "      <Hsp_query-frame>0</Hsp_query-frame>\n",
      "      <Hsp_hit-frame>0</Hsp_hit-frame>\n",
      "      <Hsp_identity>15</Hsp_identity>\n",
      "      <Hsp_positive>15</Hsp_positive>\n",
      "      <Hsp_gaps>0</Hsp_gaps>\n",
      "      <Hsp_align-len>15</Hsp_align-len>\n",
      "      <Hsp_qseq>SQETFSDLWKLLPEN</Hsp_qseq>\n",
      "      <Hsp_hseq>SQETFSDLWKLLPEN</Hsp_hseq>\n",
      "      <Hsp_midline>SQETFSDLWKLLPEN</Hsp_midline>\n",
      "    </Hsp>\n",
      "  </Hit_hsps>\n",
      "</Hit>\n",
      "<Hit>\n",
      "  <Hit_num>62</Hit_num>\n",
      "  <Hit_id>ref|NP_000537.3|</Hit_id>\n",
      "  <Hit_def>cellular tumor antigen p53 isoform a [Homo sapiens] &gt;ref|NP_001119584.1| cellular tumor antigen p53 isoform a [Homo sapiens] &gt;ref|XP_001172077.2| cellular tumor antigen p53 isoform X2 [Pan troglodytes] &gt;ref|XP_004058559.2| cellular tumor antigen p53 isoform X3 [Gorilla gorilla gorilla] &gt;sp|P04637.4| RecName: Full=Cellular tumor antigen p53; AltName: Full=Antigen NY-CO-13; AltName: Full=Phosphoprotein p53; AltName: Full=Tumor suppressor p53 [Homo sapiens] &gt;pdb|6XRE|M Chain M, Cellular tumor antigen p53 [Homo sapiens] &gt;gb|AAF64408.1| tumor suppressor protein p53 [Expression vector Ad5CMV-p53] &gt;gb|AAA59988.1| phosphoprotein p53 [Homo sapiens] &gt;gb|AFN61604.1| tumor suppressor p53 [Homo sapiens] &gt;gb|AFN61605.1| tumor suppressor p53 [Homo sapiens] &gt;gb|AYE20619.1| mutant tumor protein p53 [Homo sapiens]</Hit_def>\n",
      "  <Hit_accession>NP_000537</Hit_accession>\n",
      "  <Hit_len>393</Hit_len>\n",
      "  <Hit_hsps>\n",
      "    <Hsp>\n",
      "      <Hsp_num>1</Hsp_num>\n",
      "      <Hsp_bit-score>35.039</Hsp_bit-score>\n",
      "      <Hsp_score>79</Hsp_score>\n",
      "      <Hsp_evalue>1.26362</Hsp_evalue>\n",
      "      <Hsp_query-from>1</Hsp_query-from>\n",
      "      <Hsp_query-to>15</Hsp_query-to>\n",
      "      <Hsp_hit-from>15</Hsp_hit-from>\n",
      "      <Hsp_hit-to>29</Hsp_hit-to>\n",
      "      <Hsp_query-frame>0</Hsp_query-frame>\n",
      "      <Hsp_hit-frame>0</Hsp_hit-frame>\n",
      "      <Hsp_identity>15</Hsp_identity>\n",
      "      <Hsp_positive>15</Hsp_positive>\n",
      "      <Hsp_gaps>0</Hsp_gaps>\n",
      "      <Hsp_align-len>15</Hsp_align-len>\n",
      "      <Hsp_qseq>SQETFSDLWKLLPEN</Hsp_qseq>\n",
      "      <Hsp_hseq>SQETFSDLWKLLPEN</Hsp_hseq>\n",
      "      <Hsp_midline>SQETFSDLWKLLPEN</Hsp_midline>\n",
      "    </Hsp>\n",
      "  </Hit_hsps>\n",
      "</Hit>\n",
      "<Hit>\n",
      "  <Hit_num>63</Hit_num>\n",
      "  <Hit_id>gb|AAA59989.1|</Hit_id>\n",
      "  <Hit_def>p53 cellular tumor antigen [Homo sapiens] &gt;gb|AYE20625.1| mutant tumor protein p53 [Homo sapiens]</Hit_def>\n",
      "  <Hit_accession>AAA59989</Hit_accession>\n",
      "  <Hit_len>393</Hit_len>\n",
      "  <Hit_hsps>\n",
      "    <Hsp>\n",
      "      <Hsp_num>1</Hsp_num>\n",
      "      <Hsp_bit-score>35.039</Hsp_bit-score>\n",
      "      <Hsp_score>79</Hsp_score>\n",
      "      <Hsp_evalue>1.26362</Hsp_evalue>\n",
      "      <Hsp_query-from>1</Hsp_query-from>\n",
      "      <Hsp_query-to>15</Hsp_query-to>\n",
      "      <Hsp_hit-from>15</Hsp_hit-from>\n",
      "      <Hsp_hit-to>29</Hsp_hit-to>\n",
      "      <Hsp_query-frame>0</Hsp_query-frame>\n",
      "      <Hsp_hit-frame>0</Hsp_hit-frame>\n",
      "      <Hsp_identity>15</Hsp_identity>\n",
      "      <Hsp_positive>15</Hsp_positive>\n",
      "      <Hsp_gaps>0</Hsp_gaps>\n",
      "      <Hsp_align-len>15</Hsp_align-len>\n",
      "      <Hsp_qseq>SQETFSDLWKLLPEN</Hsp_qseq>\n",
      "      <Hsp_hseq>SQETFSDLWKLLPEN</Hsp_hseq>\n",
      "      <Hsp_midline>SQETFSDLWKLLPEN</Hsp_midline>\n",
      "    </Hsp>\n",
      "  </Hit_hsps>\n",
      "</Hit>\n",
      "<Hit>\n",
      "  <Hit_num>64</Hit_num>\n",
      "  <Hit_id>ref|XP_033040608.1|</Hit_id>\n",
      "  <Hit_def>cellular tumor antigen p53 isoform X1 [Trachypithecus francoisi]</Hit_def>\n",
      "  <Hit_accession>XP_033040608</Hit_accession>\n",
      "  <Hit_len>393</Hit_len>\n",
      "  <Hit_hsps>\n",
      "    <Hsp>\n",
      "      <Hsp_num>1</Hsp_num>\n",
      "      <Hsp_bit-score>35.039</Hsp_bit-score>\n",
      "      <Hsp_score>79</Hsp_score>\n",
      "      <Hsp_evalue>1.26362</Hsp_evalue>\n",
      "      <Hsp_query-from>1</Hsp_query-from>\n",
      "      <Hsp_query-to>15</Hsp_query-to>\n",
      "      <Hsp_hit-from>15</Hsp_hit-from>\n",
      "      <Hsp_hit-to>29</Hsp_hit-to>\n",
      "      <Hsp_query-frame>0</Hsp_query-frame>\n",
      "      <Hsp_hit-frame>0</Hsp_hit-frame>\n",
      "      <Hsp_identity>15</Hsp_identity>\n",
      "      <Hsp_positive>15</Hsp_positive>\n",
      "      <Hsp_gaps>0</Hsp_gaps>\n",
      "      <Hsp_align-len>15</Hsp_align-len>\n",
      "      <Hsp_qseq>SQETFSDLWKLLPEN</Hsp_qseq>\n",
      "      <Hsp_hseq>SQETFSDLWKLLPEN</Hsp_hseq>\n",
      "      <Hsp_midline>SQETFSDLWKLLPEN</Hsp_midline>\n",
      "    </Hsp>\n",
      "  </Hit_hsps>\n",
      "</Hit>\n",
      "<Hit>\n",
      "  <Hit_num>65</Hit_num>\n",
      "  <Hit_id>gb|AFN61606.1|</Hit_id>\n",
      "  <Hit_def>tumor suppressor p53 [Homo sapiens]</Hit_def>\n",
      "  <Hit_accession>AFN61606</Hit_accession>\n",
      "  <Hit_len>393</Hit_len>\n",
      "  <Hit_hsps>\n",
      "    <Hsp>\n",
      "      <Hsp_num>1</Hsp_num>\n",
      "      <Hsp_bit-score>35.039</Hsp_bit-score>\n",
      "      <Hsp_score>79</Hsp_score>\n",
      "      <Hsp_evalue>1.26362</Hsp_evalue>\n",
      "      <Hsp_query-from>1</Hsp_query-from>\n",
      "      <Hsp_query-to>15</Hsp_query-to>\n",
      "      <Hsp_hit-from>15</Hsp_hit-from>\n",
      "      <Hsp_hit-to>29</Hsp_hit-to>\n",
      "      <Hsp_query-frame>0</Hsp_query-frame>\n",
      "      <Hsp_hit-frame>0</Hsp_hit-frame>\n",
      "      <Hsp_identity>15</Hsp_identity>\n",
      "      <Hsp_positive>15</Hsp_positive>\n",
      "      <Hsp_gaps>0</Hsp_gaps>\n",
      "      <Hsp_align-len>15</Hsp_align-len>\n",
      "      <Hsp_qseq>SQETFSDLWKLLPEN</Hsp_qseq>\n",
      "      <Hsp_hseq>SQETFSDLWKLLPEN</Hsp_hseq>\n",
      "      <Hsp_midline>SQETFSDLWKLLPEN</Hsp_midline>\n",
      "    </Hsp>\n",
      "  </Hit_hsps>\n",
      "</Hit>\n",
      "<Hit>\n",
      "  <Hit_num>66</Hit_num>\n",
      "  <Hit_id>ref|XP_033040609.1|</Hit_id>\n",
      "  <Hit_def>cellular tumor antigen p53 isoform X2 [Trachypithecus francoisi]</Hit_def>\n",
      "  <Hit_accession>XP_033040609</Hit_accession>\n",
      "  <Hit_len>344</Hit_len>\n",
      "  <Hit_hsps>\n",
      "    <Hsp>\n",
      "      <Hsp_num>1</Hsp_num>\n",
      "      <Hsp_bit-score>35.039</Hsp_bit-score>\n",
      "      <Hsp_score>79</Hsp_score>\n",
      "      <Hsp_evalue>1.27247</Hsp_evalue>\n",
      "      <Hsp_query-from>1</Hsp_query-from>\n",
      "      <Hsp_query-to>15</Hsp_query-to>\n",
      "      <Hsp_hit-from>15</Hsp_hit-from>\n",
      "      <Hsp_hit-to>29</Hsp_hit-to>\n",
      "      <Hsp_query-frame>0</Hsp_query-frame>\n",
      "      <Hsp_hit-frame>0</Hsp_hit-frame>\n",
      "      <Hsp_identity>15</Hsp_identity>\n",
      "      <Hsp_positive>15</Hsp_positive>\n",
      "      <Hsp_gaps>0</Hsp_gaps>\n",
      "      <Hsp_align-len>15</Hsp_align-len>\n",
      "      <Hsp_qseq>SQETFSDLWKLLPEN</Hsp_qseq>\n",
      "      <Hsp_hseq>SQETFSDLWKLLPEN</Hsp_hseq>\n",
      "      <Hsp_midline>SQETFSDLWKLLPEN</Hsp_midline>\n",
      "    </Hsp>\n",
      "  </Hit_hsps>\n",
      "</Hit>\n",
      "<Hit>\n",
      "  <Hit_num>67</Hit_num>\n",
      "  <Hit_id>gb|PNI36859.1|</Hit_id>\n",
      "  <Hit_def>TP53 isoform 17 [Pan troglodytes]</Hit_def>\n",
      "  <Hit_accession>PNI36859</Hit_accession>\n",
      "  <Hit_len>343</Hit_len>\n",
      "  <Hit_hsps>\n",
      "    <Hsp>\n",
      "      <Hsp_num>1</Hsp_num>\n",
      "      <Hsp_bit-score>35.039</Hsp_bit-score>\n",
      "      <Hsp_score>79</Hsp_score>\n",
      "      <Hsp_evalue>1.27268</Hsp_evalue>\n",
      "      <Hsp_query-from>1</Hsp_query-from>\n",
      "      <Hsp_query-to>15</Hsp_query-to>\n",
      "      <Hsp_hit-from>15</Hsp_hit-from>\n",
      "      <Hsp_hit-to>29</Hsp_hit-to>\n",
      "      <Hsp_query-frame>0</Hsp_query-frame>\n",
      "      <Hsp_hit-frame>0</Hsp_hit-frame>\n",
      "      <Hsp_identity>15</Hsp_identity>\n",
      "      <Hsp_positive>15</Hsp_positive>\n",
      "      <Hsp_gaps>0</Hsp_gaps>\n",
      "      <Hsp_align-len>15</Hsp_align-len>\n",
      "      <Hsp_qseq>SQETFSDLWKLLPEN</Hsp_qseq>\n",
      "      <Hsp_hseq>SQETFSDLWKLLPEN</Hsp_hseq>\n",
      "      <Hsp_midline>SQETFSDLWKLLPEN</Hsp_midline>\n",
      "    </Hsp>\n",
      "  </Hit_hsps>\n",
      "</Hit>\n",
      "<Hit>\n",
      "  <Hit_num>68</Hit_num>\n",
      "  <Hit_id>ref|XP_011813948.1|</Hit_id>\n",
      "  <Hit_def>PREDICTED: cellular tumor antigen p53 isoform X1 [Colobus angolensis palliatus] &gt;ref|XP_011813949.1| PREDICTED: cellular tumor antigen p53 isoform X1 [Colobus angolensis palliatus]</Hit_def>\n",
      "  <Hit_accession>XP_011813948</Hit_accession>\n",
      "  <Hit_len>393</Hit_len>\n",
      "  <Hit_hsps>\n",
      "    <Hsp>\n",
      "      <Hsp_num>1</Hsp_num>\n",
      "      <Hsp_bit-score>35.039</Hsp_bit-score>\n",
      "      <Hsp_score>79</Hsp_score>\n",
      "      <Hsp_evalue>1.27603</Hsp_evalue>\n",
      "      <Hsp_query-from>1</Hsp_query-from>\n",
      "      <Hsp_query-to>15</Hsp_query-to>\n",
      "      <Hsp_hit-from>15</Hsp_hit-from>\n",
      "      <Hsp_hit-to>29</Hsp_hit-to>\n",
      "      <Hsp_query-frame>0</Hsp_query-frame>\n",
      "      <Hsp_hit-frame>0</Hsp_hit-frame>\n",
      "      <Hsp_identity>15</Hsp_identity>\n",
      "      <Hsp_positive>15</Hsp_positive>\n",
      "      <Hsp_gaps>0</Hsp_gaps>\n",
      "      <Hsp_align-len>15</Hsp_align-len>\n",
      "      <Hsp_qseq>SQETFSDLWKLLPEN</Hsp_qseq>\n",
      "      <Hsp_hseq>SQETFSDLWKLLPEN</Hsp_hseq>\n",
      "      <Hsp_midline>SQETFSDLWKLLPEN</Hsp_midline>\n",
      "    </Hsp>\n",
      "  </Hit_hsps>\n",
      "</Hit>\n",
      "<Hit>\n",
      "  <Hit_num>69</Hit_num>\n",
      "  <Hit_id>ref|NP_001040616.1|</Hit_id>\n",
      "  <Hit_def>cellular tumor antigen p53 [Macaca mulatta] &gt;ref|XP_005582844.1| PREDICTED: cellular tumor antigen p53 isoform X1 [Macaca fascicularis] &gt;ref|XP_028691546.1| cellular tumor antigen p53 isoform X1 [Macaca mulatta] &gt;sp|P56423.2| RecName: Full=Cellular tumor antigen p53; AltName: Full=Tumor suppressor p53 [Macaca fascicularis] &gt;sp|P56424.1| RecName: Full=Cellular tumor antigen p53; AltName: Full=Tumor suppressor p53 [Macaca mulatta] &gt;sp|P61260.1| RecName: Full=Cellular tumor antigen p53; AltName: Full=Tumor suppressor p53 [Macaca fuscata fuscata] &gt;gb|AAN64028.1| p53 [Macaca fuscata] &gt;gb|AAA17994.1| p53 [Macaca mulatta] &gt;gb|AAB91534.1| P53 [Macaca mulatta] &gt;gb|AAN64027.1| p53 [Macaca fascicularis] &gt;gb|ADO78765.1| cellular tumor antigen p53 [Macaca mulatta]</Hit_def>\n",
      "  <Hit_accession>NP_001040616</Hit_accession>\n",
      "  <Hit_len>393</Hit_len>\n",
      "  <Hit_hsps>\n",
      "    <Hsp>\n",
      "      <Hsp_num>1</Hsp_num>\n",
      "      <Hsp_bit-score>35.039</Hsp_bit-score>\n",
      "      <Hsp_score>79</Hsp_score>\n",
      "      <Hsp_evalue>1.27603</Hsp_evalue>\n",
      "      <Hsp_query-from>1</Hsp_query-from>\n",
      "      <Hsp_query-to>15</Hsp_query-to>\n",
      "      <Hsp_hit-from>15</Hsp_hit-from>\n",
      "      <Hsp_hit-to>29</Hsp_hit-to>\n",
      "      <Hsp_query-frame>0</Hsp_query-frame>\n",
      "      <Hsp_hit-frame>0</Hsp_hit-frame>\n",
      "      <Hsp_identity>15</Hsp_identity>\n",
      "      <Hsp_positive>15</Hsp_positive>\n",
      "      <Hsp_gaps>0</Hsp_gaps>\n",
      "      <Hsp_align-len>15</Hsp_align-len>\n",
      "      <Hsp_qseq>SQETFSDLWKLLPEN</Hsp_qseq>\n",
      "      <Hsp_hseq>SQETFSDLWKLLPEN</Hsp_hseq>\n",
      "      <Hsp_midline>SQETFSDLWKLLPEN</Hsp_midline>\n",
      "    </Hsp>\n",
      "  </Hit_hsps>\n",
      "</Hit>\n",
      "<Hit>\n",
      "  <Hit_num>70</Hit_num>\n",
      "  <Hit_id>gb|AAA61212.1|</Hit_id>\n",
      "  <Hit_def>p53 antigen [Homo sapiens]</Hit_def>\n",
      "  <Hit_accession>AAA61212</Hit_accession>\n",
      "  <Hit_len>393</Hit_len>\n",
      "  <Hit_hsps>\n",
      "    <Hsp>\n",
      "      <Hsp_num>1</Hsp_num>\n",
      "      <Hsp_bit-score>35.039</Hsp_bit-score>\n",
      "      <Hsp_score>79</Hsp_score>\n",
      "      <Hsp_evalue>1.27603</Hsp_evalue>\n",
      "      <Hsp_query-from>1</Hsp_query-from>\n",
      "      <Hsp_query-to>15</Hsp_query-to>\n",
      "      <Hsp_hit-from>15</Hsp_hit-from>\n",
      "      <Hsp_hit-to>29</Hsp_hit-to>\n",
      "      <Hsp_query-frame>0</Hsp_query-frame>\n",
      "      <Hsp_hit-frame>0</Hsp_hit-frame>\n",
      "      <Hsp_identity>15</Hsp_identity>\n",
      "      <Hsp_positive>15</Hsp_positive>\n",
      "      <Hsp_gaps>0</Hsp_gaps>\n",
      "      <Hsp_align-len>15</Hsp_align-len>\n",
      "      <Hsp_qseq>SQETFSDLWKLLPEN</Hsp_qseq>\n",
      "      <Hsp_hseq>SQETFSDLWKLLPEN</Hsp_hseq>\n",
      "      <Hsp_midline>SQETFSDLWKLLPEN</Hsp_midline>\n",
      "    </Hsp>\n",
      "  </Hit_hsps>\n",
      "</Hit>\n",
      "<Hit>\n",
      "  <Hit_num>71</Hit_num>\n",
      "  <Hit_id>ref|XP_030717427.1|</Hit_id>\n",
      "  <Hit_def>cellular tumor antigen p53 isoform X3 [Globicephala melas]</Hit_def>\n",
      "  <Hit_accession>XP_030717427</Hit_accession>\n",
      "  <Hit_len>393</Hit_len>\n",
      "  <Hit_hsps>\n",
      "    <Hsp>\n",
      "      <Hsp_num>1</Hsp_num>\n",
      "      <Hsp_bit-score>35.039</Hsp_bit-score>\n",
      "      <Hsp_score>79</Hsp_score>\n",
      "      <Hsp_evalue>1.27603</Hsp_evalue>\n",
      "      <Hsp_query-from>1</Hsp_query-from>\n",
      "      <Hsp_query-to>15</Hsp_query-to>\n",
      "      <Hsp_hit-from>15</Hsp_hit-from>\n",
      "      <Hsp_hit-to>29</Hsp_hit-to>\n",
      "      <Hsp_query-frame>0</Hsp_query-frame>\n",
      "      <Hsp_hit-frame>0</Hsp_hit-frame>\n",
      "      <Hsp_identity>15</Hsp_identity>\n",
      "      <Hsp_positive>15</Hsp_positive>\n",
      "      <Hsp_gaps>0</Hsp_gaps>\n",
      "      <Hsp_align-len>15</Hsp_align-len>\n",
      "      <Hsp_qseq>SQETFSDLWKLLPEN</Hsp_qseq>\n",
      "      <Hsp_hseq>SQETFSDLWKLLPEN</Hsp_hseq>\n",
      "      <Hsp_midline>SQETFSDLWKLLPEN</Hsp_midline>\n",
      "    </Hsp>\n",
      "  </Hit_hsps>\n",
      "</Hit>\n",
      "<Hit>\n",
      "  <Hit_num>72</Hit_num>\n",
      "  <Hit_id>emb|CAA42631.1|</Hit_id>\n",
      "  <Hit_def>p53 transformation suppressor, partial [Homo sapiens]</Hit_def>\n",
      "  <Hit_accession>CAA42631</Hit_accession>\n",
      "  <Hit_len>393</Hit_len>\n",
      "  <Hit_hsps>\n",
      "    <Hsp>\n",
      "      <Hsp_num>1</Hsp_num>\n",
      "      <Hsp_bit-score>35.039</Hsp_bit-score>\n",
      "      <Hsp_score>79</Hsp_score>\n",
      "      <Hsp_evalue>1.27603</Hsp_evalue>\n",
      "      <Hsp_query-from>1</Hsp_query-from>\n",
      "      <Hsp_query-to>15</Hsp_query-to>\n",
      "      <Hsp_hit-from>15</Hsp_hit-from>\n",
      "      <Hsp_hit-to>29</Hsp_hit-to>\n",
      "      <Hsp_query-frame>0</Hsp_query-frame>\n",
      "      <Hsp_hit-frame>0</Hsp_hit-frame>\n",
      "      <Hsp_identity>15</Hsp_identity>\n",
      "      <Hsp_positive>15</Hsp_positive>\n",
      "      <Hsp_gaps>0</Hsp_gaps>\n",
      "      <Hsp_align-len>15</Hsp_align-len>\n",
      "      <Hsp_qseq>SQETFSDLWKLLPEN</Hsp_qseq>\n",
      "      <Hsp_hseq>SQETFSDLWKLLPEN</Hsp_hseq>\n",
      "      <Hsp_midline>SQETFSDLWKLLPEN</Hsp_midline>\n",
      "    </Hsp>\n",
      "  </Hit_hsps>\n",
      "</Hit>\n",
      "<Hit>\n",
      "  <Hit_num>73</Hit_num>\n",
      "  <Hit_id>gb|AYE20611.1|</Hit_id>\n",
      "  <Hit_def>mutant tumor protein p53 [Homo sapiens]</Hit_def>\n",
      "  <Hit_accession>AYE20611</Hit_accession>\n",
      "  <Hit_len>393</Hit_len>\n",
      "  <Hit_hsps>\n",
      "    <Hsp>\n",
      "      <Hsp_num>1</Hsp_num>\n",
      "      <Hsp_bit-score>35.039</Hsp_bit-score>\n",
      "      <Hsp_score>79</Hsp_score>\n",
      "      <Hsp_evalue>1.27603</Hsp_evalue>\n",
      "      <Hsp_query-from>1</Hsp_query-from>\n",
      "      <Hsp_query-to>15</Hsp_query-to>\n",
      "      <Hsp_hit-from>15</Hsp_hit-from>\n",
      "      <Hsp_hit-to>29</Hsp_hit-to>\n",
      "      <Hsp_query-frame>0</Hsp_query-frame>\n",
      "      <Hsp_hit-frame>0</Hsp_hit-frame>\n",
      "      <Hsp_identity>15</Hsp_identity>\n",
      "      <Hsp_positive>15</Hsp_positive>\n",
      "      <Hsp_gaps>0</Hsp_gaps>\n",
      "      <Hsp_align-len>15</Hsp_align-len>\n",
      "      <Hsp_qseq>SQETFSDLWKLLPEN</Hsp_qseq>\n",
      "      <Hsp_hseq>SQETFSDLWKLLPEN</Hsp_hseq>\n",
      "      <Hsp_midline>SQETFSDLWKLLPEN</Hsp_midline>\n",
      "    </Hsp>\n",
      "  </Hit_hsps>\n",
      "</Hit>\n",
      "<Hit>\n",
      "  <Hit_num>74</Hit_num>\n",
      "  <Hit_id>ref|NP_001166211.1|</Hit_id>\n",
      "  <Hit_def>cellular tumor antigen p53 [Cavia porcellus] &gt;sp|Q9WUR6.1| RecName: Full=Cellular tumor antigen p53; AltName: Full=Tumor suppressor p53 [Cavia porcellus] &gt;emb|CAB43196.1| p53 protein [Cavia porcellus]</Hit_def>\n",
      "  <Hit_accession>NP_001166211</Hit_accession>\n",
      "  <Hit_len>391</Hit_len>\n",
      "  <Hit_hsps>\n",
      "    <Hsp>\n",
      "      <Hsp_num>1</Hsp_num>\n",
      "      <Hsp_bit-score>35.039</Hsp_bit-score>\n",
      "      <Hsp_score>79</Hsp_score>\n",
      "      <Hsp_evalue>1.27635</Hsp_evalue>\n",
      "      <Hsp_query-from>1</Hsp_query-from>\n",
      "      <Hsp_query-to>15</Hsp_query-to>\n",
      "      <Hsp_hit-from>15</Hsp_hit-from>\n",
      "      <Hsp_hit-to>29</Hsp_hit-to>\n",
      "      <Hsp_query-frame>0</Hsp_query-frame>\n",
      "      <Hsp_hit-frame>0</Hsp_hit-frame>\n",
      "      <Hsp_identity>15</Hsp_identity>\n",
      "      <Hsp_positive>15</Hsp_positive>\n",
      "      <Hsp_gaps>0</Hsp_gaps>\n",
      "      <Hsp_align-len>15</Hsp_align-len>\n",
      "      <Hsp_qseq>SQETFSDLWKLLPEN</Hsp_qseq>\n",
      "      <Hsp_hseq>SQETFSDLWKLLPEN</Hsp_hseq>\n",
      "      <Hsp_midline>SQETFSDLWKLLPEN</Hsp_midline>\n",
      "    </Hsp>\n",
      "  </Hit_hsps>\n",
      "</Hit>\n",
      "<Hit>\n",
      "  <Hit_num>75</Hit_num>\n",
      "  <Hit_id>ref|XP_007457852.1|</Hit_id>\n",
      "  <Hit_def>PREDICTED: cellular tumor antigen p53 isoform X1 [Lipotes vexillifer]</Hit_def>\n",
      "  <Hit_accession>XP_007457852</Hit_accession>\n",
      "  <Hit_len>387</Hit_len>\n",
      "  <Hit_hsps>\n",
      "    <Hsp>\n",
      "      <Hsp_num>1</Hsp_num>\n",
      "      <Hsp_bit-score>35.039</Hsp_bit-score>\n",
      "      <Hsp_score>79</Hsp_score>\n",
      "      <Hsp_evalue>1.277</Hsp_evalue>\n",
      "      <Hsp_query-from>1</Hsp_query-from>\n",
      "      <Hsp_query-to>15</Hsp_query-to>\n",
      "      <Hsp_hit-from>15</Hsp_hit-from>\n",
      "      <Hsp_hit-to>29</Hsp_hit-to>\n",
      "      <Hsp_query-frame>0</Hsp_query-frame>\n",
      "      <Hsp_hit-frame>0</Hsp_hit-frame>\n",
      "      <Hsp_identity>15</Hsp_identity>\n",
      "      <Hsp_positive>15</Hsp_positive>\n",
      "      <Hsp_gaps>0</Hsp_gaps>\n",
      "      <Hsp_align-len>15</Hsp_align-len>\n",
      "      <Hsp_qseq>SQETFSDLWKLLPEN</Hsp_qseq>\n",
      "      <Hsp_hseq>SQETFSDLWKLLPEN</Hsp_hseq>\n",
      "      <Hsp_midline>SQETFSDLWKLLPEN</Hsp_midline>\n",
      "    </Hsp>\n",
      "  </Hit_hsps>\n",
      "</Hit>\n",
      "<Hit>\n",
      "  <Hit_num>76</Hit_num>\n",
      "  <Hit_id>ref|XP_012631512.1|</Hit_id>\n",
      "  <Hit_def>cellular tumor antigen p53 [Microcebus murinus] &gt;ref|XP_012631521.1| cellular tumor antigen p53 [Microcebus murinus]</Hit_def>\n",
      "  <Hit_accession>XP_012631512</Hit_accession>\n",
      "  <Hit_len>385</Hit_len>\n",
      "  <Hit_hsps>\n",
      "    <Hsp>\n",
      "      <Hsp_num>1</Hsp_num>\n",
      "      <Hsp_bit-score>35.039</Hsp_bit-score>\n",
      "      <Hsp_score>79</Hsp_score>\n",
      "      <Hsp_evalue>1.27734</Hsp_evalue>\n",
      "      <Hsp_query-from>1</Hsp_query-from>\n",
      "      <Hsp_query-to>15</Hsp_query-to>\n",
      "      <Hsp_hit-from>15</Hsp_hit-from>\n",
      "      <Hsp_hit-to>29</Hsp_hit-to>\n",
      "      <Hsp_query-frame>0</Hsp_query-frame>\n",
      "      <Hsp_hit-frame>0</Hsp_hit-frame>\n",
      "      <Hsp_identity>15</Hsp_identity>\n",
      "      <Hsp_positive>15</Hsp_positive>\n",
      "      <Hsp_gaps>0</Hsp_gaps>\n",
      "      <Hsp_align-len>15</Hsp_align-len>\n",
      "      <Hsp_qseq>SQETFSDLWKLLPEN</Hsp_qseq>\n",
      "      <Hsp_hseq>SQETFSDLWKLLPEN</Hsp_hseq>\n",
      "      <Hsp_midline>SQETFSDLWKLLPEN</Hsp_midline>\n",
      "    </Hsp>\n",
      "  </Hit_hsps>\n",
      "</Hit>\n",
      "<Hit>\n",
      "  <Hit_num>77</Hit_num>\n",
      "  <Hit_id>ref|XP_002747994.1|</Hit_id>\n",
      "  <Hit_def>cellular tumor antigen p53 isoform X1 [Callithrix jacchus]</Hit_def>\n",
      "  <Hit_accession>XP_002747994</Hit_accession>\n",
      "  <Hit_len>395</Hit_len>\n",
      "  <Hit_hsps>\n",
      "    <Hsp>\n",
      "      <Hsp_num>1</Hsp_num>\n",
      "      <Hsp_bit-score>35.039</Hsp_bit-score>\n",
      "      <Hsp_score>79</Hsp_score>\n",
      "      <Hsp_evalue>1.28825</Hsp_evalue>\n",
      "      <Hsp_query-from>1</Hsp_query-from>\n",
      "      <Hsp_query-to>15</Hsp_query-to>\n",
      "      <Hsp_hit-from>15</Hsp_hit-from>\n",
      "      <Hsp_hit-to>29</Hsp_hit-to>\n",
      "      <Hsp_query-frame>0</Hsp_query-frame>\n",
      "      <Hsp_hit-frame>0</Hsp_hit-frame>\n",
      "      <Hsp_identity>15</Hsp_identity>\n",
      "      <Hsp_positive>15</Hsp_positive>\n",
      "      <Hsp_gaps>0</Hsp_gaps>\n",
      "      <Hsp_align-len>15</Hsp_align-len>\n",
      "      <Hsp_qseq>SQETFSDLWKLLPEN</Hsp_qseq>\n",
      "      <Hsp_hseq>SQETFSDLWKLLPEN</Hsp_hseq>\n",
      "      <Hsp_midline>SQETFSDLWKLLPEN</Hsp_midline>\n",
      "    </Hsp>\n",
      "  </Hit_hsps>\n",
      "</Hit>\n",
      "<Hit>\n",
      "  <Hit_num>78</Hit_num>\n",
      "  <Hit_id>gb|AAX42852.1|</Hit_id>\n",
      "  <Hit_def>tumor protein p53, partial [synthetic construct] &gt;gb|AAX43170.1| tumor protein p53, partial [synthetic construct]</Hit_def>\n",
      "  <Hit_accession>AAX42852</Hit_accession>\n",
      "  <Hit_len>394</Hit_len>\n",
      "  <Hit_hsps>\n",
      "    <Hsp>\n",
      "      <Hsp_num>1</Hsp_num>\n",
      "      <Hsp_bit-score>35.039</Hsp_bit-score>\n",
      "      <Hsp_score>79</Hsp_score>\n",
      "      <Hsp_evalue>1.28841</Hsp_evalue>\n",
      "      <Hsp_query-from>1</Hsp_query-from>\n",
      "      <Hsp_query-to>15</Hsp_query-to>\n",
      "      <Hsp_hit-from>15</Hsp_hit-from>\n",
      "      <Hsp_hit-to>29</Hsp_hit-to>\n",
      "      <Hsp_query-frame>0</Hsp_query-frame>\n",
      "      <Hsp_hit-frame>0</Hsp_hit-frame>\n",
      "      <Hsp_identity>15</Hsp_identity>\n",
      "      <Hsp_positive>15</Hsp_positive>\n",
      "      <Hsp_gaps>0</Hsp_gaps>\n",
      "      <Hsp_align-len>15</Hsp_align-len>\n",
      "      <Hsp_qseq>SQETFSDLWKLLPEN</Hsp_qseq>\n",
      "      <Hsp_hseq>SQETFSDLWKLLPEN</Hsp_hseq>\n",
      "      <Hsp_midline>SQETFSDLWKLLPEN</Hsp_midline>\n",
      "    </Hsp>\n",
      "  </Hit_hsps>\n",
      "</Hit>\n",
      "<Hit>\n",
      "  <Hit_num>79</Hit_num>\n",
      "  <Hit_id>gb|AYE20614.1|</Hit_id>\n",
      "  <Hit_def>mutant tumor protein p53 [Homo sapiens]</Hit_def>\n",
      "  <Hit_accession>AYE20614</Hit_accession>\n",
      "  <Hit_len>393</Hit_len>\n",
      "  <Hit_hsps>\n",
      "    <Hsp>\n",
      "      <Hsp_num>1</Hsp_num>\n",
      "      <Hsp_bit-score>35.039</Hsp_bit-score>\n",
      "      <Hsp_score>79</Hsp_score>\n",
      "      <Hsp_evalue>1.28857</Hsp_evalue>\n",
      "      <Hsp_query-from>1</Hsp_query-from>\n",
      "      <Hsp_query-to>15</Hsp_query-to>\n",
      "      <Hsp_hit-from>15</Hsp_hit-from>\n",
      "      <Hsp_hit-to>29</Hsp_hit-to>\n",
      "      <Hsp_query-frame>0</Hsp_query-frame>\n",
      "      <Hsp_hit-frame>0</Hsp_hit-frame>\n",
      "      <Hsp_identity>15</Hsp_identity>\n",
      "      <Hsp_positive>15</Hsp_positive>\n",
      "      <Hsp_gaps>0</Hsp_gaps>\n",
      "      <Hsp_align-len>15</Hsp_align-len>\n",
      "      <Hsp_qseq>SQETFSDLWKLLPEN</Hsp_qseq>\n",
      "      <Hsp_hseq>SQETFSDLWKLLPEN</Hsp_hseq>\n",
      "      <Hsp_midline>SQETFSDLWKLLPEN</Hsp_midline>\n",
      "    </Hsp>\n",
      "  </Hit_hsps>\n",
      "</Hit>\n",
      "<Hit>\n",
      "  <Hit_num>80</Hit_num>\n",
      "  <Hit_id>gb|AYE20618.1|</Hit_id>\n",
      "  <Hit_def>mutant tumor protein p53 [Homo sapiens]</Hit_def>\n",
      "  <Hit_accession>AYE20618</Hit_accession>\n",
      "  <Hit_len>393</Hit_len>\n",
      "  <Hit_hsps>\n",
      "    <Hsp>\n",
      "      <Hsp_num>1</Hsp_num>\n",
      "      <Hsp_bit-score>35.039</Hsp_bit-score>\n",
      "      <Hsp_score>79</Hsp_score>\n",
      "      <Hsp_evalue>1.28857</Hsp_evalue>\n",
      "      <Hsp_query-from>1</Hsp_query-from>\n",
      "      <Hsp_query-to>15</Hsp_query-to>\n",
      "      <Hsp_hit-from>15</Hsp_hit-from>\n",
      "      <Hsp_hit-to>29</Hsp_hit-to>\n",
      "      <Hsp_query-frame>0</Hsp_query-frame>\n",
      "      <Hsp_hit-frame>0</Hsp_hit-frame>\n",
      "      <Hsp_identity>15</Hsp_identity>\n",
      "      <Hsp_positive>15</Hsp_positive>\n",
      "      <Hsp_gaps>0</Hsp_gaps>\n",
      "      <Hsp_align-len>15</Hsp_align-len>\n",
      "      <Hsp_qseq>SQETFSDLWKLLPEN</Hsp_qseq>\n",
      "      <Hsp_hseq>SQETFSDLWKLLPEN</Hsp_hseq>\n",
      "      <Hsp_midline>SQETFSDLWKLLPEN</Hsp_midline>\n",
      "    </Hsp>\n",
      "  </Hit_hsps>\n",
      "</Hit>\n",
      "<Hit>\n",
      "  <Hit_num>81</Hit_num>\n",
      "  <Hit_id>gb|AYE20623.1|</Hit_id>\n",
      "  <Hit_def>mutant tumor protein p53 [Homo sapiens]</Hit_def>\n",
      "  <Hit_accession>AYE20623</Hit_accession>\n",
      "  <Hit_len>393</Hit_len>\n",
      "  <Hit_hsps>\n",
      "    <Hsp>\n",
      "      <Hsp_num>1</Hsp_num>\n",
      "      <Hsp_bit-score>35.039</Hsp_bit-score>\n",
      "      <Hsp_score>79</Hsp_score>\n",
      "      <Hsp_evalue>1.28857</Hsp_evalue>\n",
      "      <Hsp_query-from>1</Hsp_query-from>\n",
      "      <Hsp_query-to>15</Hsp_query-to>\n",
      "      <Hsp_hit-from>15</Hsp_hit-from>\n",
      "      <Hsp_hit-to>29</Hsp_hit-to>\n",
      "      <Hsp_query-frame>0</Hsp_query-frame>\n",
      "      <Hsp_hit-frame>0</Hsp_hit-frame>\n",
      "      <Hsp_identity>15</Hsp_identity>\n",
      "      <Hsp_positive>15</Hsp_positive>\n",
      "      <Hsp_gaps>0</Hsp_gaps>\n",
      "      <Hsp_align-len>15</Hsp_align-len>\n",
      "      <Hsp_qseq>SQETFSDLWKLLPEN</Hsp_qseq>\n",
      "      <Hsp_hseq>SQETFSDLWKLLPEN</Hsp_hseq>\n",
      "      <Hsp_midline>SQETFSDLWKLLPEN</Hsp_midline>\n",
      "    </Hsp>\n",
      "  </Hit_hsps>\n",
      "</Hit>\n",
      "<Hit>\n",
      "  <Hit_num>82</Hit_num>\n",
      "  <Hit_id>gb|ABB72446.1|</Hit_id>\n",
      "  <Hit_def>tumor protein p53 [Homo sapiens]</Hit_def>\n",
      "  <Hit_accession>ABB72446</Hit_accession>\n",
      "  <Hit_len>393</Hit_len>\n",
      "  <Hit_hsps>\n",
      "    <Hsp>\n",
      "      <Hsp_num>1</Hsp_num>\n",
      "      <Hsp_bit-score>35.039</Hsp_bit-score>\n",
      "      <Hsp_score>79</Hsp_score>\n",
      "      <Hsp_evalue>1.28857</Hsp_evalue>\n",
      "      <Hsp_query-from>1</Hsp_query-from>\n",
      "      <Hsp_query-to>15</Hsp_query-to>\n",
      "      <Hsp_hit-from>15</Hsp_hit-from>\n",
      "      <Hsp_hit-to>29</Hsp_hit-to>\n",
      "      <Hsp_query-frame>0</Hsp_query-frame>\n",
      "      <Hsp_hit-frame>0</Hsp_hit-frame>\n",
      "      <Hsp_identity>15</Hsp_identity>\n",
      "      <Hsp_positive>15</Hsp_positive>\n",
      "      <Hsp_gaps>0</Hsp_gaps>\n",
      "      <Hsp_align-len>15</Hsp_align-len>\n",
      "      <Hsp_qseq>SQETFSDLWKLLPEN</Hsp_qseq>\n",
      "      <Hsp_hseq>SQETFSDLWKLLPEN</Hsp_hseq>\n",
      "      <Hsp_midline>SQETFSDLWKLLPEN</Hsp_midline>\n",
      "    </Hsp>\n",
      "  </Hit_hsps>\n",
      "</Hit>\n",
      "<Hit>\n",
      "  <Hit_num>83</Hit_num>\n",
      "  <Hit_id>ref|XP_002827020.2|</Hit_id>\n",
      "  <Hit_def>cellular tumor antigen p53 [Pongo abelii] &gt;gb|PNJ12854.1| TP53 isoform 1 [Pongo abelii] &gt;gb|PNJ12856.1| TP53 isoform 3 [Pongo abelii]</Hit_def>\n",
      "  <Hit_accession>XP_002827020</Hit_accession>\n",
      "  <Hit_len>393</Hit_len>\n",
      "  <Hit_hsps>\n",
      "    <Hsp>\n",
      "      <Hsp_num>1</Hsp_num>\n",
      "      <Hsp_bit-score>35.039</Hsp_bit-score>\n",
      "      <Hsp_score>79</Hsp_score>\n",
      "      <Hsp_evalue>1.28857</Hsp_evalue>\n",
      "      <Hsp_query-from>1</Hsp_query-from>\n",
      "      <Hsp_query-to>15</Hsp_query-to>\n",
      "      <Hsp_hit-from>15</Hsp_hit-from>\n",
      "      <Hsp_hit-to>29</Hsp_hit-to>\n",
      "      <Hsp_query-frame>0</Hsp_query-frame>\n",
      "      <Hsp_hit-frame>0</Hsp_hit-frame>\n",
      "      <Hsp_identity>15</Hsp_identity>\n",
      "      <Hsp_positive>15</Hsp_positive>\n",
      "      <Hsp_gaps>0</Hsp_gaps>\n",
      "      <Hsp_align-len>15</Hsp_align-len>\n",
      "      <Hsp_qseq>SQETFSDLWKLLPEN</Hsp_qseq>\n",
      "      <Hsp_hseq>SQETFSDLWKLLPEN</Hsp_hseq>\n",
      "      <Hsp_midline>SQETFSDLWKLLPEN</Hsp_midline>\n",
      "    </Hsp>\n",
      "  </Hit_hsps>\n",
      "</Hit>\n",
      "<Hit>\n",
      "  <Hit_num>84</Hit_num>\n",
      "  <Hit_id>gb|AYE20622.1|</Hit_id>\n",
      "  <Hit_def>mutant tumor protein p53 [Homo sapiens]</Hit_def>\n",
      "  <Hit_accession>AYE20622</Hit_accession>\n",
      "  <Hit_len>393</Hit_len>\n",
      "  <Hit_hsps>\n",
      "    <Hsp>\n",
      "      <Hsp_num>1</Hsp_num>\n",
      "      <Hsp_bit-score>35.039</Hsp_bit-score>\n",
      "      <Hsp_score>79</Hsp_score>\n",
      "      <Hsp_evalue>1.28857</Hsp_evalue>\n",
      "      <Hsp_query-from>1</Hsp_query-from>\n",
      "      <Hsp_query-to>15</Hsp_query-to>\n",
      "      <Hsp_hit-from>15</Hsp_hit-from>\n",
      "      <Hsp_hit-to>29</Hsp_hit-to>\n",
      "      <Hsp_query-frame>0</Hsp_query-frame>\n",
      "      <Hsp_hit-frame>0</Hsp_hit-frame>\n",
      "      <Hsp_identity>15</Hsp_identity>\n",
      "      <Hsp_positive>15</Hsp_positive>\n",
      "      <Hsp_gaps>0</Hsp_gaps>\n",
      "      <Hsp_align-len>15</Hsp_align-len>\n",
      "      <Hsp_qseq>SQETFSDLWKLLPEN</Hsp_qseq>\n",
      "      <Hsp_hseq>SQETFSDLWKLLPEN</Hsp_hseq>\n",
      "      <Hsp_midline>SQETFSDLWKLLPEN</Hsp_midline>\n",
      "    </Hsp>\n",
      "  </Hit_hsps>\n",
      "</Hit>\n",
      "<Hit>\n",
      "  <Hit_num>85</Hit_num>\n",
      "  <Hit_id>ref|NP_001075873.1|</Hit_id>\n",
      "  <Hit_def>cellular tumor antigen p53 [Oryctolagus cuniculus] &gt;sp|Q95330.1| RecName: Full=Cellular tumor antigen p53; AltName: Full=Tumor suppressor p53 [Oryctolagus cuniculus] &gt;emb|CAA62216.1| p53 protein [Oryctolagus cuniculus]</Hit_def>\n",
      "  <Hit_accession>NP_001075873</Hit_accession>\n",
      "  <Hit_len>391</Hit_len>\n",
      "  <Hit_hsps>\n",
      "    <Hsp>\n",
      "      <Hsp_num>1</Hsp_num>\n",
      "      <Hsp_bit-score>35.039</Hsp_bit-score>\n",
      "      <Hsp_score>79</Hsp_score>\n",
      "      <Hsp_evalue>1.28889</Hsp_evalue>\n",
      "      <Hsp_query-from>1</Hsp_query-from>\n",
      "      <Hsp_query-to>15</Hsp_query-to>\n",
      "      <Hsp_hit-from>15</Hsp_hit-from>\n",
      "      <Hsp_hit-to>29</Hsp_hit-to>\n",
      "      <Hsp_query-frame>0</Hsp_query-frame>\n",
      "      <Hsp_hit-frame>0</Hsp_hit-frame>\n",
      "      <Hsp_identity>15</Hsp_identity>\n",
      "      <Hsp_positive>15</Hsp_positive>\n",
      "      <Hsp_gaps>0</Hsp_gaps>\n",
      "      <Hsp_align-len>15</Hsp_align-len>\n",
      "      <Hsp_qseq>SQETFSDLWKLLPEN</Hsp_qseq>\n",
      "      <Hsp_hseq>SQETFSDLWKLLPEN</Hsp_hseq>\n",
      "      <Hsp_midline>SQETFSDLWKLLPEN</Hsp_midline>\n",
      "    </Hsp>\n",
      "  </Hit_hsps>\n",
      "</Hit>\n",
      "<Hit>\n",
      "  <Hit_num>86</Hit_num>\n",
      "  <Hit_id>ref|XP_008060532.1|</Hit_id>\n",
      "  <Hit_def>cellular tumor antigen p53 [Carlito syrichta]</Hit_def>\n",
      "  <Hit_accession>XP_008060532</Hit_accession>\n",
      "  <Hit_len>388</Hit_len>\n",
      "  <Hit_hsps>\n",
      "    <Hsp>\n",
      "      <Hsp_num>1</Hsp_num>\n",
      "      <Hsp_bit-score>35.039</Hsp_bit-score>\n",
      "      <Hsp_score>79</Hsp_score>\n",
      "      <Hsp_evalue>1.28938</Hsp_evalue>\n",
      "      <Hsp_query-from>1</Hsp_query-from>\n",
      "      <Hsp_query-to>15</Hsp_query-to>\n",
      "      <Hsp_hit-from>14</Hsp_hit-from>\n",
      "      <Hsp_hit-to>28</Hsp_hit-to>\n",
      "      <Hsp_query-frame>0</Hsp_query-frame>\n",
      "      <Hsp_hit-frame>0</Hsp_hit-frame>\n",
      "      <Hsp_identity>15</Hsp_identity>\n",
      "      <Hsp_positive>15</Hsp_positive>\n",
      "      <Hsp_gaps>0</Hsp_gaps>\n",
      "      <Hsp_align-len>15</Hsp_align-len>\n",
      "      <Hsp_qseq>SQETFSDLWKLLPEN</Hsp_qseq>\n",
      "      <Hsp_hseq>SQETFSDLWKLLPEN</Hsp_hseq>\n",
      "      <Hsp_midline>SQETFSDLWKLLPEN</Hsp_midline>\n",
      "    </Hsp>\n",
      "  </Hit_hsps>\n",
      "</Hit>\n",
      "<Hit>\n",
      "  <Hit_num>87</Hit_num>\n",
      "  <Hit_id>gb|ABL09846.1|</Hit_id>\n",
      "  <Hit_def>nonfunctional tumor suppressor p53 [Homo sapiens] &gt;emb|CAJ28922.1| tumor protein p53 mutant form, partial [Homo sapiens]</Hit_def>\n",
      "  <Hit_accession>ABL09846</Hit_accession>\n",
      "  <Hit_len>382</Hit_len>\n",
      "  <Hit_hsps>\n",
      "    <Hsp>\n",
      "      <Hsp_num>1</Hsp_num>\n",
      "      <Hsp_bit-score>35.039</Hsp_bit-score>\n",
      "      <Hsp_score>79</Hsp_score>\n",
      "      <Hsp_evalue>1.29039</Hsp_evalue>\n",
      "      <Hsp_query-from>1</Hsp_query-from>\n",
      "      <Hsp_query-to>15</Hsp_query-to>\n",
      "      <Hsp_hit-from>15</Hsp_hit-from>\n",
      "      <Hsp_hit-to>29</Hsp_hit-to>\n",
      "      <Hsp_query-frame>0</Hsp_query-frame>\n",
      "      <Hsp_hit-frame>0</Hsp_hit-frame>\n",
      "      <Hsp_identity>15</Hsp_identity>\n",
      "      <Hsp_positive>15</Hsp_positive>\n",
      "      <Hsp_gaps>0</Hsp_gaps>\n",
      "      <Hsp_align-len>15</Hsp_align-len>\n",
      "      <Hsp_qseq>SQETFSDLWKLLPEN</Hsp_qseq>\n",
      "      <Hsp_hseq>SQETFSDLWKLLPEN</Hsp_hseq>\n",
      "      <Hsp_midline>SQETFSDLWKLLPEN</Hsp_midline>\n",
      "    </Hsp>\n",
      "  </Hit_hsps>\n",
      "</Hit>\n",
      "<Hit>\n",
      "  <Hit_num>88</Hit_num>\n",
      "  <Hit_id>ref|XP_023049083.1|</Hit_id>\n",
      "  <Hit_def>cellular tumor antigen p53 isoform X2 [Piliocolobus tephrosceles]</Hit_def>\n",
      "  <Hit_accession>XP_023049083</Hit_accession>\n",
      "  <Hit_len>344</Hit_len>\n",
      "  <Hit_hsps>\n",
      "    <Hsp>\n",
      "      <Hsp_num>1</Hsp_num>\n",
      "      <Hsp_bit-score>35.039</Hsp_bit-score>\n",
      "      <Hsp_score>79</Hsp_score>\n",
      "      <Hsp_evalue>1.29759</Hsp_evalue>\n",
      "      <Hsp_query-from>1</Hsp_query-from>\n",
      "      <Hsp_query-to>15</Hsp_query-to>\n",
      "      <Hsp_hit-from>15</Hsp_hit-from>\n",
      "      <Hsp_hit-to>29</Hsp_hit-to>\n",
      "      <Hsp_query-frame>0</Hsp_query-frame>\n",
      "      <Hsp_hit-frame>0</Hsp_hit-frame>\n",
      "      <Hsp_identity>15</Hsp_identity>\n",
      "      <Hsp_positive>15</Hsp_positive>\n",
      "      <Hsp_gaps>0</Hsp_gaps>\n",
      "      <Hsp_align-len>15</Hsp_align-len>\n",
      "      <Hsp_qseq>SQETFSDLWKLLPEN</Hsp_qseq>\n",
      "      <Hsp_hseq>SQETFSDLWKLLPEN</Hsp_hseq>\n",
      "      <Hsp_midline>SQETFSDLWKLLPEN</Hsp_midline>\n",
      "    </Hsp>\n",
      "  </Hit_hsps>\n",
      "</Hit>\n",
      "<Hit>\n",
      "  <Hit_num>89</Hit_num>\n",
      "  <Hit_id>ref|XP_003929235.1|</Hit_id>\n",
      "  <Hit_def>cellular tumor antigen p53 isoform X2 [Saimiri boliviensis boliviensis]</Hit_def>\n",
      "  <Hit_accession>XP_003929235</Hit_accession>\n",
      "  <Hit_len>393</Hit_len>\n",
      "  <Hit_hsps>\n",
      "    <Hsp>\n",
      "      <Hsp_num>1</Hsp_num>\n",
      "      <Hsp_bit-score>35.039</Hsp_bit-score>\n",
      "      <Hsp_score>79</Hsp_score>\n",
      "      <Hsp_evalue>1.30123</Hsp_evalue>\n",
      "      <Hsp_query-from>1</Hsp_query-from>\n",
      "      <Hsp_query-to>15</Hsp_query-to>\n",
      "      <Hsp_hit-from>15</Hsp_hit-from>\n",
      "      <Hsp_hit-to>29</Hsp_hit-to>\n",
      "      <Hsp_query-frame>0</Hsp_query-frame>\n",
      "      <Hsp_hit-frame>0</Hsp_hit-frame>\n",
      "      <Hsp_identity>15</Hsp_identity>\n",
      "      <Hsp_positive>15</Hsp_positive>\n",
      "      <Hsp_gaps>0</Hsp_gaps>\n",
      "      <Hsp_align-len>15</Hsp_align-len>\n",
      "      <Hsp_qseq>SQETFSDLWKLLPEN</Hsp_qseq>\n",
      "      <Hsp_hseq>SQETFSDLWKLLPEN</Hsp_hseq>\n",
      "      <Hsp_midline>SQETFSDLWKLLPEN</Hsp_midline>\n",
      "    </Hsp>\n",
      "  </Hit_hsps>\n",
      "</Hit>\n",
      "<Hit>\n",
      "  <Hit_num>90</Hit_num>\n",
      "  <Hit_id>sp|P13481.1|</Hit_id>\n",
      "  <Hit_def>RecName: Full=Cellular tumor antigen p53; AltName: Full=Tumor suppressor p53 [Chlorocebus aethiops] &gt;emb|CAA34420.1| unnamed protein product [Chlorocebus aethiops]</Hit_def>\n",
      "  <Hit_accession>P13481</Hit_accession>\n",
      "  <Hit_len>393</Hit_len>\n",
      "  <Hit_hsps>\n",
      "    <Hsp>\n",
      "      <Hsp_num>1</Hsp_num>\n",
      "      <Hsp_bit-score>35.039</Hsp_bit-score>\n",
      "      <Hsp_score>79</Hsp_score>\n",
      "      <Hsp_evalue>1.30123</Hsp_evalue>\n",
      "      <Hsp_query-from>1</Hsp_query-from>\n",
      "      <Hsp_query-to>15</Hsp_query-to>\n",
      "      <Hsp_hit-from>15</Hsp_hit-from>\n",
      "      <Hsp_hit-to>29</Hsp_hit-to>\n",
      "      <Hsp_query-frame>0</Hsp_query-frame>\n",
      "      <Hsp_hit-frame>0</Hsp_hit-frame>\n",
      "      <Hsp_identity>15</Hsp_identity>\n",
      "      <Hsp_positive>15</Hsp_positive>\n",
      "      <Hsp_gaps>0</Hsp_gaps>\n",
      "      <Hsp_align-len>15</Hsp_align-len>\n",
      "      <Hsp_qseq>SQETFSDLWKLLPEN</Hsp_qseq>\n",
      "      <Hsp_hseq>SQETFSDLWKLLPEN</Hsp_hseq>\n",
      "      <Hsp_midline>SQETFSDLWKLLPEN</Hsp_midline>\n",
      "    </Hsp>\n",
      "  </Hit_hsps>\n",
      "</Hit>\n",
      "<Hit>\n",
      "  <Hit_num>91</Hit_num>\n",
      "  <Hit_id>gb|PNI36857.1|</Hit_id>\n",
      "  <Hit_def>LOW QUALITY PROTEIN: TP53 isoform 15 [Pan troglodytes]</Hit_def>\n",
      "  <Hit_accession>PNI36857</Hit_accession>\n",
      "  <Hit_len>382</Hit_len>\n",
      "  <Hit_hsps>\n",
      "    <Hsp>\n",
      "      <Hsp_num>1</Hsp_num>\n",
      "      <Hsp_bit-score>35.039</Hsp_bit-score>\n",
      "      <Hsp_score>79</Hsp_score>\n",
      "      <Hsp_evalue>1.30307</Hsp_evalue>\n",
      "      <Hsp_query-from>1</Hsp_query-from>\n",
      "      <Hsp_query-to>15</Hsp_query-to>\n",
      "      <Hsp_hit-from>15</Hsp_hit-from>\n",
      "      <Hsp_hit-to>29</Hsp_hit-to>\n",
      "      <Hsp_query-frame>0</Hsp_query-frame>\n",
      "      <Hsp_hit-frame>0</Hsp_hit-frame>\n",
      "      <Hsp_identity>15</Hsp_identity>\n",
      "      <Hsp_positive>15</Hsp_positive>\n",
      "      <Hsp_gaps>0</Hsp_gaps>\n",
      "      <Hsp_align-len>15</Hsp_align-len>\n",
      "      <Hsp_qseq>SQETFSDLWKLLPEN</Hsp_qseq>\n",
      "      <Hsp_hseq>SQETFSDLWKLLPEN</Hsp_hseq>\n",
      "      <Hsp_midline>SQETFSDLWKLLPEN</Hsp_midline>\n",
      "    </Hsp>\n",
      "  </Hit_hsps>\n",
      "</Hit>\n",
      "<Hit>\n",
      "  <Hit_num>92</Hit_num>\n",
      "  <Hit_id>pdb|5HOU|A</Hit_id>\n",
      "  <Hit_def>Solution Structure of p53TAD-TAZ1 [Homo sapiens]</Hit_def>\n",
      "  <Hit_accession>5HOU_A</Hit_accession>\n",
      "  <Hit_len>171</Hit_len>\n",
      "  <Hit_hsps>\n",
      "    <Hsp>\n",
      "      <Hsp_num>1</Hsp_num>\n",
      "      <Hsp_bit-score>35.039</Hsp_bit-score>\n",
      "      <Hsp_score>79</Hsp_score>\n",
      "      <Hsp_evalue>1.30718</Hsp_evalue>\n",
      "      <Hsp_query-from>1</Hsp_query-from>\n",
      "      <Hsp_query-to>15</Hsp_query-to>\n",
      "      <Hsp_hit-from>18</Hsp_hit-from>\n",
      "      <Hsp_hit-to>32</Hsp_hit-to>\n",
      "      <Hsp_query-frame>0</Hsp_query-frame>\n",
      "      <Hsp_hit-frame>0</Hsp_hit-frame>\n",
      "      <Hsp_identity>15</Hsp_identity>\n",
      "      <Hsp_positive>15</Hsp_positive>\n",
      "      <Hsp_gaps>0</Hsp_gaps>\n",
      "      <Hsp_align-len>15</Hsp_align-len>\n",
      "      <Hsp_qseq>SQETFSDLWKLLPEN</Hsp_qseq>\n",
      "      <Hsp_hseq>SQETFSDLWKLLPEN</Hsp_hseq>\n",
      "      <Hsp_midline>SQETFSDLWKLLPEN</Hsp_midline>\n",
      "    </Hsp>\n",
      "  </Hit_hsps>\n",
      "</Hit>\n",
      "<Hit>\n",
      "  <Hit_num>93</Hit_num>\n",
      "  <Hit_id>gb|ABL09847.1|</Hit_id>\n",
      "  <Hit_def>nonfunctional tumor suppressor p53 [Homo sapiens] &gt;emb|CAJ28923.1| tumor protein p53 mutant form, partial [Homo sapiens]</Hit_def>\n",
      "  <Hit_accession>ABL09847</Hit_accession>\n",
      "  <Hit_len>343</Hit_len>\n",
      "  <Hit_hsps>\n",
      "    <Hsp>\n",
      "      <Hsp_num>1</Hsp_num>\n",
      "      <Hsp_bit-score>35.039</Hsp_bit-score>\n",
      "      <Hsp_score>79</Hsp_score>\n",
      "      <Hsp_evalue>1.31055</Hsp_evalue>\n",
      "      <Hsp_query-from>1</Hsp_query-from>\n",
      "      <Hsp_query-to>15</Hsp_query-to>\n",
      "      <Hsp_hit-from>15</Hsp_hit-from>\n",
      "      <Hsp_hit-to>29</Hsp_hit-to>\n",
      "      <Hsp_query-frame>0</Hsp_query-frame>\n",
      "      <Hsp_hit-frame>0</Hsp_hit-frame>\n",
      "      <Hsp_identity>15</Hsp_identity>\n",
      "      <Hsp_positive>15</Hsp_positive>\n",
      "      <Hsp_gaps>0</Hsp_gaps>\n",
      "      <Hsp_align-len>15</Hsp_align-len>\n",
      "      <Hsp_qseq>SQETFSDLWKLLPEN</Hsp_qseq>\n",
      "      <Hsp_hseq>SQETFSDLWKLLPEN</Hsp_hseq>\n",
      "      <Hsp_midline>SQETFSDLWKLLPEN</Hsp_midline>\n",
      "    </Hsp>\n",
      "  </Hit_hsps>\n",
      "</Hit>\n",
      "<Hit>\n",
      "  <Hit_num>94</Hit_num>\n",
      "  <Hit_id>gb|PNJ12867.1|</Hit_id>\n",
      "  <Hit_def>TP53 isoform 15 [Pongo abelii]</Hit_def>\n",
      "  <Hit_accession>PNJ12867</Hit_accession>\n",
      "  <Hit_len>343</Hit_len>\n",
      "  <Hit_hsps>\n",
      "    <Hsp>\n",
      "      <Hsp_num>1</Hsp_num>\n",
      "      <Hsp_bit-score>35.039</Hsp_bit-score>\n",
      "      <Hsp_score>79</Hsp_score>\n",
      "      <Hsp_evalue>1.31055</Hsp_evalue>\n",
      "      <Hsp_query-from>1</Hsp_query-from>\n",
      "      <Hsp_query-to>15</Hsp_query-to>\n",
      "      <Hsp_hit-from>15</Hsp_hit-from>\n",
      "      <Hsp_hit-to>29</Hsp_hit-to>\n",
      "      <Hsp_query-frame>0</Hsp_query-frame>\n",
      "      <Hsp_hit-frame>0</Hsp_hit-frame>\n",
      "      <Hsp_identity>15</Hsp_identity>\n",
      "      <Hsp_positive>15</Hsp_positive>\n",
      "      <Hsp_gaps>0</Hsp_gaps>\n",
      "      <Hsp_align-len>15</Hsp_align-len>\n",
      "      <Hsp_qseq>SQETFSDLWKLLPEN</Hsp_qseq>\n",
      "      <Hsp_hseq>SQETFSDLWKLLPEN</Hsp_hseq>\n",
      "      <Hsp_midline>SQETFSDLWKLLPEN</Hsp_midline>\n",
      "    </Hsp>\n",
      "  </Hit_hsps>\n",
      "</Hit>\n",
      "<Hit>\n",
      "  <Hit_num>95</Hit_num>\n",
      "  <Hit_id>ref|XP_033278862.1|</Hit_id>\n",
      "  <Hit_def>cellular tumor antigen p53 isoform X1 [Orcinus orca]</Hit_def>\n",
      "  <Hit_accession>XP_033278862</Hit_accession>\n",
      "  <Hit_len>400</Hit_len>\n",
      "  <Hit_hsps>\n",
      "    <Hsp>\n",
      "      <Hsp_num>1</Hsp_num>\n",
      "      <Hsp_bit-score>35.039</Hsp_bit-score>\n",
      "      <Hsp_score>79</Hsp_score>\n",
      "      <Hsp_evalue>1.31288</Hsp_evalue>\n",
      "      <Hsp_query-from>1</Hsp_query-from>\n",
      "      <Hsp_query-to>15</Hsp_query-to>\n",
      "      <Hsp_hit-from>28</Hsp_hit-from>\n",
      "      <Hsp_hit-to>42</Hsp_hit-to>\n",
      "      <Hsp_query-frame>0</Hsp_query-frame>\n",
      "      <Hsp_hit-frame>0</Hsp_hit-frame>\n",
      "      <Hsp_identity>15</Hsp_identity>\n",
      "      <Hsp_positive>15</Hsp_positive>\n",
      "      <Hsp_gaps>0</Hsp_gaps>\n",
      "      <Hsp_align-len>15</Hsp_align-len>\n",
      "      <Hsp_qseq>SQETFSDLWKLLPEN</Hsp_qseq>\n",
      "      <Hsp_hseq>SQETFSDLWKLLPEN</Hsp_hseq>\n",
      "      <Hsp_midline>SQETFSDLWKLLPEN</Hsp_midline>\n",
      "    </Hsp>\n",
      "  </Hit_hsps>\n",
      "</Hit>\n",
      "<Hit>\n",
      "  <Hit_num>96</Hit_num>\n",
      "  <Hit_id>ref|XP_025219300.1|</Hit_id>\n",
      "  <Hit_def>cellular tumor antigen p53 isoform X1 [Theropithecus gelada]</Hit_def>\n",
      "  <Hit_accession>XP_025219300</Hit_accession>\n",
      "  <Hit_len>394</Hit_len>\n",
      "  <Hit_hsps>\n",
      "    <Hsp>\n",
      "      <Hsp_num>1</Hsp_num>\n",
      "      <Hsp_bit-score>35.039</Hsp_bit-score>\n",
      "      <Hsp_score>79</Hsp_score>\n",
      "      <Hsp_evalue>1.31385</Hsp_evalue>\n",
      "      <Hsp_query-from>1</Hsp_query-from>\n",
      "      <Hsp_query-to>15</Hsp_query-to>\n",
      "      <Hsp_hit-from>15</Hsp_hit-from>\n",
      "      <Hsp_hit-to>29</Hsp_hit-to>\n",
      "      <Hsp_query-frame>0</Hsp_query-frame>\n",
      "      <Hsp_hit-frame>0</Hsp_hit-frame>\n",
      "      <Hsp_identity>15</Hsp_identity>\n",
      "      <Hsp_positive>15</Hsp_positive>\n",
      "      <Hsp_gaps>0</Hsp_gaps>\n",
      "      <Hsp_align-len>15</Hsp_align-len>\n",
      "      <Hsp_qseq>SQETFSDLWKLLPEN</Hsp_qseq>\n",
      "      <Hsp_hseq>SQETFSDLWKLLPEN</Hsp_hseq>\n",
      "      <Hsp_midline>SQETFSDLWKLLPEN</Hsp_midline>\n",
      "    </Hsp>\n",
      "  </Hit_hsps>\n",
      "</Hit>\n",
      "<Hit>\n",
      "  <Hit_num>97</Hit_num>\n",
      "  <Hit_id>ref|XP_008008385.1|</Hit_id>\n",
      "  <Hit_def>cellular tumor antigen p53 isoform X2 [Chlorocebus sabaeus] &gt;ref|XP_011849930.1| PREDICTED: cellular tumor antigen p53 isoform X1 [Mandrillus leucophaeus] &gt;ref|XP_011849931.1| PREDICTED: cellular tumor antigen p53 isoform X1 [Mandrillus leucophaeus] &gt;ref|XP_011907813.1| PREDICTED: cellular tumor antigen p53 isoform X1 [Cercocebus atys]</Hit_def>\n",
      "  <Hit_accession>XP_008008385</Hit_accession>\n",
      "  <Hit_len>393</Hit_len>\n",
      "  <Hit_hsps>\n",
      "    <Hsp>\n",
      "      <Hsp_num>1</Hsp_num>\n",
      "      <Hsp_bit-score>35.039</Hsp_bit-score>\n",
      "      <Hsp_score>79</Hsp_score>\n",
      "      <Hsp_evalue>1.31401</Hsp_evalue>\n",
      "      <Hsp_query-from>1</Hsp_query-from>\n",
      "      <Hsp_query-to>15</Hsp_query-to>\n",
      "      <Hsp_hit-from>15</Hsp_hit-from>\n",
      "      <Hsp_hit-to>29</Hsp_hit-to>\n",
      "      <Hsp_query-frame>0</Hsp_query-frame>\n",
      "      <Hsp_hit-frame>0</Hsp_hit-frame>\n",
      "      <Hsp_identity>15</Hsp_identity>\n",
      "      <Hsp_positive>15</Hsp_positive>\n",
      "      <Hsp_gaps>0</Hsp_gaps>\n",
      "      <Hsp_align-len>15</Hsp_align-len>\n",
      "      <Hsp_qseq>SQETFSDLWKLLPEN</Hsp_qseq>\n",
      "      <Hsp_hseq>SQETFSDLWKLLPEN</Hsp_hseq>\n",
      "      <Hsp_midline>SQETFSDLWKLLPEN</Hsp_midline>\n",
      "    </Hsp>\n",
      "  </Hit_hsps>\n",
      "</Hit>\n",
      "<Hit>\n",
      "  <Hit_num>98</Hit_num>\n",
      "  <Hit_id>ref|XP_036692565.1|</Hit_id>\n",
      "  <Hit_def>cellular tumor antigen p53 [Balaenoptera musculus]</Hit_def>\n",
      "  <Hit_accession>XP_036692565</Hit_accession>\n",
      "  <Hit_len>387</Hit_len>\n",
      "  <Hit_hsps>\n",
      "    <Hsp>\n",
      "      <Hsp_num>1</Hsp_num>\n",
      "      <Hsp_bit-score>35.039</Hsp_bit-score>\n",
      "      <Hsp_score>79</Hsp_score>\n",
      "      <Hsp_evalue>1.31501</Hsp_evalue>\n",
      "      <Hsp_query-from>1</Hsp_query-from>\n",
      "      <Hsp_query-to>15</Hsp_query-to>\n",
      "      <Hsp_hit-from>15</Hsp_hit-from>\n",
      "      <Hsp_hit-to>29</Hsp_hit-to>\n",
      "      <Hsp_query-frame>0</Hsp_query-frame>\n",
      "      <Hsp_hit-frame>0</Hsp_hit-frame>\n",
      "      <Hsp_identity>15</Hsp_identity>\n",
      "      <Hsp_positive>15</Hsp_positive>\n",
      "      <Hsp_gaps>0</Hsp_gaps>\n",
      "      <Hsp_align-len>15</Hsp_align-len>\n",
      "      <Hsp_qseq>SQETFSDLWKLLPEN</Hsp_qseq>\n",
      "      <Hsp_hseq>SQETFSDLWKLLPEN</Hsp_hseq>\n",
      "      <Hsp_midline>SQETFSDLWKLLPEN</Hsp_midline>\n",
      "    </Hsp>\n",
      "  </Hit_hsps>\n",
      "</Hit>\n",
      "<Hit>\n",
      "  <Hit_num>99</Hit_num>\n",
      "  <Hit_id>ref|XP_012511070.1|</Hit_id>\n",
      "  <Hit_def>PREDICTED: cellular tumor antigen p53 [Propithecus coquereli]</Hit_def>\n",
      "  <Hit_accession>XP_012511070</Hit_accession>\n",
      "  <Hit_len>385</Hit_len>\n",
      "  <Hit_hsps>\n",
      "    <Hsp>\n",
      "      <Hsp_num>1</Hsp_num>\n",
      "      <Hsp_bit-score>35.039</Hsp_bit-score>\n",
      "      <Hsp_score>79</Hsp_score>\n",
      "      <Hsp_evalue>1.31535</Hsp_evalue>\n",
      "      <Hsp_query-from>1</Hsp_query-from>\n",
      "      <Hsp_query-to>15</Hsp_query-to>\n",
      "      <Hsp_hit-from>15</Hsp_hit-from>\n",
      "      <Hsp_hit-to>29</Hsp_hit-to>\n",
      "      <Hsp_query-frame>0</Hsp_query-frame>\n",
      "      <Hsp_hit-frame>0</Hsp_hit-frame>\n",
      "      <Hsp_identity>15</Hsp_identity>\n",
      "      <Hsp_positive>15</Hsp_positive>\n",
      "      <Hsp_gaps>0</Hsp_gaps>\n",
      "      <Hsp_align-len>15</Hsp_align-len>\n",
      "      <Hsp_qseq>SQETFSDLWKLLPEN</Hsp_qseq>\n",
      "      <Hsp_hseq>SQETFSDLWKLLPEN</Hsp_hseq>\n",
      "      <Hsp_midline>SQETFSDLWKLLPEN</Hsp_midline>\n",
      "    </Hsp>\n",
      "  </Hit_hsps>\n",
      "</Hit>\n",
      "<Hit>\n",
      "  <Hit_num>100</Hit_num>\n",
      "  <Hit_id>ref|XP_037665508.1|</Hit_id>\n",
      "  <Hit_def>cellular tumor antigen p53 [Choloepus didactylus] &gt;ref|XP_037665509.1| cellular tumor antigen p53 [Choloepus didactylus]</Hit_def>\n",
      "  <Hit_accession>XP_037665508</Hit_accession>\n",
      "  <Hit_len>385</Hit_len>\n",
      "  <Hit_hsps>\n",
      "    <Hsp>\n",
      "      <Hsp_num>1</Hsp_num>\n",
      "      <Hsp_bit-score>35.039</Hsp_bit-score>\n",
      "      <Hsp_score>79</Hsp_score>\n",
      "      <Hsp_evalue>1.31535</Hsp_evalue>\n",
      "      <Hsp_query-from>1</Hsp_query-from>\n",
      "      <Hsp_query-to>15</Hsp_query-to>\n",
      "      <Hsp_hit-from>15</Hsp_hit-from>\n",
      "      <Hsp_hit-to>29</Hsp_hit-to>\n",
      "      <Hsp_query-frame>0</Hsp_query-frame>\n",
      "      <Hsp_hit-frame>0</Hsp_hit-frame>\n",
      "      <Hsp_identity>15</Hsp_identity>\n",
      "      <Hsp_positive>15</Hsp_positive>\n",
      "      <Hsp_gaps>0</Hsp_gaps>\n",
      "      <Hsp_align-len>15</Hsp_align-len>\n",
      "      <Hsp_qseq>SQETFSDLWKLLPEN</Hsp_qseq>\n",
      "      <Hsp_hseq>SQETFSDLWKLLPEN</Hsp_hseq>\n",
      "      <Hsp_midline>SQETFSDLWKLLPEN</Hsp_midline>\n",
      "    </Hsp>\n",
      "  </Hit_hsps>\n",
      "</Hit>\n",
      "</Iteration_hits>\n",
      "  <Iteration_stat>\n",
      "    <Statistics>\n",
      "      <Statistics_db-num>436338278</Statistics_db-num>\n",
      "      <Statistics_db-len>161860501762</Statistics_db-len>\n",
      "      <Statistics_hsp-len>0</Statistics_hsp-len>\n",
      "      <Statistics_eff-space>0</Statistics_eff-space>\n",
      "      <Statistics_kappa>0.041</Statistics_kappa>\n",
      "      <Statistics_lambda>0.267</Statistics_lambda>\n",
      "      <Statistics_entropy>0.14</Statistics_entropy>\n",
      "    </Statistics>\n",
      "  </Iteration_stat>\n",
      "</Iteration>\n",
      "</BlastOutput_iterations>\n",
      "</BlastOutput>\n",
      "\n"
     ]
    }
   ],
   "source": [
    "print(result)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "slideshow": {
     "slide_type": "skip"
    }
   },
   "source": [
    "# Exercise: Anything new?\n",
    "\n",
    "Write a script to see if a website has changed since the last time you checked.  The script will  save the text of the website in the current directory and compare the previously saved text to the current website."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "slideshow": {
     "slide_type": "slide"
    }
   },
   "source": [
    "# Exercise: BLAST IT!\n",
    "\n",
    "Use the BLAST URLAPI to find structures with similar sequence to a user-supplied FASTA protein sequence. Your script will take the name of a FASTA file as its only argument. The contents of this file should be provided as the QUERY parameter in a BLAST URL PUT request querying the pdb database (DATABASE=pdb) using blastp (PROGRAM=blastp).  You should extract the RID from the response using a regular expression.\n",
    "\n",
    "Using the RID, you then submit a GET request. Your GET request may either return the desired data, or it may return a status HTML page (even if you request XML) if the request hasn't finished. You should look for the presence of the string Status=WAITING in the response. If this string is present, you should repeat your GET request every 5 seconds (`time.sleep(5)`) until you get a response without it.  It is typical for it to take 30 seconds.\n",
    "\n",
    "Print out the final XML response.\n",
    "\n",
    "Example fasta file: http://mscbio2025.net/files/brca.fasta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "slideshow": {
     "slide_type": "notes"
    }
   },
   "outputs": [],
   "source": [
    "import requests,re,sys,time\n",
    "import xml.etree.ElementTree as ET\n",
    "\n",
    "if len(sys.argv) < 2:\n",
    "\tprint(\"Need fasta file\")\n",
    "\tsys.exit(1)\n",
    "\n",
    "f = open(sys.argv[1])\n",
    "fasta = f.read()\n",
    "\n",
    "values = {'CMD': 'PUT', 'DATABASE':'pdb', 'PROGRAM': 'blastp','QUERY':fasta, 'PROGRAM_NAME':'blastp','BLAST_PROGRAM':'blastp'}\n",
    "res = requests.get('http://www.ncbi.nlm.nih.gov/blast/Blast.cgi',values)\n",
    "response = res.text\n",
    "m = re.search(r'RID = (\\S+)', response)\n",
    "\n",
    "rid = m.group(1)\n",
    "result = 'Status=WAITING'\n",
    "values = {'CMD':'GET','RID': rid, 'FORMAT_TYPE':'XML'}\n",
    "while re.search('Status=WAITING',result):\n",
    "    time.sleep(5)\n",
    "    res = requests.get('http://www.ncbi.nlm.nih.gov/blast/Blast.cgi',values)\n",
    "    result = res.text\n",
    "    #print(result)\n",
    "    #print(res.url)\n",
    "\n",
    "root = ET.fromstring(result)\n",
    "cnt = 0\n",
    "#print out some summary information from the hits\n",
    "for hit in root.iter('Hit'):\n",
    "\thsp = hit.find('Hit_hsps').find('Hsp')\n",
    "\tevalue = hsp.find('Hsp_evalue').text\n",
    "\tident = float(hsp.find('Hsp_identity').text)\n",
    "\tlength = float(hsp.find('Hsp_align-len').text)\n",
    "\tpdb_ch = hit.find('Hit_accession').text\n",
    "\tm = re.search(r'(\\S+)_', pdb_ch)\n",
    "\tresolution = '0'\n",
    "\tif m:\n",
    "\t\tpdb = m.group(1)\n",
    "\t\tresponse = requests.get('http://www.pdb.org/pdb/rest/describePDB',{'structureId': pdb}).text\n",
    "\t\tm = re.search(r'resolution=\"(\\S+)\"',response)\n",
    "\t\tif m:\n",
    "\t\t\tresolution = m.group(1)\n",
    "\tprint('%s %s %.2f %s' % (pdb_ch, evalue, ident/length,resolution))\n",
    "\tcnt += 1\n",
    "\tif cnt >= 10:\n",
    "\t\tbreak\n"
   ]
  }
 ],
 "metadata": {
  "celltoolbar": "Slideshow",
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.10"
  },
  "widgets": {
   "state": {},
   "version": "1.1.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
